<h3>About the Medaka genome</h3>


<h4>Assembly</h4>

<p><img src="/img/species/pic_Oryzias_latipes.png" style="width:120px;height:75px" style="float:left;margin-right:10px;margin-bottom:10px" alt="Oryzias latipes" title="Medaka"/> 
The Japanese medaka (<i>Oryzias latipes</i>) genome, which has been  sequenced 
by whole-genome
shotgun sequencing and by using Ramen assembler, is approximately  700Mb 
in length and
comprises 24 pairs of chromosomes; 2 metacentric, 8 submetacentric, 1  
subtelocentric
and 13 acrocentric chromosomes in haploid condition.
The genome sequencing and assembly for MEDAKA1 (October 2005) were  
provided by
the <a href=" http://dolphin.lab.nig.ac.jp/medaka/">National Institute of Genetics</a> (NIG)
and the <a href="http://medaka.utgenome.org/">University of Tokyo</a>.</p>

<p>This Ensembl website presents sequence data provided by the two  
institutes. It also displays
their gene set together with the Ensembl gene set and related analysis.
See the <a href="http://medaka.utgenome.org/#policy">data release policy</a>
for the medaka genome.</p>

<h4>Annotation</h4>

<p>The gene set for Medaka was built using a modified version of the 
standard Ensembl genebuild pipeline. The species-specific sequence 
resources (medaka cDNA and protein) are very limited, so the majority of 
gene models are based on genewise alignments of proteins from other 
species that are genetically distant to medaka. To improve accuracy, 
genewise alignments were made to stretches of genomic sequence rather 
than to 'miniseqs'. Medaka cDNAs were aligned and used to add UTRs to 
the genewise predictions where possible. The gene models were then 
assessed by generating sets of potential orthologs to genes from other 
species. Potentially missing predictions and partial gene predictions 
were identified by examining the orthologs and used to improve the gene 
models. Finally, additional genes were predicted using medaka EST 
evidence, and medaka ESTs were used to add UTRs to the gene set where 
possible.</p>
