<html>
<head>

  <meta name="navigation" content="Comparative Genomics">
  <title>Multiple genome alignments</title>


</head>

<body>


<h1 id="multiplegenomealignments">Multiple genome alignments</h1>

<p>Multiple alignments are calculated between groups of genomes. These are used to calculate <a href="http://www.ensembl.org/info/genome/compara/ancestral_sequences.html">ancestral sequences</a>, <a href="http://www.ensembl.org/info/genome/compara/age_of_base.html">age of base</a>, <a href="http://www.ensembl.org/info/genome/compara/conservation_and_constrained.html">conservation scores and constrained elements</a>.</p>

<h2 id="alignmentsavailable">Alignments available</h2>

  <table>
<thead>
<tr>
<th>Name</th>
<th>Genomes</th>
<th>Method used</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="/info/genome/compara/mlss.html?mlss=1104">31 amniota vertebrates</a></td>
<td>Human, Chimpanzee, Bonobo, Gorilla, Orangutan, Gibbon, Olive baboon, Vervet-AGM, Macaque, Crab-eating macaque, Marmoset, Mouse lemur, Mouse, Algerian mouse, Ryukyu mouse, Shrew mouse, Rat, Prairie vole, Rabbit, Cow, Sheep, Pig, Dog, Cat, Horse, Opossum, Platypus, Chicken, Turkey, Zebra finch, Anole lizard</td>
<td>PECAN</td>
</tr>
<tr>
<td><a href="/info/genome/compara/mlss.html?mlss=768">5 teleost fish</a></td>
<td>Zebrafish, Medaka, Tetraodon, Stickleback, Spotted gar</td>
<td>EPO</td>
</tr>
<tr>
<td><a href="/info/genome/compara/mlss.html?mlss=1101">12 primates</a></td>
<td>Human, Gorilla, Chimpanzee, Bonobo, Orangutan, Gibbon, Crab-eating macaque, Macaque, Olive baboon, Vervet-AGM, Marmoset, Mouse lemur</td>
<td>EPO</td>
</tr>
<tr>
<td><a href="/info/genome/compara/mlss.html?mlss=825">4 sauropsids</a></td>
<td>Anole lizard, Zebra finch, Chicken, Turkey</td>
<td>EPO</td>
</tr>
<tr>
<td><a href="/info/genome/compara/mlss.html?mlss=1102">25 eutherian mammals</a></td>
<td>Human, Chimpanzee, Bonobo, Gorilla, Orangutan, Gibbon, Olive baboon, Vervet-AGM, Macaque, Crab-eating macaque, Marmoset, Mouse lemur, Mouse, Algerian mouse, Ryukyu mouse, Shrew mouse, Rat, Prairie vole, Rabbit, Cow, Sheep, Pig, Dog, Cat, Horse</td>
<td>EPO</td>
</tr>
<tr>
<td><a href="/info/genome/compara/mlss.html?mlss=793">11 fish</a></td>
<td>Zebrafish, Cave fish, Cod, Tilapia, Medaka, Platyfish, Amazon molly, Fugu, Tetraodon, Stickleback, Spotted gar</td>
<td>EPO-LOW-COVERAGE</td>
</tr>
<tr>
<td><a href="/info/genome/compara/mlss.html?mlss=826">7 sauropsids</a></td>
<td>Duck, Chicken, Turkey, Chinese softshell turtle, Anole lizard, Zebra finch, Flycatcher</td>
<td>EPO-LOW-COVERAGE</td>
</tr>
<tr>
<td><a href="/info/genome/compara/mlss.html?mlss=1103">67 eutherian mammals</a></td>
<td>Human, Chimpanzee, Bonobo, Gorilla, Orangutan, Gibbon, Angola colobus, Golden snub-nosed monekey, Black snub-nosed monkey, Pig-tailed macaque, Macaque, Crab-eating macaque, Olive baboon, Vervet-AGM, Sooty mangabey, Drill, Marmoset, Capuchin, Bolivian squirrel monkey, Ma&#39;s night monkey, Tarsier, Coquerel&#39;s sifaka, Mouse lemur, Bushbaby, Mouse, Algerian mouse, Ryukyu mouse, Shew mouse, Rat, Upper Gallilee mountains blnd mole rat, Chinese hamster CriGri, Chinese hamster CHOK1GS, Golden Hamster, Northern American deer mouse, Prairie vole, Lesser Egyptian jerboa, Kangaroo rat, Long-tailed chinchilla, Guinea pig, Brazilian guinea pig, Degu, Naked mole rate female, Naked mole rat male, Damara mole rat, Squirrel, Pika, Rabbit, Tree shrew, Horse, Dog, Panda, Ferret, Cat, Hedgehog, Shrew, Cow, Sheep, Dolphin, Alpaca, Pig, Microbat, Megabat, Sloth, Armadillo, Lesser hedgehog tenrec, Hyrax, Elephant</td>
<td>EPO-LOW-COVERAGE</td>
</tr>
<tr>
<td><a href="/info/genome/compara/mlss.html?mlss=835">18 murinae</a></td>
<td>Mouse , Mouse DBA/2J, Mouse CAST/EiJ, Mouse C57BL/6NJ, Mouse C3H/HeJ, Mouse WSB/EiJ, Mouse A/J, Mouse BALB/cJ, Mouse LP/J, Mouse NOD/ShiLtJ, Mouse 129S1/SvImJ, Mouse CBA/J, Mouse AKR/J, Mouse NZO/HlLtJ, Mouse FVB/NJ, Mouse PWK/PhJ, Mouse SPRET/EiJ, Rat</td>
<td>Cactus alignment</td>
</tr>
</tbody>
</table>

<h2 id="alignmentmethods">Alignment methods</h2>

<h3 id="pecanmultiplealignment">PECAN Multiple Alignment</h3>

<p><a href="http://hgwdev.cse.ucsc.edu/~benedict/code/Pecan.html">Pecan</a> is used to provide global multiple genomic alignments. First, Mercator is used to build a synteny map between the genomes and then Pecan builds alignments in these syntenic regions.</p>

<p>Pecan is a global multiple sequence alignment program that makes practical the probabilistic consistency methodology for significant numbers of sequences of practically arbitrary length. As input it takes a set of sequences and a phylogenetic tree. The parameters and heuristics it employs are highly user configurable, it is written entirely in Java and also requires the installation of Exonerate.</p>

<h3 id="epomultiplealignment">EPO Multiple Alignment</h3>

<p>The <a href="EPO_pipeline.md">EPO (Enredo, Pecan, Ortheus) pipeline</a> is a three step pipeline for whole-genome multiple alignments.</p>

<ol>
<li>Enredo produces colinear segments from extant genomes handling both rearrangements, deletions and duplications.</li>

<li>Pecan, as described above, is used to align these segments.</li>

<li>Finally, Ortheus is used to create genome-wide ancestral sequence reconstructions.</li>
</ol>

<p>Further details on these methods can be found at: <a href="http://europepmc.org/abstract/MED/18849524">Enredo and Pecan: Genome-wide mammalian consistency-based multiple alignment with paralogs</a></p>

<p>Due to difficulties with running Ortheus on the fragmented assemblies, we have two flavours of the pipeline.</p>

<ol>
<li>The plain EPO pipeline is available on the chromosome-level genomes, listed as EPO in the table above</li>

<li>The scaffold-level genomes are then projected onto the EPO alignments using LastZ-net alignments, listed as EPO-LOW-COVERAGE.</li>
</ol>

<p>By construction, each pair of EPO and EPO-LOW-COVERAGE alignments represent the exact same alignment of chromosome-level genomes.</p>

<h3 id="progressivecactus">Progressive Cactus</h3>

<p>Progressive-Cactus is a next-generation aligner that stores whole-genome alignments in a graph structure. Genomes can be added incrementally, which makes it scalable to hundreds of genomes. Further details on these methods can be found in <a href="http://europepmc.org/articles/PMC3166836">Algorithms for genome multiple sequence alignment</a> and <a href="http://europepmc.org/abstract/MED/21385048">Cactus graphs for genome comparisons</a>.</p>

</body>
</html>
