<html>
<head>

  <meta name="navigation" content="Comparative Genomics">
  <title>Orthology quality-controls</title>


</head>

<body>

<h1>Orthology quality-controls</h1>

<p>
The orthology QC is generated by 2 pipelines that focus on different levels of genetic difference: nucleotide-level mutations and larger-scale rearrangements.
Both approaches are completely independent from the inference of orthologies itself and are used as external evidence to confirm them.
</p>

<h2 id="goc">Gene Order Conservation score</h2>

<p>
The pipeline that focuses on large-scale rearrangements generates the gene order conservation (GOC) scores. This pipeline makes the assumption that rearrangements are likely to happen to a group of contiguous genes, rather than genes in isolation - hence we expect that the gene neighbourhoods surrounding each gene of the pair would be well conserved.
</p>

<strong>The GOC pipeline can be divided into the following steps:</strong><br />

 <ol>
      <li>Load all the predicted orthologs for a pair of species</li>
      <li>Separate the orthologs into their respective chromosomes</li>
      <li>Discard any ortholog that is by itself (usually in a scaffold). As these orthologs automatically get a NULL score for having no neighbours</li>
      <li>Order the set of orthologs in each chromosome by their start positions on the chosen reference genome</li>
      <li>For each orthologous pair, fetch the two genes upstream and downstream</li>
      <li>Check whether they are also identified as orthologues and in the same orientation</li>
      <li>Each match is scored as 25% meaning if all four neighbouring genes match that ortholog gets a GOC score of 100% for this reference genome</li>
      <li>Go back to step 4 and repeat using the alternative species as the reference genome</li>
      <li>Now we have 2 GOC scores for each other. We currently report the max of these scores</li>
</ol>

<h3>Example comparison</h3>
<img style="width: 187px; height: 349px;" alt="ortholog_qc_goc" src="ortholog_qc_goc_thumb.png" />
<p>Of the 4 neighbouring genes, 3 are orthologues and in conserved order and position, resulting in a GOC score of 75.</p>

<div class="info left-margin right-margin">
    <h3>Availability</h3>
    <div class="message-pad">
        <p>
        This pipeline on run on all bony vertebrates (<em>Euteleostomi</em>)
        </p>
    </div>
</div>

<h2 id="wga">Whole Genome Alignment score</h2>

<p>
The pipeline that focuses on the nucleotide level mutations generates the whole genome alignment (WGA) score. It assumes that high-quality “true” orthologs should be well aligned to each other. This approach is based on the nucleotide level differences, taking advantage of the wealth of alignment information available through EnsEMBL’s comparative genetics analyses.
</p>


<strong>The main steps of this WGA pipeline are:</strong><br />

 <ol>
      <li>First, as the coverage over exonic regions is taken into account by this pipeline, exon boundaries are fetched for all genes in all species of interest. These are stored in a local table for efficiency gains later</li>
      <li>Next, the species are paired off and all alignments between each pair are detected. All predicted orthologs between the pair are fetched and batched (default = 10)</li>
      <li>The coverage over each member of the orthology is calculated using every available alignment. Coverage over exons is regarded as a higher importance than intronic regions, so a weighted score is generated. The score takes the coverage over exons as a base, with bonus points given for coverage over the introns (normalized by the proportion of intronic sequence in the gene). Scores for every alignment over both genes is stored in an intermediate table</li>
      <li>Finally, an overall score for the homology prediction, as a whole is computed. This can be defined as the maximum score, after the score for the pair of genes has been averaged for each alignment i.e. we report the average score for the greatest-coverage alignment</li>
</ol>

<h3>Example comparison</h3>
<img style="width: 470px; height: 274px;" alt="ortholog_qc_wga" src="ortholog_qc_wga_thumb.png" />

<div class="info left-margin right-margin">
    <h3>Availability</h3>
    <div class="message-pad">
        <p>
        This pipeline on run on all LastZ and EPO alignments.
        </p>
    </div>
</div>


<h2 id="hc">High-confidence orthologies</h2>

<p>
We have defined for several taxonomic groups stringent thresholds to flag orthologies as <em>high-confidence</em>.
For each pair of species, we select the thresholds corresponding to their most-recent common ancestor.
To be flagged as <em>high-confidence</em>, a pair of orthologue must have a sufficient %identity, and have a GOC score or WGA coverage greater or equal to the threshold. If none of the latter two metrics are available for this pair of species, we fall back to the tree-compliance metric explained in <a href="homology_method.html#homology_types">this document</a>.
</p>

<table class="ss">
    <tr>
        <th>Clades</th>
        <th>Min. GOC score</th>
        <th>Min. WGA score</th>
        <th>Min. %identity</th>
    </tr>
    <tr>
        <td><em>Apes</em>, <em>Murinae</em></td>
        <td>75</td><td>75</td><td>80</td>
    </tr>
    <tr>
        <td><em>Mammalia</em>, <em>Aves</em>, <em>Percomorpha</em></td>
        <td>75</td><td>75</td><td>50</td>
    </tr>
    <tr>
        <td><em>Euteleostomi</em></td>
        <td>50</td><td>50</td><td>25</td>
    </tr>
    <tr>
        <td>Others</td>
        <td>No threshold used</td><td>No threshold used</td><td>25</td>
    </tr>
</table>

<p>

</p>

</body>
</html>
