<h3 class="boxed">About the <i>X. tropicalis</i> genome</h3>

<h4>Assembly</h4>

<p><img src="/img/species/pic_Xenopus_tropicalis.png" height="90" width="100" class="float-right" alt="Xenopus" title="Bwarp!" />The <i>Xenopus tropicalis</i> genome assembly <a href="http://genome.jgi-psf.org/Xentr4/Xentr4.home.html">version 4.0</a> [June 2005] is the fourth of a series of preliminary assembly releases by the JGI that are planned as part of the ongoing <i>X. tropicalis</i> genome project. This Ensembl website presents the sequence data provided by the JGI.</p>
         
<p>The assembly was constructed with the JGI assembler, Jazz, using paired end sequencing reads at a coverage of 7.65X. After trimming for vector and quality, 22.5 Million reads assembled into 19,759 scaffolds totaling 1.51 Gbp. Roughly half of the genome is contained in 272 scaffolds, all at least 1.56 Mb in length.</p>

<h4>Annotation</h4>

<p>The gene set for <i>Xenopus tropicalis</i> was built using a modified 
version of the standard Ensembl pipeline. Both <i>Xenopus tropicalis</i> and 
<i>Xenopus laevis</i> proteins were used to predict gene structures. 
cDNAs from both species were used to add UTRs. Gene predictions were also 
based on <i>Xenopus tropicalis</i> cDNAs manually annotated in the July 2005 
jamboree. Finally, gene predictions based on UniProt proteins were used to fill
gaps in the annotation.</p>
