<html>
<head>
<meta name="author" content="helpdesk" />
<meta name="navigation" content="Software" />
<title>The Ensembl API</title>
</head>

<body>

<h2 class="breadcrumbs">The Ensembl Perl API</h2>


<h3 class="boxed">Ensembl Databases and Application Programme Interfaces (APIs)</h3>

<p>
Ensembl uses <a href="http://www.mysql.com/" target="external">MySQL</a> relational databases to store its information.
A comprehensive set of Application Programme Interfaces (APIs) serve as a middle-layer between underlying database schemes and more specific application programmes.
The APIs aim to encapsulate the database layout by providing efficient high-level access to data tables and isolate applications from data layout changes.
Ensembl provides an API in 
<a href="./Pdoc/index.html">Perl</a>. 
</p>

<p>
The main Ensembl databases are introduced below.
<a href="/info/downloads/index.html">Data</a> releases for these databases can be obtained from the Ensembl
<a href="ftp://ftp.ensembl.org/pub/">FTP site</a>.
For installation of the Perl APIs see the <a href="./api_installation.html">installation</a> instructions.
</p>


<h4>Ensembl Core Databases and APIs</h4>

<p>
The set of species-specific Ensembl Core databases stores genome sequences and most of the annotation information.
This includes the gene, transcript and protein models  annotated by the Ensembl automated genome analysis and
<a href="/info/about/docs/genome_annotation.html">annotation</a> pipeline.
Ensembl Core databases also store assembly information, cDNA and 
protein alignments, external references, markers and repeat regions data sets. 
</p>

<p><a href="./core/index.html">More about the Core databases and APIs...</a></p>


<h4>Ensembl OtherFeatures Databases and APIs</h4>

<p>
Species-specific Ensembl OtherFeatures databases hold an independent EST gene set provided for all well-characterised species with a suitable amount of biological evidence.
The layout of Ensembl OtherFeatures Databases is identical to the Ensembl Core Database schema so that schema descriptions and API access are equally applicable.
</p>

<p><a href="./core/index.html#est">More about the OtherFeatures databases and APIs...</a></p>


<h4>Ensembl Compara Database and APIs</h4>

<p>
The Ensembl Compara multi-species database stores the results of genome-wide species comparisons re-calculated for each release.
The comparative genomics set includes pairwise whole genome alignments and synteny regions.  The comparative proteomics data set contains orthologue predictions and protein family clusters.</p>

<p><a href="./compara/index.html">More about the Compara database and API...</a></p>


<h4>Ensembl Variation Databases and APIs</h4>

<p>
The large amount of genetic variation information is organised in a set of species-specific Ensembl Variation databases.
</p>

<p><a href="./variation/index.html#est">More about the Variation databases and APIs...</a></p>

<h4>Ensembl Funcgen Databases and APIs</h4>

<p>
The funcgen databases store genome wide functional genomics and regulatory information.  These data are used to produce the Ensembl 'Regulatory Build'.
</p>

<p><a href="./funcgen/index.html#est">More about the Funcgen databases and APIs...</a></p>


<h3 class="boxed">Ensembl Registry</h3>

<p>
The Registry system allows to tell your programs where to find the EnsEMBL databases and how to connect to them.
It has been implemented for the Ensembl Core and Compara APIs. 
</p>

<p><a href="./registry/index.html">More about the Registry...</a></p>


<h3 class="boxed" id="support">Ensembl Software Support</h3>

<p>
Ensembl is an open project and we would like to encourage correspondence and discussions on any subject 
on any aspect of Ensembl. Please see the <a href="/contact.html">Ensembl Contacts</a> page for suitable 
options for getting in touch with us.
</p>

<p>If you are interested in undertaking a short-term collaborative project, our 
"<a href="/info/about/geek4aweek.html">Geek for a Week</a>" scheme allows developers and researchers to work 
alongside Ensembl team members. <a href="geek4aweek.html">More information</a>...</p>


</body>
</html>
