<html>
<head>
<meta name="order" content="1" />
<title>Ensembl Stable IDs</title>
</head>

<body>

<h1>Ensembl stable IDs</h1>

<p>Ensembl annotation uses a system of stable IDs that have prefixes based on 
the species scientific name plus the feature type, followed by a series of digits and a version 
e.g. ENSG00000139618.1. The version may be omitted.</p>

<h2>Feature Stable ID prefixes</h2>

<table style="margin-bottom:1em; width:50%;">
<tr>
  <th>E</th>
  <td>exon</td>
</tr>
<tr>
  <th>FM</th>
  <td>Ensembl protein family</td>
</tr>
<tr>
  <th>G</th>
  <td>gene</td>
</tr>
<tr>
  <th>GT</th>
  <td>gene tree</td>
</tr>
<tr>
  <th>P</th>
  <td>protein</td>
</tr>
<tr>
  <th>R</th>
  <td>regulatory feature</td>
</tr>
<tr>
  <th>T</th>
  <td>transcript</td>
</tr>
</table>

<h2>Ensembl Gene Stable IDs</h2>

<p>
All genes, transcripts, translations (proteins) and exons (referred to here as 'entities') are assigned stable
identifiers. When annotating a species for the first time, these
identifiers are auto-generated. For subsequent new gene annotations for a
species, the entity identifiers are mapped between re-annotation
processes using a combination of location based mappings and
those generated by <a href='http://www.ebi.ac.uk/about/vertebrate-genomics/software/exonerate'>
Exonerate</a> so that equivalent genes for a species
can be linked between releases. The process performs exon based mapping and deriving
subsequent identifier mappings based upon these findings. Further details about our methods is available from <a href="https://doi.org/10.1101/087239">Ruffier M. <i>et al.</i> 2016</a>.
</p>

<h2>Ensembl Stable ID Versioning</h2>
<p>
When reassigning stable identifiers between reannotation we can optionally choose to increment the version number assigned with a stable identifier. We do so to indicate an underlying change in the entity. The rules for incrementing the version depends on the entity in question and detailed below:

<ul>
  <li><b>Genes</b>: increments when the set of transcripts linked to a gene changes</li>
  <li><b>Transcripts</b>: increments when there is a change in a transcript's splicing pattern, chromsome location or a sequence change in the cDNA</li>
  <li><b>Proteins</b>: increments when there is a sequence change in the peptide sequence</li>
  <li><b>Exons</b>: increments when there is a sequence change in the exon genomic sequence</li>
</ul>
</p>


<p>Stable identifier versions do not increment when there are changes to the annotation linked to an entity.</p>

<h2>Species prefixes</h2>

[[SCRIPT::EnsEMBL::Web::Document::HTML::StableidPrefixes]]

</body>
</html>
