<html>
<head>
<meta name="order" content="1" />
<title>Ensembl stable IDs</title>
</head>

<body>

<h1>Ensembl stable IDs</h1>

<p>Ensembl annotation uses a system of stable IDs that have prefixes based on 
the species scientific name plus the feature type, followed by a series of digits, 
e.g. ENSG00000139618.</p>

<p>All genes, transcripts, translations (proteins) and exons (referred to here as 'entities') are assigned stable
identifiers. When annotating a species for the first time, these
identifiers are auto-generated. For subsequent new gene annotations for a
species, the entity identifiers are mapped between re-annotation
processes using a combination of location based mappings and
those generated by <a href='http://www.ebi.ac.uk/about/vertebrate-genomics/software/exonerate'>
Exonerate</a> so that equivalent genes for a species
can be linked between releases. The process performs exon based mapping and deriving
subsequent identifier mappings based upon these findings. Stable
identifiers can also be versioned depending upon the presence of
change between an old and new entity deemed similar enough to have
the identifier transfered. These rules are elaborated in
our <a href='versions.html'>stable id versioning document</a>.
</p>

<p>The prefixes in current use are:</p>

<h2>Feature prefixes</h2>

<table style="margin-bottom:1em; width:50%;">
<tr>
  <th>E</th>
  <td>exon</td>
</tr>
<tr>
  <th>FM</th>
  <td>Ensembl protein family</td>
</tr>
<tr>
  <th>G</th>
  <td>gene</td>
</tr>
<tr>
  <th>GT</th>
  <td>gene tree</td>
</tr>
<tr>
  <th>P</th>
  <td>protein</td>
</tr>
<tr>
  <th>R</th>
  <td>regulatory feature</td>
</tr>
<tr>
  <th>T</th>
  <td>transcript</td>
</tr>
</table>

<h2>Species prefixes</h2>

[[SCRIPT::EnsEMBL::Web::Document::HTML::StableidPrefixes]]

</body>
</html>
