<html>
<head>
<meta name="order" content="6" />
<title>BiomaRt, Bioconductor R package</title>
</head>

<body>

<h1>BiomaRt, Bioconductor R package</h1>
<p>The Bioconductor BiomaRt R package is a quick, easy and powerful way to access BioMart right from your R software terminal</p>
<h3>Summary</h3>
<ol>
        <li><a href="#biomartinstall">How to install the Bioconductor BiomaRt R package</a></li>
        <li><a href="#biomartdoc">Bioconductor BiomaRt R package documentation</a></li>
        <li><a href="#biomartexamples">Bioconductor BiomaRt R examples with the Ensembl Gene mart</a></li>
</ol>

<a name="biomartinstall"></a>
<h2>How to install the Bioconductor BiomaRt R package</h2>
<p>First make sure you have installed the <a href="http://www.r-project.org/">R software</a> on your computer.</p>
<p>Then, run the following commands to install the Bioconductor BiomaRt R package</p>
<pre class="code">
source("http://bioconductor.org/biocLite.R")
biocLite("biomaRt")
</pre>

<a name="biomartdoc"></a>
<h2>Bioconductor BiomaRt R package documentation</h2>
<p>More information regarding the Bioconductor BiomaRt R package and documenation can be found on the <a href="http://www.bioconductor.org/packages/release/bioc/html/biomaRt.html">BiomaRt Bioconductor page</a></p> 

<a name="biomartexamples"></a>
<h2>Bioconductor BiomaRt R examples with the Ensembl Gene mart</h2>
<h3>listMarts & listDatasets</h3>
<p>To get the list of all the Ensembl mart availables on the ensembl.org website, run the "listMarts" function</p>
<pre class="code">
&gt; library(biomaRt)

&gt; listMarts(host='www.ensembl.org')

               biomart               version
1 ENSEMBL_MART_ENSEMBL      Ensembl Genes 79
2     ENSEMBL_MART_SNP  Ensembl Variation 79
3 ENSEMBL_MART_FUNCGEN Ensembl Regulation 79
4    ENSEMBL_MART_VEGA               Vega 59
5                pride        PRIDE (EBI UK)
</pre>
<p>The "listDatasets" function will give you the list of all the species available (mart datasets) for a given mart</p>
<pre class="code">
&gt; library(biomaRt)
&gt; ensembl = useMart(biomart="ENSEMBL_MART_ENSEMBL", host="www.ensembl.org")
&gt; head(listDatasets(ensembl))
                         dataset                                description version
1         oanatinus_gene_ensembl     Ornithorhynchus anatinus genes (OANA5)   OANA5
2        cporcellus_gene_ensembl            Cavia porcellus genes (cavPor3) cavPor3
3        gaculeatus_gene_ensembl     Gasterosteus aculeatus genes (BROADS1) BROADS1
4         lafricana_gene_ensembl         Loxodonta africana genes (loxAfr3) loxAfr3
5 itridecemlineatus_gene_ensembl Ictidomys tridecemlineatus genes (spetri2) spetri2
6        choffmanni_gene_ensembl        Choloepus hoffmanni genes (choHof1) choHof1
</pre>
<h3>useMart</h3>
<p>The "useMart" function allow you to connect to a mart by specifying a biomart, host and dataset parameters. For example, to connect to the human dataset of the Ensembl gene mart:</p>
<pre class="code">
&gt; library(biomaRt)
&gt; ensembl = useMart(biomart="ENSEMBL_MART_ENSEMBL", host="www.ensembl.org",dataset="hsapiens_gene_ensembl")
</pre>
<h3>listFilters & listAttributes</h3>
<p>The "listFilters" function will give you the list of available filters for a given mart and species</p>
<pre class="code">
&gt; library(biomaRt)
&gt; ensembl = useMart(biomart="ENSEMBL_MART_ENSEMBL", host="www.ensembl.org",dataset="hsapiens_gene_ensembl")
&gt; head(listFilters(ensembl))
             name     description
1 chromosome_name Chromosome name
2           start Gene Start (bp)
3             end   Gene End (bp)
4      band_start      Band Start
5        band_end        Band End
6    marker_start    Marker Start
</pre> 
<p>The "listAttributes" function will give you the list of the available attributes for a given mart and species</p>
<pre class="code">
&gt; library(biomaRt)
&gt; ensembl = useMart(biomart="ENSEMBL_MART_ENSEMBL", host="www.ensembl.org",dataset="hsapiens_gene_ensembl")
&gt; head(listAttributes(ensembl))
                   name           description
1       ensembl_gene_id       Ensembl Gene ID
2 ensembl_transcript_id Ensembl Transcript ID
3    ensembl_peptide_id    Ensembl Protein ID
4       ensembl_exon_id       Ensembl Exon ID
5           description           Description
6       chromosome_name       Chromosome Name
</pre>
<h3>getBM</h3>
<p>The "getBM" function allow you to build a BioMart query using a list of mart filters and attributes. The function is used in the next example</p>
<h3>Example query: fetch all the Ensembl gene, transcript IDs, HGNC symbols and chromosome locations located on the human chromosome 1</h3>
<pre class="code">
&gt; library(biomaRt)
&gt; ensembl = useMart(biomart="ENSEMBL_MART_ENSEMBL", host="www.ensembl.org",dataset="hsapiens_gene_ensembl")
&gt; chr1_genes <- getBM(attributes=c(â€™ensembl_gene_id',
'ensembl_transcript_id','hgnc_symbol','chromosome_name','start_position','end_position'), filters =
'chromosome_name', values ="1", mart = ensembl)

> head(chr1_gene)

ensembl_gene_id ensembl_transcript_id hgnc_symbol chromosome_name start_position end_position
1 ENSG00000231510       ENST00000443270                           1        5086459      5090899
2 ENSG00000162444       ENST00000315901        RBP7               1        9997206     10016020
3 ENSG00000162444       ENST00000294435        RBP7               1        9997206     10016020
4 ENSG00000270171       ENST00000602640                           1        7693124      7694844
5 ENSG00000225643       ENST00000412797                           1       25581478     25590356
6 ENSG00000116497       ENST00000530710     S100PBP               1       32816767     32858879
</pre>
<h3>Example query: Fetch Ensembl Gene, Transcript IDs, HGNC IDs and symbols and Uniprot Swissprot accessions mapped to the human Ensembl Gene ID "ENSG00000139618"</h3>
<pre class="code">
&gt; library(biomaRt)
&gt; ensembl = useMart(biomart="ENSEMBL_MART_ENSEMBL", host="www.ensembl.org",dataset="hsapiens_gene_ensembl")
&gt; hgnc_swissprot <- getBM(attributes=c('ensembl_gene_id','ensembl_transcript_id','hgnc_symbol','hgnc_id','uniprot_swissprot'),filters = 'ensembl_gene_id', values = 'ENSG00000139618', mart = ensembl)
&gt; hgnc_swissprot
  ensembl_gene_id ensembl_transcript_id hgnc_symbol   hgnc_id uniprot_swissprot
1 ENSG00000139618       ENST00000380152       BRCA2 HGNC:1101            P51587
2 ENSG00000139618       ENST00000528762       BRCA2 HGNC:1101                  
3 ENSG00000139618       ENST00000470094       BRCA2 HGNC:1101                  
4 ENSG00000139618       ENST00000544455       BRCA2 HGNC:1101            P51587
</pre>
</body>
</html>
