<html>
<head>
<title>Variant Effect Predictor</title>
</head>

<body>

<h1 id="top">Variant Effect Predictor script</h1>

<a href="index.html#about">About</a> |
<a href="index.html#formats">Data formats</a> |
<a href="index.html#web">Web version</a> |
Perl script |
<a href="vep_faq.html">Frequently asked questions</a>

<h2 id="contents"> Contents </h2>
<ul>
	<li><a href="#new">What's new</a></li>
	<li><a href="#requirements">Requirements</a></li>
	<li><a href="#download">Download</a></li>
	<li>
		<a href="#running">Running the script</a>
		<ul>
			<li><a href="#basic">Basic options</a></li>
			<li><a href="#input">Input options</a></li>
			<li><a href="#dbopt">Database options</a></li>
			<li><a href="#output">Output options</a></li>
			<li><a href="#filt">Filtering and QC</a></li>
			<li><a href="#cacheopt">Caching and advanced options</a></li>
			<li><a href="#examples">Examples</a></li>
		</ul>
	</li>
	<li>
		<a href="#db">Databases and caching</a>
		<ul>
			<li><a href="#public">Public database servers</a></li>
			<li><a href="#local">Using a local database</a></li>
			<li><a href="#cache">Using the cache</a></li>
			<li><a href="#pre">Pre-built caches</a></li>
			<li><a href="#build">Building your own cache</a></li>
			<li><a href="#limitations">Limitations of the cache</a></li>
			<li><a href="#privacy">Data privacy</a></li>
			<!--<li><a href="#standalone">Standalone mode</a></li>-->
		</ul>
	</li>	
</ul>


<h2 id="new"> What's new </h2>

<h3> New in version 2.1 </h3>

<ul>
    <li>ability to use local file <a href="#cache">cache</a> in place of or
    alongside connecting to an Ensembl database</li>
	<li>significant improvements to speed of script</li>
    <li>whole-genome mode now default (no disadvantage for smaller
    datasets)</li>
	<li>improved status output with progress bars</li>
	<li>regulatory region consequences now reinstated and improved</li>
    <li>modification to output file - Transcript column is now Feature, and is
    followed by a Feature_type column</li>
</ul>

<h3> New in version 2.0 </h3>

<ul>
	<li>support for SIFT, PolyPhen and Condel non-synonymous predictions in human</li>
	<li>per-allele and compound consequence types</li>
	<li>support for Sequence Ontology (SO) and NCBI consequence terms</li>
	<li>
		modified output format
		<ul>
			<li>support for new output fields in Extra column</li>
			<li>header section contains information on database and software versions</li>
			<li>codon change shown in output</li>
			<li>CDS position shown in output</li>
			<li>option to output Ensembl protein identifiers</li>
			<li>option to output HGVS nomenclature for variants</li>
		</ul>
	</li>
	<li>support for gzipped input files</li>
	<li>enhanced configuration options, including the ability to read configuration from a file</li>
	<li>verbose output now much more useful</li>
	<li>whole-genome mode now more stable</li>
	<li>finding existing co-located variations now ~5x faster</li>
</ul>




<a href="#top">[Back to top]</a>

<h2 id="requirements"> Requirements </h2>

<p>
Version 2.1 of the script requires at least version 63 of the Ensembl Core and
Variation APIs and their relevant dependencies to be installed to use the script
in normal mode. See <a href="http://www.ensembl.org/info/docs/api/index.html"
target="_blank">instructions</a> for details. To use the <a
href="#cache">cache</a>, the gzip and zcat utilities are required. No explicit
installation of the script is necessary.
</p>



<a href="#top">[Back to top]</a>

<h2 id="download"> Download </h2>

<p> The Variant Effect Predictor script can be downloaded from Ensembl's FTP
site. </p>

<a href="ftp://ftp.ensembl.org/pub/misc-scripts/"> Go to download site </a> </p>

<p> It is also included as part of the ensembl-variation module of the Ensembl
API - you can find it in the ensembl-variation/scripts/examples/ directory. </p>



<a href="#top">[Back to top]</a>

<h2 id="running"> Running the script </h2>

<p> The script is run on the command line as follows: </p>

<pre class="code"> perl variant_effect_predictor.pl [options] </pre>

<p> where [options] represent a set of flags and options to the script. These
can be listed using the flag --help: </p>

<pre class="code"> perl variant_effect_predictor.pl --help </pre>

<p> By default the script connects to the public Ensembl database server at
ensembldb.ensembl.org; <a href="#dbopt">other connection options</a> are
available. </p>

<p> Most users will need to use only a few of the options described below; for
most the following command will be enough to get started with: </p>

<pre class="code"> perl variant_effect_predictor.pl -i input.txt -o output.txt </pre>

<p> where input.txt contains data in one of the compatible <a
href="/info/website/upload/var.html#input">input formats</a>, and output.txt is
the <a href="/info/website/upload/var.html#output">output file</a> created by
the script. </p>

<p> Options can be passed as the full string (e.g. --format), or as the shortest
unique string among the options (e.g. --form for --format, since there is
another option --force_overwrite). </p>

<p> <span style="color:red;">NOTE</span> Whole-genome mode is now the default
run mode for the VEP script. In the rare case that you would prefer to run the
script in the old per-variant mode, you can force this with --no_whole_genome
</p>



<a href="#top">[Back to top]</a>

<h3 id="basic"> Basic options </h3>

<table class="ss" style="width:75%;">
	<tr>
		<th>Flag</th>
		<th>Alternate</th>
		<th>Description</th>
	</tr>
	<tr>
		<td><pre>--help</pre></td>
		<td>&nbsp;</td>
		<td>Display help message and quit</td>
	</tr>
	<tr class="bg2">
		<td><pre>--verbose</td>
		<td><pre>-v</pre></td>
		<td>
			Output longer status messages as the script runs. This option can be
			used to generate the basis of a configuration file - see --config
			below. <i>Not used by default</i>
		</td>
	</tr>
	<tr>
		<td><pre>--quiet</pre></td>
		<td><pre>-q</pre></td>
		<td>Suppress status and warning messages. <i>Not used by default</i></td>
	</tr>
	<tr class="bg2">
		<td><pre>--no_progress</pre></td>
		<td>&nbsp;</td>
		<td>Don't show progress bars. <i>Progress bars shown by default</i></td>
	</tr>
	<tr>
		<td><pre>--config [filename]</pre></td>
		<td>&nbsp;</td>
		<td>
			Load configuration options from a config file. The config file
			should consist of whitespace-separated pairs of option names and
			settings e.g.: 
<pre>output_file   my_output.txt
species       mus_musculus
format        vcf
host          useastdb.ensembl.org</pre>
			This is useful if you find yourself using the same configuration
			options each time. You can create a quick version file of this by
			setting the flags as normal and running the script in verbose (-v)
			mode. This will output lines that can be copied to a config file
			that can be loaded in on the next run using -config. Note that any
			options specified in the normal way overwrite those in the config
			file. <i>Not used by default</i>
		</td>
</table>


<p><a href="#top">[Back to top]</a></p>


<h3 id="input"> Input options </h3>

<table class="ss" style="width:75%;">
	<tr>
		<th>Flag</th>
		<th>Alternate</th>
		<th>Description</th>
	</tr>
	<tr>
		<td><pre>--species [species]</pre></td>
		<td>&nbsp;</td>
		<td>
			Species for your data. This can be the latin name e.g.
			"homo_sapiens" or any Ensembl alias e.g. "mouse". Specifying the
			latin name can speed up initial database connection as the registry
			does not have to load all available database aliases on the server.
			<i>Default = "homo_sapiens"</i>
		</td>
	</tr>
	<tr class="bg2">
		<td><pre>--input_file [filename]</pre></td>
		<td><pre>-i</pre></td>
		<td>Input file name. If not specified, the script will attempt to read from STDIN.</td>
	</tr>
	<tr>
		<td><pre>--format [vcf<b>|</b>pileup]</pre></td>
		<td>&nbsp;</td>
		<td>
			Input file format. By default, the script auto-detects the input
			file format. Using this option you can force the script to read the
			input file as VCF or pileup format. <i>Not used by default</i>
		</td>
	</tr>
	<tr class="bg2">
		<td><pre>--output_file [filename]</pre></td>
		<td><pre>-o</pre></td>
		<td>
			Output file name. The script can write to STDOUT by specifying
			STDOUT as the output file name - this will force --quiet mode.
			<i>Default = "variant_effect_output.txt"</i>
		</td>
	</tr>
	<tr>
		<td><pre>--force_overwrite</pre></td>
		<td><pre>--force</pre></td>
		<td>
			By default, the script will fail with an error if the output file
			already exists. You can force the overwrite of the existing file by
			using this flag. <i>Not used by default</i>
		</td>
	</tr>
</table>



<p><a href="#top">[Back to top]</a></p>

<h3 id="dbopt"> Database options </h3>

<table class="ss" style="width:75%;">
	<tr>
		<th>Flag</th>
		<th>Alternate</th>
		<th>Description</th>
	</tr>
	<tr>
		<td><pre>--host [hostname]</pre></td>
		<td>&nbsp;</td>
		<td>
			Manually define the database host to connect to. Users in the US may
			find connection and transfer speeds quicker using our East coast
			mirror, useastdb.ensembl.org. <i>Default =
			"ensembldb.ensembl.org"</i>
		</td>
	</tr>
	<tr class="bg2">
		<td><pre>--user [username]</pre></td>
		<td><pre>-u</pre></td>
		<td>Manually define the database username. <i>Default = "anonymous"</i></td>
	</tr>
	<tr>
		<td><pre>--password [password]</pre></td>
		<td><pre>--pass</td>
		<td>Manually define the database password. <i>Not used by default</i></td>
	</tr>
	<tr class="bg2">
		<td><pre>--port [number]</pre></td>
		<td>&nbsp;</td>
		<td>Manually define the database port. <i>Default = 5306</i></td>
	</tr>
	<tr>
		<td><pre>--genomes</pre></td>
		<td>&nbsp;</td>
		<td>
			Override the default connection settings with those for the Ensembl
			Genomes public MySQL server. Required when using any of the <a
			href="http://www.ensemblgenomes.org/" target="_blank">Ensembl
			Genomes</a> species. <i>Not used by default</i>
		</td>
	</tr>
	<tr class="bg2">
		<td><pre>--db_version [number]</pre></td>
		<td><pre>--db</td>
		<td>
			Force the script to connect to a specific version of the Ensembl
			databases. Not recommended as there will usually be conflicts
			between software and database versions. <i>Not used by default</i>
		</td>
	</tr>
	<tr>
		<td><pre>--registry [filename]</pre></td>
		<td>&nbsp;</td>
		<td>
			Defining a registry file overwrites other connection settings and
			uses those found in the specified registry file to connect.
			<i>Not used by default</i>
		</td>
	</tr>
</table>


<p><a href="#top">[Back to top]</a></p>

<h3 id="output"> Output options </h3>

<table class="ss" style="width:75%;">
	<tr>
		<th>Flag</th>
		<th>Alternate</th>
		<th>Description</th>
	</tr>
	<tr>
		<td><pre>--terms [ensembl<b>|</b>so<b>|</b>ncbi]</pre></td>
		<td><pre>-t</pre></td>
		<td>
			The type of consequence terms to output. The Ensembl terms are
			described <a href="/info/docs/variation/index.html"
			target="_blank">here</a>. The <a
			href="http://www.sequenceontology.org/" target="_blank">Sequence
			Ontology</a> is a joint effort by genome annotation centres to
			standardise descriptions of biological sequences. The NCBI terms are
			those used by <a
			href="http://www.ncbi.nlm.nih.gov/projects/SNP/">dbSNP</a>, and are
			the least complete set - where no NCBI term is available, the script
			will output the Ensembl term. <i>Default = "ensembl"</i>
		</td>
	</tr>
	<tr class="bg2">
		<td><pre>--sift [p<b>|</b>s<b>|</b>b]</pre></td>
		<td>&nbsp</td>
		<td>
			<span style="color:red;">Human only</span>
			<a href="http://sift.jcvi.org/" target="_blank">SIFT</a> predicts
			whether an amino acid substitution affects protein function based on
			sequence homology and the physical properties of amino acids. The
			VEP can output the <b>p</b>rediction term, <b>s</b>core or
			<b>b</b>oth. Using SIFT requires a database connection - while it
			can be used with --cache, the database will still be accessed to
			retrieve SIFT data. <i>Not used by default</i>
		</td>
	</tr>
	<tr>
		<td><pre>--polyphen [p<b>|</b>s<b>|</b>b]</pre></td>
		<td><pre>--poly</td>
		<td>
			<span style="color:red;">Human only</span>
			<a href="http://genetics.bwh.harvard.edu/pph2/"
			target="_blank">PolyPhen</a> is a tool which predicts possible
			impact of an amino acid substitution on the structure and function
			of a human protein using straightforward physical and comparative
			considerations. The VEP can output the <b>p</b>rediction term,
			<b>s</b>core or <b>b</b>oth. Using PolyPhen requires a database
			connection - while it can be used with --cache, the database will
			still be accessed to retrieve PolyPhen data. <i>Not used by
			default</i>
		</td>
	</tr>
	<tr class="bg2">
		<td><pre>--condel [p<b>|</b>s<b>|</b>b]</pre></td>
		<td>&nbsp</td>
		<td>
			<span style="color:red;">Human only</span>
			<a href="https://bg.upf.edu/forge/wiki/condel/Condel"
			target="_blank">Condel</a> computes a weighed average of the scores
			(WAS) of several computational tools aimed at classifying missense
			mutations as likely deleterious or likely neutral. The VEP currently
			presents a Condel WAS from SIFT and PolyPhen. The VEP can output the
			<b>p</b>rediction term, <b>s</b>core or <b>b</b>oth. Using Condel
			requires a database connection - while it can be used with --cache,
			the database will still be accessed to retrieve Condel data. <i>Not
			used by default</i>
		</td>
	</tr>
	<tr>
		<td><pre>--regulatory</pre></td>
		<td>&nbsp;</td>
		<td>
			Look for overlaps with regulatory regions. The script can also call
			if a variant falls in a high information position within a
			transcription factor binding site. Output lines have a Feature type
			of RegulatoryFeature or MotifFeature. Using this option requires a
			database connection - while it can be used with --cache, the
			database will still be accessed to retrieve regulatory data.
			<i>Not used by default</i>
		</td>
	</tr>
	<tr>
		<td><pre>--hgvs</pre></td>
		<td>&nbsp;</td>
		<td>
			Add <a href="http://www.hgvs.org/mutnomen/" target="_blank">HGVS</a>
			nomenclature based on Ensembl stable identifiers to the output. Both
			coding and protein sequence names are added where appropriate.
			Currently it is not possible to generate HGVS identifiers from the
			cache; a database connection must be made.
			<i>Not used by default</i>
		</td>
	</tr>
	<tr class="bg2">
		<td><pre>--gene</pre></td>
		<td>&nbsp</td>
		<td>
			Force the gene column to be populated. This is disabled by default
			unless using --cache. <i>Gene column not populated by default</i>
		</td>
	</tr>
	<tr>
		<td><pre>--protein</pre></td>
		<td>&nbsp</td>
		<td>
			Add the Ensembl protein identifier to the output where appropriate.
			<i>Not used by default</i>
		</td>
	</tr>
	<tr class="bg2">
		<td><pre>--hgnc</pre></td>
		<td>&nbsp;</td>
		<td>
			Adds the HGNC gene identifer (where available) to the output.
			<i>Not used by default</i>
		</td>
	</tr>
	<tr>
		<td><pre>--most_severe</pre></td>
		<td>&nbsp</td>
		<td>
			Output only the most severe consequence per variation.
			Transcript-specific columns will be left blank. <i>Not used by
			default</i>
		</td>
	</tr>
	<tr class="bg2">
		<td><pre>--summary</pre></td>
		<td>&nbsp;</td>
		<td>
			Output only a comma-separated list of all observed consequences per
			variation. Transcript-specific columns will be left blank. <i>Not
			used by default</i>
		</td>
	</tr>
</table>


<p><a href="#top">[Back to top]</a></p>

<h3 id="filt"> Filtering and QC options </h3>

<table class="ss" style="width:75%;">
	<tr>
		<th>Flag</th>
		<th>Alternate</th>
		<th>Description</th>
	</tr>
	<tr>
		<td><pre>--check_ref</pre></td>
		<td>&nbsp;</td>
		<td>
			Force the script to check the supplied reference allele against the
			sequence stored in the Ensembl Core database. Lines that do not
			match are skipped. <i>Not used by default</i>
		</td>
	</tr>
	<tr class="bg2">
		<td><pre>--coding_only</pre></td>
		<td>&nbsp</td>
		<td>
			Only return consequences that fall in the coding regions of
			transcripts. <i>Not used by default</i>
		</td>
	</tr>
	<tr>
		<td><pre>--check_existing</pre></td>
		<td>&nbsp;</td>
		<td>
			Checks for the existence of variants that are co-located with your
			input. By default the alleles are not compared - to do so, use
			--check_alleles. <i>Not used by default</i>
		</td>
	</tr>
	<tr class="bg2">
		<td><pre>--check_alleles</pre></td>
		<td>&nbsp</td>
		<td>
			When checking for existing variants, only report a co-located
			variant if none of the alleles supplied are novel. For example,
			if the user input has alleles A/G, and an existing co-located
			variant has alleles A/C, the co-located variant will not be
			reported.<br/><br/>Strand is also taken into account - in the same
			example, if the user input has alleles T/G but on the negative
			strand, then the co-located variant <b>will</b> be reported since
			its alleles match the reverse complement of user input. <i>Not used
			by default</i>
		</td>
	</tr>
	<tr>
		<td><pre>--chr [list]</pre></td>
		<td>&nbsp;</td>
		<td>
			Select a subset of chromosomes to analyse from your file. Any data
			not on this chromosome in the input will be skipped. The list can be
			comma separated, with "-" characters representing an interval. For
			example, to include chromsomes 1, 2, 3, 10 and X you could use
			<pre>--chr 1-3,10,X</pre> <i>Not used by default</i>
		</td>
	</tr>
	<!--<tr class="bg2">
		<td><pre>--failed [0<b>|</b>1]</pre></td>
		<td>&nbsp;</td>
		<td>
			When checking for co-located variants, by default the script will
			include variants that have been flagged as failed. Set this flag to
			0 to exclude such variants. <i>Not used by default</i>
		</td>
	</tr>-->
</table>



<p><a href="#top">[Back to top]</a></p>

<h3 id="cacheopt">Caching and advanced options</h3>

<table class="ss" style="width:75%;">
	<tr>
		<th>Flag</th>
		<th>Alternate</th>
		<th>Description</th>
	</tr>
	<tr>
		<td><pre>--no_whole_genome</pre></td>
		<td>&nbsp;</td>
		<td>
			Force the script to run in non-whole-genome mode. This was the
			original default mode for the VEP script, but has now been
			superceded by whole-genome mode, which is the default. In this mode,
			variants are analysed one at a time, with no caching of transcript
			data. <i>Not used by default</i>
		</td>
	</tr>
	<tr class="bg2">
		<td><pre>--cache</pre></td>
		<td>&nbsp</td>
		<td>
            Enables use of the <a href="#cache">cache</a>. By default the VEP
            will only read from the cache - use --write_cache to enable writing.
            <i>Not used by default</i>
		</td>
	</tr>
	<tr>
		<td><pre>--dir [directory]</pre></td>
		<td>&nbsp</td>
		<td>
			Specify the base cache directory to use. This should be on a
			filesystem with around 600MB free (for human, other species may
			vary). <i>Default = "$HOME/.vep/"</i>
		</td>
	</tr>
	<!--<tr class="bg2">
		<td><pre>--standalone</pre></td>
		<td>&nbsp;</td>
		<td>
			Enable standalone mode. No database connections will be made, and
			only a complete cache (either downloaded or built using --build) can
			be used for this mode. No installation of the Ensembl API is
			required - a minimal set of modules is used that reside in the
			"Bio/" sub-directory found with the script. <i>Not used by
			default</i>
		</td>
	</tr>-->
	<tr>
		<td><pre>--buffer [number]</pre></td>
		<td>&nbsp</td>
		<td>
			Sets the internal buffer size, corresponding to the number of
			variations that are read in to memory simultaneously. Set this lower
			to use less memory at the expense of longer run time, and higher to
			use more memory with a faster run time. <i>Default = 5000</i>
		</td>
	</tr>
	<tr class="bg2">
		<td><pre>--write_cache</pre></td>
		<td>&nbsp;</td>
		<td>
			Enable writing to the cache. <i>Not used by default</i>
		</td>
	</tr>
	<tr>
		<td><pre>--build [all<b>|</b>list]</pre></td>
		<td>&nbsp</td>
		<td>
			Build a complete cache for the selected species from the database.
			Either specify a list of chromosomes (see --chr for how to do this),
			or use <pre>--build all</pre> to build for all top-level
			chromosomes. <span style="color:red;">WARNING: Do not use this flag
			when connected to one of the public databases - please instead
			download a pre-built cache or build against a local database.
			</span><i>Not used by default</i>
		</td>
	</tr>
	<tr class="bg2">
		<td><pre>--compress [command]</pre></td>
		<td>&nbsp;</td>
		<td>
			By default the VEP uses the utility zcat to decompress cached files.
			On some systems zcat may not be installed or may misbehave; by
			specifying one of <pre>--compress gzcat</pre> or
			<pre>--compress "gzip -dc"</pre> you may be able to bypass these
			problems. <i>Not used by default</i>
		</td>
	</tr>
	<tr>
		<td><pre>--skip_db_check</pre></td>
		<td>&nbsp</td>
		<td>
			<span style="color:red;">ADVANCED</span> Force the script to use a
			cache built from a different host than specified with --host. Only
            use this if you are sure the two hosts are compatible (e.g.
            ensembldb.ensembl.org can be considered compatible with
            useastdb.ensembl.org as the data is mirrored between the two).
            <i>Not used by default</i>
		</td>
	</tr>
	<tr class="bg2">
		<td><pre>--cache_region_size [size]</pre></td>
		<td>&nbsp;</td>
		<td>
			<span style="color:red;">ADVANCED</span> The size in base-pairs of
			the region covered by one file in the cache. By default this is 1MB,
			which produces approximately ~500 files maximum per sub-directory in
			human. Reducing this can reduce the amount of memory and decrease
			the run-time when you use a cache built this way. Note that you must
			specify the same --cache_region_size when both building/writing to
			the cache and reading from it. <i>Not used by default</i>
		</td>
	</tr>
</table>
  


<p><a href="#top">[Back to top]</a></p>

<h3 id="examples"> Examples </h3>

<ul>
	<li>
		Read input from STDIN, output to STDOUT
<pre class="code">perl variant_effect_predictor.pl -o stdout</pre>
	</li>
	<li>
		Add regulatory region consequences
<pre class="code">perl variant_effect_predictor.pl -i variants.txt -regulatory</pre>
	</li>
	<li>
		Input file variants.vcf.txt, input file format VCF, add HGNC gene
		identifiers, output SO consequence terms
<pre class="code">perl variant_effect_predictor.pl -i variants.vcf.txt -format vcf -hgnc -t so</pre>
	</li>
	<li>
		Force overwrite of output file variants_output.txt, check for existing
		co-located variants, output only coding sequence consequences, output
		HGVS names
<pre class="code">perl variant_effect_predictor.pl -i variants.txt -o variants_output.txt -force -check_existing -coding_only -hgvs</pre>
	</li>
	<li>
		Specify DB connection parameters in registry file ensembl.registry, add SIFT
		score and prediction, PolyPhen prediction, Condel score
<pre class="code">perl variant_effect_predictor.pl -i variants.txt -r ensembl.registry -sift b -polyphen p -condel s</pre>
	</li>
	<li>
		Connect to Ensembl Genomes db server for A.thaliana, run with buffer
		size of 10000
<pre class="code">perl variant_effect_predictor.pl -i variants.txt -genomes -species arabidopsis_thaliana -b 10000</pre>
	</li>
	<li>
		Load config from ini file, run in quiet mode
<pre class="code">perl variant_effect_predictor.pl -config vep.ini -i variants.txt -q</pre>
	</li>
	<li>
		Use cache in /home/vep/mycache/, use gzcat instead of zcat
<pre class="code">perl variant_effect_predictor.pl -cache -dir /home/vep/mycache/ -i variants.txt -compress gzcat</pre>
	</li>
</ul>


<p><a href="#top">[Back to top]</a></p>

<h2 id="db"> Databases and caching </h2>

<p> The VEP script can use a variety of data sources to retrieve transcript
information that is used to predict consequence types. Which one you choose to
use should depend on your requirements and available resources. </p>


<a href="#top">[Back to top]</a>

<h3 id="public"> Public database servers </h3>

<p> By default, the script is configured to connect to Ensembl's public MySQL
instance at ensembldb.ensembl.org. For users in the US (or for any user
geographically closer to the East coast of the USA than to Ensembl's data centre
in Cambridge, UK), a mirror server is available at useastdb.ensembl.org. To use
the mirror, use the flag --host useastdb.ensembl.org </p>

<p> Users of Ensembl Genomes species (e.g. plants, fungi, microbes) should use
their public MySQL instance; the connection parameters for this can be
automatically loaded by using the flag --genomes </p>

<p> Users with small data sets (100s of variants) should find using the default
connection settings adequate. Those with larger data sets, or those who wish to
use the script in a batch manner, should consider one of the alternatives below.
</p>


<a href="#top">[Back to top]</a>

<h3 id="local"> Using a local database </h3>

<p> It is possible to set up a local MySQL mirror with the databases for your
species of interest installed. For instructions on installing a local mirror,
see <a href="/info/docs/webcode/install/ensembl-data.html"
target="_blank">here</a>. You will need a MySQL server that you can connect to
from the machine where you will run the script (this can be the same machine).
For most of the functionality of the VEP, you will only need the Core database
(e.g. homo_sapiens_core_63_37) installed. In order to find co-located variations
or to use SIFT, PolyPhen or Condel, it is also necessary to install the relevant
variation database (e.g. homo_sapiens_variation_63_37). </p>

<p> To connect to your mirror, you can either set the connection parameters
using --host, --port, --user and --password, or use a registry file. Registry
files contain all the connection parameters for your database, as well as any
species aliases you wish to set up: </p>

<pre class="code">
use Bio::EnsEMBL::DBSQL::DBAdaptor;
use Bio::EnsEMBL::Variation::DBSQL::DBAdaptor;
use Bio::EnsEMBL::Registry;

Bio::EnsEMBL::DBSQL::DBAdaptor->new(
  '-species' => "Homo_sapiens",
  '-group'   => "core",
  '-port'    => 5306,
  '-host'    => 'ensembldb.ensembl.org',
  '-user'    => 'anonymous',
  '-pass'    => '',
  '-dbname'  => 'homo_sapiens_core_63_37'
);

Bio::EnsEMBL::Variation::DBSQL::DBAdaptor->new(
  '-species' => "Homo_sapiens",
  '-group'   => "variation",
  '-port'    => 5306,
  '-host'    => 'ensembldb.ensembl.org',
  '-user'    => 'anonymous',
  '-pass'    => '',
  '-dbname'  => 'homo_sapiens_variation_63_37'
);

Bio::EnsEMBL::Registry->add_alias("Homo_sapiens","human");
</pre>

<p> For more information on the registry and registry files, see <a
href="/info/docs/api/registry.html" target="_blank">here</a>. </p>


<a href="#top">[Back to top]</a>

<h3 id="cache"> Using the cache </h3>

<p> From version 2.1 onwards, the VEP is able to use cached data on disk in
place of reading from the database. Using the cache is probably the fastest and
most efficient way to use the VEP script, as in most cases no network
connections are made and most data is read from local disk. The diagrams below
illustrate the model of caching that is used. </p>

<table class="ss" style="width:75%;">
    <tr>
        <th>Normal mode</th>
        <th>Cache mode</th>
        <th>Build mode</th>
        <!--<th>Standalone mode</th>-->
    </tr>
    <tr>
        <td><a href="/img/vep_cache_normal.png"><img src="/img/vep_cache_normal_thumb.png" /></a></td>
        <td><a href="/img/vep_cache_cache.png"><img src="/img/vep_cache_cache_thumb.png" /></a></td></td>
        <td valign="middle"><a href="/img/vep_cache_build.png"><img src="/img/vep_cache_build_thumb.png" /></a></td></td>
        <!--<td><a href="/img/vep_cache_standalone.png"><img src="/img/vep_cache_standalone_thumb.png" /></a></td></td>-->
    </tr>
</table>

<p> It is possible to use any combination of cache and database; when using the
cache, the cache will take preference, with the database being used when the
relevant data is not found in the cache. </p>

<p> Cache files are compressed using the gzip utility. By default zcat is used
to decompress the files, although gzcat or gzip itself can be used to decompress
also - you must have one of these utilities installed in your path to use the
cache. </p>


<a href="#top">[Back to top]</a>

<h3 id="pre">Pre-built caches</h3>

<p> The easiest solution is to download a pre-built cache for your species; this
eliminates the need to connect to the database while the script is running
(except when using <a href="#limitations">certain output options</a>). <p>

<ol>
	<li>
		Download the archive file for your species:
        <p>
            <table class="ss" style="width:25%">
                <tr>
                    <th>Human <i>(Homo sapiens)</i></th>
                    <td><a href="ftp://ftp.ensembl.org/pub/release-63/variation/homo_sapiens/homo_sapiens_vep_63.tar.gz">Download</a></td>
                </tr>
                <tr>
                    <th>Mouse <i>(Mus musculus)</i></th>
                    <td><a href="ftp://ftp.ensembl.org/pub/release-63/variation/mus_musculus/mus_musculus_vep_63.tar.gz">Download</a></td>
                </tr>
                <tr>
                    <th>Rat <i>(Rattus norvegicus)</i></th>
                    <td><a href="ftp://ftp.ensembl.org/pub/release-63/variation/rattus_norvegicus/rattus_norvegicus_vep_63.tar.gz">Download</a></td>
                </tr>
                <tr>
                    <th>Zebrafish <i>(Danio rerio)</i></th>
                    <td><a href="ftp://ftp.ensembl.org/pub/release-63/variation/danio_rerio/danio_rerio_vep_63.tar.gz">Download</a></td>
                </tr>
                <tr>
                    <th>Cow <i>(Bos taurus)</i></th>
                    <td><a href="ftp://ftp.ensembl.org/pub/release-63/variation/bos_taurus/bos_taurus_vep_63.tar.gz">Download</a></td>
                </tr>
            </table>
        </p>
	</li>
	<li>
		Extract the archive in your cache directory. By default the VEP uses
		$HOME/.vep/ as the cache directory, where $HOME is your UNIX home
		directory.
        <pre class="code">
mv homo_sapiens_vep_62.tar.gz ~/.vep/
tar xfz homo_sapiens_vep_62.tar.gz</pre>
	</li>
	<li>
		Run the VEP with the --cache option
	</li>
</ol>

<p> Caches for several species, and indeed different Ensembl releases of the
same species, can be stored in the same cache base directory. The files are
stored in the following directory hierarchy: $HOME -> .vep -> species -> version
-> chromosome </p>


<a href="#top">[Back to top]</a>

<h3 id="build"> Building your own cache </h3>

<p> It is possible to build your own cache using the VEP script. <span
style="color:red;">You should <b>NOT</b> use this command when connected to the
public MySQL instances</span> - the process takes a long time, meaning the
connection can break unexpectedly and you will be violating Ensembl's reasonable
use policy on the public servers. You should either download one of the
pre-built caches, or create a <a href="#local">local</a> copy of your database
of interest to build the cache from. </p>

<p> You may wish to build a full cache if you have a custom Ensembl database
with data not found on the public servers, or you may wish to create a minimal
cache covering only a certain set of chromosome regions. Cache files are
compressed using the gzip utility; this must be installed in your path to write
cache files. </p>

<p> To build a cache "on-the-fly", use the --cache and --write_cache flags when
you run the VEP with your input. Only cache files overlapping your input
variants will be created; the next time you run the script with this cache, the
data will be read from the cache instead of the database. Any data not found in
the cache will be read from the database (and then written to the cache if
--write_cache is enabled). If your data covers a relatively small proportion of
your genome of interest (for example, a few genes of interest), it can be OK to
use the public MySQL servers when building a partial cache. </p>

<pre class="code">perl variant_effect_predictor.pl -cache -dir /my/cache/dir/ -write_cache -i input.txt</pre>

<p> To build a cache from scratch, use the flag <pre>--build all</pre> or e.g.
<pre>--build 1-5,X</pre> to build just a subset of chromosomes. You do not need
to specify any of the usual input options when building a cache:</p>

<pre class="code">perl variant_effect_predictor.pl -host dbhost -user username -pass password -port 3306 -build 21 -dir /my/cache/dir/</pre>

<p><a href="#top">[Back to top]</a></p>

<h3 id="limitations">Limitations of the cache</h3>

<p> The cache stores the following information: </p>

<ul>
	<li>Transcript location, sequence, exons and other attributes</li>
    <li>Gene, protein and HGNC identifiers for each transcript (where
    applicable)</li>
	<li>Location and alleles of existing variations</li>
</ul>

<p> It does not store any information pertaining to, and therefore cannot be
used for, the following: </p>

<ul>
    <li>SIFT</li>
    <li>PolyPhen</li>
    <li>Condel</li>
    <li>Regulatory regions</li>
    <li>HGVS names - due to a limitation in the API, it is currently not
    possible to create HGVS identifiers without using the database</li>
</ul>

<p> Enabling one of these options with --cache will cause the script to warn you
in its status output with something like the following: </p>

<pre class="code"> 2011-06-16 16:24:51 - INFO: Database will be accessed for SIFT/PolyPhen/Condel and HGVS </pre>

<a href="#top">[Back to top]</a>

<h3 id="privacy"> Data privacy </h3>

<p> When using the public database servers, the VEP script requests transcript
and variation data that overlap the loci in your input file. As such, these
coordinates are transmitted over the network to a public server, which may not
be suitable for those with sensitive or private data. Users should note that
<b>only</b> the coordinates are transmitted to the server; no other information
is sent. </p>

<p> By using a full downloaded cache or a local database, it is possible to
avoid completely any network connections to public servers, thus preserving
absolutely the privacy of your data. </p>

<a href="#top">[Back to top]</a>



<!--<h3 id="standalone"> Standalone mode </h3>

<p> <span style="color:red;">WARNING: EXPERIMENTAL!</span> It is possible to run
the VEP script in a standalone mode that does not require the installation of
any of the usual dependencies. This means users require only perl (version 5.8
or greater) and the either zcat, gzcat or gzip utilities. To enable this mode,
use the flag --standalone. You will need to have downloaded a <a
href="#pre">complete cache</a> for your species of interest, since in standalone
mode no database connections can be made. You will also need the mini package of
modules contained in <a
href="ftp://ftp.ensembl.org/pub/misc-scripts/Variant_effect_predictor_2.0/standalone_Bio.zip">this
archive</a>, which you should decompress into the same directory as the script.
Note that if you run the script from another directory,
you will need to tell perl where to find the modules using the -I flag: </p>

<pre> perl -I path/to/vep/ path/to/vep/variant_effect_predictor.pl </pre>

<p> This mode is relatively untested, as such should be treated as "alpha" code.
Feedback, however, would be very useful for the development of the script. </p>

<a href="#top">[Back to top]</a>-->


</body>
</html>
