<html>
<head>
  <meta name="navigation" content="Comparative Genomics">
  <title>Pairwise Alignments</title>
  <link rel="stylesheet" type="text/css" media="all" href="85_spreadsheet_table.css" />
  <link rel="stylesheet" type="text/css" media="all" href="95_static_content.css" />
</head>

<body>

<h2>Human vs Nile tilapia Translated Blat alignments</h2>

<p>Human (<i>Homo sapiens</i>, GRCh37) and Nile tilapia (<i>Oreochromis niloticus</i>, Orenil1.0) were aligned using the Translated Blat alignment algorithm (<a href="http://www.genome.org/cgi/content/abstract/12/4/656">Kent W, Genome Res., 2002;12(4):656-64</a>)
in Ensembl release 67. Human  was used as the reference species. After running Translated Blat, the raw Translated Blat  alignment blocks are chained according to their location in both genomes. During the final netting process, the best sub-chain is chosen in each region on the reference species.</p>

<a href="/info/docs/compara/analyses.html" class="cp-external">Full list of 
pairwise alignments</a>

<h3>Configuration parameters</h3>

<table class="ss tint species-stats">
<tr class="bg2">
<td class="data" style="width:75%">Minimum score (minScore)</td>
<td class="value">30</td>
</tr>
<tr>
<td class="data">Database type (t)</td>
<td class="value">dnax</td>
</tr>
<tr class="bg2">
<td class="data">Query type (q)</td>
<td class="value">dnax</td>
</tr>
<tr>
<td class="data">Mask out repeats (mask)</td>
<td class="value">lower</td>
</tr>
<tr class="bg2">
<td class="data">Mask out repeats on query (qmask)
</td>
<td class="value">lower
</td>
</tr>
</table>

<h3 class="boxed">Chunking parameters</h3>
<table style="text-align: left; width: 100%;" border="0" cellpadding="2"
cellspacing="2">
<tbody>
<tr>
<td style="vertical-align: top;">
</td>
<th style="vertical-align: top;">Human
</th>
<th style="vertical-align: top;">Nile tilapia
</th>
</tr>
<tr>
<th style="vertical-align: middle;">Chunk size
</th>
<td style="vertical-align: top;">1,000,000
</td>
<td style="vertical-align: top;">25,000
</td>
</tr>
<tr>
<th style="vertical-align: middle;">Overlap
</th>
<td style="vertical-align: top;">10,000
</td>
<td style="vertical-align: top;">10,000
</td>
</tr>
<tr>
<th style="vertical-align: middle;">Group set size
</th>
<td style="vertical-align: top;">100,000,000
</td>
<td style="vertical-align: top;">10,000,000
</td>
</tr>
<tr>
<th style="vertical-align: middle;">Masking options
</th>
<td style="vertical-align: top;">$ENSEMBL_CVS_ROOT_DIR/ensembl-compara/scripts/pipeline/human36.spec
</td>
<td style="vertical-align: top;">{default_soft_masking => 1}
</td>
</tr>
</tbody>
</table>

<h3>Results</h3>

<p>Number of alignment blocks: 197965</p>
<table style="text-align: left; width: 100%;" border="0" cellpadding="2"
cellspacing="2">
<tbody>
<tr>
<td style="vertical-align: top;">
</td>
<th style="vertical-align: top;">Genome coverage(bp)
</th>
<th style="vertical-align: top;">Coding exon coverage (bp)
</th>
</tr>
<tr>
<th style="vertical-align: middle;">Human</th>
<td style="vertical-align: top;"><img style="width: 200px; height: 200px;" alt=""
src="./pie_ref_align_573.png"/>
</td>
<td style="vertical-align: top;"><img style="width: 200px; height: 200px;" alt=""
src="./pie_ref_cod_align_573.png"/>
</td>
</tr>
<tr>
<td></td>
<td style="vertical-align: top;">21,577,475 out of 3,098,825,702
</td>
<td style="vertical-align: top;">15,467,010 out of 35,155,918
</td>
</tr>
<tr>
<th style="vertical-align: middle;">Nile tilapia
</th>
<td style="vertical-align: top;"><img style="width: 200px; height: 200px;" alt=""
src="./pie_non_ref_align_573.png"/>
</td>
<td style="vertical-align: top;"><img style="width: 200px; height: 200px;" alt=""
src="./pie_non_ref_cod_align_573.png"/>
</td>
</tr>
<tr>
<td></td>
<td style="vertical-align: top;">16,262,654 out of 927,383,394
</td>
<td style="vertical-align: top;">13,865,296 out of 37,304,121
</td>
</tr>
</tbody>
</table>


</body>
</html>
