<html>
<head>
  <meta name="navigation" content="Comparative Genomics">
  <title>Pairwise Alignments</title>
  <link rel="stylesheet" type="text/css" media="all" href="85_spreadsheet_table.css" />
  <link rel="stylesheet" type="text/css" media="all" href="95_static_content.css" />
</head>

<body>

<h2>Medaka vs Platyfish Translated Blat alignments</h2>

<p>Medaka (<i>Oryzias latipes</i>, MEDAKA1) and Platyfish (<i>Xiphophorus maculatus</i>, Xipmac4.4.2) were aligned using the Translated Blat alignment algorithm (<a href="http://www.genome.org/cgi/content/abstract/12/4/656">Kent W, Genome Res., 2002;12(4):656-64</a>)
in Ensembl release 69. Medaka  was used as the reference species. After running Translated Blat, the raw Translated Blat  alignment blocks are chained according to their location in both genomes. During the final netting process, the best sub-chain is chosen in each region on the reference species.</p>

<a href="/info/docs/compara/analyses.html" class="cp-external">Full list of 
pairwise alignments</a>

<h3>Configuration parameters</h3>

<table class="ss tint species-stats">
<tr class="bg2">
<td class="data" style="width:75%">Minimum score (minScore)</td>
<td class="value">30</td>
</tr>
<tr>
<td class="data">Database type (t)</td>
<td class="value">dnax</td>
</tr>
<tr class="bg2">
<td class="data">Query type (q)</td>
<td class="value">dnax</td>
</tr>
<tr>
<td class="data">Mask out repeats (mask)</td>
<td class="value">lower</td>
</tr>
<tr class="bg2">
<td class="data">Mask out repeats on query (qmask)
</td>
<td class="value">lower
</td>
</tr>
</table>

<h3 class="boxed">Chunking parameters</h3>
<table style="text-align: left; width: 100%;" border="0" cellpadding="2"
cellspacing="2">
<tbody>
<tr>
<td style="vertical-align: top;">
</td>
<th style="vertical-align: top;">Medaka
</th>
<th style="vertical-align: top;">Platyfish
</th>
</tr>
<tr>
<th style="vertical-align: middle;">Chunk size
</th>
<td style="vertical-align: top;">1,000,000
</td>
<td style="vertical-align: top;">25,000
</td>
</tr>
<tr>
<th style="vertical-align: middle;">Overlap
</th>
<td style="vertical-align: top;">10,000
</td>
<td style="vertical-align: top;">10,000
</td>
</tr>
<tr>
<th style="vertical-align: middle;">Group set size
</th>
<td style="vertical-align: top;">100,000,000
</td>
<td style="vertical-align: top;">10,000,000
</td>
</tr>
<tr>
<th style="vertical-align: middle;">Masking options
</th>
<td style="vertical-align: top;">{default_soft_masking => 1}
</td>
<td style="vertical-align: top;">{default_soft_masking => 1}
</td>
</tr>
</tbody>
</table>

<h3>Results</h3>

<p>Number of alignment blocks: 770368</p>
<table style="text-align: left; width: 100%;" border="0" cellpadding="2"
cellspacing="2">
<tbody>
<tr>
<td style="vertical-align: top;">
</td>
<th style="vertical-align: top;">Genome coverage(bp)
</th>
<th style="vertical-align: top;">Coding exon coverage (bp)
</th>
</tr>
<tr>
<th style="vertical-align: middle;">Medaka</th>
<td style="vertical-align: top;"><img style="width: 200px; height: 200px;" alt=""
src="./pie_ref_align_608.png"/>
</td>
<td style="vertical-align: top;"><img style="width: 200px; height: 200px;" alt=""
src="./pie_ref_cod_align_608.png"/>
</td>
</tr>
<tr>
<td></td>
<td style="vertical-align: top;">75,595,093 out of 868,983,502
</td>
<td style="vertical-align: top;">25,211,022 out of 30,193,605
</td>
</tr>
<tr>
<th style="vertical-align: middle;">Platyfish
</th>
<td style="vertical-align: top;"><img style="width: 200px; height: 200px;" alt=""
src="./pie_non_ref_align_608.png"/>
</td>
<td style="vertical-align: top;"><img style="width: 200px; height: 200px;" alt=""
src="./pie_non_ref_cod_align_608.png"/>
</td>
</tr>
<tr>
<td></td>
<td style="vertical-align: top;">62,611,838 out of 729,662,853
</td>
<td style="vertical-align: top;">26,911,611 out of 33,480,785
</td>
</tr>
</tbody>
</table>


</body>
</html>
