
<h3 class="boxed">About the <i>Echinops telfairi</i> genome</h3>

<h4>Assembly</h4>
<p><img src="/img/species/pic_Echinops_telfairi.png" height="100" width="100" class="float-left" alt="Tenrec" title="ssssssssssSSSSSSSSS!!" />
This is the first release of the low-coverage 2X assembly of the small Madagascar hedgehog (<em>Echinops telfairi</em>).
The genome sequencing and assembly are provided by the <a href="http://www.broad.mit.edu/mammals/">Broad Institute</a>.
</p>
<p>
The N50 size is the length such that 50% of the assembled genome lies in blocks of the N50 size or longer.
The N50 length for supercontigs is 48 kb and is 3.1 kb for contigs. The total number of bases in supercontigs is
3.81 Gb and in contigs is 2.11 Gb. 
</p>

<h4>Mammalian Genome Project</h4>
<p>
<i>Echinops telfairi</i> is one of 16 mammals that will be sequenced as part of the <a href="http://www.broad.mit.edu/mammals/">Mammalian Genome Project</a>, funded by the National Institutes of Health (<a href="http://www.nih.gov/">NIH</a>).
The species were chosen to maximise the branch length of the evolutionary tree while representing the diversity of mammalian species.</p>
                                                                                
<p>Low-coverage 2X assemblies will be produced for these mammals and used in alignments for cross-species comparison.  The aim is to increase our understanding of functional elements, especially in the human genome. 
</p>

