<html>
<head>
<title>Biotypes</title>
</head>

<body>


<h1 id="geneandtranscripttypes">Gene and transcript types</h1>

<h2 id="genes">Genes</h2>

<ul>
<li><strong>Protein coding</strong> Contains an open reading frame (ORF).


<ul>
<li><strong>Polymorphic</strong> A protein coding gene that has at least one transcript with a valid ORF and one or more coding transcripts that contain a polymorphism.</li></ul>
</li>

<li><strong>Processed transcript</strong> Doesn't contain an ORF. Divided into three major categories.


<ul>
<li><strong>Long non-coding RNA (lncRNA)</strong> Subclassified into one of the following types:</li>

<li><strong>Non coding</strong> Contains transcripts which are known from the literature to not be protein coding.</li>

<li><strong>3prime<em>overlapping</em>ncRNA</strong> Has transcripts where ditag and/or published experimental data strongly supports the existence of long (>200bp) non-coding transcripts that overlap the 3'UTR of a protein-coding locus on the same strand.</li>

<li><strong>Antisense</strong> Has transcripts that overlap the genomic span (i.e. exon or introns) of a protein-coding locus on the opposite strand.</li>

<li><strong>lincRNA (long interspersed ncRNA)</strong> Has transcripts that are long intergenic non-coding RNA locus with a length >200bp. Requires lack of coding potential and may not be conserved between species.</li>

<li><strong>Retained<em>intron</strong> Has an alternatively spliced transcript believed to contain intronic sequence relative to other, coding, variants.</li>

<li><strong>Sense</em>intronic</strong> Has a long non-coding transcript in introns of a coding gene that does not overlap any exons.</li>

<li><strong>Sense<em>overlapping</strong> Has a long non-coding transcript that contains a coding gene in its intron on the same strand.</li>

<li><strong>Macro</em>lncRNA</strong> Unspliced lncRNAs that are several kb in size.
Bidirectional lncRNA. A non-coding locus that originates from within the promoter region of a protein-coding gene, with transcription proceeding in the opposite direction on the other strand.</li>

<li><strong>ncRNA</strong></li>

<li><strong>miRNA</strong> microRNA.</li>

<li><strong>piRNA</strong> piwi-interacting RNA.</li>

<li><strong>rRNA</strong> ribsosomal RNA.</li>

<li><strong>siRNA</strong> small interfering RNA.</li>

<li><strong>snRNA</strong> small nuclear RNA.</li>

<li><strong>snoRNA</strong> small nucleolar RNA.</li>

<li><strong>tRNA</strong> transfer RNA.</li>

<li><strong>vaultRNA</strong> Short non coding RNA genes that form part of the vault ribonucleoprotein complex.</li>

<li><strong>Unclassified processed transcript</strong> Cannot be placed in one of the other categories.</li></ul>
</li>

<li><strong>Pseudogene</strong> Similar to known proteins but contain a frameshift and/or stop codon(s) which disrupts the ORF. These can be classified into the following:


<ul>
<li><strong>Processed pseudogene</strong> Pseudogene that lack introns and is thought to arise from reverse transcription of mRNA followed by reinsertion of DNA into the genome.
Unprocessed pseudogene. Pseudogene that can contain introns since produced by gene duplication.</li>

<li><strong>Transcribed pseudogene</strong> Pseudogene where protein homology or genomic structure indicates a pseudogene, but the presence of locus-specific transcripts indicates expression. These can be classified into 'Processed', 'Unprocessed' and 'Unitary'.</li>

<li><strong>Translated pseudogene</strong> Pseudogenes that have mass spec data suggesting that they are also translated. These can be classified into 'Processed', 'Unprocessed'</li>

<li><strong>Polymorphic pseudogene</strong> Pseudogene owing to a SNP/indel but in other individuals/haplotypes/strains the gene is translated.</li>

<li><strong>Unitary pseudogene</strong> A species specific unprocessed pseudogene without a parent gene, as it has an active orthologue in another    species.</li>

<li><strong>IG Pseudogene</strong> Inactivated immunoglobulin gene.</li></ul>
</li>

<li><strong>IG Gene</strong> Immunoglobulin gene.</li>

<li><strong>TR Gene</strong> T cell receptor gene.</li>

<li><strong>TEC (To be Experimentally Confirmed)</strong> This is used for non-spliced EST clusters that have polyA features. This category has been specifically created for the ENCODE project to highlight regions that could indicate the presence of protein coding genes that require experimental validation, either by 5' RACE or RT-PCR to extend the transcripts, or by confirming expression of the putatively-encoded peptide with specific antibodies.</li>
</ul>

<h2 id="transcriptclassification">Transcript Classification</h2>

<ul>
<li><strong>Protein coding</strong> Protein coding transcripts. These can be further classified as follows:


<ul>
<li><strong>Known Protein coding</strong> 100% Identical to RefSeq NP or Swiss-Prot entry.</li>

<li><strong>Novel Protein coding</strong> Shares >60% length with known coding sequence from RefSeq or Swiss-Prot or has cross-species/family support or domain evidence.</li>

<li><strong>Putative Protein coding</strong> Shares &lt;60% length with known coding sequence from RefSeq or Swiss-Prot, or has an alternative first or last coding exon.</li>

<li><strong>Nonsense mediated decay</strong> If the coding sequence (following the appropriate reference) of a transcript finishes >50bp from a downstream splice site then it is tagged as NMD. If the variant does not cover the full reference coding sequence then it is annotated as NMD if NMD is unavoidable i.e. no matter what the exon structure of the missing portion is the transcript will be subject to NMD.</li>

<li><strong>Nonstop decay</strong> Transcripts that have polyA features (including signal) without a prior stop codon in the CDS, i.e. a non-genomic polyA tail attached directly to the CDS without 3' UTR. These transcripts are subject to degradation.</li></ul>
</li>

<li><strong>Processed transcripts</strong> Doesn't contain an ORF. Divided into three major categories.


<ul>
<li><strong>Long non-coding RNA (lncRNA)</strong> Subclassified into one of the following types:</li>

<li><strong>Non coding</strong> Known from the literature to not be protein coding.</li>

<li><strong>3prime<em>overlapping</em>ncRNA</strong> Ditag and/or published experimental data strongly supports the existence of long (>200bp) non-coding transcripts that overlap the 3'UTR of a protein-coding locus on the same strand.</li>

<li><strong>Antisense</strong> Overlap the genomic span (i.e. exon or introns) of a protein-coding locus on the opposite strand.</li>

<li><strong>lincRNA (long interspersed ncRNA)</strong> Long intergenic non-coding RNA locus with a length >200bp. Requires lack of coding potential and may not be conserved between species.</li>

<li><strong>Retained<em>intron</strong> An alternatively spliced transcript believed to contain intronic sequence relative to other, coding, variants.</li>

<li><strong>Sense</em>intronic</strong> A long non-coding transcript in introns of a coding gene that does not overlap any exons.</li>

<li><strong>Sense<em>overlapping</strong> Has a long non-coding transcript that contains a coding gene in its intron on the same strand.</li>

<li><strong>macro</em>lncRNA</strong> unspliced lncRNAs that are several kb in size.</li>

<li><strong>ncRNA</strong></li>

<li><strong>miRNA</strong> microRNA.</li>

<li><strong>piRNA</strong> piwi-interacting RNA.</li>

<li><strong>rRNA</strong> ribsosomal RNA.</li>

<li><strong>siRNA</strong> small interfering RNA.</li>

<li><strong>snRNA</strong> small nuclear RNA.</li>

<li><strong>snoRNA</strong> small nucleolar RNA.</li>

<li><strong>tRNA</strong> transfer RNA.</li>

<li><strong>vaultRNA</strong> Short non coding RNA genes that form part of the vault ribonucleoprotein complex.</li>

<li><strong>Unclassified processed transcript</strong> Cannot be placed in one of the other categories.</li></ul>
</li>

<li><strong>Pseudogene</strong> Have homology to proteins but generally suffer from a disrupted coding sequence and    an active homologous gene can be found at another locus. Sometimes these entries have an intact coding sequence or an open but truncated ORF, in which case there is other evidence used (for example genomic polyA stretches at the 3' end) to classify them as a pseudogene. Can be further classified as follows:


<ul>
<li><strong>Processed pseudogene</strong> Pseudogene that appears to have been produced by integration of a reverse transcribed mRNA into the genome.</li>

<li><strong>Unprocessed pseudogene</strong> Pseudogene that shows evidence of loss of function, but has exon-intron structure.</li>

<li><strong>Transcribed pseudogene</strong> Pseudogene where protein homology or genomic structure indicates a pseudogene, but the presence of locus-specific transcripts indicates expression. These can be classified into 'Processed' and 'Unprocessed'.</li>

<li><strong>Translated pseudogene</strong> Pseudogenes that have mass spec data suggesting that they are also translated. These can be classified into 'Processed' and 'Unprocessed'</li>

<li><strong>Polymorphic pseudogene</strong> Pseudogene owing to a SNP/indel but in other individuals/haplotypes/strains the gene is translated.</li>

<li><strong>Unitary pseudogene</strong> A species specific unprocessed pseudogene without a parent gene, as it has an active orthologue in another    species.</li>

<li><strong>IG Pseudogene</strong> Inactivated immunoglobulin gene.</li></ul>
</li>

<li><strong>TEC (To be Experimentally Confirmed)</strong> This is used for non-spliced EST clusters that have polyA features. This category has been specifically created for the ENCODE project to highlight regions that could indicate the presence of novel protein coding genes that require experimental validation, either by 5' RACE or RT-PCR to extend the transcripts, or by confirming expression of the putatively-encoded peptide with specific antibodies.</li>

<li><strong>Artifact</strong> Used to tag mistakes in the public databases (Ensembl/SwissProt/ Trembl). Usually these arise from high-throughput cDNA sequencing projects, which submit automatic annotation sometimes resulting in erroneous coding sequences that are, for example, 3' UTRs.</li>

<li><strong>IG Gene</strong> Immunoglobulin gene.</li>

<li><strong>TR Gene</strong> T cell receptor gene.</li>
</ul>


</body>
</html>
