<html>
<head>
<title>Non-coding RNAs</title>
</head>

<body>

<h1>Automatic annotation of non-coding genes</h1>
  
<p>Non-coding RNAs (ncRNAs) are involved in many biological processes and are increasingly seen as important. As is the case with proteins, it is the overall structure of the molecule which imparts function. However, while similar protein structures are often reflected in a conserved amino acid sequence, sequences underlying RNA secondary structure are very variable; this makes ncRNAs difficult to detect using sequence alone.</p>

<h2 id="types-of-ncrna">Types of ncRNA</h2>
 
<table>
<thead>
<tr>
<th>Abbreviation</th>
<th>Definition</th>
</tr>
</thead>
<tbody>
<tr>
<td>tRNA</td>
<td>transfer RNA</td>
</tr>
<tr>
<td>Mt-tRNA</td>
<td>transfer RNA located in the mitochondrial genome</td>
</tr>
<tr>
<td>rRNA</td>
<td>ribosomal RNA</td>
</tr>
<tr>
<td>scRNA</td>
<td>small cytoplasmic RNA</td>
</tr>
<tr>
<td>snRNA</td>
<td>small nuclear RNA</td>
</tr>
<tr>
<td>snoRNA</td>
<td>small nucleolar RNA</td>
</tr>
<tr>
<td>miRNA</td>
<td>microRNA precursors</td>
</tr>
<tr>
<td>misc_RNA</td>
<td>miscellaneous other RNA</td>
</tr>
<tr>
<td>lincRNA</td>
<td>Long intergenic non-coding RNAs</td>
</tr>
</tbody>
</table>
  
<h2 id="annotation-details">Annotation Details</h2>
  
<p>Most ncRNAs are annotated by aligning genomic sequence against <a href="http://rfam.xfam.org/">RFAM</a> using <a href="http://blast.wustl.edu/">BLASTN</a>. The BLAST hits are clustered and filtered by E-value and are used to seed Infernal searches of the locus with the corresponding RFAM covariance models. The purpose of this is to reduce the search space required, as to scan the entire genome with all the RFAM covariance models would be extremely CPU-intensive. The resulting BLAST hits are then used as supporting evidence for ncRNA genes.</p>
<p>Some ncRNAs have specialised annotation.</p>
  
<h3 id="mirnas">miRNAs</h3>
<p>miRNAs are imported from <a href="http://www.mirbase.org/">miRBase</a>. All species are used.</p>

  <h3 id="trnas">tRNAs</h3>
<p>tRNAs are annotated as part of the raw compute process using <a href="http://lowelab.ucsc.edu/tRNAscan-SE/">tRNAscan-SE</a>. Because of this, they are not included as genes in the database, but as Simple Features instead.</p>

 <h3 id="lincrna">lincRNA</h3>
<p>lincRNA (Long intergenic non-coding RNAs) Ensembl gene annotation, cDNA alignments and chromatin-state map data from the Ensembl regulatory build are used to predict lincRNAs for human and mouse. We do not import the lincRNAs identified by Guttman et al [1], but their publication guided us to our current approach for automatically annotating lincRNAs. First, regions of chromatin methylation (H3K4me3 and H3K36me3) outside known protein-coding loci are identified. Next, cDNAs which overlap with H3K4me3 or H3K36me3 features are identified as candidate lincRNAs. A final evaluation step investigates if each candidate lincRNA has any protein-coding potential. Any candidate lincRNA containing a substantial open reading frame (ORF) covering 35% or more of its length and containing PFAM/tigrfam protein domains will be rejected. Candidate lincRNAs that pass the final evaluation step are included in the human or mouse gene set as lincRNA genes.</p>

 <h2>References</h2>
<p>Guttman M, Amit I, Garber M, French C, Lin MF, Feldser D, Huarte M, Zuk O, Carey BW, Cassady JP, Cabili MN, Jaenisch R, Mikkelsen TS, Jacks T, Hacohen N, Bernstein BE, Kellis M, Regev A, Rinn JL, Lander ES
<a href = "http://europepmc.org/articles/PMC2754849">Chromatin signature reveals over a thousand highly conserved large non-coding RNAs in mammals.</a>
Nature 2009 458:223-227
doi:10.1038/nature07672</p>


</body>
</html>
