
  <h3>About the <i>Cavia porcellus</i> genome</h3>

<h4>Assembly</h4>

<p><img src="/img/species/pic_Cavia_porcellus.png" height="91" width="100" class="float-left" alt="Guinea Pig" title="ooeeeeek!" />
This is the first release of the low-coverage 1.92X assembly of the
Guinea Pig (<i>Cavia porcellus</i>). The genome sequencing
and assembly is provided by the 
<a href="http://www.broad.mit.edu/">Broad Institute</a>.
</p>

<p>The N50 size is the length such that 50% of the assembled genome lies
in blocks of the N50 size or longer. The N50 length for supercontigs
is 14.57 kb and is 2.80 kb for contigs. The total number of bases in
supercontigs is 3.37 Gb and in contigs is 1.95 Gb.
</p>

<h4>Annotation</h4>
<p>
Owing to the fragmentary nature of this preliminary assembly, it was necessary to arrange some scaffolds into "gene-scaffold" super-structures,in order to present complete genes. There are 8,008 such gene scaffolds, with identifiers of the form "GeneScaffold_1".
</p>

<p>In addition to the projection based genebuild, genewise was used to align
Cavia porcellus proteins to the genome, including the gene-scaffolds and Cavia
porcellus cDNAs and ESTs were aligned using exonerate.

<ul><li><a href="/info/about/docs/2x_genomes.html">Details of the gene-scaffold construction and subsequent gene-build</a></li></ul>

</p>

<h4>Mammalian Genome Project</h4>
<p><i>Cavia porcellus</i> is one of 16 mammals that will be sequenced as part
of the <a href="http://www.broad.mit.edu/mammals/">Mammalian Genome Project</a>, funded by the National Institutes of
Health (<a href="http://www.nih.gov/">NIH</a>). A group of species were chosen to maximise the branch
length of the evolutionary tree while representing the diversity of
mammalian species. Low-coverage 2X assemblies will be produced for
these mammals and used in alignments for cross-species comparison. The
aim is to increase our understanding of functional elements, especially
in the human genome.
</p>

