<h1>Regulatory Build</h1>
<p>
The regulatory build provides a single "best guess" set of regulatory elements.  These elements are based on the information contained within the ensembl <a href="/info/docs/api/index.html" class="cp-external">functional genomics</a> database.
</p>





<h2>Regulatory Feature Construction</h2>
<p>
The main principle of building regulatory features is simply performing overlap analysis of annoations in each data set. Core regions are identified using the focus sets, which are chosen to define a set of potentially regions.  These tend to be broad coverage narrowly focused marks which are likely candidates for different types of regulatory elements or motifs.  Historically we have used DNase1 which is known to mark for accessible chromatin, H3K4me3 which has been associated with promoter regions, and CTCF which characterises 'insulator/enhancer' element. As such the core regions of regulatory features are likely to be positioned on or around any potential regualtory motif.  Core regions are extended only in the case of direct overlap with another focus feature.  To maintain resolution and to avoid chaining of regulatory features across regions of dense regulatory elements we have imposed a 2KB cut off for focus features.   Exceeding this cutoff causes the focus feature to be treated as an attribute feature and so does not extend the core region.</p>

<p>
The arms or bounds of a regulatory feature are defined by overlap of attribute features with respect to the core region.  Attribute feature sets are generally less discriminatory and less focused annotations which are used extend the immediate structure of the regulatory feature and to inform the annotation analysis. Directly overlapping attribute features are said to have one degree of sepration.  Attributes with two degrees of separation or only included is they are within 2KB of the core region and they are entirely contained within another longer attribute feature. This is done to capture information adjacent and indirectly associated with the core region, whilst avoiding longer range and potentially anomalous associations.</p>

<!-- Do we really want to only include those attr features which are entirely contained? -->
<!-- So long as they overlap and farthest bound is within 2KB? Cut off for attr feat length? -->
<!-- Degradation of chained core regions greater than? -->

</p>

<h2>Annotation</h2>
<p>
Regulatory features are classified by analysing the whole set of patterns of consituent attribute and focus features with respect to several classes of genomic feature. This enables us to identify clusters of patterns which are significantly over represented and allows us to generate annotations based on these patterns. Some preliminary analysis has identified patterns for the following annotations that are both common and statistically significant:
</p>
<ul>
<li>non-gene-associated = patterns under-represented in genic regions</li>
<li>gene-associated = patterns over-represented in genic regions</li>

<li>promoter-associated = patterns over-represented in the region of the first exon and 2500bp upstream of protein coding genes, but not in the downstream 'gene-body'. </li>
</ul>

<p>
These data sets can be displayed along the chromosome in <a href="/Help/View?id=140" class="cp-internal">'Region in Detail'</a>, displayed for a gene in the <a href="/Help/View?id=142" class="cp-internal">'GeneRegulation'</a> view or mined from the <a href="/info/docs/api/index.html" class="cp-external">functional genomics</a> database.
<p>

<h2>Source data</h2>
<p>
Regulatory features are generated by using as a variety genome wide epigenomic data sets, currently comprising the following:
</p>

<table class="ss tint">
<tr><th colspan=3>Regulatory Build Version 4</th></tr>
<tr><th>Focus Sets</th><th class="center">Data type</th><th class="center">Reference</th></tr>
<tr>
<td>DNase1 Hypersensitivity site (GM06990 and CD4+ T cells) </td>
<td>ChIP-Seq</td>
<td>1</td>
</tr>
<tr>
<td class="bg2">CCCTC-binding factor (CTCF)</td>
<td>ChIP-Chip*</td>
<td>2</td>
</tr>
<tr>
<td>CCTC-binding factor (CTCF)</td>
<td>ChIP-Seq</td>
<td>4</td>
</tr>
</table>

<p>

<table class="ss tint">
<tr><th>Attribute Sets</th><th>Data type</th><th>Reference</th></tr>
<tr>
<td>H4K20me3</td>
<td>ChIP-Chip</td>
<td>3</td>
</tr>
<tr class="bg2">
<td>H3K27me3</td>
<td>ChIP-Chip</td>
<td>3</d>
<tr>
<tr>
<td>H3K36me3</td>								<td>ChIP-Chip</td>
<td>3</td>
</tr>
<tr class="bg2">
<td>H3K79me3</td>									<td>ChIP-Chip</td>
<td>3</td>
</tr>
<tr>
<td>H3K9me3</td>									<td>ChIP-Chip</td>
<td>3</td>
</tr>

<tr class="bg2">
<td>H2AZ</td>
<td>ChIP-Seq</td>
<td>4</td>
</tr>

<tr>
<td>H2BK5me1</td>
<td>ChIP-Seq</td>
<td>4</td>
</tr>

<tr class="bg2">
<td>H3K27me1, me2, me3</td>
<td>ChIP-Seq</td>
<td>4</td>
</tr>

<tr>
<td>H3K36me1, me3</td>
<td>ChIP-Seq</td>
<td>4</td>
</tr>

<tr class="bg2">
<td>H3K4me1,me2,me3</td>
<td>ChIP-Seq</td>
<td>4</td>
</tr>

<tr>
<td>H3K79me1,me2,me3</td>
<td>ChIP-Seq</td>
<td>4</td>
</tr>

<tr class="bg2">
<td>H3K9me1,me2,me3</td>
<td>ChIP-Seq</td>
<td>4</td>
</tr>

<tr>
<td>H3R2me1, me2</td>
<td>ChIP-Seq</td>
<td>4</td>
</tr>

<tr class="bg2">
<td>H4K20me1, me3</td>
<td>ChIP-Seq</td>
<td>4</td>
</tr>

<tr>
<td>H4R3me2</td>
<td>ChIP-Seq</td>
<td>4</td>
</tr>

<tr class="bg2">
<td>PolII</td>
<td>ChIP-Seq</td>
<td>4</td>
</tr>


<tr>
<td>CD4_H2AK5ac</td>
<td>ChIP-Seq</td>
<td>5</td>
</tr>

<tr class="bg2">
<td>CD4_H2AK9ac</td>
<td>ChIP-Seq</td>
<td>5</td>
</tr>

<tr>
<td>CD4_H2BK120ac</td>
<td>ChIP-Seq</td>
<td>5</td>
</tr>

<tr class="bg2">
<td>CD4_H2BK12ac</td>
<td>ChIP-Seq</td>
<td>5</td>
</tr>

<tr>
<td>CD4_H2BK20ac</td>
<td>ChIP-Seq</td>
<td>5</td>
</tr>

<tr class="bg2">
<td>CD4_H2BK5ac</td>
<td>ChIP-Seq</td>
<td>5</td>
</tr>

<tr>
<td>CD4_H3K14ac</td>
<td>ChIP-Seq</td>
<td>5</td>
</tr>

<tr class="bg2">>
<td>CD4_H3K18ac</td>
<td>ChIP-Seq</td>
<td>5</td>
</tr>

<tr>
<td>CD4_H3K23ac</td>
<td>ChIP-Seq</td>
<td>5</td>
</tr>

<tr class="bg2">
<td>CD4_H3K27ac</td>
<td>ChIP-Seq</td>
<td>5</td>
</tr>

<tr>
<td>CD4_H3K36ac</td>
<td>ChIP-Seq</td>
<td>5</td>
</tr>

<tr class="bg2">
<td>CD4_H3K4ac</td>
<td>ChIP-Seq</td>
<td>5</td>
</tr>

<tr>
<td>CD4_H3K9ac</td>
<td>ChIP-Seq</td>
<td>5</td>
</tr>

<tr class="bg2">
<td>CD4_H4K12ac</td>
<td>ChIP-Seq</td>
<td>5</td>
</tr>

<tr>
<td>CD4_H4K16ac</td>
<td>ChIP-Seq</td>
<td>5</td>
</tr>

<tr class="bg2">
<td>CD4_H4K5ac</td>
<td>ChIP-Seq</td>
<td>5</td>
</tr>

<tr>
<td>CD4_H4K8ac</td>
<td>ChIP-Seq</td>
<td>5</td>
</tr>

<tr class="bg2">
<td>CD4_H4K91ac</td>
<td>ChIP-Seq</td>
<td>5</td>
</tr>
</table>


<!-- Add section on peak calling SWEmbl etc -->

<strong>References</strong>
</p>
<p>
1.  Genome-wide identification of DNaseI hypersensitive sites was performed by Greg Crawford and Terry Furey (Duke University) using a whole genome DNase-sequencing protocol (Crawford et al., Genome Research 2006).<br> DNase-sequencing was performed using the Illumina (Solexa) sequencing by synthesis method from a DNase treated library generated from the GM06990 cell line (Crawford and Furey, unpublished).
</p>
<p>
2.  Kim, T.H.; Abdullaev, Z.K.; Smith, A.D.; Ching, K.A.; Loukinov, D.I.; Green, R.D.; Zhang, M.Q.; Lobanenkov, V.V. & Ren, B.<br>
Analysis of the vertebrate insulator protein CTCF-binding sites in the human genome.<br>
Cell, 2007 , 128 , 1231-1245
</p>
<p>
3.  Hirst, M; Hurd, P.J.; Bainbridge, M.; Robertson, G.; Kirmizis, A.; Nelson, C.; Zhao, Y.; Zeng, T.; Pandoh, P.; Tam, A.; Prabhu, A.; Dhalla, N.; Sa, D.; Delaney, A.; Bilenky, M.; Jones, S.; Kouzarides, T.; Marra, M.  (In preparation)
</p>
<p>
4. A. Barski, S. Cuddapah, K. Cui, T.Y. Roh, D.E. Schones, Z. Wang, G. Wei, I. Chepelev and K. Zhao, (2007). High-resolution profiling of histone methylations in the human genome, Cell 129 (2007), pp. 823-837.
</p>
<p>
5. Wang Z, Zang C, Rosenfeld JA, Schones DE, Barski A, Cuddapah S, Cui K, Roh TY, Peng W, Zhang MQ, Zhao K. Combinatorial patterns of histone acetylations and methylations in the human genome. Nat Genet. 2008 Jul;40(7):897-903. Epub 2008 Jun 15. PMID: 18552846
<p>
* The CTCF data was processed with the Nessie HMM (Flicek, unpublished).
</p>
