<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="CONTENT-TYPE" CONTENT="text/html; charset=utf-8">
	<TITLE>Regulatory Build</TITLE>
	<META NAME="GENERATOR" CONTENT="OpenOffice.org 3.1  (Unix)">
	<META NAME="CREATED" CONTENT="0;0">
	<META NAME="CHANGEDBY" CONTENT="Damian Keefe">
	<META NAME="CHANGED" CONTENT="20100511;12503800">
	<META NAME="order" CONTENT="2">
	<STYLE TYPE="text/css">
	<!--
		H2.ctl { font-family: "Lohit Hindi" }
	-->
	</STYLE>
</HEAD>
<BODY LANG="en-GB" DIR="LTR">
<H1>Regulatory Build</H1>
<P>The regulatory build provides a single &quot;best guess&quot; set
of regulatory elements. These elements are based on the information
contained within the ensembl <A HREF="/info/docs/api/index.html">functional
genomics</A> database. 
</P>
<H2 CLASS="western">Regulatory Feature Construction</H2>
<P>The 'RegulatoryBuild' is performed by overlap analysis of
annotations from data sets in a two stage cell type aware manner. 
</P>
<P>In stage one, core regions are identified across all available
cell types using 'focus' features, which are chosen to define a set
of potential binding sites. These tend to be broad coverage, narrowly
focused marks which are likely candidates for different types of
regulatory elements or motifs. Focus feature types include DNase1
which is known to mark for accessible chromatin, transcription factor
binding sites (TFBSs) and CTCF, which characterises
'insulator/enhancer' elements. As such the core regions of regulatory
features are likely to be positioned on or around any potential
regulatory motif. Core regions are extended only in the case of
direct overlap with another focus feature. To maintain resolution and
to avoid chaining of regulatory features across regions of dense
regulatory elements a 2KB cut-off is imposed. Exceeding this cut-off
causes the offending focus feature to be treated as an attribute
feature and so does not extend the core region. 
</P>
<P><!-- as it is assumed that these will exist, either active or silenced, across all cell types 
irrepective of the presence of supporting focus data for a given cell type. -->Stage
two extends the structure in a cell type specific manner, using
'attribute' features. Attribute features do not define a binding site
and are some times longer ranging feature types which are useful for
classification, such as Histone modifications. If core data exists
for a given cell type, a RegulatoryFeature is seeded using the core
region defined in stage one. The arms or bounds are defined by
overlap of attribute features with respect to the core region.
Directly overlapping attribute features are said to have one degree
of separation. Attributes with two degrees of separation are only
included if they are within 2KB of the core region and they are
entirely contained within another longer associated attribute
feature. This is done to capture information adjacent and indirectly
associated with the core region, whilst avoiding longer range and
potentially anomalous associations. 
</P>
<P>These two stages give rise to regulatory FeatureSets for the core
'MultiCell' features and for each available cell type. 
</P>
<H2 CLASS="western"><!-- Do we really want to only include those attribute features which are
entirely contained? --><!-- So long as they overlap and farthest
bound is within 2KB? Cut-off for attribute feature length? --><!-- Degradation of chained core regions greater than? -->Annotation</H2>
<P>Regulatory Features (regfeats) are classified by considering their
position on the genome in relation to other classes of feature on the
genome (eg genes, repeats, intergenic regions) together with the
combination of regulatory attributes they possess as coded in their
display_label. The display label is a binary string with each
position corresponding to a particular focus or attribute feature and
a value of 1 indicating that the regulatory feature overlaps this
particular type of focus or attribute feature. A set of randomly
distributed features (mockfeats) corresponding to the regfeats in
terms of length and chromosome are also generated so that we can
judge if the placement of regfeats in relation to the genomic
features is non-random.</P>
<P>The first step in the procedure is to determine which genomic
features (genfeats) each regfeat overlaps. A single common basepair
is sufficient to consider two features overlapping. We do the same
with the mockfeats. (Strictly speaking this is not the first step, as
we know from experience that certain regulatory features are most
probably artefacts and that others contain no useable information so
these are filtered out before the procedure begins and the mockfeats
correspond to only the filtered set of regfeats).</P>
<P>Next we create a set of patterns of attributes we wish to
evaluate. Currently this is all the patterns which occur in the
display labels more than once, plus all the patterns which can be
created by re-setting one bit of the existing patterns from 1 to 0.</P>
<P>For each pattern, we look at all the regulatory features which
have the same or more bits set. If there are more than 100 such
regfeats we count the number of times these features overlap each
class of genfeat. We do the same count with the set of mockfeats
which correspond to the regfeats. If &gt;50% of the regfeats overlap
a particular class of genfeat and the chi-squared calculated using
the mockfeat count as the 'expected' value is &gt;8.0 (P0.005) we
record that this pattern is associated with this class of genfeat.</P>
<P>If the pattern IS associated with a genfeat we collect a second
set of patterns which have this pattern's PLUS any other bits set.
For each of these patterns we look at all the regulatory features
which have the same or more bits set and we count the number of times
these features overlap each class of genfeat. If less than 50% of the
regfeats overlap we record that this second pattern is not associated
with the class of genfeat involved.</P>
<P>Having determined all the associated and not-associated patterns
for each class of genfeat, we look at all the regfeats and use the
'associated' and then 'not-associated' patterns to set or unset a
flag indicating whether the particular regfeat is associated with a
particular class of genfeat. During this process it is possible for a
given regfeat to be associated with more than one class of genfeat
and some of these can be contradictory. This is particularly the case
where all or nearly all the bits are set.</P>
<P>Finally, for the purposes of the regulatory build, there is a set
of rules which 1. resolve conflicts amongst the above flags and 2.
assign a regulatory feature_type to the regfeat. The following types
are currently in use :-</P>
<UL>
	<LI><P STYLE="margin-bottom: 0cm">non-gene-associated = patterns
	under-represented in gene regions 
	</P>
	<LI><P STYLE="margin-bottom: 0cm">gene-associated = patterns
	over-represented in gene regions 
	</P>
	<LI><P>promoter-associated = patterns over-represented in the region
	of the first exon and 2500 bp upstream of protein coding genes, but
	not in the downstream 'gene-body'. 
	</P>
</UL>
<P>At present only cell-type specific regulatory features are
classified as different cell types may give conflicting signals
reflecting their unique combination of regulatory and transcriptional
states.</P>
<P>These data sets can be displayed along the chromosome in <A HREF="/Help/View%3Fid=140">'Region
in Detail'</A>, displayed for a gene in the <A HREF="/Help/View%3Fid=142">'GeneRegulation'</A>
view or mined from the <A HREF="/info/docs/api/index.html">functional
genomics</A> database. 
</P>
<H2 CLASS="western">Source data</H2>
<P>Regulatory features are generated by using a variety of genome
wide epigenomic data sets, currently comprising the following: 
</P>

<table class="ss tint">

<tr>
<th colspan="3">Human Regulatory Build Version 7 (GM06990)</th>
</tr>
<tr>
<th>Focus Sets</th>
<th class="center">Data type</th>
<th class="center">Reference</th>
</tr>

<tr>
<td>DNase1</td>
<td>ChIP-Seq</td>
<td>1</td>
</tr>

<tr>
<th colspan="3">Human Regulatory Build Version 7 (IMR90)</th>
</tr>
<tr>
<th>Focus Sets</th>
<th class="center">Data type</th>
<th class="center">Reference</th>
</tr>

<tr>
<td class="bg2">CTCF</td>
<td>ChIP-Chip*</td>
<td>2</td>
</tr>

<tr>
<th colspan="3">Human Regulatory Build Version 7 (CD4)</th>
</tr>
<tr>
<th>Focus Sets</th>
<th class="center">Data type</th>
<th class="center">Reference</th>
</tr>

<tr>
<td>DNase1</td>
<td>ChIP-Seq</td>
<td>1</td>
</tr>

<tr class="bg2">
<td>CTCF</td>
<td>ChIP-Seq</td>
<td>3</td>
</tr>

<tr>
<th>Attribute Sets</th>
<th>Data type</th>
<th>Reference</th>
</tr>

<tr class="bg2">
<td>H2AZ</td>
<td>ChIP-Seq</td>
<td>3</td>
</tr>

<tr>
<td>H2BK5me1</td>
<td>ChIP-Seq</td>
<td>3</td>
</tr>

<tr class="bg2">
<td>H3K27me1, me2, me3</td>
<td>ChIP-Seq</td>
<td>3</td>
</tr>

<tr>
<td>H3K36me1, me3</td>
<td>ChIP-Seq</td>
<td>3</td>
</tr>

<tr class="bg2">
<td>H3K4me1,me2,me3</td>
<td>ChIP-Seq</td>
<td>3</td>
</tr>

<tr>
<td>H3K79me1,me2,me3</td>
<td>ChIP-Seq</td>
<td>3</td>
</tr>

<tr class="bg2">
<td>H3K9me1,me2,me3</td>
<td>ChIP-Seq</td>
<td>3</td>
</tr>

<tr>
<td>H3R2me1, me2</td>
<td>ChIP-Seq</td>
<td>3</td>
</tr>

<tr class="bg2">
<td>H4K20me1, me3</td>
<td>ChIP-Seq</td>
<td>3</td>
</tr>

<tr>
<td>H4R3me2</td>
<td>ChIP-Seq</td>
<td>3</td>
</tr>

<tr class="bg2">
<td>PolII</td>
<td>ChIP-Seq</td>
<td>3</td>
</tr>

<tr>
<td>H2AK5ac</td>
<td>ChIP-Seq</td>
<td>4</td>
</tr>

<tr class="bg2">
<td>H2AK9ac</td>
<td>ChIP-Seq</td>
<td>4</td>
</tr>

<tr>
<td>H2BK120ac</td>
<td>ChIP-Seq</td>
<td>4</td>
</tr>

<tr class="bg2">
<td>H2BK12ac</td>
<td>ChIP-Seq</td>
<td>4</td>
</tr>

<tr>
<td>H2BK20ac</td>
<td>ChIP-Seq</td>
<td>4</td>
</tr>

<tr class="bg2">
<td>H2BK5ac</td>
<td>ChIP-Seq</td>
<td>4</td>
</tr>

<tr>
<td>H3K14ac</td>
<td>ChIP-Seq</td>
<td>4</td>
</tr>

<tr class="bg2">
<td>H3K18ac</td>
<td>ChIP-Seq</td>
<td>4</td>
</tr>

<tr>
<td>H3K23ac</td>
<td>ChIP-Seq</td>
<td>4</td>
</tr>

<tr class="bg2">
<td>H3K27ac</td>
<td>ChIP-Seq</td>
<td>4</td>
</tr>

<tr>
<td>H3K36ac</td>
<td>ChIP-Seq</td>
<td>4</td>
</tr>

<tr class="bg2">
<td>H3K4ac</td>
<td>ChIP-Seq</td>
<td>4</td>
</tr>

<tr>
<td>H3K9ac</td>
<td>ChIP-Seq</td>
<td>4</td>
</tr>

<tr class="bg2">
<td>H4K12ac</td>
<td>ChIP-Seq</td>
<td>4</td>
</tr>

<tr>
<td>H4K16ac</td>
<td>ChIP-Seq</td>
<td>4</td>
</tr>

<tr class="bg2">
<td>H4K5ac</td>
<td>ChIP-Seq</td>
<td>4</td>
</tr>

<tr>
<td>H4K8ac</td>
<td>ChIP-Seq</td>
<td>4</td>
</tr>

<tr class="bg2">
<td>H4K91ac</td>
<td>ChIP-Seq</td>
<td>4</td>
</tr>

<tr>
<th colspan="3">Human Regulatory Build Version 7 (K562)</th>
</tr>
<tr>
<th>Focus Sets</th>
<th class="center">Data type</th>
<th class="center">Reference</th>
</tr>

<tr>
<td>DNase1</td>
<td>ChIP-Seq</td>
<td>5</td>
</tr>

<tr class="bg2">
<td>DNase1</td>
<td>ChIP-Seq</td>
<td>6</td>
</tr>

<tr>
<td>CTCF</td>
<td>ChIP-Seq</td>
<td>7</td>
</tr>

<tr class="bg2">
<td>c-Myc</td>
<td>ChIP-Seq</td>
<td>8</td>
</tr>

<tr>
<td>Jund</td>
<td>ChIP-Seq</td>
<td>8</td>
</tr>

<tr class="bg2">
<td>c-Fos</td>
<td>ChIP-Seq</td>
<td>8</td>
</tr>

<tr>
<td>Max</td>
<td>ChIP-Seq</td>
<td>9</td>
</tr>

<tr class="bg2">
<td>Nfe2</td>
<td>ChIP-Seq</td>
<td>9</td>
</tr>

<tr>
<td>Gata1</td>
<td>ChIP-Seq</td>
<td>9</td>
</tr>

<tr class="bg2">
<td>Znf263</td>
<td>ChIP-Seq</td>
<td>9</td>
</tr>

<tr>
<td>c-Jun</td>
<td>ChIP-Seq</td>
<td>9</td>
</tr>

<tr class="bg2">
<td>Gabp</td>
<td>ChIP-Seq</td>
<td>10</td>
</tr>

<tr>
<td>Nrsf</td>
<td>ChIP-Seq</td>
<td>10</td>
</tr>

<tr class="bg2">
<td>Srf</td>
<td>ChIP-Seq</td>
<td>10</td>
</tr>


<tr>
<th>Attribute Sets</th>
<th>Data type</th>
<th>Reference</th>
</tr>

<tr class="bg2">
<td>H3K27ac</td>
<td>ChIP-Seq</td>
<td>7</td>
</tr>

<tr>
<td>H3K27me3</td>
<td>ChIP-Seq</td>
<td>7</td>
</tr>

<tr class="bg2">
<td>H3K36me3</td>
<td>ChIP-Seq</td>
<td>7</td>
</tr>

<tr>
<td>H3K4me1, me2, me3</td>
<td>ChIP-Seq</td>
<td>7</td>
</tr>

<tr class="bg2">
<td>H3K9ac</td>
<td>ChIP-Seq</td>
<td>7</td>
</tr>

<tr>
<td>H3K9me1</td>
<td>ChIP-Seq</td>
<td>7</td>
</tr>

<tr class="bg2">
<td>H4K20me1</td>
<td>ChIP-Seq</td>
<td>7</td>
</tr>

<tr>
<td>PolII</td>
<td>ChIP-Seq</td>
<td>7</td>
</tr>

<tr class="bg2">
<td>PolII</td>
<td>ChIP-Seq</td>
<td>10</td>
</tr>

<tr>
<td>PolII</td>
<td>ChIP-Seq</td>
<td>9</td>
</tr>


</table>

<TABLE CELLPADDING=2 CELLSPACING=2>
<tr>
<th colspan="3">Mouse Regulatory Build Version 2 (ES)</th>
</tr>

<tr>
<th>Focus Sets</th>
<th class="center">Data type</th>
<th class="center">Reference</th>
</tr>

<tr class="bg2">
<td>DNase1</td>
<td>ChIP-Seq</td>
<td>11</td>
</tr>

<tr>
<th>Attribute Sets</th>
<th>Data type</th>
<th>Reference</th>
</tr>

<tr class="bg2">
<td>H3me</td>
<td>ChIP-Seq</td>
<td>12</td>
</tr>

<tr>
<td>H3K4me3</td>
<td>ChIP-Seq</td>
<td>12</td>
</tr>

<tr class="bg2">
<td>H3K9me3</td>
<td>ChIP-Seq</td>
<td>12</td>
</tr>

<tr>
<td>H3K27me3</td>
<td>ChIP-Seq</td>
<td>12</td>
</tr>

<tr class="bg2">
<td>H3K36me3</td>
<td>ChIP-Seq</td>
<td>12</td>
</tr>

<tr>
<td>H4K20me3</td>
<td>ChIP-Seq</td>
<td>12</td>
</tr>

<tr class="bg2">
<td>PolII</td>
<td>ChIP-Seq</td>
<td>12</td>
</tr>
</TABLE>
<P><!-- Add section on peak calling SWEmbl etc --><STRONG>References</STRONG>
</P>
<p>
1. Genome-wide identification of DNaseI hypersensitive sites was
performed by Greg Crawford and Terry Furey (Duke University) using a
whole genome DNase-sequencing protocol (Crawford et al., Genome
Research 2006).<br />DNase-sequencing was performed using the Illumina
(Solexa) sequencing by synthesis method from a DNase treated library
generated from the GM06990 cell line (Crawford and Furey,
unpublished).
</p>

<p>
2. Kim, T.H.; Abdullaev, Z.K.; Smith, A.D.; Ching, K.A.; Loukinov,
D.I.; Green, R.D.; Zhang, M.Q.; Lobanenkov, V.V. &amp; Ren, B.<br />
Analysis of the vertebrate insulator protein CTCF-binding sites in the
human genome.<br />Cell, 2007 , 128 , 1231-1245
</p>

<p>
3. A. Barski, S. Cuddapah, K. Cui, T.Y. Roh, D.E. Schones, Z. Wang,
G. Wei, I. Chepelev and K. Zhao, (2007). High-resolution profiling of
histone methylations in the human genome, Cell 129 (2007),
pp. 823-837.
</p>

<p>
4. Wang Z, Zang C, Rosenfeld JA, Schones DE, Barski A, Cuddapah S, Cui
K, Roh TY, Peng W, Zhang MQ, Zhao K. Combinatorial patterns of histone
acetylations and methylations in the human genome. Nat Genet. 2008
Jul;40(7):897-903. Epub 2008 Jun 15. PMID: 18552846
</p>


<p>
5. This data was produced as part of the ENCODE project. These data were 
generated by the UW ENCODE group.
More information <A HREF='http://genome.ucsc.edu/cgi-bin/hgTrackUi?hgsid=159137986&c=chrX&g=wgEncodeUwDnaseSeq'>here</A> 
</p>

<p>
6. This data was produced as part of the ENCODE project. These data and 
annotations were created by a collaboration of multiple institutions.
More information <A HREF='http://genome.ucsc.edu/cgi-bin/hgTrackUi?hgsid=159137986&c=chrX&g=wgEncodeChromatinMap'>here</A> 
</p>

<p>
7. This data was produced as part of the ENCODE project. The ChIP-seq data were 
generated at the Broad Institute and in the Bradley E. Bernstein lab at the 
Massachusetts General Hospital/Harvard Medical School. 
More information <A HREF='http://genome.ucsc.edu/cgi-bin/hgTrackUi?hgsid=159137986&c=chrX&g=wgEncodeBroadChipSeq'>here</A> 
</p>

<p>
8. Raha D, Wang Z, Moqtaderi Z, Wu L et al. Close association of RNA polymerase 
II and many transcription factors with Pol III genes. Proc Natl Acad Sci USA 
2010 Feb 23;107(8):3639-44. PMID: 20139302
</p>

<p>
9. This data was produced as part of the ENCODE project. These data were generated 
and analyzed by the labs of Michael Snyder, Mark Gerstein and Sherman Weissman at 
Yale University; Peggy Farnham at UC Davis; and Kevin Struhl at Harvard.    
More information <A HREF='http://genome.ucsc.edu/cgi-bin/hgTrackUi?hgsid=159137986&c=chrX&g=wgEncodeYaleChIPseq'>here</A> 
</p>

<p>
10. This data was produced as part of the ENCODE project. These data were 
provided by the Myers Lab at the HudsonAlpha Institute for Biotechnology.
More information <A HREF='http://genome.ucsc.edu/cgi-bin/hgTrackUi?hgsid=159137986&c=chrX&g=wgEncodeHudsonalphaChipSeq'>here</A> 
</p>

<p>
11. Dnase1-sequencing was produced as a collaboration between Ensembl,
David Adams (Wellcome Trust Sanger Institute), and Greg Crawford (Duke
University).
</p>

<p>
12. Mikkelsen TS, Ku M, Jaffe DB, Issac B, Lieberman E, Giannoukos G,
Alvarez P, Brockman W, Kim TK, Koche RP, Lee W, Mendenhall E,
O'Donovan A, Presser A, Russ C, Xie X, Meissner A, Wernig M, Jaenisch
R, Nusbaum C, Lander ES, Bernstein BE. Genome-wide maps of chromatin
state in pluripotent and lineage-committed cells. Nature. 2007 Aug
2;448(7153):548-9. PMID: 17603471
</p>

<p>
	All Chip-Seq data was processed according to the following protocol: 
	The raw sequence reads were aligned to the Human GRCh37 genome assembly 
	using the bwa aligner software (Li H and Durbin R, 2010). All matches to 
	mitochondria were filtered out, and the resulting alignments were passed to 
	the SWEMBL peak caller software (S. Wilder et al, in preparation). The 
	resulting peaks were then filtered out to avoid problematic areas identified 
	in the ENCODE project. DNAse sets were called using less stringent parameters 
	to accommodate for its broader distribution. 
</p>

<P>* The CTCF Chip-ChIP data was processed with the Nessie HMM (Flicek,
unpublished). 
</P>
</BODY>
</HTML>
