<HTML>

<head>
<meta name="order" content="2" />
  <meta name="navigation" content="Regulation">
  <title>Regulation</title>
</head>


<BODY LANG="en-GB" DIR="LTR">
<H1>Regulation</H1>

<a href="/info/docs/funcgen/regulatory_build.html">Regulatory Build</a> |
<a href="/info/docs/funcgen/regulation_sources.html">Regulation Sources</a> |
<a href="/info/docs/funcgen/regulatory_segmentation.html">Regulatory Segmentation</a> |
<a href="/info/docs/funcgen/microarray_probe_set_mapping.html">Microarray Probeset Mapping</a> |
<a href="/info/docs/api/funcgen/index.html">Perl API</a> |
<a href="/info/docs/api/funcgen/funcgen_schema.html">Database Description</a>

<br /><br />

<h2 id="classes">Regulatory features</h2>

<p>
Ensembl stores regulatory regions for human and mouse, inferred from experimental data through the 
<a href="/info/docs/funcgen/regulatory_build.html">Regulatory Build</a> process. It also provides the
experimental evidence supporting the regulatory features, which is derived from publicly
available <a href="/info/docs/funcgen/regulation_sources.html">data sources</a>.
</p>

<br />

<h3 id="classes">Cell-specific regulatory feature classification</h3>

<p>
We provide cell-specific classifications for the regulatory features 
(see more details of this process <a href="regulatory_build.html#regannot">here</a>). 
</p>

<table id="regulation_classifications" class="ss" style="width:50%;">

<tr>
	<th>Regulatory feature classification</th>
	<th>Meaning</th>
</tr>
<tr>
	<td>promoter-associated</td>
	<td>The regulatory feature is likely to correspond to a proximal regulatory element such as a promoter.</td>
</tr>
<tr class="bg2">
	<td>non-gene-associated</td>
	<td>The regulatory feature is likely to correspond to a distal regulatory element such as an insulator or distal enhancer.</td>
</tr>
<tr>
	<td>gene-associated</td>
	<td>These regulatory features are associated to gene bodies. They often seem to represent marks of a gene's transcriptional activity (expressed/repressed gene)</td>
</tr>
<tr class="bg2">
	<td>Polymerase III-associated</td>
	<td>The regulatory feature is likely to correspond to a proximal regulatory element specifically associated to Polymerase III transcription.</td>
</tr>
</table>

<br/>

<h3 id="classes">Transcription Factor binding sites</h3>

We annotate regulatory features with putative Transcription Factor binding sites (TFBS). 
These are visualized in the website as black boxes within regulatory features.
Clicking on the black box will provide extra details on the TFBS, as the position weight matrix (PWM) and the matching score (relatively to the optimal site). 
More information on the TFBS annotation process can be seen <a href="regulatory_build.html#tfbs">here</a>.
Information regarding the PWMs used and their correspondence to transcription factors can be seen in the data source <a href="regulation_sources.html#tfgenehuman">documentation</a>.
<br />

<h2 id="classes">Regulatory Segmentation</h2>
<p>
Some summary text in here about the <a href="/info/docs/funcgen/regulatory_segmentation.html">regulatory genome segmentation.</a> 
</p>

<h2 id="classes">Other Regulatory data</h2>

<p>
Ensembl Regulation databases also store data directly imported from external sources (identified as <i>'Other regulatory regions'</i>):
<ul>
<li>Micro RNA target predictions for human and mouse (a stringent subset of MiRanda described <a rel="external" href='http://www.ebi.ac.uk/enright-srv/microcosm/htdocs/targets/v5/consset.html'>here</a>).</li>
<li>Experimentally validated human non-coding fragments from the <a rel="external" href='http://enhancer.lbl.gov/'>VISTA Enhancer Browser</a>.</li>
<li>Genome-wide regulatory module and element prediction from <a rel="external" href='http://www.cisred.org'>cisRED</a> for human and mouse.</li>
</ul>
</p>


<p>
Ensembl also provides access, via <a href='/info/data/ensembl_das.html'>DAS</a> tracks, to information regarding Human DNA Methylation. 
This includes MeDIP-Chip methylation datasets for 17 cell lines and tissues from <a rel="external" href='http://www.ncbi.nlm.nih.gov/pubmed?term=18577705'>Rakyan et al (2008)</a>, and 
RRBS (Reduced Representation Bisulfite Sequencing) data for 44 cell lines from <a rel="external" href='http://www.genome.gov/10005107'>ENCODE</a>. 
Replicates were merged by averaging the percentage of methylation in each replicate, but only taking positions with at least 20 reads in the combined set.
The methylation state is indicated by a color gradient which goes from dark blue, indicating highly methylated areas, through green and then towards yellow which indicates low methylation.
</p>
<br />

<h2 id="microarray">Microarray Probe Mappings</h2>

<p>
Ensembl stores <a href="/info/docs/funcgen/microarray_probe_set_mapping.html">microarray probe mappings</a> for several species [link to list of species and available arrays].
</p>

<br />
<h2 id="displays">Regulation displays</h2>
<p>
Regulation data can be viewed in the browser through pages such as:
</p>

<ul>
<li>
<strong>Gene:</strong> Regulatory Regions surrounding a gene (e.g. <a href="/Homo_sapiens/Gene/Regulation?db=core;g=ENSG00000159197">for all regulatory features around KCNE2</a>).
Note that even if a regulatory region is near or within a gene, this does not mean that it is acting on that gene.
</li>

<li><strong>Location:</strong> Region in Detail (by default 'MultiCell' regulatory features are displayed.
Cell-specific annotations and evidence tracks can also be drawn using &quot;Configure this page&quot; at the left. 
The menu allows display of information in Ensembl databases along with external sources in DAS format such as 
Methylation data.)</li>
</ul>

<p>
Clicking on any regulatory feature on an Ensembl page will open a Regulation tab with information about the 
evidence supporting the selected regulatory feature, as well as cell-specific annotations.
You can choose different views in the Regulation tab:

<ul>
<li>
The <i>'Details by cell line'</i> view is the default view, where you get the generic 'MultiCell' regulatory feature 
and cell-specific classifications, as well as part of their supporting evidence displayed as tracks. To display more evidence tracks, 
you can go to the <i>'Configure this page'</i> link and select tracks within the <i>'Regulation'</i> subsection.
'DNA Methylation' and external sources ('Other regulatory regions') are not used as evidence for regulatory regions.
</li>
<li>
The <i>'Summary'</i> view will display the 'MultiCell' regulatory feature, along with it's core evidence and cell-specific annotations.
By default, this view does not display cell-specific evidence.
</li>
<li>
The <i>'Feature Context'</i> view displays the regulatory features in a wider context around the chosen regulatory feature. 
</li>
<li>
The complete list of supporting core evidence (TF binding and open chromatin) for a regulatory feature can be obtained 
via the <i>'Evidence'</i> view. 
</li>
</ul>
</p>

<h3>Display configuration</h3>

Configuration of the various regulation displays is available under the 'Regulation' section, with the following subsections:

<ul>
<li><b>Regulatory Features</b> These are the output of the Regulatory Build. You can specifiy regulatory feature tracks for individual cell lines</li>
<li><b>Open Chromatin and TFBS</b> Here you can configure the evidence tracks used in the Regulatory Build to build regulatory features. </li>
<li><b>Histones and Polymerases</b> Here you can configure the evidence tracks used in the Regulatory Build to establish cell-specific boundaries for regulatory features. </li>
<li><b>DNA Methylation</b> Here you can configure the methylation tracks you wish to visualise. </li>
<li><b>Other regulatory regions</b> Here you can configure the external tracks you wish to visualise. </li>
</ul>

<p>
When diplaying the regulatory feature evidence, there are two visualisations available:<br/>
<ul>
<li><b>Peaks</b> - Shows significantly enriched 'peak' regions</li>
<li><b>Signal</b> - Shows a plot or wiggle representing the raw signal.</li>
</ul>
</p>

<p> 
Species-specific microarray probe mappings can be visualized on the Location view by turning on tracks from the
<i>'Oligo Probes'</i> section in the configuration panel.
</p>

<h2>Data Access</h2>

<p>
Regulation information stored in Ensembl databases can be accessed through the website, but also using <a href="/info/data/biomart.html"
class="cp-external">BioMart</a> (regulation database), the <a href="/info/docs/api/index.html" class="cp-external">Perl API</a> 
(<a href="/info/docs/api/funcgen/index.html">regulation API</a>) or by directly accessing the databases. 
More information about programmatic access is available in the <a href="/info/docs/api/funcgen/regulation_tutorial.html">regulation API tutorial</a> page.
A more detailed description of the regulation databases can be seen <a href='/info/docs/api/funcgen/funcgen_schema.html'>here</a>.
</p>


<P><BR><BR>
</P>
</BODY>
</HTML>
