
<html>
<head>
<meta name="order" content="5" />
  <title>Roadmap</title>
</head>

<body>

<h1>Ensembl Roadmap</h1>

<p>The following updates are planned for upcoming releases of Ensembl.</p>

<p><strong>Please note that we have no fixed timeline for most of these items</strong></p>

<h2>Gene annotation</h2>
<ul>
<li>Builds in progress: Human, Western clawed frog (<i>Xenopus tropicalis</i>), White-cheeked gibbon (<i>Nomascus leucogenys</i>)</li>
<li>New species expected: Sea lamprey (<i>Petromyzon marinus</i>), Atlantic cod (<i>Gadus morhua</i>), Cynomolgus macaque (<i>Macaca fascicularis</i>), Turquoise killifish (<i>Nothobranchius furzeri</i>), Tasmanian devil (<i>Sarcophilus harrisii</i>)</li>
<li>Updated assemblies expected for: mouse, rat, pig, microbat, bushbaby and gorilla</li>
<li>Incorporation of new non-reference sequence for human, with basic gene annotation</li>
<li>Incorporation of HAVANA manual annotation into the human gene set for every release, mouse gene set for every second release, and zebrafish gene set at least once a year</li>
<li>Tissue specific transcript models based on RNASeq data for human, based on the Illumina Human Body Map 2.0 data</li>
</ul>

<h2>Comparative Genomics</h2>
<ul>
<li>Bird multiple alignments</li>
<li>Extend the EPO multiple alignment pipeline to all vertebrates</li>
<li>Display super-protein tree</li>
<li>HMM-based classification of proteins for Families and Gene Trees</li>
<li>Annotate ohnologs</li>
<li>Display gene loss and gain events</li>
</ul>

<h2>Variation updates</h2>
<ul>
  <li>continue to import new variation data from dbSNP and DGVa where available </li>
  <li>improve variation annotation using data from the 1000 Genomes Project</li>
  <li> continue to import genome wide association study phenotypes for variants from the NHGRI catalog</li>
  <li>database schema changes to associate variation consequence is a specific variant alleles rather than just a locus </li>
  <li>incorporate sequence ontology terms in order to add an ontology to the variation consequences</li>
  <li>extend variation consequence pipeline to consider specific allele changes within regulatory regions </li>
  <li>
extend variation consequence pipeline to include predictions using SIFT and/or PolyPHEN</li>
</ul>

<h2>Core API and schema</h2>
<ul>
  <li>Support for alternative initiation</li>
  <li>Support for operons</li>
  <li>Refinements to stable ID mapping</li>
  <li>Improved support for assembly exceptions (including haplotypes and "patch" sequences)</li>
  <li>Support for alternative alleles</li>
  <li>Improved Perl API documentation</li>
  <li>Refinements to gene name and description assignment</li>
</ul>

<h2>Functional Genomics updates</h2>
<ul>
<!--
  <li>Incorporation of CD133+ and CD36+ Haematopoietic Stem Cell
  histone modification data into Ensembl under the current schema (Cui
  et al 2009).</li>
  <li>Incorporation of the first ENCODE histone modification, TFBS and DNase1 data into Ensembl under the current schema.</li>
  <li>Incorporation of at least one set of Epigenomics Roadmap project data of histone modification data for human ES cells.</li>
  <li>Improve documentation of eFG in the user interface, including
  updated API tutorial, and describe developments through the blog.</li>
  <li>Perform cell-specific regulatory builds</li>
  <li>Make (some) data visible in the region in detail view of the interface as wiggle type tracks.</li>
  <li>Development of the regulatory feature view</li>
 <li>Establishment of <i>de novo</i> PWM identification pipeline for TFs without
known binding motifs.</li>
  <li>Improved database storage of the relationships between ChIP-seq data,
TF, ensembl gene and PWM.</li>
  <li>Enabling regulatory build information to be used by the SNP
consequence predictor.</li>
<li>TF display within Regulatory displays including PWM mappings.</li>
  <li>Establishment of transcription factor (TF) binding site (position
weight matrix (PWM)) mapping pipeline for regulatory features. JASPAR
and other PWMs for TFs used in ChIP-seq datasets will be identified
within binding regions defined in the regulatory features.</li>
-->
 <li>Incorporating DNase1 footprinting and sequence conservation in to
TFBS identification</li>
 <li>Integrating RNA-seq data with regulatory element annotation</li>
 <li>Developing meta-annotation of Multi-cell regulatory elements.</li>
 <li>Development of data tracking database.</li>
  <li>Fine tuning of peak-calling parameters for histone modification
ChIP-seq, including investigation of alternate methods.</li>
  <li>Further development of uniform signal processing.</li>
  <li>Removal of collections data from the functional genomics database.</li>

  <li>eFG Tools:
  <ul>
    <li>Nearest gene</li>
    <li>PWM consequences</li>
  </ul>
  </li>
  <li>Web display developments:
  <ul>
    <li>Further refinements of wiggle track config/display including
  track highlighting.</li>
    <li>Data source display, including links to outside data sources for
ChIP-seq data, antibodies, PWMs, etc.</li>
  </ul>
  </li>
  <li>Incorporating ChIP-seq data from further species for possible
additional regulatory builds e.g. Schmidt et al (PMID: 20378774).
  </li>
</ul>

<h2>New web features</h2>
<ul>
<li>'Export image' buttons on karyotypes and other 'vertical' images</li>
<li>Ability to save user configurations</li>
<li>Ability to create a custom page from a selection of existing components</li>
</ul>

<h2>Biomart</h2>
<ul>
  <li>Test new interface</li>
  <li>Test new configurator tool</li>
  <li>Migrate to new building code and tools if testing is successful</li>
</ul>

</body>
</html>
