
<html>
<head>
<meta name="order" content="5" />
  <title>Roadmap</title>
</head>

<body>

<h1>Ensembl Roadmap</h1>

<p>The following updates are planned for upcoming releases of Ensembl.</p>

<p><strong>Please note that we have no fixed timeline for most of these items</strong></p>

<h2>Gene annotation</h2>
<ul>
<li>Builds in progress: duck, rabbit</li>
<li>New species expected: lamprey, gibbon, cod</li>
<li>Updated assemblies expected for mouse, rat, marmoset, bat and gorilla</li>
<li>Refinement of LincRNA prediction pipeline and extension to mouse</li>
<li>Incorporation of RNASeq based models in zebrafish gene set</li>
<li>Tissue specific transcript models based on RNASeq data</li>
<li>Improvements to selenocysteine containing gene predictions</li>
</ul>

<h2>Comparative Genomics</h2>
<ul>
<li>Fish multiple alignments</li>
<li>Bird multiple alignments</li>
<li>Homology predictions for ncRNAs</li>
<li>HMM-based classification of proteins for Families and Gene Trees</li>
</ul>

<h2> Variation updates</h2>
<ul>
  <li>import of malarial dataset from the EGA (European Genome-phenome Archive) </li>
  <li>update list of variants with UniProt annotation </li>
  <li>new import of variants with phenotype annotation from NHGRI </li>
  <li>new import of dbSNP 131 databases when available, including data from the 1000 Genomes project </li>
  <li>support for indels on the resequencing website page </li>
  <li>web displays based on LRG sequences </li>
  <li>karyotype display with all variant positions associated with a particular phenotype </li>
</ul>

<h2>Core API and schema</h2>
<ul>
  <li>Support for alternative initiation</li>
  <li>Support for operons</li>
  <li>Refinements to stable ID mapping</li>
  <li>Support for LRG (Locus Reference Genomic) sequences</li>
  <li>Improved support for assembly exceptions (including haplotypes and "patch" sequences)</li>
  <li>Refinements to gene name and description assignment</li>
</ul>

<h2>Biomart</h2>
<ul>
  <li>Test new interface</li>
  <li>Test new configurator tool</li>
  <li>Migrate to new building code and tools if testing is successful</li>
</ul>

<h2>eFG updates</h2>
<ul>
  <li>Incorporation of CD133+ and CD36+ Haematopoietic Stem Cell
  histone modification data into Ensembl under the current schema (Cui
  et al 2009).</li>
  <li>Incorporation of the first ENCODE histone modification, TFBS and DNase1 data into Ensembl under the current schema.</li>
  <li>Incorporation of at least one set of Epigenomics Roadmap project data of histone modification data for human ES cells.</li>
  <li>Improve documentation of eFG in the user interface, including
  updated API tutorial, and describe developments through the blog.</li>
  <li>Perform cell-specific regulatory builds</li>
  <li>Make (some) data visible in the region in detail view of the interface as wiggle type tracks.</li>
  <li>Development of the regulatory feature view</li>

<h2>New web features</h2>
<ul>
<li>'Export image' buttons on karyotypes and other 'vertical' images</li>
<li>Replacement for GOView, using our new GO schema and API</li>
<li>Ability to save user configurations</li>
<li>Ability to create a custom page from a selection of existing components</li>
</ul>

</body>
</html>
