
<h2>About the <i>Pongo pygmaeus abelii</i> genome</h2>

<h3>Assembly</h3>

<p><img src="/img/species/pic_Pongo_pygmaeus.png" height="100" width="108" class="float-left" alt="Orangutan" title="Ook!" />
This site presents the whole genome shotgun assembly from a female Sumatran 
orangutan named Susie, housed at the Gladys Porter Zoo (Brownsville, TX). 
The primary donor-derived reads were assembled using PCAP (Huang, 2006) 
using stringent parameters; by aligning the orangutan genome against the 
human genome, it was possible to identify interchromosomal cross-overs and
thus eliminate global mis-assemblies larger than 50kb. 
</p>
<p>
Of the 3.09Gb of total sequence, 3.08Gb are ordered and oriented along
the chromosomes. Gap sizes between supercontigs were estimated based on
their size in human, with a maximum allowed gap size of 30kb.
</p>

<p>The Orangutan genome has been released in pre-publication
 status from the Genome Sequencing Center from Washington
 University, St Louis. This is provided freely to be used
 by anyone, but they have requested that the scientific
 ethics of other groups publishing on this pre-publication
 data are respected. This is outlined in detail in the 
 <a href="http://www.wellcome.ac.uk/stellent/groups/corporatesite/@policy_communications/documents/web_document/wtd003207.pdf">Fort Lauradale
 agreement</a>; in brief, small scale analysis, eg,
 the analysis of a single locus is an expected use of
 the data which can be published on without any expectation
 of coordination. In contrast, large scale, genomewide analysis
 is expected to be either coordinated with the Orangutan
 Analysis group in some manner or published after the initial
 paper. More details on the reasoning for this and details
 are given in the Fort Lauradale document.</p>

<h3>Annotation</h3>
<p>
Due to the high sequence similarity to the human genome, the Orangutan 
genebuild was based on a projection of human gene structures 
(Ensembl Human build 36i). The projections were made through chained whole 
genome BLASTz alignments. These projected genes were combined with 
orangutan-specific proteins, and additional human genes were added using 
exonerate where the projection was unable to make satisfactory gene 
models. UTRs were added using orangutan-specific ESTs and cDNAs as well as 
human cDNAs.
</p>
