<html>
<head>
<title>Accessing Ensembl Data</title>
</head>

<body>

<h1>Accessing Ensembl Data</h1>

<p>Ensembl data is available through a number of routes - which you choose depends on
the amount and type of data you wish to fetch.</p>

<h2>Small quantities of data</h2>
<p>Many of the pages displaying Ensembl genomic data offer an export 
option, suitable for small amounts of data, e.g. a single gene sequence. <a href="export.html">More...</a></p>
<h2>Large datasets/complex analyses</h2>
<p>If you require larger amounts of data (e.g. all the genes on a chromosome) or are 
conducting a more detailed analysis, we recommend using our 
publicly-accessible MySQL server. <a href="mysql.html">More...</a></p>
<p>Alternatively, you can access a subset of Ensembl data using the DAS (Distributed Annotation System) 
protocol, via a simple URL. <a href="ensembl_das.html">More...</a></p>
<h2>Complex cross-database queries</h2>
<p>More complex datasets can be retrieved using the BioMart data-mining tool.
<a href="biomart.html">More...</a></p>
<h2>Whole databases</h2>
<p>If required, entire databases can be downloaded from our FTP site 
in a variety of formats, from flat files to MySQL dumps. <a href="ftp.html">More...</a></p>

</body>
</html>
