<html>
<head>
  <meta name="order" content="1" />
  <title>Data Description</title>
  <script type="text/javascript">
    window.onload = function() {
      $('.ht').helptip({'track': true});
      $('._ht').helptip({'track': true});
    };
  </script>
</head>

<body>

<h1 style="margin-top:15px">Ensembl Variation - Data description</h1>

<p>
  Below is a description of the data we store in the databases for Ensembl Variation.<br />For several different species in Ensembl, 
  we import variant data (<span class="ht" title="Single Nucleotide Polymorphisms">SNPs</span>, <span class="ht" title="Copy Number Variations">CNVs</span>, allele frequencies, genotypes, etc) from a variety of 
  <a href="sources_documentation.html">sources</a> (e.g. dbSNP). Imported variants and alleles are subjected to a <a href="#quality_control">quality control</a> process to flag suspect data.
  <br />We classify the variants into different
  <a href="#classes">classes</a> and calculate the <a href="predicted_data.html">predicted consequence(s)</a>
  of the variant and we have also created variant <a href="#variation_sets">sets</a> to help people retrieve a specific
  group of variants from a particular dataset.<br />
  In human, we calculate the linkage disequilibrium for each variant, by <a href="#populations">population</a>.
</p>
  
<p>See some examples of imported data on the Ensembl website (Human):</p>
<div class="portal" style="margin-bottom:5px;width:auto">
  <div>
    <a href="/Homo_sapiens/Variation/Context?v=rs1333049" title="Location of a variant in the genome"><img src="/i/96/var_genomic_context.png" alt="Location of a variant in the genome"/></a>
  </div>
  <div>
    <a href="/Homo_sapiens/Variation/Population?v=rs699" title="Population genotypes and frequencies of a variant"><img src="/i/96/var_population_genetics.png" alt="Population genotypes and frequencies of a variant"/></a>
  </div>
  <div>
    <a href="/Homo_sapiens/Variation/Sample?v=rs699" title="Sample genotypes of a variant"><img src="/i/96/var_sample_genotypes.png" alt="Sample genotypes of a variant"/></a>
  </div>
  <div>
    <a href="/Homo_sapiens/Variation/Phenotype?v=rs1333049" title="Phenotype(s) associated with a variant"><img src="/i/96/var_phenotype_data.png"alt="Phenotype(s) associated with a variant"/></a>
  </div>
  <div>
    <a href="/Homo_sapiens/Variation/Citations?v=rs1333049" title="Citations"><img src="/i/96/var_citations.png" alt="Citations"/></a>
  </div>

</div>
<div style="clear:both"></div>

<hr style="margin-bottom:4px" />

<h2 id="displays">Variant displays</h2>

<p>
Variant data can be viewed in the browser through pages such as:
</p>

<ul>
<li><b>Gene:</b> Variation Table and Variation Image (for all variants in a gene) 
e.g. <a href="/Homo_sapiens/Gene/Variation_Gene/Table?db=core;g=ENSG00000159197">for all variants in KCNE2</a>. 
Structural Variation to see all structural variants overlapping the gene.</li>
<li><b>Transcript:</b> Population comparison, Comparison
image (for comparing variants in a transcript across different individual or strain
sequences) e.g. <a href="/Mus_musculus/Transcript/Population/Image?t=ENSMUST00000050949"> compare Tmco4 in different mouse strains </a> </li>
<li><b>Transcript: </b> Sequence, protein: list of the coding variants in protein coordinates.</li>
<li><b>Location:</b> Region in Detail (Variants can be
drawn using &quot;Configure this page&quot; at the left. The menu allows
display of information in Ensembl databases along with external
sources in DAS format such as <a rel="external"
href="http://projects.tcag.ca/variation/">DGV loci</a>.)</li>
<li><b>Phenotype: </b> A karyotype view to display the variants associated with a certain phenotype, e.g. <a href="/Homo_sapiens/Phenotype/Locations?ph=5815">phenotype "Glaucoma"</a>.</li>
</ul>

<h3>Examples:</h3>
<div style="margin-bottom:25px;margin-left:15px">
  <div style="float:left;margin-right:30px;text-align:center">
    <a class="nodeco _ht _ht_track" href="/Homo_sapiens/Gene/Variation_Gene/Table?db=core;g=ENSG00000159197" title="Example of Gene Variation Table page"><img src="/i/96/gene.png" alt="" class="bordered"><div style="font-weight:bold">Gene</div><div>Variation Table</div></a>
  </div>
  <div style="float:left;margin-right:30px;text-align:center">
    <a class="nodeco _ht _ht_track" href="/Homo_sapiens/Transcript/Variation_Transcript/Table?db=core;t=ENST00000403559" title="Example of Transcript Variation Table page"><img src="/i/96/transcript.png" alt="" class="bordered"><div style="font-weight:bold">Transcript</div><div>Variation Table</div></a>
  </div>
  <div style="float:left;margin-right:30px;text-align:center">
    <a class="nodeco _ht _ht_track" href="/Homo_sapiens/Transcript/ProtVariations?db=core;t=ENST00000403559" title="Example of Protein Variation page"><img src="/i/96/transcript.png" alt="" class="bordered"><div style="font-weight:bold">Transcript</div><div>Protein Variation</div></a>
  </div>
  <div style="float:left;margin-right:30px;text-align:center">
    <a class="nodeco _ht _ht_track" href="/Homo_sapiens/Location/View?r=17:63973115-64437414" title="Example of 'Region in detail' page"><img src="/i/96/region.png" alt="" class="bordered"><div style="font-weight:bold">Location</div><div>Region in detail</div></a>
  </div>
  <div style="float:left;margin-right:30px;text-align:center">
    <a class="nodeco _ht _ht_track" href="/Homo_sapiens/Phenotype/Locations?ph=5815" title="Example of Phenotype page"><img src="/i/96/phenotype.png" alt="" class="bordered"><div style="font-weight:bold">Phenotype</div><div>Karyotype view</div></a>
  </div>
  <div style="clear:both"></div>
</div>

<p>
Clicking on any variant on an Ensembl page will open a Variation tab
with information about the flanking sequence and source for the
selected variant. Links to linkage disequilibrium (LD) plots,
phenotype information (for human) from EGA, OMIM and <a rel="external"
href="https://www.ebi.ac.uk/gwas/">NHGRI-EBI</a> and Ensembl genes and
transcripts that include the variant can be found at the left of
this tab. You may also view multiple genome alignments of various
species, highlighting the
variant. <a href="/info/genome/compara/analyses.html#ancestral"
class="cp-external">Ancestral sequences</a> are included in this
display. 
</p>

<p>
Variant information can also be accessed
using <a href="/info/data/biomart/"
class="cp-external">BioMart</a> (gene or variation database), and
the <a href="/info/docs/api/index.html" class="cp-external">Perl
API</a> (<a href="/info/docs/api/variation/index.html"
class="cp-external">variation API</a>).
</p>


<br />
<hr style="margin-bottom:4px" />
<h2 id="classes">Variant classes</h2>

<p>
We call the class of a variant according to its component alleles and its mapping 
to the reference genome, and then display this information on the website. Internally 
we use <a href="http://www.sequenceontology.org/">Sequence Ontology</a> terms, but we 
map these to our own 'display' terms where common usage differs from the SO 
definition (e.g. our term SNP is closer to the SO term SNV). All the classes we call, 
along with their equivalent SO term are shown in the table below. We also differentiate 
somatic mutations from germline variants in the display term, prefixing the term with
'somatic'. API users can fetch either the SO term or the display term.
</p>

<br />
<!-- Data classes - start -->

<table id="variation_classes" class="ss">
  <tr>
    <th style="width:8px;padding-left:0px;padding-right:0px;text-align:center">*</th>
    <th>SO term</th>
    <th>SO description</th>
    <th>SO accession</th>
    <th colspan="2">Called for (e.g.)</th>
  </tr>

  <tr>
    <td style="padding:0px;margin:0px;"></td>
    <td style="font-weight:bold">SNV</td>
    <td>SNVs are single nucleotide positions in genomic DNA at which different sequence alternatives exist.</td>
    <td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001483">SO:0001483</a></td>
    <td>
      <ul style="margin:0px;padding-left:1em">
        <li style="margin:0px">Variant</li>
      </ul>
    </td>
    <td style="padding-left:0px;width:16px"><div><a href="/Homo_sapiens/Variation/Explore?v=rs3" target="_blank" title="See a variant example"><img src="/i/16/internal_link.png" alt="Link"/></a></div></td>
  </tr>
  <tr class="bg2">
    <td style="padding:0px;margin:0px;"></td>
    <td style="font-weight:bold">genetic_marker</td>
    <td>A measurable sequence feature that varies within a population.</td>
    <td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001645">SO:0001645</a></td>
    <td>
      <ul style="margin:0px;padding-left:1em">
        <li style="margin:0px">Variant</li>
      </ul>
    </td>
    <td style="padding-left:0px;width:16px"><div><a href="/Homo_sapiens/Variation/Explore?v=rs111226187" target="_blank" title="See a variant example"><img src="/i/16/internal_link.png" alt="Link"/></a></div></td>
  </tr>
  <tr>
    <td style="padding:0px;margin:0px;"></td>
    <td style="font-weight:bold">substitution</td>
    <td>A sequence alteration where the length of the change in the variant is the same as that of the reference.</td>
    <td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:1000002">SO:1000002</a></td>
    <td>
      <ul style="margin:0px;padding-left:1em">
        <li style="margin:0px">Variant</li>
      </ul>
    </td>
    <td style="padding-left:0px;width:16px"><div><a href="/Homo_sapiens/Variation/Explore?v=rs1800995" target="_blank" title="See a variant example"><img src="/i/16/internal_link.png" alt="Link"/></a></div></td>
  </tr>
  <tr class="bg2">
    <td style="padding:0px;margin:0px;"></td>
    <td style="font-weight:bold">tandem_repeat</td>
    <td>Two or more adjacent copies of a region (of length greater than 1).</td>
    <td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0000705">SO:0000705</a></td>
    <td>
      <ul style="margin:0px;padding-left:1em">
        <li style="margin:0px">Variant</li>
      </ul>
    </td>
    <td style="padding-left:0px;width:16px"><div><a href="/Homo_sapiens/Variation/Explore?v=rs1805180" target="_blank" title="See a variant example"><img src="/i/16/internal_link.png" alt="Link"/></a></div></td>
  </tr>
  <tr>
    <td style="padding:0px;margin:0px;;background-color:#000080;border-top:1px solid #FFF"></td>
    <td style="font-weight:bold">Alu_insertion</td>
    <td>An insertion of sequence from the Alu family of mobile elements.</td>
    <td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0002063">SO:0002063</a></td>
    <td>
      <ul style="margin:0px;padding-left:1em">
        <li style="margin:0px"><span class="_ht ht" title="Structural variant">SV</span></li>
      </ul>
    </td>
    <td style="padding-left:0px;width:16px"><div><a href="/Homo_sapiens/StructuralVariation/Explore?sv=nsv1162620" target="_blank" title="See a structural variant example"><img src="/i/16/internal_link.png" alt="Link"/></a></div></td>
  </tr>
  <tr class="bg2">
    <td style="padding:0px;margin:0px;;background-color:#007FFF;border-top:1px solid #FFF"></td>
    <td style="font-weight:bold">complex_structural_alteration</td>
    <td>A structural sequence alteration or rearrangement encompassing one or more genome fragments, with 4 or more breakpoints.</td>
    <td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001784">SO:0001784</a></td>
    <td>
      <ul style="margin:0px;padding-left:1em">
        <li style="margin:0px"><span class="_ht ht" title="Structural variant">SV</span></li>
      </ul>
    </td>
    <td style="padding-left:0px;width:16px"><div><a href="/Homo_sapiens/StructuralVariation/Explore?sv=esv275549" target="_blank" title="See a structural variant example"><img src="/i/16/internal_link.png" alt="Link"/></a></div></td>
  </tr>
  <tr>
    <td style="padding:0px;margin:0px;;background-color:#007FFF;border-top:1px solid #FFF"></td>
    <td style="font-weight:bold">complex_substitution</td>
    <td>When no simple or well defined DNA mutation event describes the observed DNA change, the keyword \"complex\" should be used. Usually there are multiple equally plausible explanations for the change.</td>
    <td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:1000005">SO:1000005</a></td>
    <td>
      <ul style="margin:0px;padding-left:1em">
        <li style="margin:0px"><span class="_ht ht" title="Structural variant">SV</span></li>
      </ul>
    </td>
    <td style="padding-left:0px;width:16px"><div><a href="/Homo_sapiens/StructuralVariation/Explore?sv=nsv511117" target="_blank" title="See a structural variant example"><img src="/i/16/internal_link.png" alt="Link"/></a></div></td>
  </tr>
  <tr class="bg2">
    <td style="padding:0px;margin:0px;;background-color:#0000CC;border-top:1px solid #FFF"></td>
    <td style="font-weight:bold">copy_number_gain</td>
    <td>A sequence alteration whereby the copy number of a given regions is greater than the reference sequence.</td>
    <td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001742">SO:0001742</a></td>
    <td>
      <ul style="margin:0px;padding-left:1em">
        <li style="margin:0px"><span class="_ht ht" title="Structural variant">SV</span></li>
      </ul>
    </td>
    <td style="padding-left:0px;width:16px"><div><a href="/Homo_sapiens/StructuralVariation/Explore?sv=esv1004360" target="_blank" title="See a structural variant example"><img src="/i/16/internal_link.png" alt="Link"/></a></div></td>
  </tr>
  <tr>
    <td style="padding:0px;margin:0px;;background-color:#CC0000;border-top:1px solid #FFF"></td>
    <td style="font-weight:bold">copy_number_loss</td>
    <td>A sequence alteration whereby the copy number of a given region is less than the reference sequence.</td>
    <td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001743">SO:0001743</a></td>
    <td>
      <ul style="margin:0px;padding-left:1em">
        <li style="margin:0px"><span class="_ht ht" title="Structural variant">SV</span></li>
      </ul>
    </td>
    <td style="padding-left:0px;width:16px"><div><a href="/Homo_sapiens/StructuralVariation/Explore?sv=esv991724" target="_blank" title="See a structural variant example"><img src="/i/16/internal_link.png" alt="Link"/></a></div></td>
  </tr>
  <tr class="bg2">
    <td style="padding:0px;margin:0px;;background-color:#8601AF;border-top:1px solid #FFF"></td>
    <td style="font-weight:bold">copy_number_variation</td>
    <td>A variation that increases or decreases the copy number of a given region.</td>
    <td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001019">SO:0001019</a></td>
    <td>
      <ul style="margin:0px;padding-left:1em">
        <li style="margin:0px"><span class="_ht ht" title="Structural variant">SV</span></li>
      </ul>
    </td>
    <td style="padding-left:0px;width:16px"><div><a href="/Homo_sapiens/StructuralVariation/Explore?sv=esv275461" target="_blank" title="See a structural variant example"><img src="/i/16/internal_link.png" alt="Link"/></a></div></td>
  </tr>
  <tr>
    <td style="padding:0px;margin:0px;;background-color:#0000CC;border-top:1px solid #FFF"></td>
    <td style="font-weight:bold">duplication</td>
    <td>An insertion which derives from, or is identical in sequence to, nucleotides present at a known location in the genome.</td>
    <td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:1000035">SO:1000035</a></td>
    <td>
      <ul style="margin:0px;padding-left:1em">
        <li style="margin:0px"><span class="_ht ht" title="Structural variant">SV</span></li>
      </ul>
    </td>
    <td style="padding-left:0px;width:16px"><div><a href="/Homo_sapiens/StructuralVariation/Explore?sv=esv2830426" target="_blank" title="See a structural variant example"><img src="/i/16/internal_link.png" alt="Link"/></a></div></td>
  </tr>
  <tr class="bg2">
    <td style="padding:0px;margin:0px;;background-color:#000000;border-top:1px solid #FFF"></td>
    <td style="font-weight:bold">interchromosomal_breakpoint</td>
    <td>A rearrangement breakpoint between two different chromosomes.</td>
    <td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001873">SO:0001873</a></td>
    <td>
      <ul style="margin:0px;padding-left:1em">
        <li style="margin:0px"><span class="_ht ht" title="Structural variant">SV</span></li>
      </ul>
    </td>
    <td style="padding-left:0px;width:16px"><div><a href="/Homo_sapiens/StructuralVariation/Explore?sv=esv2713382" target="_blank" title="See a structural variant example"><img src="/i/16/internal_link.png" alt="Link"/></a></div></td>
  </tr>
  <tr>
    <td style="padding:0px;margin:0px;;background-color:#C3A4FF;border-top:1px solid #FFF"></td>
    <td style="font-weight:bold">interchromosomal_translocation</td>
    <td>A translocation where the regions involved are from different chromosomes.</td>
    <td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0002060">SO:0002060</a></td>
    <td>
      <ul style="margin:0px;padding-left:1em">
        <li style="margin:0px"><span class="_ht ht" title="Structural variant">SV</span></li>
      </ul>
    </td>
    <td style="padding-left:0px;width:16px"><div><a href="/Homo_sapiens/StructuralVariation/Explore?sv=nsv1150170" target="_blank" title="See a structural variant example"><img src="/i/16/internal_link.png" alt="Link"/></a></div></td>
  </tr>
  <tr class="bg2">
    <td style="padding:0px;margin:0px;;background-color:#000000;border-top:1px solid #FFF"></td>
    <td style="font-weight:bold">intrachromosomal_breakpoint</td>
    <td>A rearrangement breakpoint within the same chromosome.</td>
    <td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001874">SO:0001874</a></td>
    <td>
      <ul style="margin:0px;padding-left:1em">
        <li style="margin:0px"><span class="_ht ht" title="Structural variant">SV</span></li>
      </ul>
    </td>
    <td style="padding-left:0px;width:16px"><div><a href="/Homo_sapiens/StructuralVariation/Explore?sv=esv1914887" target="_blank" title="See a structural variant example"><img src="/i/16/internal_link.png" alt="Link"/></a></div></td>
  </tr>
  <tr>
    <td style="padding:0px;margin:0px;;background-color:#C3A4FF;border-top:1px solid #FFF"></td>
    <td style="font-weight:bold">intrachromosomal_translocation</td>
    <td>A translocation where the regions involved are from the same chromosome.</td>
    <td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0002061">SO:0002061</a></td>
    <td>
      <ul style="margin:0px;padding-left:1em">
        <li style="margin:0px"><span class="_ht ht" title="Structural variant">SV</span></li>
      </ul>
    </td>
    <td style="padding-left:0px;width:16px"><div><a href="/Homo_sapiens/StructuralVariation/Explore?sv=nsv1152656" target="_blank" title="See a structural variant example"><img src="/i/16/internal_link.png" alt="Link"/></a></div></td>
  </tr>
  <tr class="bg2">
    <td style="padding:0px;margin:0px;;background-color:#C530FF;border-top:1px solid #FFF"></td>
    <td style="font-weight:bold">inversion</td>
    <td>A continuous nucleotide sequence is inverted in the same position.</td>
    <td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:1000036">SO:1000036</a></td>
    <td>
      <ul style="margin:0px;padding-left:1em">
        <li style="margin:0px"><span class="_ht ht" title="Structural variant">SV</span></li>
      </ul>
    </td>
    <td style="padding-left:0px;width:16px"><div><a href="/Homo_sapiens/StructuralVariation/Explore?sv=esv996586" target="_blank" title="See a structural variant example"><img src="/i/16/internal_link.png" alt="Link"/></a></div></td>
  </tr>
  <tr>
    <td style="padding:0px;margin:0px;;background-color:#8601AF;border-top:1px solid #FFF"></td>
    <td style="font-weight:bold">loss_of_heterozygosity</td>
    <td>A functional variant whereby the sequence alteration causes a loss of function of one allele of a gene.</td>
    <td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001786">SO:0001786</a></td>
    <td>
      <ul style="margin:0px;padding-left:1em">
        <li style="margin:0px"><span class="_ht ht" title="Structural variant">SV</span></li>
      </ul>
    </td>
    <td style="padding-left:0px;width:16px"><div><a href="/Homo_sapiens/StructuralVariation/Explore?sv=nsv2768232" target="_blank" title="See a structural variant example"><img src="/i/16/internal_link.png" alt="Link"/></a></div></td>
  </tr>
  <tr class="bg2">
    <td style="padding:0px;margin:0px;;background-color:#CC0000;border-top:1px solid #FFF"></td>
    <td style="font-weight:bold">mobile_element_deletion</td>
    <td>A deletion of a mobile element when comparing a reference sequence (has mobile element) to a individual sequence (does not have mobile element).</td>
    <td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0002066">SO:0002066</a></td>
    <td>
      <ul style="margin:0px;padding-left:1em">
        <li style="margin:0px"><span class="_ht ht" title="Structural variant">SV</span></li>
      </ul>
    </td>
    <td style="padding-left:0px;width:16px"><div><a href="/Homo_sapiens/StructuralVariation/Explore?sv=nsv3077686" target="_blank" title="See a structural variant example"><img src="/i/16/internal_link.png" alt="Link"/></a></div></td>
  </tr>
  <tr>
    <td style="padding:0px;margin:0px;;background-color:#000080;border-top:1px solid #FFF"></td>
    <td style="font-weight:bold">mobile_element_insertion</td>
    <td>A kind of insertion where the inserted sequence is a mobile element.</td>
    <td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001837">SO:0001837</a></td>
    <td>
      <ul style="margin:0px;padding-left:1em">
        <li style="margin:0px"><span class="_ht ht" title="Structural variant">SV</span></li>
      </ul>
    </td>
    <td style="padding-left:0px;width:16px"><div><a href="/Homo_sapiens/StructuralVariation/Explore?sv=esv3309266" target="_blank" title="See a structural variant example"><img src="/i/16/internal_link.png" alt="Link"/></a></div></td>
  </tr>
  <tr class="bg2">
    <td style="padding:0px;margin:0px;;background-color:#000080;border-top:1px solid #FFF"></td>
    <td style="font-weight:bold">novel_sequence_insertion</td>
    <td>An insertion the sequence of which cannot be mapped to the reference genome.</td>
    <td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001838">SO:0001838</a></td>
    <td>
      <ul style="margin:0px;padding-left:1em">
        <li style="margin:0px"><span class="_ht ht" title="Structural variant">SV</span></li>
      </ul>
    </td>
    <td style="padding-left:0px;width:16px"><div><a href="/Homo_sapiens/StructuralVariation/Explore?sv=esv3310477" target="_blank" title="See a structural variant example"><img src="/i/16/internal_link.png" alt="Link"/></a></div></td>
  </tr>
  <tr>
    <td style="padding:0px;margin:0px;;background-color:#732E00;border-top:1px solid #FFF"></td>
    <td style="font-weight:bold">short_tandem_repeat_variation</td>
    <td>A variation that expands or contracts a tandem repeat with regard to a reference.</td>
    <td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0002096">SO:0002096</a></td>
    <td>
      <ul style="margin:0px;padding-left:1em">
        <li style="margin:0px"><span class="_ht ht" title="Structural variant">SV</span></li>
      </ul>
    </td>
    <td style="padding-left:0px;width:16px"><div><a href="/Homo_sapiens/StructuralVariation/Explore?sv=nsv2726918" target="_blank" title="See a structural variant example"><img src="/i/16/internal_link.png" alt="Link"/></a></div></td>
  </tr>
  <tr class="bg2">
    <td style="padding:0px;margin:0px;;background-color:#732E00;border-top:1px solid #FFF"></td>
    <td style="font-weight:bold">tandem_duplication</td>
    <td>A duplication consisting of 2 identical adjacent regions.</td>
    <td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:1000173">SO:1000173</a></td>
    <td>
      <ul style="margin:0px;padding-left:1em">
        <li style="margin:0px"><span class="_ht ht" title="Structural variant">SV</span></li>
      </ul>
    </td>
    <td style="padding-left:0px;width:16px"><div><a href="/Homo_sapiens/StructuralVariation/Explore?sv=esv3302857" target="_blank" title="See a structural variant example"><img src="/i/16/internal_link.png" alt="Link"/></a></div></td>
  </tr>
  <tr>
    <td style="padding:0px;margin:0px;;background-color:#C3A4FF;border-top:1px solid #FFF"></td>
    <td style="font-weight:bold">translocation</td>
    <td>A region of nucleotide sequence that has translocated to a new position. The observed adjacency of two previously separated regions.</td>
    <td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0000199">SO:0000199</a></td>
    <td>
      <ul style="margin:0px;padding-left:1em">
        <li style="margin:0px"><span class="_ht ht" title="Structural variant">SV</span></li>
      </ul>
    </td>
    <td style="padding-left:0px;width:16px"><div><a href="/Homo_sapiens/StructuralVariation/Explore?sv=nsv1157267" target="_blank" title="See a structural variant example"><img src="/i/16/internal_link.png" alt="Link"/></a></div></td>
  </tr>
  <tr class="bg2">
    <td style="padding:0px;margin:0px;;background-color:#CC0000;border-top:1px solid #FFF"></td>
    <td style="font-weight:bold">deletion</td>
    <td>The point at which one or more contiguous nucleotides were excised.</td>
    <td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0000159">SO:0000159</a></td>
    <td>
      <ul style="margin:0px;padding-left:1em">
        <li style="margin:0px">Variant</li><li style="margin:4px 0px 0px"><span class="_ht ht" title="Structural variant">SV</span></li>
      </ul>
    </td>
    <td style="padding-left:0px;width:16px"><div><a href="/Homo_sapiens/Variation/Explore?v=rs9" target="_blank" title="See a variant example"><img src="/i/16/internal_link.png" alt="Link"/></a></div><div style="padding-top:1px"><a href="/Homo_sapiens/StructuralVariation/Explore?sv=esv989436" target="_blank" title="See a structural variant example"><img src="/i/16/internal_link.png" alt="Link"/></a></div></td>
  </tr>
  <tr>
    <td style="padding:0px;margin:0px;;background-color:gold;border-top:1px solid #FFF"></td>
    <td style="font-weight:bold">indel</td>
    <td>A sequence alteration which included an insertion and a deletion, affecting 2 or more bases.</td>
    <td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:1000032">SO:1000032</a></td>
    <td>
      <ul style="margin:0px;padding-left:1em">
        <li style="margin:0px">Variant</li><li style="margin:4px 0px 0px"><span class="_ht ht" title="Structural variant">SV</span></li>
      </ul>
    </td>
    <td style="padding-left:0px;width:16px"><div><a href="/Homo_sapiens/Variation/Explore?v=rs3906" target="_blank" title="See a variant example"><img src="/i/16/internal_link.png" alt="Link"/></a></div><div style="padding-top:1px"><a href="/Homo_sapiens/StructuralVariation/Explore?sv=esv3627597" target="_blank" title="See a structural variant example"><img src="/i/16/internal_link.png" alt="Link"/></a></div></td>
  </tr>
  <tr class="bg2">
    <td style="padding:0px;margin:0px;;background-color:#000099;border-top:1px solid #FFF"></td>
    <td style="font-weight:bold">insertion</td>
    <td>The sequence of one or more nucleotides added between two adjacent nucleotides in the sequence.</td>
    <td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0000667">SO:0000667</a></td>
    <td>
      <ul style="margin:0px;padding-left:1em">
        <li style="margin:0px">Variant</li><li style="margin:4px 0px 0px"><span class="_ht ht" title="Structural variant">SV</span></li>
      </ul>
    </td>
    <td style="padding-left:0px;width:16px"><div><a href="/Homo_sapiens/Variation/Explore?v=rs109" target="_blank" title="See a variant example"><img src="/i/16/internal_link.png" alt="Link"/></a></div><div style="padding-top:1px"><a href="/Homo_sapiens/StructuralVariation/Explore?sv=esv991325" target="_blank" title="See a structural variant example"><img src="/i/16/internal_link.png" alt="Link"/></a></div></td>
  </tr>
  <tr>
    <td style="padding:0px;margin:0px;;background-color:#000000;border-top:1px solid #FFF"></td>
    <td style="font-weight:bold">sequence_alteration</td>
    <td>A sequence_alteration is a sequence_feature whose extent is the deviation from another sequence.</td>
    <td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001059">SO:0001059</a></td>
    <td>
      <ul style="margin:0px;padding-left:1em">
        <li style="margin:0px">Variant</li><li style="margin:4px 0px 0px"><span class="_ht ht" title="Structural variant">SV</span></li>
      </ul>
    </td>
    <td style="padding-left:0px;width:16px"><div><a href="/Homo_sapiens/Variation/Explore?v=rs4340" target="_blank" title="See a variant example"><img src="/i/16/internal_link.png" alt="Link"/></a></div><div style="padding-top:1px"><a href="/Homo_sapiens/StructuralVariation/Explore?sv=esv24160" target="_blank" title="See a structural variant example"><img src="/i/16/internal_link.png" alt="Link"/></a></div></td>
  </tr>
  <tr class="bg2">
    <td style="padding:0px;margin:0px;;background-color:#4682B4;border-top:1px solid #FFF"></td>
    <td style="font-weight:bold">probe</td>
    <td>A DNA sequence used experimentally to detect the presence or absence of a complementary nucleic acid.</td>
    <td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0000051">SO:0000051</a></td>
    <td>
      <ul style="margin:0px;padding-left:1em">
        <li style="margin:0px">CNV probe</li>
      </ul>
    </td>
    <td style="padding-left:0px;width:16px"><div><a href="/Homo_sapiens/StructuralVariation/Explore?sv=cnvi0111251" target="_blank" title="See a structural variant example"><img src="/i/16/internal_link.png" alt="Link"/></a></div></td>
  </tr></table>

<p>
<b>*</b> Corresponding colours for the Ensembl web displays (only for Structural variants). 
The colours were originally based on the <a rel="external" href="http://www.ncbi.nlm.nih.gov/dbvar/content/overview/">dbVar</a> displays.
<p>

<div class="js_panel" id="pie_chart_panel" style="margin-top:15px;margin-bottom:20px">
  <input class="panel_type" type="hidden" value="Piechart">
  <input class="graph_config" type="hidden" name="legendpos" value="'east'" />
  <input class="graph_dimensions" type="hidden" value="[90,80,80]" />
  
  <input type="hidden" class="graph_data" value="[[309685982,'SNV'],[11908423,'deletion'],[7019011,'insertion'],[847848,'sequence_alteration'],[228404,'substitution'],[178101,'indel'],[6808,'genetic_marker'],[4623,'tandem_repeat']]" />
  
  <div class="pie_chart_classes" title="classes" style="width:400px;height:220px;border:1px solid #CCC;border-radius:8px;box-shadow:0 1px 3px #666">
     <h3 style="padding:10px 10px 5px">Human variant class distribution - Ensembl 92</h4>
     <div id="graphHolder0"></div>
  </div>
</div>
  <!-- Data classes - end -->
 
 
<br />
<hr style="margin-bottom:4px" />
<h2 id="indel">Insertion and Deletion coordinates </h2>

<p> In Ensembl, an insertion is indicated by start coordinate = end coordinate +
1. For example, an insertion of 'C' between nucleotides 12600 and
12601 on the forward strand is indicated with start and end coordinates as
follows:</p>

<pre>
   12601     12600   
</pre>
<p>A deletion is indicated by the exact nucleotide coordinates. For
example, a three base pair deletion of nucleotides 12600, 12601, and
12602 of the reverse strand will have start and end coordinates of :</p>

<pre>
   12600     12602    
</pre>

<br />
<hr style="margin-bottom:4px" />
<h2 id="maf">Minor Alleles</h2>

<p>
Minor alleles and their frequencies are available for variants discovered in the 1000 Genomes Project.<br />
These are calculated by dbSNP. If there are more than <b>two</b> alleles, the <b>second most common</b> is reported 
(See example: <a href="/Homo_sapiens/Variation/Population?v=rs200077393">rs200077393</a>).<br />
This allows <b>common variants</b> to be distinguished from <b>rare variants</b> in situations where deep sequencing 
has identified a third rare allele. 
</p> 

<br />
<hr style="margin-bottom:4px" />
<h2 id="populations">Populations</h2>

<p>For each variant we store population data (allele
frequencies, genotypes) available from different projects (e.g. 1000
Genomes, HapMap). You can see this on our variation pages, for
example: <a href="/Homo_sapiens/Variation/Population?v=rs699">rs699</a>.
Frequencies displayed (to 3 decimal places) may not add up to 1 due to rounding.
<br />
Below we list the population names associated with the main genotyping projects available in Ensembl Variation:
</p>

<!-- Data populations - start -->

  <ul style="padding-left:1em;margin-left:2px">
    <li><h3>Populations from the <a href="http://www.1000genomes.org" target="_blank" style="text-decoration:none">1000 Genomes Project</a> (<i>Homo sapiens</i>)</h3></li>
  </ul>
  <table id="1000_genomes_project" class="ss" style="margin-bottom:4px">
  
  <tr>
    <th style="width:200px">Name</th>
    <th style="width:36px">Size</th>
    <th>Description</th>
  </tr>

  <tr class="supergroup">
    <td>1000GENOMES:phase_3:<b>ALL</b></td>
    <td style="text-align:right">2504</td>
    <td>All phase 3 individuals</td>
  </tr>
  <tr class="subgroup">
    <td>1000GENOMES:phase_3:<b>AFR</b></td>
    <td style="text-align:right">661</td>
    <td>African</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>ACB</b></li></ul></td>
    <td style="text-align:right">96</td>
    <td>African Caribbean in Barbados</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>ASW</b></li></ul></td>
    <td style="text-align:right">61</td>
    <td>African Ancestry in Southwest US</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>ESN</b></li></ul></td>
    <td style="text-align:right">99</td>
    <td>Esan in Nigeria</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>GWD</b></li></ul></td>
    <td style="text-align:right">113</td>
    <td>Gambian in Western Division, The Gambia</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>LWK</b></li></ul></td>
    <td style="text-align:right">99</td>
    <td>Luhya in Webuye, Kenya</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>MSL</b></li></ul></td>
    <td style="text-align:right">85</td>
    <td>Mende in Sierra Leone</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>YRI</b></li></ul></td>
    <td style="text-align:right">108</td>
    <td>Yoruba in Ibadan, Nigeria</td>
  </tr>
  <tr class="subgroup">
    <td>1000GENOMES:phase_3:<b>AMR</b></td>
    <td style="text-align:right">347</td>
    <td>American</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>CLM</b></li></ul></td>
    <td style="text-align:right">94</td>
    <td>Colombian in Medellin, Colombia</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>MXL</b></li></ul></td>
    <td style="text-align:right">64</td>
    <td>Mexican Ancestry in Los Angeles, California</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>PEL</b></li></ul></td>
    <td style="text-align:right">85</td>
    <td>Peruvian in Lima, Peru</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>PUR</b></li></ul></td>
    <td style="text-align:right">104</td>
    <td>Puerto Rican in Puerto Rico</td>
  </tr>
  <tr class="subgroup">
    <td>1000GENOMES:phase_3:<b>EAS</b></td>
    <td style="text-align:right">504</td>
    <td>East Asian</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>CDX</b></li></ul></td>
    <td style="text-align:right">93</td>
    <td>Chinese Dai in Xishuangbanna, China</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>CHB</b></li></ul></td>
    <td style="text-align:right">103</td>
    <td>Han Chinese in Bejing, China</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>CHS</b></li></ul></td>
    <td style="text-align:right">105</td>
    <td>Southern Han Chinese, China</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>JPT</b></li></ul></td>
    <td style="text-align:right">104</td>
    <td>Japanese in Tokyo, Japan</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>KHV</b></li></ul></td>
    <td style="text-align:right">99</td>
    <td>Kinh in Ho Chi Minh City, Vietnam</td>
  </tr>
  <tr class="subgroup">
    <td>1000GENOMES:phase_3:<b>EUR</b></td>
    <td style="text-align:right">503</td>
    <td>European</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>CEU</b></li></ul></td>
    <td style="text-align:right">99</td>
    <td>Utah residents with Northern and Western European ancestry</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>FIN</b></li></ul></td>
    <td style="text-align:right">99</td>
    <td>Finnish in Finland</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>GBR</b></li></ul></td>
    <td style="text-align:right">91</td>
    <td>British in England and Scotland</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>IBS</b></li></ul></td>
    <td style="text-align:right">107</td>
    <td>Iberian populations in Spain</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>TSI</b></li></ul></td>
    <td style="text-align:right">107</td>
    <td>Toscani in Italy</td>
  </tr>
  <tr class="subgroup">
    <td>1000GENOMES:phase_3:<b>SAS</b></td>
    <td style="text-align:right">489</td>
    <td>South Asian</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>BEB</b></li></ul></td>
    <td style="text-align:right">86</td>
    <td>Bengali in Bangladesh</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>GIH</b></li></ul></td>
    <td style="text-align:right">103</td>
    <td>Gujarati Indian in Houston, TX</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>ITU</b></li></ul></td>
    <td style="text-align:right">102</td>
    <td>Indian Telugu in the UK</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>PJL</b></li></ul></td>
    <td style="text-align:right">96</td>
    <td>Punjabi in Lahore, Pakistan</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>STU</b></li></ul></td>
    <td style="text-align:right">102</td>
    <td>Sri Lankan Tamil in the UK</td>
  </tr>
</table>


<p style="margin-bottom:35px">
Variants which have been discovered in this project have the "evidence status" <a href="#evidence_status"><b>1000Genomes</b></a>.
On the website this corresponds to the icon <a href="#evidence_status"><img class="_ht" src="/i/val/evidence_1000Genomes.png" title="1000Genomes" style="vertical-align:bottom"/></a>.
</p>
    
  <ul style="padding-left:1em;margin-left:2px">
    <li><h3>Populations from the <a href="http://gnomad.broadinstitute.org/" target="_blank" style="text-decoration:none">gnomAD exomes</a> (<i>Homo sapiens</i>)</h3></li>
  </ul>
  <table id="gnomad_exomes" class="ss" style="margin-bottom:4px">
  
  <tr>
    <th style="width:200px">Name</th>
    <th style="width:36px">Size</th>
    <th>Description</th>
  </tr>

  <tr>
    <td>gnomADe:<b>ALL</b></td>
    <td style="text-align:right">-</td>
    <td>All gnomAD exomes individuals</td>
  </tr>
  <tr class="bg2">
    <td>gnomADe:<b>AFR</b></td>
    <td style="text-align:right">-</td>
    <td>African/African American</td>
  </tr>
  <tr>
    <td>gnomADe:<b>AMR</b></td>
    <td style="text-align:right">-</td>
    <td>Latino</td>
  </tr>
  <tr class="bg2">
    <td>gnomADe:<b>ASJ</b></td>
    <td style="text-align:right">-</td>
    <td>Ashkenazi Jewish</td>
  </tr>
  <tr>
    <td>gnomADe:<b>EAS</b></td>
    <td style="text-align:right">-</td>
    <td>East Asian</td>
  </tr>
  <tr class="bg2">
    <td>gnomADe:<b>FIN</b></td>
    <td style="text-align:right">-</td>
    <td>Finnish</td>
  </tr>
  <tr>
    <td>gnomADe:<b>NFE</b></td>
    <td style="text-align:right">-</td>
    <td>Non-Finnish European</td>
  </tr>
  <tr class="bg2">
    <td>gnomADe:<b>OTH</b></td>
    <td style="text-align:right">-</td>
    <td>Other</td>
  </tr>
  <tr>
    <td>gnomADe:<b>SAS</b></td>
    <td style="text-align:right">-</td>
    <td>South Asian</td>
  </tr>
</table>


<div style="margin-bottom:35px"></div>

  <ul style="padding-left:1em;margin-left:2px">
    <li><h3>Populations from the <a href="http://gnomad.broadinstitute.org/" target="_blank" style="text-decoration:none">gnomAD genomes</a> (<i>Homo sapiens</i>)</h3></li>
  </ul>
  <table id="gnomad_genomes" class="ss" style="margin-bottom:4px">
  
  <tr>
    <th style="width:200px">Name</th>
    <th style="width:36px">Size</th>
    <th>Description</th>
  </tr>

  <tr>
    <td>gnomADg:<b>ALL</b></td>
    <td style="text-align:right">-</td>
    <td>All gnomAD genomes individuals</td>
  </tr>
  <tr class="bg2">
    <td>gnomADg:<b>AFR</b></td>
    <td style="text-align:right">-</td>
    <td>African/African American</td>
  </tr>
  <tr>
    <td>gnomADg:<b>AMR</b></td>
    <td style="text-align:right">-</td>
    <td>Latino</td>
  </tr>
  <tr class="bg2">
    <td>gnomADg:<b>ASJ</b></td>
    <td style="text-align:right">-</td>
    <td>Ashkenazi Jewish</td>
  </tr>
  <tr>
    <td>gnomADg:<b>EAS</b></td>
    <td style="text-align:right">-</td>
    <td>East Asian</td>
  </tr>
  <tr class="bg2">
    <td>gnomADg:<b>FIN</b></td>
    <td style="text-align:right">-</td>
    <td>Finnish</td>
  </tr>
  <tr>
    <td>gnomADg:<b>NFE</b></td>
    <td style="text-align:right">-</td>
    <td>Non-Finnish European</td>
  </tr>
  <tr class="bg2">
    <td>gnomADg:<b>OTH</b></td>
    <td style="text-align:right">-</td>
    <td>Other</td>
  </tr>
</table>


<div style="margin-bottom:35px"></div>

  <ul style="padding-left:1em;margin-left:2px">
    <li><h3>Population from the <a href="https://www.nhlbi.nih.gov/research/resources/nhlbi-precision-medicine-initiative/topmed" target="_blank" style="text-decoration:none">TOPMed</a> (<i>Homo sapiens</i>)</h3></li>
  </ul>
  <table id="topmed" class="ss" style="margin-bottom:4px">
  
  <tr>
    <th style="width:200px">Name</th>
    <th style="width:36px">Size</th>
    <th>Description</th>
  </tr>

  <tr>
    <td><b>TOPMed</b></td>
    <td style="text-align:right">-</td>
    <td>Trans-Omics for Precision Medicine (TOPMed) Program</td>
  </tr>
</table>


<div style="margin-bottom:35px"></div>

  <ul style="padding-left:1em;margin-left:2px">
    <li><h3>Populations from the <a href="https://www.uk10k.org/" target="_blank" style="text-decoration:none">UK10K</a> (<i>Homo sapiens</i>)</h3></li>
  </ul>
  <table id="uk10k" class="ss" style="margin-bottom:4px">
  
  <tr>
    <th style="width:200px">Name</th>
    <th style="width:36px">Size</th>
    <th>Description</th>
  </tr>

  <tr>
    <td>UK10K:<b>ALSPAC</b></td>
    <td style="text-align:right">-</td>
    <td>ALSPAC cohort</td>
  </tr>
  <tr class="bg2">
    <td>UK10K:<b>TWINSUK</b></td>
    <td style="text-align:right">-</td>
    <td>TWINSUK cohort excluding 67 samples where a monozygotic or dyzygotic twin was included in the release</td>
  </tr>
</table>


<div style="margin-bottom:35px"></div>

  <ul style="padding-left:1em;margin-left:2px">
    <li><h3>Populations from the <a href="https://www.genome.gov/10001688/international-hapmap-project/" target="_blank" style="text-decoration:none">HapMap Project</a> (<i>Homo sapiens</i>)</h3></li>
  </ul>
  <table id="hapmap_project" class="ss" style="margin-bottom:4px">
  
  <tr>
    <th style="width:200px">Name</th>
    <th style="width:36px">Size</th>
    <th>Description</th>
  </tr>

  <tr>
    <td>CSHL-HAPMAP:<b>HAPMAP-ASW</b></td>
    <td style="text-align:right">90</td>
    <td>African ancestry in Southwest USA. ASW is one of the 11 populations in HapMap phase 3.</td>
  </tr>
  <tr class="bg2">
    <td>CSHL-HAPMAP:<b>HAPMAP-CHB</b></td>
    <td style="text-align:right">90</td>
    <td>Han Chinese in Beijing, China. CHB is one of the 11 populations in HapMap phase 3.</td>
  </tr>
  <tr>
    <td>CSHL-HAPMAP:<b>HAPMAP-CHD</b></td>
    <td style="text-align:right">100</td>
    <td>Chinese in Metropolitan Denver, Colorado. CHD is one of the 11 populations in HapMap phase 3.</td>
  </tr>
  <tr class="bg2">
    <td>CSHL-HAPMAP:<b>HAPMAP-GIH</b></td>
    <td style="text-align:right">100</td>
    <td>Gujarati Indians in Houston, Texas. GIH is one of the 11 populations in HapMap phase 3.</td>
  </tr>
  <tr>
    <td>CSHL-HAPMAP:<b>HAPMAP-LWK</b></td>
    <td style="text-align:right">100</td>
    <td>Luhya in Webuye, Kenya. LWK is one of the 11 populations in HapMap phase 3.</td>
  </tr>
  <tr class="bg2">
    <td>CSHL-HAPMAP:<b>HAPMAP-MEX</b></td>
    <td style="text-align:right">90</td>
    <td>Mexican ancestry in Los Angeles, California. MEX is one of the 11 populations in HapMap phase 3.</td>
  </tr>
  <tr>
    <td>CSHL-HAPMAP:<b>HAPMAP-MKK</b></td>
    <td style="text-align:right">180</td>
    <td>Maasai in Kinyawa, Kenya. MKK is one of the 11 populations in HapMap phase 3.</td>
  </tr>
  <tr class="bg2">
    <td>CSHL-HAPMAP:<b>HAPMAP-TSI</b></td>
    <td style="text-align:right">100</td>
    <td>Toscans in Italy. TSI is one of the 11 populations in HapMap phase 3.</td>
  </tr>
  <tr>
    <td>CSHL-HAPMAP:<b>HapMap-CEU</b></td>
    <td style="text-align:right">185</td>
    <td>Utah residents with Northern and Western European ancestry from the CEPH collection. CEU is one of the 11 populations in HapMap phase 3.</td>
  </tr>
  <tr class="bg2">
    <td>CSHL-HAPMAP:<b>HapMap-HCB</b></td>
    <td style="text-align:right">48</td>
    <td>45 unrelated Han Chinese in Beijing, China, representing one of the populations studied in the International HapMap project (<a href="http://www.hapmap.org"> http://www.hapmap.org</a>). See <a href="http://www.hapmap.org/citinghapmap.html.en">http://www.hapmap.org/citinghapmap.html.en</a> for further information about,this population and others studied in the project. <a href="http://www.hapmap.org/hapmappopulations.html.en">http://www.hapmap.org/hapmappopulations.html.en</a>,also has relevant information.</td>
  </tr>
  <tr>
    <td>CSHL-HAPMAP:<b>HapMap-JPT</b></td>
    <td style="text-align:right">93</td>
    <td>Japanese in Tokyo, Japan. JPT is one of the 11 populations in HapMap phase 3.</td>
  </tr>
  <tr class="bg2">
    <td>CSHL-HAPMAP:<b>HapMap-YRI</b></td>
    <td style="text-align:right">185</td>
    <td>Yoruba in Ibadan, Nigeria. YRI is one of the 11 populations in HapMap phase 3.</td>
  </tr>
</table>


<p style="margin-bottom:35px">
Variants which have been discovered in this project have the "evidence status" <a href="#evidence_status"><b>HapMap</b></a>.
On the website this corresponds to the icon <a href="#evidence_status"><img class="_ht" src="/i/val/evidence_HapMap.png" title="HapMap" style="vertical-align:bottom"/></a>.
</p>
    
  <ul style="padding-left:1em;margin-left:2px">
    <li><h3>Populations from the <a href="http://evs.gs.washington.edu/EVS/" target="_blank" style="text-decoration:none">Exome Sequencing Project (ESP)</a> (<i>Homo sapiens</i>)</h3></li>
  </ul>
  <table id="exome_sequencing_project_(esp)" class="ss" style="margin-bottom:4px">
  
  <tr>
    <th style="width:200px">Name</th>
    <th style="width:36px">Size</th>
    <th>Description</th>
  </tr>

  <tr>
    <td>ESP6500:<b>African_American</b></td>
    <td style="text-align:right">-</td>
    <td>-</td>
  </tr>
  <tr class="bg2">
    <td>ESP6500:<b>European_American</b></td>
    <td style="text-align:right">-</td>
    <td>-</td>
  </tr>
</table>


<p style="margin-bottom:35px">
Variants which have been discovered in this project have the "evidence status" <a href="#evidence_status"><b>ESP</b></a>.
On the website this corresponds to the icon <a href="#evidence_status"><img class="_ht" src="/i/val/evidence_ESP.png" title="ESP" style="vertical-align:bottom"/></a>.
</p>
    
  <ul style="padding-left:1em;margin-left:2px">
    <li><h3>Population from the <a href="http://www.sanger.ac.uk/resources/mouse/genomes/" target="_blank" style="text-decoration:none">Mouse Genomes Project (MGP)</a> (<i>Mus musculus</i>)</h3></li>
  </ul>
  <table id="mouse_genomes_project_(mgp)" class="ss" style="margin-bottom:4px">
  
  <tr>
    <th style="width:200px">Name</th>
    <th style="width:36px">Size</th>
    <th>Description</th>
  </tr>

  <tr>
    <td><b>Mouse Genomes Project</b></td>
    <td style="text-align:right">18</td>
    <td>18 mouse strains whole-genome sequenced by the Mouse Genomes Project</td>
  </tr>
</table>


<div style="margin-bottom:35px"></div>

  <ul style="padding-left:1em;margin-left:2px">
    <li><h3>Populations from the <a href="http://projects.ensembl.org/nextgen/" target="_blank" style="text-decoration:none">NextGen Project - Sheep</a> (<i>Ovis aries</i>)</h3></li>
  </ul>
  <table id="nextgen_project_-_sheep" class="ss" style="margin-bottom:4px">
  
  <tr>
    <th style="width:200px">Name</th>
    <th style="width:36px">Size</th>
    <th>Description</th>
  </tr>

  <tr>
    <td>NextGen:<b>IROA</b></td>
    <td style="text-align:right">20</td>
    <td>Iranian Ovis aries (sheep) from the NextGen Project</td>
  </tr>
  <tr class="bg2">
    <td>NextGen:<b>MOOA</b></td>
    <td style="text-align:right">160</td>
    <td>Moroccan Ovis aries (sheep) from the NextGen Project</td>
  </tr>
</table>


<div style="margin-bottom:35px"></div>

  <ul style="padding-left:1em;margin-left:2px">
    <li><h3>Population from the <a href="http://projects.ensembl.org/nextgen/" target="_blank" style="text-decoration:none">NextGen Project - Cow</a> (<i>Bos taurus</i>)</h3></li>
  </ul>
  <table id="nextgen_project_-_cow" class="ss" style="margin-bottom:4px">
  
  <tr>
    <th style="width:200px">Name</th>
    <th style="width:36px">Size</th>
    <th>Description</th>
  </tr>

  <tr>
    <td>NextGen:<b>IRBT</b></td>
    <td style="text-align:right">8</td>
    <td>Iranian Bos taurus (cow) from the NextGen Project</td>
  </tr>
</table>


<div style="margin-bottom:35px"></div>
<!-- Data populations - end -->

<br />
<hr style="margin-bottom:4px" />
<h2 id="variation_sets">Variant sets</h2>
    
<p>
We use the concept of variant sets to group variants that share some property together. For example, we have grouped the variants identified in the 1000 Genomes phase 3 sets, these are divided into subsets based on population. For example, the set representing variants identified in the 1000 Genomes phase 3 study is named 
'1000 Genomes 3 - All' and has several subsets like: '1000 Genomes 3 - AFR', '1000 Genomes 3 - AMR', '1000 Genomes 3 - EAS', '1000 Genomes 3 - EUR' and '1000 Genomes 3 - SAS'. 
The variant sets can be displayed as separate tracks on the location view. This behaviour is controlled from the 'Variation' section on the configuration panel which is accessed by 
clicking the 'Configure this page' link in the left hand side navigation.
</p>

<p>
The sets are constructed during production and are stored in the database. The table below lists the available
variant sets in the Ensembl variation database (subsets are indicated by bullet points).
</p>

<!-- Data variation_sets - start -->
<h4>Variant sets common to all species</h4>
<table id="variation_set_table" class="ss">

  <tr>
    <th>Name</th>
    <th>Short name</th>
    <th>Description</th>
  </tr>

  <tr>
    <td><b>All failed variations</b></td>
    <td>fail_all</td>
    <td>Variations that have failed the Ensembl QC checks</td>
  </tr>
</table>
<br />
<h4>Variant sets specific to Human</h4>
<table id="human_variation_set_table" class="ss">

  <tr>
    <th>Name</th>
    <th>Short name</th>
    <th>Description</th>
  </tr>
  <tr>
    <td><b>1000 Genomes 3 - All</b></td>
    <td>1kg_3</td>
    <td>Variants genotyped by the 1000 Genomes project (phase 3)</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">1000 Genomes 3 - AFR</li></ul></td>
    <td>1kg_3_afr</td>
    <td>Variants genotyped in African individuals by the 1000 Genomes project (phase 3)</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">1000 Genomes 3 - AFR - common</li></ul></td>
    <td>1kg_3_afr_com</td>
    <td>Variants genotyped in African individuals by the 1000 Genomes project (phase 3) with frequency of at least 1%</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">1000 Genomes 3 - AMR</li></ul></td>
    <td>1kg_3_amr</td>
    <td>Variants genotyped in admixed American individuals by the 1000 Genomes project (phase 3)</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">1000 Genomes 3 - AMR - common</li></ul></td>
    <td>1kg_3_amr_com</td>
    <td>Variants genotyped in admixed American individuals by the 1000 Genomes project (phase 1) with frequency of at least 1%</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">1000 Genomes 3 - All - common</li></ul></td>
    <td>1kg_3_com</td>
    <td>Variants genotyped by the 1000 Genomes project (phase 1) with frequency of at least 1%</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">1000 Genomes 3 - EAS</li></ul></td>
    <td>1kg_3_eas</td>
    <td>Variants genotyped in East Asian individuals by the 1000 Genomes project (phase 3)</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">1000 Genomes 3 - EAS - common</li></ul></td>
    <td>1kg_3_eas_com</td>
    <td>Variants genotyped in East Asian individuals by the 1000 Genomes project (phase 1) with frequency of at least 1%</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">1000 Genomes 3 - EUR</li></ul></td>
    <td>1kg_3_eur</td>
    <td>Variants genotyped in European individuals by the 1000 Genomes project (phase 3)</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">1000 Genomes 3 - EUR - common</li></ul></td>
    <td>1kg_3_eur_com</td>
    <td>Variants genotyped in European individuals by the 1000 Genomes project (phase 1) with frequency of at least 1%</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">1000 Genomes 3 - SAS</li></ul></td>
    <td>1kg_3_sas</td>
    <td>Variants genotyped in South Asian individuals by the 1000 Genomes project (phase 3)</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">1000 Genomes 3 - SAS - common</li></ul></td>
    <td>1kg_3_sas_com</td>
    <td>Variants genotyped in East Asian individuals by the 1000 Genomes project (phase 1) with frequency of at least 1%</td>
  </tr>
  <tr>
    <td><b>All HapMap</b></td>
    <td>hapmap</td>
    <td>Variants which have been assayed by The International HapMap Project [http://hapmap.ncbi.nlm.nih.gov/]</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">HapMap - CEU</li></ul></td>
    <td>hapmap_ceu</td>
    <td>Variants which have been assayed by The International HapMap Project from CEU individuals</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">HapMap - HCB</li></ul></td>
    <td>hapmap_hcb</td>
    <td>Variants which have been assayed by The International HapMap Project from HCB individuals</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">HapMap - JPT</li></ul></td>
    <td>hapmap_jpt</td>
    <td>Variants which have been assayed by The International HapMap Project from JPT individuals</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">HapMap - YRI</li></ul></td>
    <td>hapmap_yri</td>
    <td>Variants which have been assayed by The International HapMap Project from YRI individuals</td>
  </tr>
  <tr class="bg2">
    <td><b>All LSDB-associated variants</b></td>
    <td>lsdb_variants</td>
    <td>Variants association from one or several Locus Specific DataBase (LSDB)</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">HbVar</li></ul></td>
    <td>HbVar</td>
    <td>Variants for the Human Hemoglobin Variants and Thalassemias database</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">Infevers</li></ul></td>
    <td>Infevers</td>
    <td>Variants from the registry of Hereditary Auto-inflammatory Disorders Mutations</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">KAT6BDB</li></ul></td>
    <td>KAT6BDB</td>
    <td>Variants from the K(lysine) acetyltransferase 6B database, BCM</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">LMDD</li></ul></td>
    <td>LMDD</td>
    <td>Variants from the Leiden Muscular Dystrophy Database</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">LSDB</li></ul></td>
    <td>LSDB</td>
    <td>Variants dbSNP annotates as being from LSDBs</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">OIVD</li></ul></td>
    <td>OIVD</td>
    <td>Variants from the Osteogenesis Imperfecta Variant Database</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">PAHdb</li></ul></td>
    <td>PAHdb</td>
    <td>Variants from the Phenylalanine hydroxylase database</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">dbPEX</li></ul></td>
    <td>dbPEX</td>
    <td>Variants from the PEX Gene Database</td>
  </tr>
  <tr>
    <td><b>All phenotype/disease-associated variants</b></td>
    <td>ph_variants</td>
    <td>Variants that have been associated with a phenotype or a disease</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">All ClinVar</li></ul></td>
    <td>ClinVar</td>
    <td>Variants with ClinVar annotation</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">COSMIC phenotype variants</li></ul></td>
    <td>ph_cosmic</td>
    <td>Phenotype annotations of somatic mutations found in human cancers from the COSMIC project</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">Clinically associated variants</li></ul></td>
    <td>clin_assoc</td>
    <td>Variants described by ClinVar as being probable-pathogenic, pathogenic, drug-response or histocompatibility</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">HGMD-PUBLIC variants</li></ul></td>
    <td>ph_hgmd_pub</td>
    <td>Variants annotated by HGMD</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">NHGRI-EBI catalog phenotype variants</li></ul></td>
    <td>ph_nhgri</td>
    <td>Variants associated with phenotype data from the NHGRI-EBI GWAS catalog [http://www.ebi.ac.uk/gwas/]</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">OMIM phenotype variants</li></ul></td>
    <td>ph_omim</td>
    <td>Variants linked to entries in the Online Mendelian Inheritance in Man (OMIM) database</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">PhenCode</li></ul></td>
    <td>phencode</td>
    <td>Variants from the PhenCode Project</td>
  </tr>
  <tr>
    <td><b>ESP_6500</b></td>
    <td>esp_6500</td>
    <td>Variants from the NHLBI Exome Sequencing Project (investigating heart, lung and blood disorders)</td>
  </tr>
  <tr class="bg2">
    <td><b>ExAC</b></td>
    <td>exac</td>
    <td>Variants identified by the Exome Aggregation Consortium (ExAC) - release 0.3</td>
  </tr>
  <tr>
    <td><b>Genotyping chip variants</b></td>
    <td>all_chips</td>
    <td>Variants which have assays on commercial chips held in ensembl</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">Affy GeneChip 500K</li></ul></td>
    <td>Affy_500K</td>
    <td>Variants from the Affymetrix GeneChip Human Mapping 500K Array Set</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">Affy GenomeWideSNP_6.0</li></ul></td>
    <td>Affy_SNP6</td>
    <td>Variants from the Affymetrix Genome-Wide Human SNP Array 6.0</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">HumanOmniExpress</li></ul></td>
    <td>HumanOmniExpress</td>
    <td>Variants from the Illumina HumanOmniExpress 12v1-1_a whole genome genotyping array</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">Illumina_Cardio-Metabo_Chip</li></ul></td>
    <td>Cardio-Metabo_Chip</td>
    <td>Variants from the Illumina Cardio-Metabo_Chip genotyping array designed to target variants of interest for metabolic and cardiovascular disease traits</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">Illumina_CytoSNP12v1</li></ul></td>
    <td>Illumina_CytoSNP12v1</td>
    <td>Variants from the Illumina Cyto SNP-12 v1 whole genome SNP genotyping chip designed for cytogenetic analysis</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">Illumina_ExomeChip</li></ul></td>
    <td>ExomeChip</td>
    <td>Variants from the Illumina ExomeChip genotyping array designed to target variants within exons</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">Illumina_Human1M-duo</li></ul></td>
    <td>Illumina_1M-duo</td>
    <td>Variants from the Illumina Human1M-duo v3 whole genome genotyping array designed for association studies</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">Illumina_Human610_Quad</li></ul></td>
    <td>Human610_Quad</td>
    <td>Variants from the Illumina Human610_Quad v1_B whole genome genotyping array designed for association studies</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">Illumina_Human660W-quad</li></ul></td>
    <td>Illumina_660Q</td>
    <td>Variants from the Illumina Human660W-quad whole genome genotyping array designed for association studies</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">Illumina_HumanHap550</li></ul></td>
    <td>HumanHap550</td>
    <td>Variants from the Illumina Human550 v3.0 whole genome genotyping array designed for association studies</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">Illumina_HumanHap650Y</li></ul></td>
    <td>HumanHap650Y</td>
    <td>Variants from the Illumina HumanHap650Y v3.0 whole genome genotyping array designed for association studies</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">Illumina_HumanOmni1-Quad</li></ul></td>
    <td>HumanOmni1-Quad</td>
    <td>Variants from the Illumina HumanOmni1-Quad whole genome genotyping array designed for association studies</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">Illumina_HumanOmni2.5</li></ul></td>
    <td>HumanOmni2.5</td>
    <td>Variants from the Illumina HumanOmni2.5 4v1 whole genome genotyping array designed for association studies</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">Illumina_HumanOmni5</li></ul></td>
    <td>HumanOmni5</td>
    <td>Variants from the Illumina HumanOmni5v1 whole genome genotyping array designed for association studies</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">Illumina_ImmunoChip</li></ul></td>
    <td>ImmunoChip</td>
    <td>Variants from the Illumina ImmunoChip genotyping array designed to target variants of interest for autoimmune and inflammatory diseases</td>
  </tr>
  <tr>
    <td><b>HumanCoreExome-12</b></td>
    <td>HumanCoreExome</td>
    <td>Variants from the Illumina HumanCoreExome-12 v1 genotyping chip.</td>
  </tr>
  <tr class="bg2">
    <td><b>Uniprot variants</b></td>
    <td>ph_uniprot</td>
    <td>Variants with annotations provided by UniProt</td>
  </tr>
</table>
<!-- Data variation_sets - end -->


<br />
<hr style="margin-bottom:4px" />
  

</body>
</html>
