<html>
<head>
  <meta name="order" content="1" />
  <title>Data Description</title>
  <script type="text/javascript">
    window.onload = function() {
      $('.ht').helptip({'track': true});
      $('._ht').helptip({'track': true});
    };
  </script>
</head>

<body>

<h1 style="margin-top:15px">Ensembl Variation - Data description</h1>

<p>
  Below is a description of the data we store in the databases for Ensembl Variation.<br />For several different species in Ensembl, 
  we import variant data (<span class="ht" title="Single Nucleotide Polymorphisms">SNPs</span>, <span class="ht" title="Copy Number Variations">CNVs</span>, allele frequencies, genotypes, etc) from a variety of 
  <a href="sources_documentation.html">sources</a> (e.g. dbSNP). Imported variants and alleles are subjected to a <a href="#quality_control">quality control</a> process to flag suspect data.
  <br />We classify the variants into different
  <a href="#classes">classes</a> and calculate the <a href="predicted_data.html">predicted consequence(s)</a>
  of the variant and we have also created variant <a href="#variation_sets">sets</a> to help people retrieve a specific
  group of variants from a particular dataset.<br />
  In human, we calculate the linkage disequilibrium for each variant, by <a href="#populations">population</a>.
</p>
  
<p>See some examples of imported data on the Ensembl website (Human):</p>
<div class="portal" style="margin-bottom:5px;width:auto">
  <div>
    <a href="/Homo_sapiens/Variation/Context?v=rs1333049" title="Location of a variant in the genome"><img src="/i/96/var_genomic_context.png" alt="Location of a variant in the genome"/></a>
  </div>
  <div>
    <a href="/Homo_sapiens/Variation/Population?v=rs699" title="Population genotypes and frequencies of a variant"><img src="/i/96/var_population_genetics.png" alt="Population genotypes and frequencies of a variant"/></a>
  </div>
  <div>
    <a href="/Homo_sapiens/Variation/Sample?v=rs699" title="Sample genotypes of a variant"><img src="/i/96/var_sample_genotypes.png" alt="Sample genotypes of a variant"/></a>
  </div>
  <div>
    <a href="/Homo_sapiens/Variation/Phenotype?v=rs1333049" title="Phenotype(s) associated with a variant"><img src="/i/96/var_phenotype_data.png"alt="Phenotype(s) associated with a variant"/></a>
  </div>
  <div>
    <a href="/Homo_sapiens/Variation/Citations?v=rs1333049" title="Citations"><img src="/i/96/var_citations.png" alt="Citations"/></a>
  </div>

</div>
<div style="clear:both"></div>

<hr style="margin-bottom:4px" />

<h2 id="displays">Variant displays</h2>

<p>
Variant data can be viewed in the browser through pages such as:
</p>

<ul>
<li><b>Gene:</b> Variation Table and Variation Image (for all variants in a gene) 
e.g. <a href="/Homo_sapiens/Gene/Variation_Gene/Table?db=core;g=ENSG00000159197">for all variants in KCNE2</a>. 
Structural Variation to see all structural variants overlapping the gene.</li>
<li><b>Transcript:</b> Population comparison, Comparison
image (for comparing variants in a transcript across different individual or strain
sequences) e.g. <a href="/Mus_musculus/Transcript/Population/Image?t=ENSMUST00000050949"> compare Tmco4 in different mouse strains </a> </li>
<li><b>Transcript: </b> Sequence, protein: list of the coding variants in protein coordinates.</li>
<li><b>Location:</b> Region in Detail (Variants can be
drawn using &quot;Configure this page&quot; at the left. The menu allows
display of information in Ensembl databases along with external
sources in DAS format such as <a rel="external"
href="http://projects.tcag.ca/variation/">DGV loci</a>.)</li>
<li><b>Phenotype: </b> A karyotype view to display the variants associated with a certain phenotype, e.g. <a href="/Homo_sapiens/Phenotype/Locations?ph=5815">phenotype "Glaucoma"</a>.</li>
</ul>

<h3>Examples:</h3>
<div style="margin-bottom:25px;margin-left:15px">
  <div style="float:left;margin-right:30px;text-align:center">
    <a class="nodeco _ht _ht_track" href="/Homo_sapiens/Gene/Variation_Gene/Table?db=core;g=ENSG00000159197" title="Example of Gene Variation Table page"><img src="/i/96/gene.png" alt="" class="bordered"><div style="font-weight:bold">Gene</div><div>Variation Table</div></a>
  </div>
  <div style="float:left;margin-right:30px;text-align:center">
    <a class="nodeco _ht _ht_track" href="/Homo_sapiens/Transcript/Variation_Transcript/Table?db=core;t=ENST00000403559" title="Example of Transcript Variation Table page"><img src="/i/96/transcript.png" alt="" class="bordered"><div style="font-weight:bold">Transcript</div><div>Variation Table</div></a>
  </div>
  <div style="float:left;margin-right:30px;text-align:center">
    <a class="nodeco _ht _ht_track" href="/Homo_sapiens/Transcript/ProtVariations?db=core;t=ENST00000403559" title="Example of Protein Variation page"><img src="/i/96/transcript.png" alt="" class="bordered"><div style="font-weight:bold">Transcript</div><div>Protein Variation</div></a>
  </div>
  <div style="float:left;margin-right:30px;text-align:center">
    <a class="nodeco _ht _ht_track" href="/Homo_sapiens/Location/View?r=17:63973115-64437414" title="Example of 'Region in detail' page"><img src="/i/96/region.png" alt="" class="bordered"><div style="font-weight:bold">Location</div><div>Region in detail</div></a>
  </div>
  <div style="float:left;margin-right:30px;text-align:center">
    <a class="nodeco _ht _ht_track" href="/Homo_sapiens/Phenotype/Locations?ph=5815" title="Example of Phenotype page"><img src="/i/96/phenotype.png" alt="" class="bordered"><div style="font-weight:bold">Phenotype</div><div>Karyotype view</div></a>
  </div>
  <div style="clear:both"></div>
</div>

<p>
Clicking on any variant on an Ensembl page will open a Variation tab
with information about the flanking sequence and source for the
selected variant. Links to linkage disequilibrium (LD) plots,
phenotype information (for human) from EGA, OMIM and <a rel="external"
href="https://www.ebi.ac.uk/gwas/">NHGRI-EBI</a> and Ensembl genes and
transcripts that include the variant can be found at the left of
this tab. You may also view multiple genome alignments of various
species, highlighting the
variant. <a href="/info/genome/compara/analyses.html#ancestral"
class="cp-external">Ancestral sequences</a> are included in this
display. 
</p>

<p>
Variant information can also be accessed
using <a href="/info/data/biomart/"
class="cp-external">BioMart</a> (gene or variation database), and
the <a href="/info/docs/api/index.html" class="cp-external">Perl
API</a> (<a href="/info/docs/api/variation/index.html"
class="cp-external">variation API</a>).
</p>


<br />
<hr style="margin-bottom:4px" />


<h2 id="maf">Minor Alleles</h2>

<p>
Minor alleles and their frequencies are available for variants discovered in the 1000 Genomes Project.<br />
These are calculated by dbSNP. If there are more than <b>two</b> alleles, the <b>second most common</b> is reported 
(See example: <a href="/Homo_sapiens/Variation/Population?v=rs200077393">rs200077393</a>).<br />
This allows <b>common variants</b> to be distinguished from <b>rare variants</b> in situations where deep sequencing 
has identified a third rare allele. 
</p> 

<br />
<hr style="margin-bottom:4px" />
<h2 id="populations">Populations</h2>

<p>For each variant we store population data (allele
frequencies, genotypes) available from different projects (e.g. 1000
Genomes, HapMap). You can see this on our variation pages, for
example: <a href="/Homo_sapiens/Variation/Population?v=rs699">rs699</a>.
Frequencies displayed (to 3 decimal places) may not add up to 1 due to rounding.
<br />
Below we list the population names associated with the main genotyping projects available in Ensembl Variation:
</p>

<!-- Data populations - start -->

  <ul style="padding-left:1em;margin-left:2px">
    <li><h3>Populations from the <a href="http://www.1000genomes.org" target="_blank" style="text-decoration:none">1000 Genomes Project</a> (<i>Homo sapiens</i>)</h3></li>
  </ul>
  <table id="1000_genomes_project" class="ss" style="margin-bottom:4px">
  
  <tr>
    <th style="width:200px">Name</th>
    <th style="width:36px">Size</th>
    <th>Description</th>
  </tr>

  <tr class="supergroup">
    <td>1000GENOMES:phase_3:<b>ALL</b></td>
    <td style="text-align:right">2504</td>
    <td>All phase 3 individuals</td>
  </tr>
  <tr class="subgroup">
    <td>1000GENOMES:phase_3:<b>AFR</b></td>
    <td style="text-align:right">661</td>
    <td>African</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>ACB</b></li></ul></td>
    <td style="text-align:right">96</td>
    <td>African Caribbean in Barbados</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>ASW</b></li></ul></td>
    <td style="text-align:right">61</td>
    <td>African Ancestry in Southwest US</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>ESN</b></li></ul></td>
    <td style="text-align:right">99</td>
    <td>Esan in Nigeria</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>GWD</b></li></ul></td>
    <td style="text-align:right">113</td>
    <td>Gambian in Western Division, The Gambia</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>LWK</b></li></ul></td>
    <td style="text-align:right">99</td>
    <td>Luhya in Webuye, Kenya</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>MSL</b></li></ul></td>
    <td style="text-align:right">85</td>
    <td>Mende in Sierra Leone</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>YRI</b></li></ul></td>
    <td style="text-align:right">108</td>
    <td>Yoruba in Ibadan, Nigeria</td>
  </tr>
  <tr class="subgroup">
    <td>1000GENOMES:phase_3:<b>AMR</b></td>
    <td style="text-align:right">347</td>
    <td>American</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>CLM</b></li></ul></td>
    <td style="text-align:right">94</td>
    <td>Colombian in Medellin, Colombia</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>MXL</b></li></ul></td>
    <td style="text-align:right">64</td>
    <td>Mexican Ancestry in Los Angeles, California</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>PEL</b></li></ul></td>
    <td style="text-align:right">85</td>
    <td>Peruvian in Lima, Peru</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>PUR</b></li></ul></td>
    <td style="text-align:right">104</td>
    <td>Puerto Rican in Puerto Rico</td>
  </tr>
  <tr class="subgroup">
    <td>1000GENOMES:phase_3:<b>EAS</b></td>
    <td style="text-align:right">504</td>
    <td>East Asian</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>CDX</b></li></ul></td>
    <td style="text-align:right">93</td>
    <td>Chinese Dai in Xishuangbanna, China</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>CHB</b></li></ul></td>
    <td style="text-align:right">103</td>
    <td>Han Chinese in Bejing, China</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>CHS</b></li></ul></td>
    <td style="text-align:right">105</td>
    <td>Southern Han Chinese, China</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>JPT</b></li></ul></td>
    <td style="text-align:right">104</td>
    <td>Japanese in Tokyo, Japan</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>KHV</b></li></ul></td>
    <td style="text-align:right">99</td>
    <td>Kinh in Ho Chi Minh City, Vietnam</td>
  </tr>
  <tr class="subgroup">
    <td>1000GENOMES:phase_3:<b>EUR</b></td>
    <td style="text-align:right">503</td>
    <td>European</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>CEU</b></li></ul></td>
    <td style="text-align:right">99</td>
    <td>Utah residents with Northern and Western European ancestry</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>FIN</b></li></ul></td>
    <td style="text-align:right">99</td>
    <td>Finnish in Finland</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>GBR</b></li></ul></td>
    <td style="text-align:right">91</td>
    <td>British in England and Scotland</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>IBS</b></li></ul></td>
    <td style="text-align:right">107</td>
    <td>Iberian populations in Spain</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>TSI</b></li></ul></td>
    <td style="text-align:right">107</td>
    <td>Toscani in Italy</td>
  </tr>
  <tr class="subgroup">
    <td>1000GENOMES:phase_3:<b>SAS</b></td>
    <td style="text-align:right">489</td>
    <td>South Asian</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>BEB</b></li></ul></td>
    <td style="text-align:right">86</td>
    <td>Bengali in Bangladesh</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>GIH</b></li></ul></td>
    <td style="text-align:right">103</td>
    <td>Gujarati Indian in Houston, TX</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>ITU</b></li></ul></td>
    <td style="text-align:right">102</td>
    <td>Indian Telugu in the UK</td>
  </tr>
  <tr>
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>PJL</b></li></ul></td>
    <td style="text-align:right">96</td>
    <td>Punjabi in Lahore, Pakistan</td>
  </tr>
  <tr class="bg2">
    <td><ul style="margin:0px"><li style="margin:0px">1000GENOMES:phase_3:<b>STU</b></li></ul></td>
    <td style="text-align:right">102</td>
    <td>Sri Lankan Tamil in the UK</td>
  </tr>
</table>


<p style="margin-bottom:35px">
Variants which have been discovered in this project have the "evidence status" <a href="#evidence_status"><b>1000Genomes</b></a>.
On the website this corresponds to the icon <a href="#evidence_status"><img class="_ht" src="/i/val/evidence_1000Genomes.png" title="1000Genomes" style="vertical-align:bottom"/></a>.
</p>
    
  <ul style="padding-left:1em;margin-left:2px">
    <li><h3>Populations from the <a href="http://gnomad.broadinstitute.org/" target="_blank" style="text-decoration:none">gnomAD exomes</a> (<i>Homo sapiens</i>)</h3></li>
  </ul>
  <table id="gnomad_exomes" class="ss" style="margin-bottom:4px">
  
  <tr>
    <th style="width:200px">Name</th>
    <th style="width:36px">Size</th>
    <th>Description</th>
  </tr>

  <tr>
    <td>gnomADe:<b>ALL</b></td>
    <td style="text-align:right">-</td>
    <td>All gnomAD exomes individuals</td>
  </tr>
  <tr class="bg2">
    <td>gnomADe:<b>AFR</b></td>
    <td style="text-align:right">-</td>
    <td>African/African American</td>
  </tr>
  <tr>
    <td>gnomADe:<b>AMR</b></td>
    <td style="text-align:right">-</td>
    <td>Latino</td>
  </tr>
  <tr class="bg2">
    <td>gnomADe:<b>ASJ</b></td>
    <td style="text-align:right">-</td>
    <td>Ashkenazi Jewish</td>
  </tr>
  <tr>
    <td>gnomADe:<b>EAS</b></td>
    <td style="text-align:right">-</td>
    <td>East Asian</td>
  </tr>
  <tr class="bg2">
    <td>gnomADe:<b>FIN</b></td>
    <td style="text-align:right">-</td>
    <td>Finnish</td>
  </tr>
  <tr>
    <td>gnomADe:<b>NFE</b></td>
    <td style="text-align:right">-</td>
    <td>Non-Finnish European</td>
  </tr>
  <tr class="bg2">
    <td>gnomADe:<b>OTH</b></td>
    <td style="text-align:right">-</td>
    <td>Other</td>
  </tr>
  <tr>
    <td>gnomADe:<b>SAS</b></td>
    <td style="text-align:right">-</td>
    <td>South Asian</td>
  </tr>
</table>


<div style="margin-bottom:35px"></div>

  <ul style="padding-left:1em;margin-left:2px">
    <li><h3>Populations from the <a href="http://gnomad.broadinstitute.org/" target="_blank" style="text-decoration:none">gnomAD genomes</a> (<i>Homo sapiens</i>)</h3></li>
  </ul>
  <table id="gnomad_genomes" class="ss" style="margin-bottom:4px">
  
  <tr>
    <th style="width:200px">Name</th>
    <th style="width:36px">Size</th>
    <th>Description</th>
  </tr>

  <tr>
    <td>gnomADg:<b>ALL</b></td>
    <td style="text-align:right">-</td>
    <td>All gnomAD genomes individuals</td>
  </tr>
  <tr class="bg2">
    <td>gnomADg:<b>AFR</b></td>
    <td style="text-align:right">-</td>
    <td>African/African American</td>
  </tr>
  <tr>
    <td>gnomADg:<b>AMR</b></td>
    <td style="text-align:right">-</td>
    <td>Latino</td>
  </tr>
  <tr class="bg2">
    <td>gnomADg:<b>ASJ</b></td>
    <td style="text-align:right">-</td>
    <td>Ashkenazi Jewish</td>
  </tr>
  <tr>
    <td>gnomADg:<b>EAS</b></td>
    <td style="text-align:right">-</td>
    <td>East Asian</td>
  </tr>
  <tr class="bg2">
    <td>gnomADg:<b>FIN</b></td>
    <td style="text-align:right">-</td>
    <td>Finnish</td>
  </tr>
  <tr>
    <td>gnomADg:<b>NFE</b></td>
    <td style="text-align:right">-</td>
    <td>Non-Finnish European</td>
  </tr>
  <tr class="bg2">
    <td>gnomADg:<b>OTH</b></td>
    <td style="text-align:right">-</td>
    <td>Other</td>
  </tr>
</table>


<div style="margin-bottom:35px"></div>

  <ul style="padding-left:1em;margin-left:2px">
    <li><h3>Population from the <a href="https://www.nhlbi.nih.gov/research/resources/nhlbi-precision-medicine-initiative/topmed" target="_blank" style="text-decoration:none">TOPMed</a> (<i>Homo sapiens</i>)</h3></li>
  </ul>
  <table id="topmed" class="ss" style="margin-bottom:4px">
  
  <tr>
    <th style="width:200px">Name</th>
    <th style="width:36px">Size</th>
    <th>Description</th>
  </tr>

  <tr>
    <td><b>TOPMed</b></td>
    <td style="text-align:right">-</td>
    <td>Trans-Omics for Precision Medicine (TOPMed) Program</td>
  </tr>
</table>


<div style="margin-bottom:35px"></div>

  <ul style="padding-left:1em;margin-left:2px">
    <li><h3>Populations from the <a href="https://www.uk10k.org/" target="_blank" style="text-decoration:none">UK10K</a> (<i>Homo sapiens</i>)</h3></li>
  </ul>
  <table id="uk10k" class="ss" style="margin-bottom:4px">
  
  <tr>
    <th style="width:200px">Name</th>
    <th style="width:36px">Size</th>
    <th>Description</th>
  </tr>

  <tr>
    <td>UK10K:<b>ALSPAC</b></td>
    <td style="text-align:right">-</td>
    <td>ALSPAC cohort</td>
  </tr>
  <tr class="bg2">
    <td>UK10K:<b>TWINSUK</b></td>
    <td style="text-align:right">-</td>
    <td>TWINSUK cohort excluding 67 samples where a monozygotic or dyzygotic twin was included in the release</td>
  </tr>
</table>


<div style="margin-bottom:35px"></div>

  <ul style="padding-left:1em;margin-left:2px">
    <li><h3>Populations from the <a href="https://www.genome.gov/10001688/international-hapmap-project/" target="_blank" style="text-decoration:none">HapMap Project</a> (<i>Homo sapiens</i>)</h3></li>
  </ul>
  <table id="hapmap_project" class="ss" style="margin-bottom:4px">
  
  <tr>
    <th style="width:200px">Name</th>
    <th style="width:36px">Size</th>
    <th>Description</th>
  </tr>

  <tr>
    <td>CSHL-HAPMAP:<b>HAPMAP-ASW</b></td>
    <td style="text-align:right">90</td>
    <td>African ancestry in Southwest USA. ASW is one of the 11 populations in HapMap phase 3.</td>
  </tr>
  <tr class="bg2">
    <td>CSHL-HAPMAP:<b>HAPMAP-CHB</b></td>
    <td style="text-align:right">90</td>
    <td>Han Chinese in Beijing, China. CHB is one of the 11 populations in HapMap phase 3.</td>
  </tr>
  <tr>
    <td>CSHL-HAPMAP:<b>HAPMAP-CHD</b></td>
    <td style="text-align:right">100</td>
    <td>Chinese in Metropolitan Denver, Colorado. CHD is one of the 11 populations in HapMap phase 3.</td>
  </tr>
  <tr class="bg2">
    <td>CSHL-HAPMAP:<b>HAPMAP-GIH</b></td>
    <td style="text-align:right">100</td>
    <td>Gujarati Indians in Houston, Texas. GIH is one of the 11 populations in HapMap phase 3.</td>
  </tr>
  <tr>
    <td>CSHL-HAPMAP:<b>HAPMAP-LWK</b></td>
    <td style="text-align:right">100</td>
    <td>Luhya in Webuye, Kenya. LWK is one of the 11 populations in HapMap phase 3.</td>
  </tr>
  <tr class="bg2">
    <td>CSHL-HAPMAP:<b>HAPMAP-MEX</b></td>
    <td style="text-align:right">90</td>
    <td>Mexican ancestry in Los Angeles, California. MEX is one of the 11 populations in HapMap phase 3.</td>
  </tr>
  <tr>
    <td>CSHL-HAPMAP:<b>HAPMAP-MKK</b></td>
    <td style="text-align:right">180</td>
    <td>Maasai in Kinyawa, Kenya. MKK is one of the 11 populations in HapMap phase 3.</td>
  </tr>
  <tr class="bg2">
    <td>CSHL-HAPMAP:<b>HAPMAP-TSI</b></td>
    <td style="text-align:right">100</td>
    <td>Toscans in Italy. TSI is one of the 11 populations in HapMap phase 3.</td>
  </tr>
  <tr>
    <td>CSHL-HAPMAP:<b>HapMap-CEU</b></td>
    <td style="text-align:right">185</td>
    <td>Utah residents with Northern and Western European ancestry from the CEPH collection. CEU is one of the 11 populations in HapMap phase 3.</td>
  </tr>
  <tr class="bg2">
    <td>CSHL-HAPMAP:<b>HapMap-HCB</b></td>
    <td style="text-align:right">48</td>
    <td>45 unrelated Han Chinese in Beijing, China, representing one of the populations studied in the International HapMap project (<a href="http://www.hapmap.org"> http://www.hapmap.org</a>). See <a href="http://www.hapmap.org/citinghapmap.html.en">http://www.hapmap.org/citinghapmap.html.en</a> for further information about,this population and others studied in the project. <a href="http://www.hapmap.org/hapmappopulations.html.en">http://www.hapmap.org/hapmappopulations.html.en</a>,also has relevant information.</td>
  </tr>
  <tr>
    <td>CSHL-HAPMAP:<b>HapMap-JPT</b></td>
    <td style="text-align:right">93</td>
    <td>Japanese in Tokyo, Japan. JPT is one of the 11 populations in HapMap phase 3.</td>
  </tr>
  <tr class="bg2">
    <td>CSHL-HAPMAP:<b>HapMap-YRI</b></td>
    <td style="text-align:right">185</td>
    <td>Yoruba in Ibadan, Nigeria. YRI is one of the 11 populations in HapMap phase 3.</td>
  </tr>
</table>


<p style="margin-bottom:35px">
Variants which have been discovered in this project have the "evidence status" <a href="#evidence_status"><b>HapMap</b></a>.
On the website this corresponds to the icon <a href="#evidence_status"><img class="_ht" src="/i/val/evidence_HapMap.png" title="HapMap" style="vertical-align:bottom"/></a>.
</p>
    
  <ul style="padding-left:1em;margin-left:2px">
    <li><h3>Populations from the <a href="http://evs.gs.washington.edu/EVS/" target="_blank" style="text-decoration:none">Exome Sequencing Project (ESP)</a> (<i>Homo sapiens</i>)</h3></li>
  </ul>
  <table id="exome_sequencing_project_(esp)" class="ss" style="margin-bottom:4px">
  
  <tr>
    <th style="width:200px">Name</th>
    <th style="width:36px">Size</th>
    <th>Description</th>
  </tr>

  <tr>
    <td>ESP6500:<b>African_American</b></td>
    <td style="text-align:right">-</td>
    <td>-</td>
  </tr>
  <tr class="bg2">
    <td>ESP6500:<b>European_American</b></td>
    <td style="text-align:right">-</td>
    <td>-</td>
  </tr>
</table>


<p style="margin-bottom:35px">
Variants which have been discovered in this project have the "evidence status" <a href="#evidence_status"><b>ESP</b></a>.
On the website this corresponds to the icon <a href="#evidence_status"><img class="_ht" src="/i/val/evidence_ESP.png" title="ESP" style="vertical-align:bottom"/></a>.
</p>
    
  <ul style="padding-left:1em;margin-left:2px">
    <li><h3>Population from the <a href="http://www.sanger.ac.uk/resources/mouse/genomes/" target="_blank" style="text-decoration:none">Mouse Genomes Project (MGP)</a> (<i>Mus musculus</i>)</h3></li>
  </ul>
  <table id="mouse_genomes_project_(mgp)" class="ss" style="margin-bottom:4px">
  
  <tr>
    <th style="width:200px">Name</th>
    <th style="width:36px">Size</th>
    <th>Description</th>
  </tr>

  <tr>
    <td><b>Mouse Genomes Project</b></td>
    <td style="text-align:right">18</td>
    <td>18 mouse strains whole-genome sequenced by the Mouse Genomes Project</td>
  </tr>
</table>


<div style="margin-bottom:35px"></div>

  <ul style="padding-left:1em;margin-left:2px">
    <li><h3>Populations from the <a href="http://projects.ensembl.org/nextgen/" target="_blank" style="text-decoration:none">NextGen Project - Sheep</a> (<i>Ovis aries</i>)</h3></li>
  </ul>
  <table id="nextgen_project_-_sheep" class="ss" style="margin-bottom:4px">
  
  <tr>
    <th style="width:200px">Name</th>
    <th style="width:36px">Size</th>
    <th>Description</th>
  </tr>

  <tr>
    <td>NextGen:<b>IROA</b></td>
    <td style="text-align:right">20</td>
    <td>Iranian Ovis aries (sheep) from the NextGen Project</td>
  </tr>
  <tr class="bg2">
    <td>NextGen:<b>MOOA</b></td>
    <td style="text-align:right">160</td>
    <td>Moroccan Ovis aries (sheep) from the NextGen Project</td>
  </tr>
</table>


<div style="margin-bottom:35px"></div>

  <ul style="padding-left:1em;margin-left:2px">
    <li><h3>Population from the <a href="http://projects.ensembl.org/nextgen/" target="_blank" style="text-decoration:none">NextGen Project - Cow</a> (<i>Bos taurus</i>)</h3></li>
  </ul>
  <table id="nextgen_project_-_cow" class="ss" style="margin-bottom:4px">
  
  <tr>
    <th style="width:200px">Name</th>
    <th style="width:36px">Size</th>
    <th>Description</th>
  </tr>

  <tr>
    <td>NextGen:<b>IRBT</b></td>
    <td style="text-align:right">8</td>
    <td>Iranian Bos taurus (cow) from the NextGen Project</td>
  </tr>
</table>


<div style="margin-bottom:35px"></div>
<!-- Data populations - end -->

<br />
<hr style="margin-bottom:4px" />
  

</body>
</html>
