<html>
<head>
<meta name="order" content="1" />
<title>Variation</title>
</head>

<body>

  <h1 style="margin-top:15px">About Ensembl Variation</h1>


  <p style="font-weight:bold">
    The Ensembl Variation database stores areas of the genome that differ
    between individual genomes ("variants") and, where available,
    associated disease and phenotype information.
  </p>
  <p>
    There are different types of variants for several <a href="./species_data_types.html#sources">species</a>:
  </p>
  <ul style="margin-top:2px">
    <li style="margin-top:4px">single nucleotide polymorphisms (SNPs)</li>
    <li>short nucleotide insertions and/or deletions</li>
    <li>longer variants classified as structural variants (including CNVs)</li>
  </ul>

  <p>Explore the links below to learn more about Ensembl Variation:</p>

  <div style="display:table;font-size:14px;margin:10px 0px 40px">
    <div style="display:table-row">

      <div style="display:table-cell">
        <div class="round-box plain-box bordered" style="margin-right:16px">
          <h3>Data documentation</h3>
          <ul>
            <li><a href="./data_access.html">Data access</a></li>
            <li><a href="./data_description.html">Data description</a></li>
            <li><a href="./variant_quality.html">Data quality</a></li>
            <li><a href="./classification.html">Variant classification</a></li>
          </ul>
        </div>
      </div>

      <div style="display:table-cell">
        <div class="round-box plain-box bordered" style="margin-right:16px">
          <h3>Species data</h3>
          <ul>
            <li><a href="./sources_documentation.html">Data sources</a></li>
            <li><a href="./species_data_types.html">Data types & species list</a></li>
            <li><a href="./species_detailed_counts.html">Detailed data counts</a></li>
            <li><a href="./populations.html">Population allele frequencies & genotypes</a></li>
            <li><a href="./sets.html">Variant sets</a></li>
          </ul>
        </div>
      </div>

      <div style="display:table-cell">
        <div class="round-box plain-box bordered" style="margin-right:16px">
          <h3>Data prediction</h3>
          <ul>
            <li><a href="./predicted_data.html">Calculated consequences</a></li>
            <li><a href="./protein_function.html">Protein function prediction</a></li>
          </ul>
        </div>
      </div>

    </div>
    <div style="display:table-row">

      <div style="display:table-cell">
        <div class="round-box plain-box bordered" style="margin-right:16px">
          <h3>Phenotype data</h3>
          <ul>
            <li><a href="./phenotype_association.html">Phenotype & disease association</a></li>
            <li><a href="./sources_phenotype_documentation.html">Phenotype sources</a></li>
          </ul>
        </div>
      </div>

      <div style="display:table-cell">
        <div class="round-box plain-box bordered" style="margin-right:16px">
          <h3>Tools</h3>
          <ul>
            <li><a href="./variant_tools.html">Variant tools</a></li>
          </ul>
        </div>
      </div>

    </div>
  </div>

<hr style="margin-bottom:4px" />
<h2 id="references">References</h2>

<ul>

  <li>
    Daniel R Zerbino, Premanand Achuthan, Wasiu Akanni, M Ridwan Amode, Daniel Barrell, 
    Jyothish Bhai, Konstantinos Billis, Carla Cummins, Astrid Gall, Carlos Garc&iacute;a Gir&oacute;n, 
    Laurent Gil, Leo Gordon, Leanne Haggerty, Erin Haskell, Thibaut Hourlier, Osagie G Izuogu, 
    Sophie H Janacek, Thomas Juettemann, Jimmy Kiang To, Matthew R Laird, Ilias Lavidas, 
    Zhicheng Liu, Jane E Loveland, Thomas Maurel, William McLaren, Benjamin Moore, Jonathan Mudge, 
    Daniel N Murphy, Victoria Newman, Michael Nuhn, Denye Ogeh, Chuang Kee Ong, Anne Parker, 
    Mateus Patricio, Harpreet Singh Riat, Helen Schuilenburg, Dan Sheppard, Helen Sparrow, 
    Kieron Taylor, Anja Thormann, Alessandro Vullo, Brandon Walts, Amonida Zadissa, Adam Frankish, 
    Sarah E Hunt, Myrto Kostadima, Nicholas Langridge, Fergal J Martin, Matthieu Muffato, 
    Emily Perry, Magali Ruffier, Dan M Staines, Stephen J Trevanion, Bronwen L Aken, Fiona Cunningham, 
    Andrew Yates, and Paul Flicek<br />

    <strong>Ensembl 2018</strong><br />
    <em>Nucleic Acids Research</em><br />
    <a href="https://doi.org/10.1093/nar/gkx1098" rel="external">doi:10.1093/nar/gkx1098</a>    
  </li>

  <li>
    <p>
      McLaren W, Gil L, Hunt SE, Riat HS, Ritchie GRS, Thormann A, Flicek P and Cunningham F.<br />
      <strong>The Ensembl Variant Effect Predictor</strong><br />
      <em>Genome Biology</em> 17:122(2016)<br />
      <a href="http://dx.doi.org/10.1186/s13059-016-0974-4" rel="external">doi:10.1186/s13059-016-0974-4</a>
    </p>
  </li>

  <li>
    <p>
    Rios D, McLaren WM, Chen Y, Birney E, Stabenau A, Flicek P, Cunningham F.<br />
    <strong>A Database and API for variation, dense genotyping and resequencing data</strong><br />
    <em>BMC Bioinformatics</em> 11:238 (2010)<br />
    <a href="http://dx.doi.org/10.1186/1471-2105-11-238"
    rel="external">doi:10.1186/1471-2105-11-238</a></p>
  </li>

  <li>
    <p>Chen Y, Cunningham F, Rios D, McLaren WM, Smith J, Pritchard B, Spudich GM, Brent S, Kulesha E, Marin-Garcia P, Smedley D, Birney E, Flicek P.<br />
    <strong>Ensembl Variation Resources</strong><br />
    <em>BMC Genomics</em> 11(1):293 (2010)<br />
    <a href="http://dx.doi.org/10.1186/1471-2164-11-293"
    rel="external">doi:10.1186/1471-2164-11-293</a></p>
  </li>
</ul>

</body>
</html>
