<html>
<head>
<meta name="order" content="1" />
<title>Variation</title>
</head>

<body>

  <h1 style="margin-top:15px">About Ensembl Variation</h1>


  <p style="margin-bottom:0px">
    The Ensembl Variation database stores areas of the genome that differ
    between individual genomes ("variants") and, where available,
    associated disease and phenotype information.<br />
    There are different types of variants for several <a href="data_description.html#source">species</a>:
  </p>
  <ul style="margin-top:2px">
    <li style="margin-top:4px">single nucleotide polymorphisms (SNPs)</li>
    <li>short nucleotide insertions and/or deletions</li>
    <li>longer variants classified as structural variants (including CNVs)</li>
  </ul>

  <p>
    We predict the effects of variants on the Ensembl transcripts and
    regulatory features for each species. You can run the same analysis on
    your own data using the <a href="/info/docs/tools/vep/index.html">Variant Effect
    Predictor</a>.
    <br />
    These data are integrated with other data sources in Ensembl, and can
    be accessed using the API (see links on the right handside menu) or website.
  </p>

<br />

<br />
<h2 id="vcf">VCF import</h2>
<p>
The <i>import_vcf.pl</i> script populates an Ensembl Variation database from a VCF (Variant Call Format) file. A description of the VCF file format can be found on the <a href="http://www.1000genomes.org/node/101" rel="external">1000 Genomes project website</a>.<br />The script can either populate a database from scratch, or add data to an existing database.
<ul style="margin-left:6px">
  <li style="list-style-image: url(/i/16/documentation.png);vertical-align:top">
    <a href="./import_vcf.html" style="font-weight:bold;vertical-align:top">See the "Import VCF script" documentation</a>
  </li>
</ul>
</p>



<br />
<h2 id="references">References</h2>

<ul>
  <li>
    Bronwen L. Aken, Premanand Achuthan, Wasiu Akanni, M.Ridwan Amode, Friederike Bernsdorff, 
    Jyothish Bhai, Konstantinos Billis, Denise Carvalho-Silva, Carla Cummins, Peter Clapham, 
    Laurent Gil, Carlos Garc&iacute;a Gir&oacute;n, Leo Gordon, Thibaut Hourlier, Sarah E. Hunt,
    Sophie H. Janacek, Thomas Juettemann, Stephen Keenan, Matthew R. Laird, Ilias Lavidas, 
    Thomas Maurel, William McLaren, Benjamin Moore, Daniel N. Murphy, Rishi Nag, Victoria Newman, 
    Michael Nuhn, Chuang Kee Ong, Anne Parker, Mateus Patricio, Harpreet Singh Riat, Daniel Sheppard,
    Helen Sparrow, Kieron Taylor, Anja Thormann, Alessandro Vullo, Brandon Walts, Steven P. Wilder, 
    Amonida Zadissa, Myrto Kostadima, Fergal J. Martin, Matthieu Muffato, Emily Perry, Magali Ruffier,
    Daniel M. Staines, Stephen J. Trevanion, Fiona Cunningham, Andrew Yates,  Daniel R. Zerbino
    and Paul Flicek<br />

    <strong>Ensembl 2017</strong><br />
    <em>Nucleic Acids Research</em><br />
    <a href="http://dx.doi.org/10.1093/nar/gkw1104" rel="external">doi: 10.1093/nar/gkw1104</a></p>
  </li>

  <li>
    <p>
      McLaren W, Gil L, Hunt SE, Riat HS, Ritchie GRS, Thormann A, Flicek P and Cunningham F.<br />
      <strong>The Ensembl Variant Effect Predictor</strong><br />
      <em>Genome Biology</em> 17:122(2016)<br />
      <a href="http://dx.doi.org/10.1186/s13059-016-0974-4" rel="external">doi:10.1186/s13059-016-0974-4</a>
    </p>
  </li>

  <li>
    <p>
    Rios D, McLaren WM, Chen Y, Birney E, Stabenau A, Flicek P, Cunningham F.<br />
    <strong>A Database and API for variation, dense genotyping and resequencing data</strong><br />
    <em>BMC Bioinformatics</em> 11:238 (2010)<br />
    <a href="http://dx.doi.org/10.1186/1471-2105-11-238"
    rel="external">doi:10.1186/1471-2105-11-238</a></p>
  </li>

  <li>
    <p>Chen Y, Cunningham F, Rios D, McLaren WM, Smith J, Pritchard B, Spudich GM, Brent S, Kulesha E, Marin-Garcia P, Smedley D, Birney E, Flicek P.<br />
    <strong>Ensembl Variation Resources</strong><br />
    <em>BMC Genomics</em> 11(1):293 (2010)<br />
    <a href="http://dx.doi.org/10.1186/1471-2164-11-293"
    rel="external">doi:10.1186/1471-2164-11-293</a></p>
  </li>
</ul>

</body>
</html>
