<html>
<head>
<meta name="order" content="1" />
<title>Calculated Variant consequences</title>

<script type="text/javascript">
  // Function to highlight a line
  function highlight_line() {
    if(window.location.hash) {
      var hash = window.location.hash.substring(1); //Puts hash in variable, and removes the # character
      if (document.getElementById(hash)) {
        element = document.getElementById(hash);
        if (element.nodeName == "TR") {
          element.style.backgroundColor="#d6ff97";
        }
      }
    }
  }
</script>

</head>

<body>

<h1 style="margin-top:15px">Calculated variant consequences</h1>

<p>
For each variant that is mapped to the reference genome, we identify each
Ensembl transcript that overlap the variant. We then use a rule-based approach to
predict the effects that each allele of the variant may have on the transcript.
The set of consequence terms, defined by the <a rel="external"
href="http://www.sequenceontology.org">Sequence Ontology</a> (SO), that can be
currently assigned to each combination of an allele and a transcript is shown
in the table below. Note that each allele of each variant may have a different
effect in different transcripts. 
</p>

<p> 
This approach is applied to all germline variants and somatic mutations
stored in the Ensembl variation databases (though we do not yet currently calculate
consequences for structural variants). The resulting consequence type calls,
along with information determined as part of the process, such as the cDNA and
CDS coordinates, and the affected codons and amino acids in coding transcripts,
are stored in the Ensembl Variation database and displayed on the website. You can use 
this pipeline for your own data via the <a href="/info/docs/tools/vep">VEP</a>. 
</p>

<p>
We used <a rel="external" href="http://www.sequenceontology.org">SO terms</a> by default since the Ensembl release 68. There is an equivalent 
SO term for each of our old Ensembl terms but in a few cases there is a more specific 
SO term available, as shown in the table below. If you have text files or VEP outputs 
with our old Ensembl terms, you can easily update these to using the SO terms by running the following 
<a href="https://raw.github.com/Ensembl/ensembl-variation/release/[[SPECIESDEFS::ENSEMBL_VERSION]]/scripts/misc/convert_ensembl_to_SO_consequences.pl" rel="external">
script</a> e.g. <br />
</p>
<p>
<code style="border: 1px solid #CCC;padding:3px" >perl convert_ensembl_to_SO_consequences.pl input.txt > converted.txt</code>
</p>

<p style="padding-top:15px">
See below a diagram showing the location of each display 
term relative to the transcript structure:
<br />
<a href="consequences.jpg" target="_blank">
  <img src="consequences.jpg" style="width:700px;height:250px" alt="consequence diagram" />
</a>
</p>

<p style="padding-top:10px">
The terms in the table below are <b>shown in order of severity</b> (more severe to less severe) as estimated by
Ensembl, and this ordering is used on the website summary views. This ordering
is necessarily subjective and API and VEP users can always get the full set of
consequences for each allele and make their own severity judgement. The 
<a href="/Help/Glossary?id=535" class="popup">IMPACT rating</a> 
is a separate rating given for compatibility with other variant
annotation tools (e.g. <a href="http://snpeff.sourceforge.net/">snpEff</a>).
</p>

[[SCRIPT::EnsEMBL::Web::Document::HTML::ConsequenceTable]]

<script language="Javascript" type="text/javascript">
  window.onload = function() {
    $('.ht').helptip({'track': true});
    highlight_line();
  };
</script>

  <p> Missense variants may have <a href ="/variation/protein_function.html">further annotation on their effect on the protein function</a>, using a number of algorithms.

<br />

</body>
</html>

