
<html>
<head>
<title>Protein function prediction</title>

<script type="text/javascript">
  // Function to highlight a line
  function highlight_line() {
    if(window.location.hash) {
      var hash = window.location.hash.substring(1); //Puts hash in variable, and removes the # character
      if (document.getElementById(hash)) {
        element = document.getElementById(hash);
        if (element.nodeName == "TR") {
          element.style.backgroundColor="#d6ff97";
        }
      }
    }
  }
</script>

</head>

<body>

<h1 style="margin-top:15px">Ensembl Variation - Protein function predictions</h1>

<p>
For human variants that are predicted to result in an <a href ="/variation/predicted_data.html">amino acid substitution</a> we 
use SIFT and PolyPhen2 to prediction the effect of the change on
protein function. We compute predictions for each of these tools for all possible 
single amino acid substitutions in the Ensembl human proteome. This enables us to provide 
predictions for novel mutations for VEP and API users. We were able to compute predictions
from at least one tool for over 95% of the human proteins in Ensembl. 
SIFT predictions are also available for cat, chicken, cow, dog, goat, horse, mouse, pig, rat sheep and zebrafish.
</p>

<p>
These tools are developed by external groups and we provide a brief explanation of
the approach each takes below, and how we run it in Ensembl. For much more detail
please see the representative papers listed <a href="#references">below</a>, and the 
relevant publications available on each tool's website. 
</p>

<div style="border:1px solid #CCC">
<div style="padding:4px 4px 0px">
<h2 id="sift">SIFT</h2>

<p>
<a rel="external" href="http://sift.bii.a-star.edu.sg">SIFT</a> predicts whether an amino acid 
substitution is likely to affect protein function based on sequence homology and the
physico-chemical similarity between the alternate amino acids. The data we provide for each amino
acid substitution is a score and a qualitative prediction (either 'tolerated' or 'deleterious'). 
The score is the normalized probability that the amino acid change is 
tolerated so scores nearer zero are more likely to be deleterious.
The qualitative prediction is derived from this score such that substitutions with 
a score &lt; 0.05 are called 'deleterious' and all others are called 'tolerated'.
</p>

<p>
We ran <b>SIFT version <!-- sift_version - start -->5.2.2<!-- sift_version - end --></b> following the instructions from the authors and used SIFT
to choose homologous proteins rather than supplying them ourselves. We used all 
protein sequences available from <!-- sift_protein_db_version - start -->UniRef90 (release 2014_11)<!-- sift_protein_db_version - end --> as the protein database. 
</p>
</div>
<div style="border-top:1px dotted #CCC;border-bottom:1px dotted #CCC;background-color:#FAFAFA;padding:0px 4px">
  <div style="float:left;margin-right:10px;margin-top:8px">List of species with SIFT results:</div>
  <!-- sift_species - start -->

  <div style="float:left;font-style:italic">
    <ul style="margin-bottom:0px">
      <li>Bos taurus</li>
      <li>Canis familiaris</li>
      <li>Capra hircus</li>
      <li>Danio rerio</li>
    </ul>
  </div>
  <div style="float:left;font-style:italic;margin-left:10px">
    <ul style="margin-bottom:0px">
      <li>Equus caballus</li>
      <li>Felis catus</li>
      <li>Gallus gallus</li>
      <li>Homo sapiens</li>
    </ul>
  </div>
  <div style="float:left;font-style:italic;margin-left:10px">
    <ul style="margin-bottom:0px">
      <li>Mus musculus</li>
      <li>Ovis aries</li>
      <li>Rattus norvegicus</li>
      <li>Sus scrofa</li>
    </ul>
  </div><!-- sift_species - end -->
  <div class="clear"></div>
</div>
<div style="padding:8px 4px">
  <table id="sift_values" class="ss" style="min-width:500px;width:auto;margin:0px">
    <tr>
      <th>SIFT value</th>
      <th>Qualitative prediction</th>
      <th colspan="2">Website display example</th>
    </tr>
    <tr>
      <td>Less than 0.05</td>
      <td>"Deleterious"</td>
      <td>
        <div class="score score_bad" title="deleterious">0.01</div>
      </td>
      <td>
        <div class="score score_neutral" title="deleterious - low confidence">0.01</div>
      </td>
    </tr>
    <tr class="bg2">
      <td>Greater than or equal to 0.05</td>
      <td>"Tolerated"</td>
      <td>
        <div class="score score_good" title="tolerated">0.8</div>
      </td>
      <td>
        <div class="score score_neutral" title="tolerated - low confidence">0.8</div>
      </td>
    </tr>
  </table>
</div>
</div>

<br />


<div style="border:1px solid #CCC">
<div style="padding:4px 4px 0px">
<h2 id="polyphen">PolyPhen</h2>

<p>
<a rel="external" href="http://genetics.bwh.harvard.edu/pph2/">PolyPhen-2</a> predicts
the effect of an amino acid substitution on the structure and function of a protein
using sequence homology, Pfam annotations, 3D structures from PDB where available, and
a number of other databases and tools (including DSSP, ncoils etc.). As with SIFT, for each amino
acid substitution where we have been able to calculate a prediction, we provide both a
qualitative prediction (one of 'probably damaging', 'possibly damaging', 'benign'
or 'unknown') and a score. The PolyPhen score represents the probability that a substitution 
is damaging, so values nearer one are more confidently predicted to be deleterious (note 
that this the opposite to SIFT). The qualitative prediction is based on the False 
Positive Rate of the classifier model used to make the predictions.
</p>

<p>
We ran <b>PolyPhen-2 version <!-- polyphen_version - start -->2.2.2<!-- polyphen_version - end -->, release <!-- polyphen_release - start -->405c<!-- polyphen_release - end --></b> (available 
<a rel="external" href="http://genetics.bwh.harvard.edu/pph2/dokuwiki/downloads">here</a>)
following all instructions from the authors and using the UniProtKB UniRef100 (release 2013_10) 
non-redundant protein set as the protein database and DSSP (snapshot 22-Nov-2013) and PDB (snapshot 22-Nov-2013) 
as the structural databases. 
When computing the predictions we store results for the classifier models trained on the HumDiv 
and HumVar datasets. Both result sets are available through the variation API which defaults to HumVar if no selection is made.</a>
 (Please refer to the PolyPhen  <a rel="external" href="http://genetics.bwh.harvard.edu/pph2/dokuwiki/overview">website</a> or publications for more details of the classification system).
<!-- HumDiv aims to discriminate variants that alter protein function, whereas HumVar aims to discriminate functions that cause human disease.-->
</p>
</div>
<div style="border-top:1px dotted #CCC;border-bottom:1px dotted #CCC;background-color:#FAFAFA;padding:0px 4px">
  <div style="float:left;margin-right:10px;margin-top:8px">List of species with PolyPhen data:</div>
    <!-- polyphen_species - start -->

  <div style="float:left;font-style:italic">
    <ul style="margin-bottom:0px"><li>Homo sapiens</li></ul>
  </div><!-- polyphen_species - end -->
  <div class="clear"></div>
</div>
<div style="padding:8px 4px">
  <table id="polyphen_values" class="ss" style="min-width:625px;width:auto;margin:0px">
    <tr>
      <th>Polyphen value</th>
      <th>Qualitative prediction</th>
      <th>Website display example</th>
    </tr>
    <tr class="bg2">
      <td>greater than 0.908</td>
      <td>"Probably Damaging"</td>
      <td><div class="score score_bad">0.95</div></td>
    </tr>
    <tr>
      <td>greater than 0.446 and less than or equal to 0.908</td>
      <td>"Possibly Damaging"</td>
      <td><div class="score score_ok">0.5</div></td>
    </tr>
    <tr class="bg2">
      <td>less than or equal to 0.446</td>
      <td>"Benign"</td>
      <td><div class="score score_good">0.25</div></td>
    </tr>
    <tr>
      <td>unknown</td>
      <td>"Unknown"</td>
      <td><div class="score score_neutral">unknown</div></td>
    </tr>
  </table>
</div>
</div>
<br />

<div style="max-width:1000px" class="info">
  <h3>Inconsistent predictions</h3>
  <div class="message-pad">
	  <p><b>PolyPhen</b> and <b>SIFT</b> results are heavily dependant on sequence conservation estimates derived from protein sequence alignments and using different versions of the protein databases can result in substantial variance in the predictions and scores obtained. Ensembl releases are not synchronised with updates to the SIFT and PolyPhen web servers, so <b>differences are to be expected</b>. Changes in calls will also be seen between some Ensembl releases due to updates in the software and databases used.</p>
	</div>
</div>

<br />

<h4 id="nsSNP_data_format">Prediction data format</h4>

<p>
The SIFT and PolyPhen predictions are precomputed and stored in the variation
databases and predictions are accessible in the variation API by using the <i>sift_prediction</i>, 
<i>sift_score</i>, <i>polyphen_prediction</i> and <i>polyphen_score</i> methods on a 
<i>Bio::EnsEMBL::Variation::TranscriptVariationAllele</i> object. For anyone wishing
to access the complete set of predictions from the MySQL database or API, an explanation of
the format used is provided here.
</p>


<p>
The predictions and associated scores are stored as a matrix, with a column for 
each possible alternate amino acid and a row for each position in the translation. Each prediction
for a position and amino acid is stored as a 2-byte value which encodes both the qualitative
prediction and score encoded as described below. A simple example matrix is shown in the figure 
below, though note we only show the decoded score rather than the actual binary value stored
in the database.
</p>

<p>
<a href="/img/protein_function_prediction_matrix.png" target="_blank">
  <img src="/img/protein_function_prediction_matrix.png" style="width:500px;heigth:143px" alt="protein function encoding"/>
</a>
</p>

<p>
Prediction matrices can be fetched and manipulated in a user-friendly manner using the variation API, specifically
using the <i>ProteinFunctionPredictionMatrixAdaptor</i> which allows you to fetch a prediction matrix
using either a transcript or a translation stable ID. This adaptor returns a <i>ProteinFunctionPredictionMatrix</i>
object and you can use the <i>get_prediction</i> method to retrieve a prediction for a given position
and amino acid. If you want to use entire matrices, you should use the <i>deserialize</i> method to decode
an entire binary formatted matrix into a simple Perl hash. Please refer to the <a
href="/info/docs/api/variation/index.html">API documentation</a> for both 
of these classes for more details. If you require direct access to the MySQL database (for instance 
because they are accessing the database in a language other than Perl) we provide a description of the 
binary format used below.
</p>

<p>
Prediction matrices for each translation from each of SIFT and PolyPhen are stored in the 
<i>protein_function_predictions</i> table. The analysis used to calculate the predictions
is identified in the <i>analysis_attrib_id</i> column which refers to a value stored in the
<i>attrib</i> table, and the protein sequence the predictions
apply to is identified by the <i>translation_md5_id</i> column which refers to a hexadecimal 
MD5 hash of the sequence stored in the <i>translation_md5</i> table. The 
prediction matrices are stored in the <i>prediction_matrix</i> column as a gzip compressed binary string.
Once uncompressed, this string contains a 40 byte substring for each row in the matrix
concatenated together in position order. Each row is composed of 20 2-byte predictions, 
one for each possible alternative amino acid in alphabetical order, though note that the 
value for the amino acid that matches the reference amino acid is 
identified as a 2-byte value with all bits set, or <i>0xFFFF</i> in hexadecimal notation. 
The prediction itself is packed as a 16 bit little-endian 
("VAX" order, or "v" format if using the perl <i>pack</i> subroutine) unsigned short value. 
The top two bits of this short 
are used to encode the qualitative prediction (PolyPhen has four possible values, while SIFT has just
two), and the bottom ten bits are used to encode the prediction score. To decode the qualitative 
prediction you should mask off all bits except the top two, and shift the resulting short
right by 14 bits and treat this as an integer between zero and three. The corresponding prediction can then 
be looked up in the table below. To decode the prediction score you should mask off the 
top six bits and the resulting value can be treated as a number between zero and 1000, which 
should be divided by 1000 to give a three decimal place score (casting to a floating point type
if necessary). Bits 11-14 are not used, except to encode the "same as reference" dummy prediction 
<i>0xFFFF</i>.
</p>

<img src="/img/protein_function_encoding.png" style="width:500px;heigth:143px" alt="protein function encoding"/>

<p>
A diagram of the encoding scheme is shown above. In this example from a polyphen prediction, 
the top prediction bits are <i>0b01</i> which in decimal is the number 1, which corresponds to 
"possibly damaging" in the table below. The score bits are <i>0b1110001010</i> which in decimal 
is the number 906, which when divided by 1000, gives a score of 0.906.
</p>

<table id="prediction_values" class="ss" style="width:50%;">
<tr>
  <th>Tool</th>
  <th>Encoded value</th>
  <th>Qualitative prediction</th>
</tr>
<tr>
  <td>PolyPhen</td>
  <td>0</td>
  <td>"probably damaging"</td>
</tr>
<tr class="bg2">
  <td>PolyPhen</td>
  <td>1</td>
  <td>"possibly damaging"</td>
</tr>
<tr>
  <td>PolyPhen</td>
  <td>2</td>
  <td>"benign"</td>
</tr>
<tr class="bg2">
  <td>PolyPhen</td>
  <td>3</td>
  <td>"unknown"</td>
</tr>
<tr>
  <td>SIFT</td>
  <td>0</td>
  <td>"tolerated"</td>
</tr>
<tr class="bg2">
  <td>SIFT</td>
  <td>1</td>
  <td>"deleterious"</td>
</tr>
</table>

<p>
To retrieve a prediction for a particular amino acid substitution at a given position
in a translation, first you must retrieve the binary matrix from the database and
uncompress it using gzip. You can calculate the offset into this string by multiplying 
the desired position (starting at zero) by 20 and then adding the index of the desired amino acid in an alphabetical ordering of
amino acids (also starting at zero), and then multiply this value by two to take into account the fact that each 
prediction uses two bytes. Each matrix also includes a three byte header used check if the data is corrupted etc. so
you will also need to add three to the calculated offset. The two bytes at the calculated position 
can then be extracted and the approach described above can be used to retrieve the prediction and score. 
A perl implementation of this scheme can be found in the <i>Bio::EnsEMBL::Variation::ProteinFunctionPredictionMatrix</i>
module in the variation API and should hopefully inform attempts to reimplement this 
scheme in other languages.
</p>


<br />
<h2 id="references">References</h2>

<ul>
  <li>
    <p>Adzhubei IA, Schmidt S, Peshkin L, Ramensky VE, Gerasimova A, Bork P, Kondrashov AS, Sunyaev SR. <br />
    <strong>A method and server for predicting damaging missense mutations</strong><br />
    <em>Nature Methods</em> 7(4):248-249 (2010)<br />
    <a href="http://dx.doi.org/10.1038/nmeth0410-248" rel="external">doi:10.1038/nmeth0410-248</a></p> 
  </li>

  <li>
    <p>Kumar P, Henikoff S, Ng PC.<br />
    <strong>Predicting the effects of coding non-synonymous variants on protein function using the SIFT algorithm</strong><br />
    <em>Nature Protocols</em> 4(8):1073-1081 (2009)<br />
    <a href="http://dx.doi.org/10.1038/nprot.2009.86" rel="external">doi:10.1038/nprot.2009.86</a></p> 
  </li>

</ul>

</body>
</html>
