<html>
<head>
<title>Canis familiaris site map</title>
</head>

<body>

<h2><i>Canis familiaris</i> site map</h2>

    <div class="col-wrapper">
    <div class="col3">
    
            <h3 class="boxed">Find a sequence or feature</h3>
        
            <dl class="species-map">
            <dt><a href="/Canis_familiaris/mapview?chr=11">MapView</a></dt>
                <dd>Explore a chromosome</dd>
                <dt><a href="/Canis_familiaris/idhistoryview?gene=ENSCAFG00000013459">IDHistoryView</a></dt>
                <dd>Track the history of an Ensembl stable ID</dd>
                <dt><a href="/Canis_familiaris/historyview?gene=ENSCAFG00000018725">HistoryView</a></dt>
                <dd>Convert a list of old Ensembl stable IDs</dd>
                </dl>

            <h3 class="boxed">Genes and Sequences</h3>
        
            <dl class="species-map">
            <dt><a href="/Canis_familiaris/geneview?gene=ENSCAFG00000001655">GeneView</a></dt>
                <dd>Gene report</dd>
                <dt><a href="/Canis_familiaris/geneseqview?gene=ENSCAFG00000019747">GeneSeqView</a></dt>
                <dd>Displays the sequence section from GeneView</dd>
                <dt><a href="/Canis_familiaris/genespliceview?gene=ENSCAFG00000000679">GeneSpliceView</a></dt>
                <dd>Alternative splicing for a given gene</dd>
                <dt><a href="/Canis_familiaris/exonview?exon=ENSCAFE00000136253">ExonView</a></dt>
                <dd>Exon report</dd>
                <dt><a href="/Canis_familiaris/transview?transcript=ENSCAFT00000003064">TransView</a></dt>
                <dd>Transcript report</dd>
                <dt><a href="/Canis_familiaris/contigview?region=contig_25411">ContigView</a></dt>
                <dd>Small-scale Sequence display</dd>
                <dt><a href="/Canis_familiaris/cytoview?l=12:4032951-4090542">CytoView</a></dt>
                <dd>Large-scale Sequence display</dd>
                <dt><a href="/Canis_familiaris/markerview?marker=REN59H07">MarkerView</a></dt>
                <dd>Information about a chromosome marker</dd>
                </dl>
</div>

    <div class="col3">
    
            <h3 class="boxed">Proteins</h3>
        
            <dl class="species-map">
            <dt><a href="/Canis_familiaris/protview?peptide=ENSCAFP00000015447">ProtView</a></dt>
                <dd>Protein Report</dd>
                <dt><a href="/Canis_familiaris/domainview?domainentry=IPR003560">DomainView</a></dt>
                <dd>Protein Domain report</dd>
                <dt><a href="/Canis_familiaris/familyview?family=ENSF00000000111">FamilyView</a></dt>
                <dd>Protein Family report</dd>
                </dl>

            <h3 class="boxed">Variation</h3>
        
            <dl class="species-map">
            <dt><a href="/Canis_familiaris/genesnpview?gene=ENSCAFG00000000001">GeneSNPView</a></dt>
                <dd>Lists all SNPs for a gene</dd>
                <dt><a href="/Canis_familiaris/transcriptsnpview?transcript=ENSCAFT00000025284">TranscriptSNPView</a></dt>
                <dd>Compare a transcript's SNPs in individuals / strains / populations</dd>
                <dt><a href="/Canis_familiaris/snpview?snp=rs8572765">SNPView</a></dt>
                <dd>SNP report</dd>
                </dl>

            <h3 class="boxed">Comparative genomics</h3>
        
            <dl class="species-map">
            <dt><a href="/Canis_familiaris/syntenyview?chr=1;otherspecies=Mus_musculus">SyntenyView</a></dt>
                <dd>Compare syntenous regions</dd>
                <dt><a href="/Canis_familiaris/multicontigview?l=9:26256001-26307366;s1=Homo_sapiens">MultiContigView</a></dt>
                <dd>Compare syntenous sequences (assembled genomes only) - accessible through SyntenyView</dd>
                </dl>
</div>

    <div class="col3">
    
            <h3 class="boxed">Feature displays</h3>
        
            <dl class="species-map">
            <dt><a href="/Canis_familiaris/featureview?type=OligoProbe;id=Cfa.17601.1.S1_s_at">FeatureView</a></dt>
                <dd>Display Ensembl features (on a karyotype if available)</dd>
                <dt><a href="/Canis_familiaris/karyoview">KaryoView</a></dt>
                <dd>Map your own data onto a karyotype</dd>
                </dl>


    <h3 class="boxed">Import/Export data</h3>
    
    <dl class="species-map">
<dt><a href="/info/data/external_data/index.html">DAS</a> (Distributed Annotation Server)</dt>

    <dd>Import your data into Ensembl</dd>
    <dt><a href="/Canis_familiaris/exportview">ExportView</a></dt>
    <dd>Export sequence data to file</dd>
    </dl>
    </div>


    </div>
    

</body>
</html>
