<html>
<head>
<title>ENCODE data in Ensembl</title>
</head>
<body>

<div class="column-wrapper">
<h1>ENCODE data in Ensembl</h1>
</div>
<div class="column-wrapper">
  <div class="column-two">
    <div class="column-left">
      <div class="plain-box">

<h2>How to view ENCODE data</h2> 

<p>Encode data can be attached to Region in Detail via a track hub - the link below will do this automatically:</p>

<p><a href="http://www.ensembl.org/Homo_sapiens/Location/View?g=ENSG00000130544;contigviewbottom=url:http://ftp.ebi.ac.uk/pub/databases/ensembl/encode/integration_data_jan2011/hub.txt;format=DATAHUB;menu=ENCODE%20data">Link to add ENCODE integrative analysis hub</a></p>

<p>This creates a menu in the Control Panel on Region in Detail, from which you
can add individual tracks or groups of tracks using matrix selectors.
Cell type and experimental factor are the two principal axes; other dimensions
can be selected by clicking on a box to open an additional submenu (see below).</p>

<img src="/img/matrix_encode.png" class="screenshot" style="margin-bottom:20px" />
       
<h3>Removing the track hub</h3>

<p>To turn off tracks, simply return to the configuration matrix you used earlier,
or hover your mouse over the track name on the lefthand side of the Region in Detail
image and click on the X in the popup menu that appears.</p>

<p>To remove the track hub completely, go to "Manage Your Data" and click on the trash can icon
on the "ENCODE data" row.</p>
<p class="button"><a href="/Homo_sapiens/UserData/ManageData"><img src="/i/16/rev/page-user.png" style="vertical-align:middle;padding:2px" /> Manage your data</a> 
&rarr; <img src="/i/16/trash.png" style="vertical-align:middle" /></p> 
      </div>
    </div>
  </div>
  <div class="column-two">
    <div class="column-right" style="padding-left:2em">

<h2>The ENCODE project</h2>

<p><a href="http://www.encodeproject.org"><img src="/img/ENCODE_logo.jpg" style="float:left;padding:2px;" /></a>
The <a href="http://www.encodeproject.org">Encyclopedia of DNA Elements (ENCODE) Consortium</a>
is an international collaboration of research groups funded by the National Human Genome Research 
Institute (NHGRI). The goal of ENCODE is to build a comprehensive parts list of functional elements 
in the human genome, including elements that act at the protein and RNA levels, and regulatory elements 
that control cells and circumstances in which a gene is active.</p>

<p>The ENCODE Project has produced genome-wide data for over 100 different cell types for
investigating different aspects of genomic regulation, including:</p>
<ul>
<li>chromatin structure (5C)</li>
<li>open chromatin (DNase-seq and FAIRE-seq)</li>
<li>histone modifications and DNA-binding of over 100 transcription factors (ChIP-seq)</li>
<li>RNA transcription (RNA-seq and CAGE)</li>
</ul>

<h2>ENCODE and Ensembl</h2>

<p>The Ensembl team has played an active role in the ENCODE project:</p>

<ul>
<li>The GENCODE project, which creates the gene sets used by the ENCODE Project Consortium and Ensembl, is headed by Tim Hubbard at the Wellcome Trust Sanger Institute. It uses the Ensembl Genebuild pipeline, RNA data from the ENCODE page and manual annotation provided by the HAVANA team (also at the Wellcome Trust Sanger Institute).</li>

<li>Members of the Ensembl Regulation team were involved in constructing and running the ENCODE uniform peak-calling pipeline, creating standarised and reproducible peak calls for the ENCODE Transcription Factor ChIP-seq data, and in creating the ENCODE Combined genome segmentation, using open chromatin, histone modification, RNA polymerase and CTCF ChIP-seq signal to assign functional annotations to regions of the human genome for the six main cell types.</li>

<li>Members of the Ensembl Compara team were involved in the identification of conserved regions and additional analyses for the ENCODE project.</li>

<li>Members of the Ensembl team were heavily involved in the ENCODE Data Analysis Centre, headed by Ewan Birney at the European Bioinformatics Institute, which organised all the integrative analysis conducted in the Consortium and hosted the analysed data.</li>
</ul>

<p>Ensembl uses data created by the ENCODE project in many of its pipelines, including:</p>
<ul>
<li>The Genebuild pipeline and annotation</li>

<li>The Ensembl Regulation Regulatory Build</li>
</ul>

<p>The regulatory build uses many ENCODE DNase-seq, FAIRE-seq and ChIP-seq short read files, as well as data from other projects/publications, as its input, before aligning and peak-calling for identification and annotation of regulatory features, and the peak-calls and signal tracks can be viewed using the Regulation section of "Configure This Page".</p>

<p>Ensembl Regulation also incorporates the ENCODE Combined genome segmentations, which can be viewed in the Ensembl website or accessed in BioMart or programatically through the API.</p>

    </div>
  </div>
</div>
