<html>
<head>
<title>What's New in Ensembl 33</title>
</head>

</body>
        <h2>What's New in Ensembl 33</h2>
<h3 class="boxed" id="cat0">Species News</h3>

<h4 id="item35"><i>Homo sapiens</i></h4>

<ul class="spaced">
<li><strong>New Human cDNAs</strong><br />Monthly update of the human cDNA database.</li>
<li><strong>Vega Human</strong><br />Updated Ensembl-Vega database for Homo sapiens.</li>
<li><strong>Gene descriptions</strong><br />The missing gene descriptions for ncRNAs have been added to the Human core database.</li>
</ul>

<h4 id="item38"><i>Mus musculus</i></h4>

<ul class="spaced">
<li><strong>Mouse BAC Clones</strong><br />Entries have been added to the assembly so that BAC clones can be fetched.</li>
<li><strong>Transcript supporting evidence</strong><br />The supporting evidence table IDs have been adjusted so that all transcripts show the correct evidence.</li>
</ul>

<h4 id="item31"><i>Danio rerio</i></h4>

<ul class="spaced">
<li><strong>Updates to gene table</strong><br />The Zebrafish core and est databases had incorrect values for biotype and confidence in the gene table, resulting in EST genes being displayed in the wrong colour and without exons. This has been corrected in release 33.</li>
</ul>

<h4 id="item33"><i>Rattus norvegicus</i></h4>

<ul class="spaced">
<li><strong>Xrefs re-run</strong><br />Rat xrefs have been re-run to correct problems with RGD mappings.</li>
</ul>

<h4 id="item32"><i>Drosophila melanogaster</i></h4>

<ul class="spaced">
<li><strong>New Xrefs</strong><br />Additional xrefs for fly.</li>
</ul>

<h4 id="item34"><i>Gallus gallus</i></h4>

<ul class="spaced">
<li><strong>Xrefs re-run</strong><br />Chicken xrefs have been re-run to improve assignment of display xrefs, particularly for HOX genes.</li>
</ul>

<h4 id="item29"><i>Ciona intestinalis</i></h4>

<ul class="spaced">
<li><strong>New Xrefs</strong><br />Ensembl <i>Ciona</i> gene predictions which match the JGI's predictions on assembly 1.0 now have xrefs which link through to the JGI website.</li>
<li><strong>Gene descriptions</strong><br />The truncated descriptions in the Ciona gene table have been replaced with full descriptions.</li>
</ul>

<h3 class="boxed" id="cat2">Data updates</h3>

<h4>HUGO name changes</h4>
<p>The display name for the HUGO database has been changed to 'HGNC symbol' in order to match current usage by SwissProt, etc.</p>

<h4>New variation databases</h4>
<p>All species have new variation databases.</p>

<h4>Affy mapping</h4>
<p>Some AffyMetrix data was accidentally omitted from the last release of Ensembl (v32, July 2005). These data have been re-computed and added to the core databases of the affected species.</p>

<h4>Fixes to affy_feature</h4>
<p>The affy_feature table in the core database had no values in the 'mismatches' column in the last release; the correct values have been added.</p>

<h4>Compara</h4>
<p>Addition of paralogues for mouse, drosophila and zebrafish.</p>

<h3 class="boxed" id="cat4">Web features</h3>

<h4>Bug fixes</h4>
<p>A number of small bugs in the new-style view pages have been fixed, including missing popup menus in SyntenyView.</p>

<h3 class="boxed" id="cat1">API/web code changes</h3>

<h4>Perl method status</h4>
<p>The Perldoc for the Perl API has been expanded to indicate the status of each method: 'stable', 'at risk', or 'medium risk'.</p>

<h3 class="boxed" id="cat3">Schema changes</h3>

<h4>New column in external_db</h4>
<p>A new column, db_display_name, has been added to external_db, to replace the hardcoded web code hashmap previously used, and the API updated to support the change.</p>



</body>
</html>
