<html>
<head>
<title>Bos taurus site map</title>
</head>

<body>

<h2><i>Bos taurus</i> site map</h2>

    <div class="col-wrapper">
    <div class="col3">
    
            <h3 class="boxed">Find a sequence or feature</h3>
        
            <dl class="species-map">
            <dt><a href="/Bos_taurus/mapview?chr=18">MapView</a></dt>
                <dd>Explore a chromosome</dd>
                <dt><a href="/Bos_taurus/idhistoryview?gene=ENSBTAG00000021527">IDHistoryView</a></dt>
                <dd>Track the history of an Ensembl stable ID</dd>
                <dt><a href="/Bos_taurus/historyview">HistoryView</a></dt>
                <dd>Convert a list of old Ensembl stable IDs</dd>
                </dl>

            <h3 class="boxed">Genes and Sequences</h3>
        
            <dl class="species-map">
            <dt><a href="/Bos_taurus/geneview?gene=ENSBTAG00000021527">GeneView</a></dt>
                <dd>Gene report</dd>
                <dt><a href="/Bos_taurus/geneseqview?gene=ENSBTAG00000021527">GeneSeqView</a></dt>
                <dd>Displays the sequence section from GeneView</dd>
                <dt><a href="/Bos_taurus/genespliceview?gene=ENSBTAG00000013489">GeneSpliceView</a></dt>
                <dd>Alternative splicing for a given gene</dd>
                <dt><a href="/Bos_taurus/">GeneRegulationView</a></dt>
                <dd>Regulatory factors for a given gene</dd>
                <dt><a href="/Bos_taurus/exonview?exon=ENSBTAE00000004497">ExonView</a></dt>
                <dd>Exon report</dd>
                <dt><a href="/Bos_taurus/transview?transcript=ENSBTAT00000028690">TransView</a></dt>
                <dd>Transcript report</dd>
                <dt><a href="/Bos_taurus/contigview?region=AAFC03063255">ContigView</a></dt>
                <dd>Small-scale Sequence display</dd>
                <dt><a href="/Bos_taurus/cytoview?region=AAFC03063255">CytoView</a></dt>
                <dd>Large-scale Sequence display</dd>
                <dt><a href="/Bos_taurus/markerview?marker=BE217550">MarkerView</a></dt>
                <dd>Information about a chromosome marker</dd>
                </dl>
</div>

    <div class="col3">
    

    <h3 class="boxed">Import/Export data</h3>
    
    <dl class="species-map">
<dt><a href="/info/using/external_data/index.html">DAS</a> (Distributed Annotation Server)</dt>

    <dd>Import your data into Ensembl</dd>
    <dt><a href="/Bos_taurus/exportview">ExportView</a></dt>
    <dd>Export sequence data to file</dd>
    </dl>
    
            <h3 class="boxed">Proteins</h3>
        
            <dl class="species-map">
            <dt><a href="/Bos_taurus/protview?peptide=ENSBTAP00000012528">ProtView</a></dt>
                <dd>Protein Report</dd>
                <dt><a href="/Bos_taurus/domainview?domainentry=IPR000569">DomainView</a></dt>
                <dd>Protein Domain report</dd>
                <dt><a href="/Bos_taurus/familyview?family=ENSF00000000493">FamilyView</a></dt>
                <dd>Protein Family report</dd>
                </dl>

            <h3 class="boxed">Feature displays</h3>
        
            <dl class="species-map">
            <dt><a href="/Bos_taurus/featureview?id=Bt.22329.1.S1_at;type=OligoProbe">FeatureView</a></dt>
                <dd>Display Ensembl features (on a karyotype if available)</dd>
                <dt><a href="/Bos_taurus/karyoview">KaryoView</a></dt>
                <dd>Map your own data onto a karyotype</dd>
                </dl>
</div>

    <div class="col3">
    
            <h3 class="boxed">Variation</h3>
        
            <dl class="species-map">
            <dt><a href="/Bos_taurus/">GeneSNPView</a></dt>
                <dd>Lists all SNPs for a gene</dd>
                <dt><a href="/Bos_taurus/transcriptsnpview?transcript=ENSBTAT00000024948">TranscriptSNPView</a></dt>
                <dd>Compare a transcript's SNPs in individuals / strains / populations</dd>
                <dt><a href="/Bos_taurus/snpview?snp=rs42639218">SNPView</a></dt>
                <dd>SNP report</dd>
                <dt><a href="/Bos_taurus/">LDView</a></dt>
                <dd>Diagram of LD values for a SNP</dd>
                <dt><a href="/Bos_taurus/">LDTableView</a></dt>
                <dd>Table of LD values for a SNP</dd>
                </dl>

            <h3 class="boxed">Comparative genomics</h3>
        
            <dl class="species-map">
            <dt><a href="/Bos_taurus/syntenyview?chr=8;other_species=Homo_sapiens">SyntenyView</a></dt>
                <dd>Compare syntenous regions</dd>
                <dt><a href="/Bos_taurus/multicontigview?l=Un:206954408-206961364;s1=Homo_sapiens">MultiContigView</a></dt>
                <dd>Compare syntenous sequences (assembled genomes only) - accessible through SyntenyView</dd>
                <dt><a href="/Bos_taurus/genetreeview?gene=ENSBTAG00000012865">GeneTreeView</a></dt>
                <dd>Tree of orthologues for a gene</dd>
                <dt><a href="/Bos_taurus/geneseqalignview?gene=ENSBTAG00000029106">GeneSeqAlignView</a></dt>
                <dd>Compare the DNA sequence of orthologues</dd>
                </dl>
</div>


    </div>
    

</body>
</html>
