
  <h3 class="boxed">About the Opossum genome</h3>

<h4>Assembly</h4>

<p><img src="/img/species/pic_Monodelphis_domestica.png" height="87" width="140" class="float-left" alt="Opossum" title="choo choo" />

This site presents the latest genome assembly (MonDom5) of the gray short-tailed opossum, released in October 2006. Scaffold and conyigs remain the same as the previous release; only mapping changes have been made.</p>

<p>The assembly, from the <a href="http://www.broad.mit.edu">Broad Institute</a>, has a base coverage of approximately 7.33X, constructed from 5180 supercontigs, having N50 length 59,092,948.  The total contig length is 3,501,487,001, spanning 3,573,120,728 bases (including gaps).
</p>

<h4>Annotation</h4>
<p>The current gene set for Opossum is projected from the MonDom4 geneset, which was built using a modified version of the standard Ensembl genebuild pipeline. The species-specific sequence resources (opossum cDNA and protein) are very limited, so the vast majority of gene models are based on genewise alignments of proteins from other species. Most of the proteins being aligned were from species genetically distant to opossum. To improve the accuracy of models generated from these proteins, the genewise alignments were made to stretches of genomic sequence rather than to 'miniseqs'. Opossum and human cDNAs were aligned and used to add UTRs to the genewise predictions where possible. The gene models were assessed by generating sets of potential orthologs to genes from other mammalian species. Potentially missing predictions and partial gene predictions were identified by examining the orthologs, and exonerate used to build new gene models for these based on the human ortholog peptide sequence.</p>
