<html>
<head>
<title>Identifying Alternative Splicing Events</title>
</head>
<body>

<h1>Alternative Splicing Event Set</h1>

<h2 id="Introduction">Introduction</h2>
<p>

The set of alternative splicing events (AS events) is derived from Ensembl 
<a href="/info/docs/genebuild/genome_annotation.html">gene annotation</a>. 
The Ensembl gene set includes both automatic and manual annotation, with all 
transcripts based on experimental evidence. Alternatively spliced transcripts 
of the same Ensembl gene overlap in coding sequence.  These data can be found 
on the website in the gene and/or transcript tab, and can be accessed by BioMart, 
or programmatically from the core database using the Perl API.
</p>
<p>
AS events are available for the following model organisms:
<ul>
<li>Caenorhabditis elegans</li>
<li>Danio rerio</li>
<li>Drosophila melanogaster</li>
<li>Homo sapiens</li>
<li>Mus musculus</li>
<li>Rattus norvegicus</li>
</ul>
</p>

<h2 id="Definition">Common Alternative Spliced Patterns</h2>
<p>
Seven common AS patterns are automatically annotated in Ensembl for 
the species mentioned:
</p>
<table>
<tr>
<th>AS Pattern Type</th><th>Acronym</th><th>Definition</th>
</tr>
<tr><td>Cassette exon (skipped exon)</td><td>CE</td><td>One exon is spliced out of the primary transcript together with its flanking introns.</td></tr>
<tr><td>Intron retention</td><td>IR</td><td>A sequence is spliced out as an intron or remains in the mature mRNA transcript.</td></tr>
<tr><td>Mutually exclusive exons</td><td>MXE</td><td>In the simpliest case, one or two consecutive exons are retained but not both.</td></tr>
<tr><td>Alternative 3' sites</td><td>A3SS</td><td>Also called alternatively acceptor sites. Two or more splice sites are recognized at the 5' end of an exon. An alternative 3' splice junction (acceptor site) is used, changing the 5' boundary of the downstream exon.</td></tr>
<tr><td>Alternative 5' sites</td><td>A5SS</td><td>Also called alternative donor sites. Two or more splice sites are recognized at the 3' end of an exon. An alternative 5' splice junction (donor site) is used, changing the 3' boundary of the upstream exon.</td></tr>
<tr><td>Alternative first exon</td><td>AFE</td><td>A pattern of mutually exclusive exon whereby second exons of each variant have identical boundaries but the first exons are mutually exclusive. This is a to allow annotation of possible alternative promoter usage*.</td></tr>
<tr><td>Alternative last exon</td><td>ALE</td><td>Penultimate exons of each splice variant have identical boundaries, but the last exons are mutually exclusive. This is a to allow annotation of possible alternative polyadenylation usage*.</td></tr>
</table>
<p>

<p>
* AFE and ALE event types are difficult to interpret and biological evidence is needed to support these predictions.
</p>

<p>
Ensembl imports two additional types of AS events from the ASTD project ([1]).
</p>
<table>
<tr>
<th>AS Pattern Type</th><th>Acronym</th><th>Definition</th>
</tr> 
<tr><td>Intron isoform</td><td>II</td><td>Alternative donor/acceptor splice sites leads to truncation/extension of introns. This is for introns with truncation and/or extension on <b>both</b> splice sites.</td></tr>
<tr><td>Exon isoform</td><td>EI</td><td>Alternative donor/acceptor splice sites leads to truncation/extension of exons. This is for exons with truncation and/or extension on <b>both</b> splice sites. First and last exons are never in this computation. Flanking exons must overlap.</td></tr>
</table>

<p>
Different acronyms for the same type of AS event can be found in the 
scientific literature and in several online related resources. 
Ensembl uses CE, IR, MXE, A3SS, A5SS, AFE, ALE, II, EI as standard acronyms. 
Similar acronyms can be found in [1] and [2].

<p>
<strong>References</strong>
</p>

<p>
1. Koscielny G, Le Texier V, Gopalakrishnan C, Kumanduri V, Riethoven JJ, Nardone F, Stanley E, Fallsehr C, Hofmann O, Kull M, Harrington E, Boué S, Eyras E, Plass M, Lopez F, Ritchie W, Moucadel V, Ara T, Pospisil H, Herrmann A, G Reich J, Guigó R, Bork P, Doeberitz MK, Vilo J, Hide W, Apweiler R, Thanaraj TA, Gautheret D. ASTD: The Alternative Splicing and Transcript Diversity database. Genomics. 2009 Mar;93(3):213-20.
</p>

<p>
2. Wang ET, Sandberg R, Luo S, Khrebtukova I, Zhang L, Mayr C, Kingsmore SF, Schroth GP, Burge CB. Alternative Isoform Regulation in Human Tissue Transcriptomes. Nature 456, 470-476.
</p>

</body>
</html>
