<html>
<head>
<title>Genome Annotation</title>
</head>

<body>

<h1>Ensembl Gene Set</h1>

<h2 id="Introduction">Introduction</h2>
<p>
Gene annotation provided by Ensembl includes both automatic annotation, 
i.e. genome-wide determination of transcripts, and manual curation, i.e. 
reviewed determination of transcripts on a case-by-case basis (in this 
case limited to some species such as human and mouse). Furthermore, 
Ensembl imports annotation from FlyBase (http://flybase.org/), WormBase 
(http://www.wormbase.org/) and SGD (http://www.yeastgenome.org/). 
Ensembl transcripts displayed on our website are products of the Ensembl 
automatic pipeline, termed the Ensembl genebuild. All Ensembl 
transcripts are based on experimental evidence and thus the automated 
pipeline relies on the mRNAs and protein sequences deposited into public 
databases from the scientific community. Manually-curated transcripts 
produced by the <a rel="external" href="http://www.sanger.ac.uk/HGP/havana/">HAVANA</a> 
group at the Welcome Trust Sanger Institute are displayed in a separate track on our website.
</p>

<h2 id="Annotation">Genebuild</h2>
<p>
Standard Ensembl genebuilds are performed on high-coverage genomes and usually take at least four months to complete. While the genebuild process is tailored to each species according to the data that are available, the main steps followed in a standard genebuild are the same for every species.

The initial set-up for a genebuild involves loading the assembly into an Ensembl schema datababase and then running several analyses across the genome including masking of repeats and ab-initio gene predictions. The ab-initio gene predictions are run for website display purposes only and are not used as supporting evidence for the gene structures displayed on our website.

The first stage of the genebuild is known as the Targetted stage. Here, species-specific proteins are aligned to the genome and Genewise is used to build a transcript structure for the protein on the genome. 

The Targetted stage is followed by the Similarity stage in which proteins from closely related species are used to build transcript structure in regions were a Targetted transcript structure is absent. For those species having a lot of experimentally-generated protein sequences, the Targetted stage tends to contribute most of the gene structures in the genebuild. However, for species with fewer species-specific protein sequences the Similarity stage plays a much more important role in predicting gene structures.

The next stage in the genebuild is to align species-specific cDNA and EST sequences to the genome. Where cDNA alignments overlap transcripts predicted in the preceding stages, any non-translated region from the cDNA is spliced onto the transcript prediction as UTR. EST alignments are displayed on our website but are usually not used as supporting evidence in the genebuild.

The final set of gene predictions is obtained by merging identical transcripts built from different proteins sequences to produce multi-transcript gene predictions, each with a non-redundant set of transcripts models. For every transcript model, the protein and mRNA sequences used to predict the model is viewable in the browser as 'supporting evidence'.  These can also be obtained using the <a href="/info/data/api/">Perl API</a>.

Once a 'final' gene set has been obtained, a number of post-processing procedures are applied to filter and annotate the predicted genes. These include protein domain annotation, ncRNA annotation, pseudogene prediction and cross-referencing to external databases.
</p>
<p>
The human and mouse genebuilds involve additional steps not included in the standard Ensembl build. For both species, transcripts from the Consensus Coding Sequence (<a href="ccds.html">CCDS</a>) set are imported directly and not altered by the genebuild process. In addition, where manual curation is available for a transcript, the Ensembl and HAVANA transcript models are compared. The Ensembl and HAVANA models are merged when they agree on the same coding sequence. Merged models will be coloured gold in the browser.  A merged, or golden, gene indicates one or more common transcripts between Ensembl and HAVANA.

</p>
The genebuild process for <a href="2x_genomes.html">low-coverage genomes</a> does not follow that of the standard Ensembl genebuild described above. 
</p>

<ul>
<li><a href="http://www.ensembl.org/info/about/publications.html#Genome_Res_14_934">References</a></li>
</ul>

<h2 id="Sources">Sources of Supporting Evidence for a Transcript</h2>
<p>
Ensembl transcripts are based on mRNA and protein from the following databases.  
If you do not find a transcript that you expected in Ensembl, make sure there is sequence submitted into these databases.
If the sequence is missing, consider submitting your experimental evidence to <a rel="external" href="http://www.ebi.ac.uk/embl/Submission/index.html">EMBL-Bank</a>.
</p>

<ul>
<li>
<a rel="external" href="http://www.ebi.ac.uk/embl/">EMBL</a> Nucleotide Sequence Database is the European aspect of the International Nucleotide Sequence Database Consortium, <a rel="external" href="http://www.ebi.ac.uk/embl/Contact/collaboration.html">INSDC</a>, and is maintained at the European Bioinformatics Institute, the <a rel="external" href="http://www.ebi.ac.uk/">EBI</a>, near Cambridge, UK.
All sequence records are synchronised with the other partners in the INSDC, namely NCBI <a rel="external" href="http://www.ncbi.nlm.nih.gov/Genbank/">GenBank</a> in North America and <a href="http://www.ddbj.nig.ac.jp/" rel="external">DDBJ</a> in Japan.
</li>
<li>
The <a rel="external" href="http://www.uniprot.org/">UniProtKB</a> houses a collection of reviewed, manually annotated protein sequences, UniProtKB/Swiss-Prot, and a set of unreviewed proteins translated from the EMBL/GenBank/DDBJ nucleotide set, UniProtKB/TrEMBL.
</li>
<li>
NCBI <a rel="external" href="http://www.ncbi.nlm.nih.gov/RefSeq/">RefSeq</a> aims to provide a comprehensive set of mRNA and proteins.  Manually annotated, reviewed proteins have an ID beginning with NP, known protein, while mRNAs in this catagory begin with NM.  
Predicted proteins and mRNA transcripts are not used in the Ensembl Genebuild, and begin with XP and XM, respectively.
</li>
</ul>


<h2 id="Names">Gene Names</h2>
<p>
Most human genes have an associated HGNC symbol from the <a rel="external" href="http://www.genenames.org/">HUGO Gene Nomenclature Committee</a>. 
If the gene or transcript comes from the Ensembl Genebuild's automated pipeline only, the HGNC symbol will have HGNC automatic associated with it. 
If it is a manually curated transcript from Vega/Havana, it will have HGNC curated associated with it.
</p>
<p>
'Clone-based' identifiers apply to transcripts that cannot be associated with an HGNC symbol and are either assigned by Ensembl or Vega, as above.
</p>
<p>
The list of gene name catagories for human is as follows:
</p>
<ol>
<li>HGNC automatic</li>
<li>HGNC curated</li>
<li>Clone-based ensembl</li>
<li>Clone-based vega</li>
</ol>
<p>
A new number follows each transcript name. If the number starts with '0', e.g. 001, 002, etc., this is a manually curated transcript from Vega/Havana. 
If the number starts with '2', such as 200, 201,..., this is an automatically annotated transcript from Ensembl. 
This does not take the place of the Ensembl gene ID, ENSG..., which is stable from release to release, and has not changed.
The goal is to be more consistent with the naming of our manually curated imports from Vega.
</p>

<h2 id="Status and colour">Gene Status and Colours</h2>

<p>
Ensembl genes and transcripts are <strong>known, in red or gold</strong>, or <strong>novel, in black</strong>.
</p> 
<p>
A <strong>known</strong> gene or transcript is coloured <strong>red</strong> and has a sequence match in a public sequence database such as UniProtKB and NCBI RefSeq. 
The match must be for the same species, otherwise the gene is considered to be novel 
for that species.  <strong>Novel</strong> genes and transcripts are coloured <strong>black</strong>.  
<strong>Gold</strong> genes or transcripts indicate that the model produced by the Ensembl genbuild
has identical coding sequence (CDS) to a manually annotated transcript. Comparison of Ensembl-produced gene and transcript models to manually curated data is limited to human and mouse, and thus gold gene and transcript models will only be seen for these species.

Known miRNAs must have a match in miRBase for the same species. 
Matches to IDs in other databases, such as UniProtKB and NCBI EntrezGene, are listed in a link from the Transcript page. 
</p>
<p>
NOTE: Names and known/novel status are assigned to a transcript AFTER it is determined using either the Ensembl genebuild or reviewed, manual curation.
Names are obtained by comparing Ensembl transcript sequence to sequence in other databases.  If a sequence match is found for the same species, the transcript is 'known'.
If not, the transcript is 'novel'.
</p>

<h2 id="Other">Other transcripts</h2>
<p>
Ig segments, non-coding RNA and transient EST gene models are also automatically annotated by Ensembl.  The genes based on ESTs, or Expressed Sequence Tags, can be a good starting point to determine new isoforms.  However, these models are not as robust as the known and novel protein-coding set, as ESTs have a high degree of error in their sequence.  This may not be the case for some human sets, but in the majority, the mRNA and protein sequences used in the main gene set determination are more reliable.  EST genes have 'EST' in the identifier, such as 'ENSESTG' for a human Ensembl EST gene.  They can also be used to identify the presence of mRNA, as a confirmation that a transcript is expressed.
</p>
<p>
Transcripts are scanned for pseudogene behaviour or false coding sequence.  A transcript is considered a <strong>pseudogene </strong>if 1 of these 4 criteria are met:
<ol>
<li>It is a single exon transcript and it matches a multi-exon transcript elsewhere in the genome.
</li>
<li>
The transcript  is completely masked out by repeat masker.
</li>
<li>
The transcript contians no introns and multiple frameshifts.
</li>
<li>
The transcript contains frameshifts and all of the introns are &gt;80% covered by repeats.
</li>
</ol>


</body>
</html>
