<html>
<head>
<title>Zebrafish RNASeq Gene ModelsGenome Annotation</title>
</head>

<body>

<h1>Zebrafish RNASeq Gene Models</h1>

<p>This is an experimental set of gene models produced using paired end Illumina RNASeq data from the Wellcome Trust Sanger Institute Zebrafish Transcriptome Sequencing Project Ref: <a rel="external"  href="http://www.ebi.ac.uk/ena/data/view/display=html&ERP000016">ERP000016</a>.</p>

<p>The models are produced from a 2 step alignment process using <a rel="external" href="http://www.biomedcentral.com/1471-2105/6/31">Exonerate</a>. First, a local genomic alignment is created that is collapsed to create alignment blocks roughly corresponding to exons. Read pairing information is used to group exons into approximate transcript structures. Secondly, reads are realigned to the proto-transcripts using a splice model and a short word length to create a set of spliced alignments representing canonical and non-canonical introns.</p>

<p>Gene models are created by combining the proto-transcripts with the spliced reads to create all possible variants, the variant with the most read support is displayed.</p>

<h1>Exon Supporting Features</h1>

<p>These show the level of read support for each RNAseq gene model on the exon and transcript level. The features are created by realigning all the reads against the models in a tissue specific way, this is then used to create an RPKM value.  (reads per kilobase of model per million mapped reads, from Mortazavi Nature Methods 2008).</p>
<p>Not all exons or transcripts will have expression in all tissues.</p>

<h1>Intron Supporting Features</h1>

<p>These represent the collapsed set of spliced reads used to inform the gene models. The features show the number of reads from each tissue that confirm a particular intron. Not all introns show expression in all tissues. Also, not all of the introns features were used in the gene models shown.</p>
