<h3 class="boxed">What's New in Ensembl 37</h3>
<h4><i>Caenorhabditis elegans</i> News</h4><ul class="spaced flush">
<li><strong>Import of Wormbase 150</strong><br />Ensembl <i>C.elegans</i> has been updated with an import of the Wormbase release of November 2005. New data includes orthologous relationships identified by the InParanoid algorithm.</li>
<li><strong>SSAHA2</strong><br />Transcript-based SSAHA2 servers are now available for all species. Go to <a href="/Multi/blastview">BlastView</a> and select SSAHA2 from the Search Tool box.</li>
<li><strong>UniProt updates</strong><br /> Support has been added for variable splice UniProt records by adding an external db with a display label of  "UniprotKB/SpliceVariant" to the external_db table on all databases.<br /><a href="/Multi/newsview?rel=37#item603">Read more</a>...
</li>
<li><strong>GeneSeqalignView</strong><br />This new view is based on GeneSeqView, with the added ability to align the given sequence with one or more species.<br /><a href="/Multi/newsview?rel=37#item614">Read more</a>...
</li>
<li><strong>Bug fix to Transcript.pm</strong><br />Transcript.pm; is_known() was returning true if the 
transcript had a display_xref set, which is both incorrect and 
inconsistent with the is_known() method in Gene.pm. The new behaviour is to return true if the transcript's status is KNOWN.</li>
</ul>

<p><a href="/Caenorhabditis_elegans/newsview?rel=37">More news...</a></p>

