<html>
<head>
<title>Other information</title>
<meta name="order" content="9" />
</head>
<body>
<div>
  
  <div style="float:right"><img src="/img/vep_logo.png"/></div>
  
  <h1 id="top"><span style="color:#006">Variant Effect Predictor</span> <img src="/i/16/info.png"> <span style="color:#666">Other information</span></h1>
  <hr/>
  <h2 id="assembly">Species with multiple assemblies</h2>
  
  <p> With the arrival of GRCh38, Ensembl now supports two different assembly
  versions for the human genome while users transition from GRCh37. We provide a
  VEP cache download on the latest software version
  ([[SPECIESDEFS::ENSEMBL_VERSION]]) for both assembly versions. </p>
  
  <p> The <a href="vep_download.html#installer">VEP installer</a> will install
  and set up the correct cache and FASTA file for your assembly of interest. If
  using the --AUTO functionality to install without prompts, remember to add the
  assembly version required using e.g. "--ASSEMBLY GRCh37". It is also possible
  to have concurrent installations of caches from both assemblies; just use the
  <a href="vep_options.html#opt_assembly">--assembly</a> to select the correct
  one when you run the VEP script. </p>
  
  <p> Once you have installed the relevant cache and FASTA file, you are then
  able to use the VEP as normal. For those using GRCh37 and requiring database
  access in addition to the cache (for example, to look up variant identifiers
  using <a href="vep_options.html#opt_format">--format id</a>, see <a
  href="vep_cache.html#limitations">cache limitations</a>), the script will warn
  you that you must change the database port in order to connect to the correct
  database: </p>
  
  <pre class="code">ERROR: Cache assembly version (GRCh37) and database or selected assembly version (GRCh38) do not match

If using human GRCh37 add "--port 3337" to use the GRCh37 database, or --offline to avoid database connection entirely</pre>
  
  <p> For users looking to move their data between assemblies, Ensembl provides
  an assembly converter tool - if you've downloaded the VEP, then you have it
  already! The script is found in the ensembl-tools/scripts/assembly_converter
  folder. There is also an <a href="/info/docs/tools/index.html">online version
  of the tool</a> available. Both UCSC (<a
  href="https://genome.ucsc.edu/util.html" rel="external">liftOver</a>) and NCBI
  (<a href="http://www.ncbi.nlm.nih.gov/genome/tools/remap"
  rel="external">Remap</a>) also provide tools for converting data between
  assemblies. </p>
  
  <hr/>
    
  <h2 id="pick">Producing one annotation per variant</h2>
  
  <p> By default the VEP is configured to provide annotation on every genomic
  feature that each input variant overlaps. This means that if a variant
  overlaps a gene with multiple alternate splicing variants (transcripts), then
  a block of annotation for each of these transcripts is reported in the output.
  In the <a href="../vep_formats.html#output">default VEP output format</a> each
  of these blocks is written on a single line of output; in
  <a href="../vep_formats.html#vcfout">VCF output format</a> the blocks are
  separated by commas in the INFO field. </p>
  
  <p> For many users, however, this depth of annotation is not required, and to
  this end the VEP provides a number of options to reduce the amount of output
  produced. Which to choose depends on your motivations and requirements on the
  output. </p>
  
  <p> <b>NB:</b> Wherever possible we would discourage users from summarising
  data in this way. Summarising inevitably involves data loss, and invariably at
  some point this will lead to the loss of biologically relevant information. For
  example, if your variant overlaps both a regulatory feature and a transcript
  and you use one of the flags below, the overlap with the regulatory feature
  will be lost in your output, when in some cases this may be a clue to the
  "real" functional effect of your variant. For these reasons we encourage users
  to use one of the flagging options
  (<a href="vep_options.html#opt_flag_pick">--flag_pick</a> or
  <a href="vep_options.html#opt_flag_pick_allele">--flag_pick_allele</a>) and to
  post-filter results. </p>
  
  <ul>
    <li><b><a href="vep_options.html#opt_pick">--pick</a></b>: this is the optionwe anticipate will be of use to most users. The VEP chooses one block of annotation per variant, using an ordered set of criteria. This order may be customised using <a href="vep_options.html#opt_pick_order">--pick_order</a>.
      <ol>
        <li>canonical status of transcript</li>
        <li>transcript support level</li>
        <li>biotype of transcript (protein_coding preferred)</li>
        <li>consequence rank according to <a href="/info/docs/variation/predicted_data.html#consequences">this table</a></li>
        <li>transcript or feature length (longer preferred)</li>
      </ol>
    </li>
    
    <li><b><a href="vep_options.html#opt_pick_allele">--pick_allele</a></b>: as above, but chooses one consequence block per variant allele. This can be useful for <a href="../vep_formats.html#vcf">VCF input files</a> with more than one ALT allele</li>
    
    <li><b><a href="vep_options.html#opt_flag_pick">--flag_pick</a></b>: instead of choosing one block and removing the others, this option adds a flag &quot;PICK=1&quot; to picked annotation block, allowing users to easily filter on this later using the VEP's <a href="vep_filter.html">filtering script</a></li>
    
    <li><b><a href="vep_options.html#opt_flag_pick_allele">--flag_pick_allele</a></b>: as above, but flags one block per allele</li>
    
    <li><b><a href="vep_options.html#opt_per_gene">--per_gene</a></b>: as <a href="vep_options.html#opt_pick">--pick</a>, but chooses one annotation block per gene that the input variant overlaps</li>
    
    <li><b><a href="vep_options.html#opt_most_severe">--most_severe</a></b>: this flag reports only the consequence type of the block with the highest rank, according to <a href="/info/docs/variation/predicted_data.html#consequences">this table</a>. Feature-specific annotation is absent from the output using this flag, so use with caution!</li>
    
    <li><b><a href="vep_options.html#opt_summary">--summary</a></b>: this flag reports only a comma-separated list of the consequence types predicted for this variant. Feature-specific annotation is absent from the output using this flag, so use with caution!</li>
  </ul>
  
  <hr/>
    
  <h2 id="hgvs">HGVS notations</h2>
  
  <p> The VEP script supports using HGVS notations as input. This feature is
  currently under development, and not all HGVS notation types are supported.
  Specifically, only notations relative to genomic (g) or coding (c) sequences
  are currently supported; protein (p) notations are supported in limited
  fashion due to the complexity involved in determining the multiple possible
  underlying genomic sequence changes that could produce a single protein
  change. The script will warn the user if it fails to parse a particular
  notation. </p>
  
  <p> By default the VEP script uses Ensembl transcripts as its reference for
  determining consequences, and hence also for HGVS notations. However, it is
  possible to parse HGVS notations that use RefSeq transcripts as the
  reference sequence by using the <a
  href="vep_options.html#opt_refseq">--refseq</a> flag when running the script.
  Such notations must include the version number of the transcript e.g. </p>
  
  <pre>NM_080794.3:c.1001C>T</pre>
  
  <p> where ".3" denotes that this is version 3 of the transcript NM_080794.
  <a href="#refseq">See below</a> for more details on how the VEP can use
  RefSeq transcripts. </p>
  
    <hr/>
  <h2 id="refseq">RefSeq transcripts</h2>
  
  <p> Ensembl produces Core schema databases containing alignments of RefSeq
  transcript objects to the reference genome. This is the <a
  href="/info/docs/api/core/index.html#est">otherfeatures database</a>, and is
  produced for human and mouse. The database also contains alignments of CCDS
  transcripts and Ensembl EST sequences - they may be included in your
  analysis using <a href="vep_options.html#opt_all_refseq">--all_refseq</a>.
  By passing the <a href="vep_options.html#opt_refseq">--refseq</a> flag when
  running the VEP script, these alternative transcripts will be used as the
  reference for predicting variant consequences. Gene IDs given in the output
  when using this option are generally NCBI GeneIDs. </p>
  
  <p> Users should note that RefSeq sequences may disagree with the reference
  sequence to which they are aligned, hence results generated when using this
  option should be interpreted with a degree of caution. A much more complex
  and stringent process is used to produce the main Ensembl Core database, and
  this should be used in preference to the RefSeq transcripts. </p>
  
  <p> SIFT and PolyPhen predictions and scores are now calculated and referred
  to internally using the translated sequence, so predictions are available
  using the <a href="vep_options.html#opt_refseq">--refseq</a> flag where the
  RefSeq translation matches the Ensembl translation (they will match in the
  vast majority of cases - most differences between Ensembl and RefSeq
  transcripts occur in non-coding regions). </p>
  
    <hr/>
  <h2 id="convert">File conversion</h2>
  
  <p> The VEP script can be used to <a
  href="vep_options.html#opt_convert">convert</a> files between the various
  formats that it parses. This may be useful for a user with, for example, a
  number of variants given in HGVS notation against RefSeq transcript
  identifiers. The conversion process allows these notations to be converted
  into genomic reference coordinates, and then used to predict consequences in
  the VEP against Ensembl transcripts.</p>
  
</div>

</body>
</html>
