<html>
<head>
<title>Common Feature Tables</title>
</head>

<body>

<h1>Common Feature Tables</h1>


<h2>The dna_align_feature table</h2>

<ul>

<li>
<strong>seq_region_id</strong>, <strong>seq_region_start</strong>, <strong>seq_region_end</strong>, <strong>seq_region_strand</strong>: 
sequence position of the alignment.
</li>

<li>
<strong>hit_name</strong> the name or primary id of the aligned sequence in the database, described in the associated analysis.
</li>

<li>
<strong>hit_start</strong>, <strong>hit_end</strong>, <strong>hit_strand</strong>: sequence position of the aligned stretch of DNA on the hit_name sequence.
</li>

<li>
<strong>score</strong>, <strong>evalue</strong>, <strong>perc_ident</strong> selected quality numbers for the alignment. 
The API supports score as a filter criterion.
</li>

<li>
<strong>cigar_line</strong> a string that describes the alignment. 
Pieces of matching sequence <strong>M</strong>, deleted bases <strong>D</strong> and inserted bases <strong>I</strong>.
</li>

</ul>


<h2> The protein_align_feature table</h2>

<p>
Describes an alignment between an external peptide sequence and a piece of Ensembl DNA sequence. 
It is equivalent to the dna_align_feature table apart from:
</p>

<ul>

<li>
<p>
there is no <strong>hit_strand</strong> as protein sequence cannot have a negative strand
</p>
</li>

<li>
<p>
the interpretation of the hit coordinates, as they are amino acid counts.
</p>
</li>

<li>
<p>
the cigar_string uses DNA basepair counts for the alignment description.
</p>
</li>

</ul>

<h2>The repeat_features and repeat_consensus tables</h2>

<ul>

<li>
<p>
Repetitive regions on the genome are stored in this table.
All repeats are further classified by the repeat_consensus table.
</p>
</li>

<li>
<p>
<strong>repeat_start</strong>, <strong>repeat_end</strong> and <strong>score</strong> are retrievable by the API but not used by it
(the API cannot filter by this criteria).
Repeats are retrievable by their different consensus sequence (type).
</p>
</li>

</ul>


<h2>The prediction_transcripts table</h2>

<ul>

<li>
<p>
A prediction transcript is a gene prediction of low quality where there is not much information provided (compared to the information for a gene, see below).
Prediction transcripts link to one or more prediction exons and 
 an analysis, which generated this prediction (and the prediction exons).
</p>
</li>

</ul>


<h2>Prediction exons</h2>

<ul>

<li>
<p>
<strong>rank</strong>  The position of the exon, starting from 1, 
in the transcript from the 5 prime end.
</p>
</li>

<li>
<p>
<strong>start_phase</strong> describes whether the prediction exon starts with the first, second or third base in a codon. 
<ul>
<li> Phase 0: starts with the first base</li>
<li> Phase 1: starts with the second base</li>
<li> Phase 2: starts with the third base of the codon</li>
</ul>
</p>
</li>

<li>
<p>
The <srong>score</strong> and <strong>pvalue</strong> are used to store scores for the prediction algorithm (no filter criteria).
</p>
</li>

</ul>

</body>
</html>
