<html>
<head>
<title>Gene Tables</title>
</head>

<body>

<h1>Gene Tables</h1>

<p>
These tables reflect the Ensembl gene model, which states that a gene contains one or more transcripts, which in turn contain one or more exons. 
Transcripts may have a translation for protein-coding genes or may not if they are pseudogenes or RNA genes.
All of the above objects can have a <strong>stable_id</strong> that Ensembl tries to map between different genebuilds. 
This is stored in separate tables because during the gene prediction process these IDs are not known and are uploaded later (as it is easier to upload than update statements).
</p>

<p>
Some table columns need further explanation:
</p>

<ul>

<li>
<strong>gene.type</strong> currently contains either <strong>ensembl</strong> or <strong>pseudogene</strong> and is in need of standardisation (as does the analysis link).
</li>

<li>
<strong>gene.display_xref_id</strong> links to the <strong>xref</strong> entry that gives this gene an external name. 
This also defines the gene as known.
</li>

<li>
<strong>transcript.display_xref_id</strong> same as for the gene.
</li>

<li>
<strong>exon_transcript</strong> table builds a many-to-many relationship between transcripts and exons. 
The same exon can be shared by many transcripts (in the same gene). 
Rank count begins at 1 and goes from the 5 prime end of transcript.
</li>

<li>
<strong>exon.phase</strong> (values 0, 1, 2 or -1 ).  
The phase of the exon.  If the first base in the exon is not coding its phase is -1.
</li>

<li>
<strong>exon.end_phase</strong> (values 0, 1, 2 and -1).
  <ul>
  <li> Phase of -1: the last base of the exon is non coding</li>
  <li> Phase of 1 : the last codon has one base only</li>
  <li> Phase of 2 : the last codon has two bases</li>
  <li> Phase of 0 : the last codon has three bases</li>
  </ul>
</li>

</ul>

</body>
</html>
