<html>
<head>
<title>Database Schema</title>
</head>

<body>

<h1>Database Schema</h1>

<ul>

<li>
<strong>Sequence region tables</strong> hold the genome sequences for sequence-level entities (e.g. BAC clones) and assembly information for top-level entities (e.g. chromosomes).
<a href="sequence_region.html">More ...</a>
</li>

<li>
<strong>General Feature tables</strong> store generic features that are annotated on the genome sequence.
<a href="general_feature.html">More ...</a>
</li>

<li>
The <strong>Analysis table</strong> provides information on individual analysis steps that were run to annotate features on the genome sequence.
<a href="core_analysis.html">More ...</a>
</li>

<li>
<strong>Common Feature tables</strong> contain essential feature annotation relevant for most species, including cDNA and protein alignments to the genome sequence and <i>ab initio</i> predicted transcripts.
<a href="common_feature.html">More ...</a>
</li>

<li>
<strong>Gene-related tables</strong> contain Ensembl gene, transcript, protein and exon annotation.
<a href="gene.html">More ...</a>
</li>

<li>
<strong>External references</strong> link Ensembl gene, transcript and protein predictions to other biological source databases.
<a href="external_reference.html">More ...</a>
</li>

<li>
<strong>Miscellaneous Feature tables</strong> contain all those features not organised in other tables.
<a href="misc_feature.html">More ...</a>
</li>

<li>
<strong>Archive tables</strong> contain mappings from expired Ensembl stable gene, transcript and protein IDs to the newer sets.
<a href="core_archive.html">More ...</a>
</li>

<li>
<strong>Marker tables</strong> store information for STS markers with physical sequence evidence, as well as markers on genetic maps.
<a href="marker.html">More ...</a>
</li>

<li>
<strong>Quantitative Trait Loci tables</strong>
<a href="qtl.html">More ...</a>
</li>

</ul>

<p><a href="schema_description.html">Detailed description</a> of each individual table in the Ensembl core schema.</p>

</body>
</html>
