<html>
<head>
<title>Marker Feature Tables</title>
</head>

<body>

<h1>Marker Feature Tables</h1>

<p>
An Ensembl marker is uniquely identified by a pair of primers.
This is practical for Ensembl even if it deviates from the typical biological definition of a marker.
A pair of primers is used to map a marker on the genome via a <a href="schema_description.html#marker_feature">marker_feature</a>.
If a marker has more than one marker feature, it will be placed at more than 
one position on the genome.
The <strong>marker_feature.map_weight</strong> column stores the number of times the marker maps to the genome.  
(This is obviously done after all <strong>marker_features</strong> are created).
</p>

<p>
The marker's name in Ensembl is the <strong>display_marker_synonym_id</strong>.
If a marker in Ensembl is known under different names in other databases, 
these are stored in <a href="schema_description.html#marker_synonym">marker_synonym</a>.
The same identifier might be associated with different primer pairs and will have different <strong>marker_id</strong> in Ensembl.
</p>

<p>
Ensembl markers are usually part of a mapping project that uses experimental methods to place them on the genome.  
A map represents a collection of experimentally placed markers and depending on the type of experiment the marker map location may have different forms (bands, radiation hybrid, cM) depending.
An experimental mapping usually produces a chromosome_name.
If the markers used for the experiments are not defined by a primer pair,  the actual name is provided as link to the synonym table.
</p>

</body>
</html>
