<html>
<head>
<meta name="order" content="2" />
  <meta name="navigation" content="Comparative Genomics">
  <title>Comparative Genomics</title>


</head>

<body>

<h1>Comparative Genomics</h1>

<h2>Data available</h2>

<ul>
<li><strong>Protein trees</strong> are constructed using the longest protein for every gene in Ensembl.
<strong>Homologues</strong> are deduced from these trees.  
Proteins are clustered based on Best-Reciprocal Hits and Blast Score Ratios. 
Each cluster of protein is aligned using Muscle. 
NJTREE is used to get a gene tree from each multiple alignment. 
The gene tree is reconciliated with the species tree to call duplication events.
<a href="homology_method.html">More details</a>. </li> 

<li><strong>Families</strong> are constructed by clustering of all Ensembl proteins, i.e. not only the longest protein.  
Metazoan proteins from UniProtKB SwissProt and TrEMBL are added to extend the protein set.</li>

<li><strong>Whole genome alignments</strong> are performed either pairwise between two species using BlastZ-net or translated Blat analysis, or using multiple species. <a href="analyses.html">More information</a>.</li>
</ul>

<h2>Access</h2>
<p>
<strong>Data</strong> can be accessed using the <a href="/info/data/api/index.html">Compara Perl API</a>, 
<a href="info/data/biomart.html">BioMart</a>, or comparative genomics pages on the browser.
Gene trees can be viewed from any 'Gene' page on the browser, and exported via the control panel.  
</p>

</body>
</html>
