<html>
<head>
<meta name="author" content="helpdesk" />
<meta name="navigation" content="Software" />
<title>Changing the species in Ensembl</title>
</head>

<body>

<h2 class="breadcrumbs">Changing the species in Ensembl</h2>

<h3 class="boxed">Removing species</h3>

<p>It is obviously simpler to omit species from your Ensembl mirror 
than to add new ones, so we'll look at this first. Simply add the
required <kbd>delete</kbd> lines to the <kbd>update_conf</kbd> method in
<kbd>/public-plugins/mirror/conf/SiteDefs.pm</kbd>:</p>

<div class="formpanel">
<pre>
delete($SiteDefs::__species_aliases{'Mus_musculus'});
delete($SiteDefs::__species_aliases{'Rattus_norvegicus'});
</pre>
</div>

<p><strong>Important note</strong>: if you are deleting either Human
or Mouse, you need to specify a new primary or secondary species
(these are used as defaults when generating links to examples). E.g.:</p>

<div class="formpanel">
<pre>
$SiteDefs::ENSEMBL_SECONDARY_SPECIES  = 'Canis_familiaris'; 
</pre>
</div>

<h3 class="boxed">Adding species</h3>

<p>To add a species, you obviously need a complete Ensembl database
in the same schema as the rest of your site, which is outside the 
scope of this tutorial. However assuming you have the data already,
the following changes will add it to the website (substituting the
details of your own species for the ones in the examples, obviously!).</p>

<h4>Configuration files</h4>

<p>Firstly, add it to the species aliases in the <kbd>update_conf</kbd> 
method in <kbd>/public-plugins/mirror/conf/SiteDefs.pm</kbd>:</p>

<div class="formpanel">
<pre>
$SiteDefs::__species_aliases{'Canis_latrans'}           = [qw(cl coyote)];
$SiteDefs::__species_aliases{'Geococcyx_californianus'} = [qw(gc roadrunner)];
</pre>
</div>

<p>Note that all aliases should be unique, so check in 
<kbd>/public-plugins/ensembl/conf/SiteDefs.pm</kbd> to see which 
are already in use. These aliases allow quick access to pages, so that users
can type, e.g. www.ensembl.org/worm/geneview?gene=Y42H9B.1 instead of
www.ensembl.org/Caenorhabditis_elegans/geneview?gene=Y42H9B.1</p>

<p>You will also need a .ini file for the new species. Copy one of the 
existing files from <kbd>/public-plugins/ensembl/conf/ini-files/</kbd>
to <kbd>/public-plugins/mirror/conf/ini-files/</kbd> and edit it to
reflect the relevant settings. The section that will require the most
extensive updates is the [SEARCH_LINKS] list at the end of the file,
since this will need to point to useful examples from your own database.</p>

<p>If your species has chromosomes, make sure you have the correct names
in the array ENSEMBL_CHROMOSOMES, as this is used to draw karyotypes.</p>

<h4>Website database</h4>

<p>Whilst the genomic databases provide the bulk of the data about species
in Ensembl, news and some static content is provided from a separate small
database, <kbd>ensembl_website</kbd>.

<p>Using a MySQL client program, update the ensembl_website database as follows:</p>

<p><kbd>insert into species values (1001, 'cl', 'Canis_latrans', 'Coyote');</kbd></p>

<p>N.B. We suggest using a high primary key value so as to avoid conflict with 
future releases of Ensembl.</p>

<p>Then run the utility script <kbd>/utils/update_webdb.pl</kbd>, which will
link your new species into the current release.</p>

<h4>Static content</h4>

<p>Whilst the above changes will ensure that dynamic content is automatically
generated for your new species, users will need a way into that data. Each
species in Ensembl has a home page which offers entry points and general
information; for the sake of argument we will assume that you are keeping
to the standard page format.</p>

<p>First, make a directory <kbd>/public-plugins/mirror/htdocs/Canis_latrans/</kbd>,
then copy the HTML files from a similarly-assembled species. That is, if 
your data is assembled into chromosomes, choose a species with chromosomes; 
if it only has scaffolds or contigs, choose a species without chromosomes.</p>

<p>Then simply go through the copied files and change the content to suit your
data. If the species has chromosomes, you can use karyoview to generate a
blank karyotype; the clickable image map can be created in a suitable HTML
editor, or you can edit the file manually (an image manipulation program 
such as the GIMP can be used to find the coordinates of each clickable area).</p>

<p><img src="/img/info/thumb_Canis_latrans.png" class="float-left" />
Finally, you need some kind of species image for the favourites list on the
home page. This should be a 40x40 pixel PNG file named thumb_Canis_latrans.png and
placed in <kbd>/public-plugins/mirror/htdocs/img/species/</kbd>.</p>

</body>
</html>
