<html>
<head>
<title>FTP Download</title>
<meta name="order" content="7" />
</head>

<body>

<h1>The Ensembl FTP Server</h1>

<p>If required, entire databases can be downloaded from our FTP site
in a variety of formats, from flat files to MySQL dumps. Please be aware that these
files can run to many gigabytes of data.</p>

<p>To facilitate storage and download all databases are
<a href="http://directory.fsf.org/gzip.html" rel="external">GNU Zip</a> 
(gzip, *.gz) compressed.
</p>
<p>
<strong>Please note:</strong>
Ensembl supports downloading of many correlation tables via the highly
customisable
<a href="/biomart/martview">BioMart</a> data mining tool.
You may find exploring this web-based data mining tool easier than extracting
information from our database dumps.
</p>

[[SCRIPT::EnsEMBL::Web::Document::HTML::FTPtable]]

<h2>About the data</h2>

<p>
The following types of data dumps are available on the FTP site. 
</p>
<dl>
<dt>FASTA</dt>
<dd>FASTA sequence databases of Ensembl gene, transcript and protein model
predictions. 
Since the 
<a href="http://www.ebi.ac.uk/help/formats.html#fasta" rel="external">FASTA format</a>
does not permit sequence annotation, these database files are mainly intended
for use with local sequence similarity search algorithms.
Each directory has a README file with a detailed description of the header line format and the file naming conventions.

  <dl>
  <dt>DNA</dt>
  <dd><a href="http://www.repeatmasker.org/" rel="external">Masked</a> and unmasked genome sequences associated with the assembly (contigs, chromosomes etc.).</dd>
  <dt>cDNA</dt>
  <dd>cDNA sequences for Ensembl or <i>ab initio</i> <a href="/info/docs/genebuild/genome_annotation.html">predicted genes</a>.</dd>
  <dt>Peptides</dt>
  <dd>Protein sequences for Ensembl or <i>ab initio</i> <a href="/info/docs/genebuild/genome_annotation.html">predicted genes</a>.</dd>
  <dt>RNA</dt>
  <dd>Non-coding RNA gene preditions.</dd>
  </dl>
</dd>

<dt>Flatfile</dt> 
<dd>Flat files allow more extensive sequence annotation by means of feature tables
and contain thus the genome sequence as annotated by the automated Ensembl
<a href="/info/docs/genebuild/genome_annotation.html">genome annotation pipeline</a>.
Each nucleotide sequence record in a flat file represents a 1Mb slice of the
genome sequence.
Flat files are broken into chunks of 1000 sequence records for easier
downloading.
  <dl>
  <dt>EMBL</dt>
  <dd>Ensembl database dumps in <a href="http://www.ebi.ac.uk/embl/" rel="external">EMBL</a> nucleotide
sequence <a href="http://www.ebi.ac.uk/embl/Documentation/User_manual/usrman.html" rel="external">database format</a>
</dd>
  <dt>GenBank</dt>
  <dd>Ensembl database dumps in <a href="http://www.ncbi.nlm.nih.gov/Genbank/" rel="external">GenBank</a>
nucleotide sequence
<a href="http://www.ncbi.nlm.nih.gov/Sitemap/samplerecord.html" rel="external">database format</a></dd>
  </dl>
</dd>

<dt>MySQL</dt>
<dd>All Ensembl  <a href="http://www.mysql.com/" rel="external">MySQL</a> databases are 
available in text format as are the SQL table definition files.  These can be imported into to any SQL database for a local
<a href="/info/docs/webcode/install/">installation</a> of a mirror site.
Generally, the FTP directory tree contains one one directory per database.
For more information about these databases and their Application
Programming Interfaces (or APIs) see the
<a href="/info/docs/api/">API</a> section.
</dd>

<dt>GTF</dt>
<dd>Gene sets for each species. These files include annotations of both coding
and non-coding genes. This file format is described <a href="http://mblab.wustl.edu/GTF2.html">here</a>.</dd>

<dt>EMF flatfile dumps (variation and comparative data)</dt>
<dd>Alignments of resequencing data are available for several species as Ensembl
Multi Format (EMF) flatfile dumps. The accompanying README file describes the file
format.
<p>
Also, the same format is used to dump whole-genome multiple alignments as
well as gene-based multiple alignments and phylogentic trees used to infer
Ensembl orthologues and paralogues. These files are available in the ensembl_compara
database which will be found in the <a href="ftp://ftp.ensembl.org/pub/current_multi_species/">multi_species directory</a>.
</p>
</dd>
<dt>BED format files (comparative data)</dt>
<dd>Constrained elements calculated using GERP are available in BED format. For more information see the accompanying README file. 
<p>BED format is a simple line-based format. The first 3 mandatory column are:
<ul>
  <li>chromosome name, starting with chr to be compiant with UCSC browser. Example: chrX</li>
  <li>start position. This is a 0-based position</li>
  <li>end position.</li>
</ul>
More information on the BED file format is available <a href="http://genome.ucsc.edu/FAQ/FAQformat#format1">here</a>.
</p>
</dd>
</dl>

<p>
Each directory on 
<a href="ftp://ftp.ensembl.org/pub/" rel="external">ftp.ensembl.org</a> contains a
<a href="ftp://ftp.ensembl.org/pub/current_README">README</a> file.
 This additional document explains the FTP directory
<a href="index.html">structure</a>.
</p>

</body>
</html>
