<html>
<head>
<meta name="order" content="1" />
<title>Variation</title>
</head>

<body>

<h1>Ensembl Variation</h1>

<a href="/info/docs/variation/index.html">About Variation Data</a> | 
<a href="/info/docs/variation/database.html">Database Description</a> |
<a href="/info/docs/variation/variation_schema.html">Variation Tables Description</a> |
<a href="/info/docs/api/variation/index.html">Perl API</a>

<br /><br />
<p>
Ensembl stores both structural variants and sequence variants including
Single Nucleotide Polymorphisms (SNPs), insertions, deletions and somatic mutations
for several species and makes this data available on the website, via the Perl API,
and provides direct access to the underlying databases.
</p>

<p>
This data is also integrated with other sections of Ensembl, in particular we predict
the effects of variants on the Ensembl transcripts for each species.
</p>

<h2> Variation species </h2>
<p>
Ensembl stores variation data for the following species, but note that users can still use the 
<a href="#VEP">Variant Effect Predictor</a> on species for which we do not currently have 
a variation database.
</p>
<table><tr><td>
<ul>
	<li><i>Bos taurus</i></li>
	<li><i>Canis familiaris</i></li>
	<li><i>Danio rerio</i></li>
	<li><i>Drosophila melanogaster</i></li>
	<li><i>Equus caballus</i></li>
	<li><i>Felis catus</i></li>
	<li><i>Gallus gallus</i></li>
	<li><i>Homo sapiens</i></li>
	<li><i>Monodelphis domestica</i></li>
</ul>
</td><td>
<ul>
	<li><i>Mus musculus</i></li>
	<li><i>Ornithorhynchus anatinus</i></li>
	<li><i>Pan troglodytes</i></li>
	<li><i>Pongo abelii</i></li>
	<li><i>Rattus norvegicus</i></li>
	<li><i>Saccharomyces cerevisiae</i></li>
	<li><i>Sus scrofa</i></li>
	<li><i>Taeniopygia guttata</i></li>
	<li><i>Tetraodon nigroviridis</i></li>
</ul>
</td></tr></table>


<h2 id="sources"> Sources of Variation Data </h2>

<p>
The majority of variants are imported from NCBI <a rel="external"
href="http://www.ncbi.nlm.nih.gov/projects/SNP/">dbSNP</a>. 
The data is imported when it is released by dbSNP and incorporated
into the next Ensembl release. If dbSNP releases the data on a different assembly,
Ensembl will remap the variant positions onto the current assembly.<br/>
</p>

<p>
For Human, data from projects like 
the <a rel="external" href="http://www.HapMap.org/">HapMap Project</a>
 and <a rel="external" href="http://www.1000genomes.org/page.php">1000
Genomes Project</a> is imported once it has been submitted to <a rel="external"
href="http://www.ncbi.nlm.nih.gov/projects/SNP/">dbSNP</a>. 
</p>

<p>
Ensembl also includes data from other sources. For a complete list, go to
a species Location page (e.g. <a href="/Homo_sapiens/Location/View?r=6:133017695-133161157"> for human</a>),
and click on the 'Configure this page' link on the left-hand side.
The 'Germline variation' and 'Somatic mutations' sections contain a detailed description of 
all sources of variation data for that species.
</p>

<p>
For example, in human there is data from;
</p>

<ul>
<li><a rel="external" href="http://www.ncbi.nlm.nih.gov/projects/SNP/">dbSNP</a></li>
<li>Illumina chips</li>
<li><a rel="external" href="http://www.affymetrix.com/index.affx">Affymetrix GeneChip Arrays</a></li> 
<li><a rel="external" href="http://www.hgmd.cf.ac.uk/ac/index.php">Human Gene Mutation Database
(HGMD)</a> (public identifiers)</li>
<li><a rel="external" href="http://www.uniprot.org/">UniProt</a></li>
<li><a rel="external" href="http://www.sanger.ac.uk/genetics/CGP/cosmic/">Catalogue Of Somatic Mutations In
Cancer (COSMIC)</a></li>
<li><a rel="external" href="http://www.ebi.ac.uk/dgva/page.php">DGVa</a> and <a
rel="external" href="http://www.ncbi.nlm.nih.gov/dbvar/">dbVAR</a>: structural polymorphism data</li>
<li><a rel="external"
href="http://www.genome.gov/">NHGRI</a>, <a rel="external" href="http://www.ebi.ac.uk/ega/page.php">European
Genome-phenome Archive (EGA)</a>, <a rel="external" href="http://www.biomedcentral.com/1471-2350/10/6">Open
GWAS DB</a>: phenotype information from genome-wide association studies.</li>
<li><a rel="external" href="http://www.ncbi.nlm.nih.gov/omim">OMIM</a>: phenotype information.</li>
<li>Locus Reference Genomic sequences (<a rel="external"
href="http://www.lrg-sequence.org/page.php">LRG</a>)</li>
</ul>

<br />
<h2 id="displays"> Variation Displays </h2>
<p>
Variation data can be viewed in the browser through pages such as:
</p>

<ul>
<li><strong>Gene:</strong> Variation table and Variation image (for
all variations in a gene) e.g. for example <a href="/Homo_sapiens/Gene/Variation_Gene/Table?db=core;g=ENSG00000159197">for all variants in KCNE2</a> </li>
<li><strong>Transcript:</strong> Population comparison, Comparison
image (for comparing variants in a transcript across different individual or strain
sequences) e.g. <a href="/Mus_musculus/Transcript/Population/Image?g=ENSMUSG00000041143;r=4:138528820-138586500;t=ENSMUST00000084208"> compare Tmco4 in different mouse strains </a> </li>
<li> <strong> Transcript: </strong> Sequence, protein: list of the coding variants in protein coordinates.</li>
<li><strong>Location:</strong> Region in Detail (Variations can be
drawn using &quot;Configure this page&quot; at the left. The menu allows
display of information in Ensembl databases along with external
sources in DAS format such as <a rel="external"
href="http://projects.tcag.ca/variation/">DGV loci</a>.)</li>
</ul>

<p>
Clicking on any variation on an Ensembl page will open a Variation tab
with information about the flanking sequence and source for the
selected variation. Links to linkage disequilibrium (LD) plots,
phenotype information (for human) from EGA, OMIM and <a rel="external"
href="http://www.genome.gov/">NHGRI</a> and Ensembl genes and
transcripts that include the variation can be found at the left of
this tab. You may also view multiple genome alignments of various
species, highlighting the
variation. <a href="/info/docs/compara/analyses.html#ancestral"
class="cp-external">Ancestral sequences</a> are included in this
display. 
</p>

<p>
Variation information can also be accessed
using <a href="/info/data/biomart.html"
class="cp-external">BioMart</a> (gene or variation database), and
the <a href="/info/docs/api/index.html" class="cp-external">Perl
API</a> (<a href="/info/docs/api/variation/index.html"
class="cp-external">variation databases</a>).
</p>

<br />

<h2 id="classes"> Variation classes </h2>

We call the class of a variation according to its component alleles and its mapping 
to the reference genome, and then display this information on the website. Internally 
we use <a href="http://www.sequenceontology.org/">Sequence Ontology</a> terms, but we 
map these to our own 'display' terms where common usage differs from the SO 
definition (e.g. our term SNP is closer to the SO term SNV). All the classes we call, 
along with their equivalent SO term are shown in the table below. We also differentiate 
somatic mutations from germline variations in the display term, prefixing the term with
'somatic'. API users can fetch either the SO term or the display term.</p>

<br />

<table id="variation_classes" class="ss" style="width:50%;">
<tr>
	<th>Ensembl term</th>
	<th>SO term</th>
	<th>SO accession</th>
</tr>
<tr>
	<td>SNP</td>
	<td rowspan=2>SNV</td>
	<td rowspan=2><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001483">SO:0001483</a></td>
</tr>
<tr>
	<td>somatic_SNV</td>
</tr>
<tr class="bg2">
	<td>substitution</td>
	<td rowspan=2>substitution</td>
	<td rowspan=2><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:1000002">SO:1000002</a></td>
</tr>
<tr class="bg2">
	<td>somatic_substitution</td>
</tr>
<tr>
	<td>CNV</td>
	<td rowspan=2>copy_number_variation</td>
	<td rowspan=2><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001019">SO:0001019</a></td>
</tr>
<tr>
	<td>somatic_CNV</td>
</tr>
<tr class="bg2">
	<td>insertion</td>
	<td rowspan=2>insertion</td>
	<td rowspan=2><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0000667">SO:0000667</a></td>
</tr>
<tr class="bg2">
	<td>somatic_insertion</td>
</tr>
<tr>
	<td>deletion</td>
	<td rowspan=2>deletion</td>
	<td rowspan=2><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0000159">SO:0000159</a></td>
</tr>
<tr>
	<td>somatic_deletion</td>
</tr>
<tr class="bg2">
	<td>indel</td>
	<td rowspan=2>indel</td>
	<td rowspan=2><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:1000032">SO:1000032</a></td>
</tr>
<tr class="bg2">
	<td>somatic_indel</td>
</tr>
<tr>
	<td>tandem_repeat</td>
	<td rowspan=2>tandem_repeat</td>
	<td rowspan=2><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0000705">SO:0000705</a></td>
</tr>
<tr>
	<td>somatic_tandem_repeat</td>
</tr>
<tr class="bg2">
	<td>sequence_alteration</td>
	<td rowspan=2>sequence_alteration</td>
	<td rowspan=2><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001059">SO:0001059</a></td>
</tr>
<tr class="bg2">
	<td>somatic_sequence_alteration</td>
</tr>
</table>


<h3 id="indel">Insertion and Deletion coordinates </h3>

<p> In Ensembl, an insertion is indicated by start coordinate = end coordinate +
1. For example, an insertion of 'C' between nucleotides 12600 and
12601 on the forward strand is indicated with start and end coordinates as
follows:</p>

<pre>
   12601     12600   
</pre>
<p>A deletion is indicated by the exact nucleotide coordinates. For
example, a three base pair deletion of nucleotides 12600, 12601, and
12602 of the reverse strand will have start and end coordinates of :</p>

<pre>
   12600     12602    
</pre>

<br />
<h2 id="consequences">Predicted variation consequences</h2>

<p>
For each variation that is mapped to the reference genome, we identify any
Ensembl transcripts that overlap the variation and use a rule-based approach to
predict the effect that each allele of the variation may have on the transcript.
The set of consequence terms, defined by the <a rel="external"
href="http://www.sequenceontology.org">Sequence Ontology</a> (SO), that can be
currently be assigned to each combination of an allele and a transcript is shown
in the table below. Note that each allele of each variation may have a different
effect in different transcripts. 
</p>

<p> 
This approach is applied to all germline variations and somatic mutations
stored in the Ensembl variation databases (though we do not currently calculate
consequences for structural variants). The resulting consequence type calls,
along with information determined as part of the process, such as the cDNA and
CDS coordinates, and the affected codons and amino acids in coding transcripts,
are stored in the variation database and displayed on the website. 
</p>

<p>
Prior to release 62 we used our own internal terms to describe the consequence
types, and we continue to use these by default on the website. You can opt to
see SO terms on some variation views using 'Configure this page', and 
<a href="#VEP">VEP</a> and API users can choose which terms to use. The SO terms 
are more specific than our
display terms but we map all display terms to one or more SO terms, as shown in
the table below. Where the NCBI have an equivalent term we also include it in
this table (and again you can opt to use NCBI terms on some views via 'Configure
this page' and in the VEP and API).
</p>

<p>We follow the SO definition of the term as closely as possible, but in some
cases (e.g. non_synonymous_codon) we have slightly generalised the definition to
include variants that affect multiple codons as well as a single one. We are
working with the SO to refine these definitions. The descriptions shown in the
table are our own, not the SO definitions.
</p>

<p>
The terms in the table are shown in order of severity as estimated by
Ensembl, and this ordering is used on the website summary views. This ordering
is necessarily subjective and API and VEP users can always get the full set of
consequences for each allele and make their own severity judgement.
</p>

<p>
A <a href="consequences.jpg">diagram</a> showing the location of each display 
term relative to the transcript structure is available.
</p>

<br />

<table id="consequence_type_table" class="ss">
<tr>
	<th>Ensembl term</th>
	<th>Ensembl description</th>
	<th>SO term</th>
	<th>SO accession</th>
	<th>NCBI term</th>
</tr>
<tr>
	<td rowspan=2>Essential splice site</td>
	<td rowspan=2>In the first 2 or the last 2 basepairs of an intron</td>
	<td>splice_acceptor_variant</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001574">SO:0001574</a></td>
	<td>splice-3</td>
</tr>
<tr>
	<td>splice_donor_variant</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001575">SO:0001575</a></td>
	<td>splice-5</td>
</tr>
<tr class="bg2">
	<td>Stop gained</td>
	<td>In coding sequence, resulting in the gain of a stop codon</td>
	<td>stop_gained</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001587">SO:0001587</a></td>
	<td>nonsense</td>
</tr>
<tr>
	<td>Stop lost</td>
	<td>In coding sequence, resulting in the loss of a stop codon</td>
	<td>stop_lost</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001578">SO:0001578</a></td>
	<td>-</td>
</tr>
<tr class="bg2">
	<td>Complex in/del</td>
	<td>Insertion or deletion that spans an exon/intron or coding sequence/UTR border</td>
	<td>complex_change_in_transcript</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001577">SO:0001577</a></td>
	<td>-</td>
</tr>
<tr>
	<td rowspan=4>Non-synonymous coding</td>
	<td rowspan=4>In coding sequence and results in an amino acid change in the encoded peptide sequence</td>
	<td>initiator_codon_change</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001582">SO:0001582</a></td>
	<td>-</td>
</tr>
<tr>
	<td>inframe_codon_gain</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001651">SO:0001651</a></td>
	<td>-</td>
</tr>
<tr>
	<td>inframe_codon_loss</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001652">SO:0001652</a></td>
	<td>-</td>
</tr>
<tr>
	<td>non_synonymous_codon</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001583">SO:0001583</a></td>
	<td>missense</td>
</tr>

<tr class="bg2">
	<td>Frameshift coding</td>
	<td>In coding sequence, resulting in a frameshift</td>
	<td>frameshift_variant</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001589">SO:0001589</a></td>
	<td>frameshift</td>
</tr>
<tr>
	<td>Splice site</td>
	<td>1-3 bps into an exon or 3-8 bps into an intron</td>
	<td>splice_region_variant</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001630">SO:0001630</a></td>
	<td>-</td>
</tr>
<tr class="bg2">
	<td>Partial codon</td>
	<td>Located within the final, incomplete codon of a transcript whose end coordinate is unknown</td>
	<td>incomplete_terminal_codon_variant</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001626">SO:0001626</a></td>
	<td>-</td>
</tr>
<tr>
	<td rowspan=2>Synonymous coding</td>
	<td rowspan=2>In coding sequence, not resulting in an amino acid change (silent mutation)</td>
	<td>stop_retained_variant</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001567">SO:0001567</a></td>
	<td>-</td>
</tr>
<tr>
	<td>synonymous_codon</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001588">SO:0001588</a></td>
	<td>cds-synon</td>
</tr>
<tr class="bg2">
	<td>Coding unknown</td>
	<td>In coding sequence with indeterminate effect</td>
	<td>coding_sequence_variant</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001580">SO:0001580</a></td>
	<td>-</td>
</tr>
<tr>
	<td>Within mature miRNA</td>
	<td>Located within a microRNA</td>
	<td>mature_miRNA_variant</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001620">SO:0001620</a></td>
	<td>-</td>
</tr>
<tr class="bg2">
	<td>5 prime UTR</td>
	<td>In 5 prime untranslated region</td>
	<td>5_prime_UTR_variant</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001623">SO:0001623</a></td>
	<td>untranslated_5</td>
</tr>
<tr>
	<td>3 prime UTR</td>
	<td>In 3 prime untranslated region</td>
	<td>3_prime_UTR_variant</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001624">SO:0001624</a></td>
	<td>untranslated_3</td>
</tr>
<tr class="bg2">
	<td>Intronic</td>
	<td>In intron</td>
	<td>intron_variant</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001627">SO:0001627</a></td>
	<td>intron</td>
</tr>
<tr>
	<td>NMD transcript</td>
	<td>Located within a transcript predicted to undergo nonsense-mediated decay</td>
	<td>NMD_transcript_variant</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001621">SO:0001621</a></td>
	<td>-</td>
</tr>
<tr class="bg2">
	<td>Within non-coding gene</td>
	<td>Located within a gene that does not code for a protein</td>
	<td>nc_transcript_variant</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001619">SO:0001619</a></td>
	<td>-</td>
</tr>
<tr>
	<td rowspan=2>Upstream</td>
	<td rowspan=2>Within 5 kb upstream of the 5 prime end of a transcript</td>
	<td>2KB_upstream_variant</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001636">SO:0001636</a></td>
	<td>near-gene-5</td>
</tr>
<tr>
	<td>5KB_upstream_variant</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001635">SO:0001635</a></td>
	<td>-</td>
</tr>
<tr class="bg2">
	<td rowspan=2>Downstream</td>
	<td rowspan=2>Within 5 kb downstream of the 3 prime end of a transcript</td>
	<td>500B_downstream_variant</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001634">SO:0001634</a></td>
	<td>near-gene-3</td>
</tr>
<tr class="bg2">
	<td>5KB_downstream_variant</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001633">SO:0001633</a></td>
	<td>-</td>
</tr>
<tr>
	<td>Intergenic</td>
	<td>More than 5 kb either upstream or downstream of a transcript</td>
	<td>intergenic_variant</td>
	<td><a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001628">SO:0001628</a></td>
	<td>-</td>
</tr>
</table>

<br />
<h2 id="nsSNP">Protein function predictions</h2>

<p>
For human mutations that are predicted to result in an amino acid substitution we also 
provide SIFT and PolyPhen predictions for the effect of this substitution on
protein function. We compute the predictions for each of these tools for all possible 
single amino acid substitutions in the Ensembl human protein set. This means we can provide 
predictions for novel mutations for VEP and API users. We were able to compute predictions
from at least one tool for over 99% of the proteins in Ensembl. We also use the Condel tool
to provide a consensus prediction based on the SIFT and PolyPhen predictions.
</p>

<p>
These tools are developed by external groups and we provide a brief explanation of
the approach each takes below, and how we run it in Ensembl. For much more detail
please see the relevant publications listed on each tool's website. We hope to be 
able to provide amino acid substitution predictions for species other than human 
in future releases.
</p>

<h3 id="sift">SIFT</h3>

<p>
<a rel="external" href="http://sift.jcvi.org/">SIFT</a> predicts whether an amino acid 
substitution is likely to affect protein function based on sequence homology and the
physico-chemical similarity between the alternate amino acids. The data we provide for each amino
acid substitution is a score and a qualitative prediction (either 'tolerated' or 'deleterious'). 
The score is the normalized probability that the amino acid change is 
tolerated so scores nearer 0 are more likely to be deleterious.
The qualitative prediction is derived from this score such that substitutions with 
a score < 0.05 are called 'deleterious' and all others are called 'tolerated'.
</p>

<p>
We ran SIFT version 4.0.3 (available 
<a rel="external" href="http://sift.jcvi.org/www/sift4.0.3.tar.gz">here</a>) 
following the instructions from the authors and used SIFT
to choose homologous proteins rather than supplying them ourselves. We used all 
protein sequences available from UniProtKB (both the SwissProt and TrEMBL sets) as 
the protein database. All data was downloaded from UniProt in February 2011.
</p>

<h3 id="polyphen">PolyPhen</h3>

<p>
<a rel="external" href="http://genetics.bwh.harvard.edu/pph2/">PolyPhen-2</a> predicts
the effect of an amino acid substitution on the structure and function of a protein
using sequence homology, Pfam annotations, 3D structures from PDB where available, and
a number of other databases and tools (including DSSP, ncoils etc.). As with SIFT, for each amino
acid substitution where we have been able to calculate a prediction, we provide both a
qualitative prediction (one of 'probably damaging', 'possibly damaging', 'benign'
or 'unknown') and a score. The PolyPhen score represents the probability that a substitution 
is damaging, so values nearer 1 are more confidently predicted to be deleterious (note 
that this the opposite to SIFT). The qualitative prediction is based on the False 
Positive Rate of the classifier model used to make the predictions.
</p>

<p>
We ran PolyPhen-2 version 2.0.23 (available 
<a rel="external" href="http://genetics.bwh.harvard.edu/pph2/dokuwiki/_media/polyphen-2.0.23r349.tar.gz">here</a>)
and again we followed all instructions from the authors, and used the UniProtKB 
UniRef100 non-redundant protein set as the protein database, which was downloaded,
along with PDB structures, and annotations from Pfam and DSSP in February 2011. When 
computing the predictions we used the classifier model trained on the HumDiv training 
set (please refer to the PolyPhen publications for more details of the classification 
system).
</p>

<h3 id="condel">Condel</h3>

<p>
<a rel="external" href="http://bg.upf.edu/group/projects/condel.php">Condel</a> is
a general method for calculating a consensus prediction from the output of tools
designed to predict the effect of an amino acid substitution. It does so by calculating
a weighted average score of the scores of each component method. The Condel authors 
provided us with a version specialised for finding a consensus between SIFT and 
PolyPhen and we integrated this into the variation API. Tests run by the authors 
on the HumVar dataset (a test set curated by the PolyPhen team), show that Condel 
can improve both the sensitivity and specificity of predictions compared to either 
SIFT or PolyPhen used alone (please contact the authors for details). The Condel 
score, along with a qualitative prediction (one of 'neutral' or 'deleterious'), 
are available in the VEP and via the API. The Condel score is the consensus probability 
that a substitution is deleterious, so values nearer 1 are predicted with greater 
confidence to affect protein function.
</p>

<br />

<h2 id="VEP">The Variant Effect Predictor (VEP)</h2>
    
<p>
Users can run all the analyses described above on novel mutations 
using the Ensembl Variant Effect Predictor (VEP). Provided you have a 
list of allele sequences and associated genomic coordinates, you can use 
the VEP to calculate consequences with respect to the Ensembl transcript set for
your species of interest, and (for human) to retrieve SIFT, PolyPhen and Condel predictions 
for any missense mutations. The VEP can also identify if a mutation is already 
known (i.e. already found in dbSNP or any of the other sources of variation 
data listed <a href="#sources">above</a>), and so can be used to filter your 
data for novel mutations. The VEP works for any species in Ensembl and not 
just those for which we have an existing variation database.
</p>

<p>
The VEP can either be run online from the <a href="/tools.html">Tools</a> 
section of the website or you can download a script to run yourself on larger
datasets. You can also write custom scripts using the Perl API.
All of these approaches use the same backend code as is used to calculate the
consequences displayed on the website. Please refer to the 
<a href="/info/website/upload/var.html">VEP</a> documentation for more
details on running the VEP, including supported input formats, or the 
<a href="/info/docs/api/variation/index.html">variation API</a> documentation
for help writing custom scripts. The VEP is under active development and
<a href="/info/about/contact/index.html">feedback</a> is welcome!
</p>

<br />
<h2 id="notes">Notes</h2>

<p>For more detailed information on the variation specific help see <a href="/Help/View?id=168">here</a>.<br
/>
There is also more information about programmatic access in the <a
href="/info/docs/api/variation/variation_tutorial.html">variation API tutorial</a> page.
</p>


<br />
<h2 id="references">References</h2>

<p>McLaren W, Pritchard B, Rios D, Chen Y, Flicek P, Cunningham F.<br />
<strong>Deriving the consequences of genomic variants with the Ensembl API and SNP Effect Predictor.</strong><br />
<em>BMC Bioinformatics</em>26(16):2069-70(2010)<br />
<a href="http://dx.doi.org/10.1093/bioinformatics/btq330"
rel="external">doi:10.1093/bioinformatics/btq330 </a>
</p> 

<p>
Rios D, McLaren WM, Chen Y, Birney E, Stabenau A, Flicek P, Cunningham F.<br />
<strong>A Database and API for variation, dense genotyping and resequencing data</strong><br />
<em>BMC Bioinformatics</em> 11:238 (2010)<br />
<a href="http://dx.doi.org/10.1186/1471-2105-11-238"
rel="external">doi:10.1186/1471-2105-11-238</a>
</p>

<p>Chen Y, Cunningham F, Rios D, McLaren WM, Smith J, Pritchard B, Spudich GM, Brent S, Kulesha E, Marin-Garcia P, Smedley D, Birney E, Flicek P.<br />
<strong>Ensembl Variation Resources</strong><br />
<em>BMC Genomics</em> 11(1):293 (2010)<br />
<a href="http://dx.doi.org/10.1186/1471-2164-11-293"
rel="external">doi:10.1186/1471-2164-11-293</a>
</p>

<p>
Redon, R. <i>et al.</i><br />
<strong>Global variation in copy number in the human
genome</strong><br />
<em>Nature</em> 444:444-454 (2006)<br />
<a href="http://dx.doi.org/10.1038/nature05329"
rel="external">doi:10.1038/nature05329</a>
</p>

<p>
Spencer, C. C. A. <i>et al.</i><br />
<strong>The Influence of Recombination on Human Genetic
Diversity</strong><br />
<em>PLoS Genet.</em> 2(9):e148 (2006)<br />
<a href="http://dx.doi.org/10.1371/journal.pgen.0020148"
rel="external">doi:10.1371/journal.pgen.0020148</a>
</p>

<p>
Venter, J. C. <i>et al.</i><br />
<strong>The Sequence of the Human Genome</strong><br />
<em>Science</em> 291(5507):1304-51 (2001)<br />
<a href="http://dx.doi.org/10.1126/science.1058040"
rel="external">doi:10.1126/science.1058040</a>
</p>

<p>
Wheeler, D. A. <i>et al.</i><br />
<strong>The complete genome of an individual by massively parallel DNA
sequencing</strong><br />
<em>Nature</em> 452:872-876 (2008)<br />
<a href="http://dx.doi.org/10.1371/10.1038/nature06884"
rel="external">doi:10.1038/nature06884</a>
</p>

</body>
</html>
