<html>
<head>
<meta name="order" content="1" />
<title>Variation</title>
</head>

<body>

<h1>Ensembl Variation</h1>

<p>
Ensembl stores structural variants as well as sequence variants i.e.
Single Nucleotide Polymorphisms (SNPs), insertions, deletions and somatic mutations.
</p>

<h2> Variation species </h2>
<p>Ensembl has variants for the following species:</p>
<table><tr><td>
<ul>
	<li><i>Bos taurus</i></li>
	<li><i>Canis familiaris</i></li>
	<li><i>Danio rerio</i></li>
	<li><i>Drosophila melanogaster</i></li>
	<li><i>Equus caballus</i></li>
	<li><i>Felis catus</i></li>
	<li><i>Gallus gallus</i></li>
	<li><i>Homo sapiens</i></li>
	<li><i>Monodelphis domestica</i></li>
</ul>
</td><td>
<ul>
	<li><i>Mus musculus</i></li>
	<li><i>Ornithorhynchus anatinus</i></li>
	<li><i>Pan troglodytes</i></li>
	<li><i>Pongo abelii</i></li>
	<li><i>Rattus norvegicus</i></li>
	<li><i>Saccharomyces cerevisiae</i></li>
	<li><i>Sus scrofa</i></li>
	<li><i>Taeniopygia guttata</i></li>
	<li><i>Tetraodon nigroviridis</i></li>
</ul>
</td></tr></table>


<h2 id="sources"> Sources of Variation Data </h2>
<p>
The majority of variants are downloaded from NCBI <a rel="external"
href="http://www.ncbi.nlm.nih.gov/projects/SNP/">dbSNP</a>. 
The data is imported when it is released by dbSNP and incorporated
into the next Ensembl release. If dbSNP releases the data on a different assembly,
Ensembl will remap the variant positions onto the current assembly.<br>
For Human, data from projects like 
the <a rel="external" href="http://www.HapMap.org/">HapMap Project</a>
 and <a rel="external" href="http://www.1000genomes.org/page.php">1000
Genomes Project</a> is imported once it has been submitted to <a rel="external"
href="http://www.ncbi.nlm.nih.gov/projects/SNP/">dbSNP</a>. 
</p>

<p>
Ensembl also includes data from other sources. For a complete list, go to
a species Location page (e.g. <a href="/Homo_sapiens/Location/View?r=6:133017695-133161157"> for human</a>),
 and click on the &quot;Configure Your Page&quot; link on the left-hand side.
The &quot;Germline variation&quot; section of this contains a detailed description of all sources of variation data
for that species.<br>
For example, in human there is data from;
<ul>
<li><a rel="external" href="http://www.ncbi.nlm.nih.gov/projects/SNP/">dbSNP</a></li>
<li>Illumina chips</li>
<li><a rel="external" href="http://www.affymetrix.com/index.affx">Affymetrix GeneChip Arrays</a></li> 
<li><a rel="external" href="http://www.hgmd.cf.ac.uk/ac/index.php">Human Gene Mutation Database
(HGMD)</a> (public identifiers)</li>
<li><a rel="external" href="http://www.uniprot.org/">UniProt</a></li>
<li><a rel="external" href="http://www.sanger.ac.uk/genetics/CGP/cosmic/">Catalogue Of Somatic Mutations In
Cancer (COSMIC)</a></li>
<li><a rel="external" href="http://www.ebi.ac.uk/dgva/page.php">DGVa</a> and <a
rel="external" href="http://www.ncbi.nlm.nih.gov/dbvar/">dbVAR</a>: structural polymorphism data</li>
<li><a rel="external"
href="http://www.genome.gov/">NHGRI</a>, <a rel="external" href="http://www.ebi.ac.uk/ega/page.php">European
Genome-phenome Archive (EGA)</a>, <a rel="external" href="http://www.biomedcentral.com/1471-2350/10/6">Open
GWAS DB</a>: phenotype information from genome-wide association studies.</li>
<li>Locus Reference Genomic sequences (<a rel="external"
href="http://www.lrg-sequence.org/page.php">LRG</a>)</li>
</ul>

<br />
<h2 id="displays"> Variation Displays </h2>
<p>
Variation data can be viewed in the browser through pages such
as:
</p>

<ul>
<li><strong>Gene:</strong> Variation table and Variation image (for
all variations in a gene) e.g. for example <a href="/Homo_sapiens/Gene/Variation_Gene/Table?db=core;g=ENSG00000159197">for all variants in KCNE2</a> </li>
<li><strong>Transcript:</strong> Population comparison, Comparison
image (for comparing variants in a transcript across different individual or strain
sequences) e.g. <a href="/Mus_musculus/Transcript/Population/Image?g=ENSMUSG00000041143;r=4:138528820-138586500;t=ENSMUST00000084208"> compare Tmco4 in different mouse strains </a> </li>
<li> <strong> Transcript: </strong> Sequence, protein: list of the coding variants in protein coordinates.</li>
<li><strong>Location:</strong> Region in Detail (Variations can be
drawn using &quot;Configure this page&quot; at the left. The menu allows
display of information in Ensembl databases along with external
sources in DAS format such as <a rel="external"
href="http://projects.tcag.ca/variation/">DGV loci</a>.)</li>
</ul>

<p>
Clicking on any variation on an Ensembl page will open a Variation tab
with information about the flanking sequence and source for the
selected variation. Links to linkage disequilibrium (LD) plots,
phenotype information (for human) from EGA, OMIM and <a rel="external"
href="http://www.genome.gov/">NHGRI</a> and Ensembl genes and
transcripts that include the variation can be found at the left of
this tab. You may also view multiple genome alignments of various
species, highlighting the
variation. <a href="/info/docs/compara/analyses.html#ancestral"
class="cp-external">Ancestral sequences</a> are included in this
display. 
</p>

<p>
Variation information can also be accessed
using <a href="/info/data/biomart.html"
class="cp-external">BioMart</a> (gene or variation database), and
the <a href="/info/docs/api/index.html" class="cp-external">Perl
API</a> (<a href="/info/docs/api/variation/index.html"
class="cp-external">variation databases</a>).
</p>

<br />
<h2 id="classes"> Variation classes </h2>
See the different classes, with the corresponding <a rel="external" href="http://www.sequenceontology.org/">Sequence
Ontology</a> IDs in parenthesis:
<ul>
	<li>SNP (<a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001483">SO:0001483</a>)</li>
	<li>CNV (<a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001019">SO:0001019</a>)</li>
	<li>INSERTION (<a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0000667">SO:0000667</a>)</li>
	<li>DELETION (<a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0000159">SO:0000159</a>)</li>
	<li>INDEL (<a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:1000032">SO:1000032</a>)</li>
	<li>SEQUENCE ALTERATION (<a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0001059">SO:0001059</a>)</li>
	<li>SUBSTITUTION (<a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:1000002">SO:1000002</a>)</li>
	<li>TANDEM REPEAT (<a rel="external" href="http://www.sequenceontology.org/miso/current_release/term/SO:0000705">SO:0000705</a>)</li>
</ul>

<p>A variation is classified according to its component alleles and the class is
displayed on the website using the Ensembl display term. These terms are
intended to align with Sequence Ontology terms wherever possible, but some SO
definitions do not match ours (specifically our term SNP is closer to the SO
term SNV than the SO term SNP). The variation_class table in the variation
database shows the mapping between display terms and SO terms, and API users can
fetch either term from a Variation object. We also differentiate somatic
mutations from germline variations in the display term, prefixing the term with
'somatic_'.</p>


<h3 id="indel">Insertion and Deletion coordinates </h3>

<p> In Ensembl, an insertion is indicated by start coordinate = end coordinate +
1. For example, an insertion of 'C' between nucleotides 12600 and
12601 on the forward strand is indicated with start and end coordinates as
follows:</p>

<pre>
   12601     12600   
</pre>
<p>A deletion is indicated by the exact nucleotide coordinates. For
example, a three base pair deletion of nucleotides 12600, 12601, and
12602 of the reverse strand will have start and end coordinates of :</p>

<pre>
   12600     12602    
</pre>

<br />
<h2 id="consequences"> Predicted variation consequences</h2>

<p> The variation consequence types describe the location or effect of a
variation on a transcript. These are shown in order of estimated severity as assigned by
Ensembl and used by the Perl API for display purposes. Please note this order is subjective and works for the majority of cases.
Click <a href="consequences.jpg">here</a>
for a diagram showing the location of each type relative to the transcript
structure.
</p>

<ul>

<li><strong>Essential splice site</strong> - In the first 2 or the
last 2 basepairs of an intron.</li>

<li><strong>Stop gained</strong> - In coding sequence, resulting in
the gain of a stop codon (i.e. leading to a shortened peptide
sequence).</li>

<li><strong>Stop lost</strong> - In coding sequence, resulting in the
loss of a stop codon.</li>

<li><strong>Complex InDel</strong> - Insertion or deletion that spans
an exon/intron border or a coding sequence/UTR border.</li>

<li><strong>Frameshift variations</strong> - In coding sequence,
resulting in a frameshift.</li>

<li><strong>Non-synonymous SNPs</strong> - SNPs that are located in
the coding sequence and result in an amino acid change in the encoded
peptide sequence.</li>

<li><strong>Splice site</strong> - 1-3 bps into an exon or 3-8 bps
into an intron.</li>

<li><strong>Partial codon</strong> - Located within the final, incomplete
codon of a transcript with a shortened coding sequence where the end is
unknown.</li>

<li><strong>Synonymous SNPs</strong> - In coding sequence, not
resulting in an amino acid change (i.e. silent mutation).</li>

<li><strong>Regulatory region variations</strong> - In regulatory
region annotated by Ensembl.</li>

<li><strong>Within mature miRNA</strong> - Located within a
microRNA.</li>

<li><strong>5prime UTR variations</strong> - In 5prime UTR
(untranslated region).</li>

<li><strong>3prime UTR variations</strong> - In 3prime UTR.</li>

<li><strong>Intronic variations</strong> - In intron.</li>

<li><strong>NMD transcript</strong> - Located within a transcript
predicted to undergo nonsense-mediated decay.</li>

<li><strong>Within non-coding gene</strong> - Located within a gene
that does not code for a protein.</li>

<li><strong>Upstream variations</strong> - Within 5 kb upstream of the
5prime end of a transcript.</li>

<li><strong>Downstream variations</strong> - Within 5 kb downstream of
the 3prime end of a transcript.</li>

<li><strong>Intergenic variations</strong> - More than 5 kb either
upstream or downstream of a transcript.</li>

</ul>

<h3 id="input"> Calculating variation consequences for your own data </h3>

<p> You can also calculate consequences for your own data via the website. Please see
<a href="/info/website/upload/var.html"> more information here </a>
</p>


<br />
<h2 id="notes">Notes</h2>

<p>For more detailed information on the variation specific help see <a href="/Help/View?id=168">here</a>.<br
/>
There is also more information about programmatic access in the <a
href="/info/docs/api/variation/variation_tutorial.html">variation API tutorial</a> page.
</p>


<br />
<h2 id="references">References</h2>

<p>McLaren W, Pritchard B, Rios D, Chen Y, Flicek P, Cunningham F.<br />
<strong>Deriving the consequences of genomic variants with the Ensembl API and SNP Effect Predictor.</strong><br />
<em>BMC Bioinformatics</em>26(16):2069-70(2010)<br />
<a href="http://dx.doi.org/10.1093/bioinformatics/btq330"
rel="external">doi:10.1093/bioinformatics/btq330 </a>
</p> 

<p>
Rios D, McLaren WM, Chen Y, Birney E, Stabenau A, Flicek P, Cunningham F.<br />
<strong>A Database and API for variation, dense genotyping and resequencing data</strong><br />
<em>BMC Bioinformatics</em> 11:238 (2010)<br />
<a href="http://dx.doi.org/10.1186/1471-2105-11-238"
rel="external">doi:10.1186/1471-2105-11-238</a>
</p>

<p>Chen Y, Cunningham F, Rios D, McLaren WM, Smith J, Pritchard B, Spudich GM, Brent S, Kulesha E, Marin-Garcia P, Smedley D, Birney E, Flicek P.<br />
<strong>Ensembl Variation Resources</strong><br />
<em>BMC Genomics</em> 11(1):293 (2010)<br />
<a href="http://dx.doi.org/10.1186/1471-2164-11-293"
rel="external">doi:10.1186/1471-2164-11-293</a>
</p>

<p>
Redon, R. <i>et al.</i><br />
<strong>Global variation in copy number in the human
genome</strong><br />
<em>Nature</em> 444:444-454 (2006)<br />
<a href="http://dx.doi.org/10.1038/nature05329"
rel="external">doi:10.1038/nature05329</a>
</p>

<p>
Spencer, C. C. A. <i>et al.</i><br />
<strong>The Influence of Recombination on Human Genetic
Diversity</strong><br />
<em>PLoS Genet.</em> 2(9):e148 (2006)<br />
<a href="http://dx.doi.org/10.1371/journal.pgen.0020148"
rel="external">doi:10.1371/journal.pgen.0020148</a>
</p>

<p>
Venter, J. C. <i>et al.</i><br />
<strong>The Sequence of the Human Genome</strong><br />
<em>Science</em> 291(5507):1304-51 (2001)<br />
<a href="http://dx.doi.org/10.1126/science.1058040"
rel="external">doi:10.1126/science.1058040</a>
</p>

<p>
Wheeler, D. A. <i>et al.</i><br />
<strong>The complete genome of an individual by massively parallel DNA
sequencing</strong><br />
<em>Nature</em> 452:872-876 (2008)<br />
<a href="http://dx.doi.org/10.1371/10.1038/nature06884"
rel="external">doi:10.1038/nature06884</a>
</p>

</body>
</html>
