<html>
<head>
<meta name="order" content="1" />
<title>Variation</title>
</head>

<body>

<h1>Ensembl Variation</h1>

<p>
Ensembl imports variations including Single Nucleotide Polymorphisms
(SNPs) and insertion-deletion mutations (Indels) and their flanking
sequence from various sources. These sequences are aligned to the
reference sequence. Variation positions are calculated in this way
along with any effects on transcripts in the area. The majority of
variations are obtained from NCBI <a rel="external"
href="http://www.ncbi.nlm.nih.gov/projects/SNP/">dbSNP</a>. For human,
other sources include <a rel="external"
href="http://www.affymetrix.com/index.affx">Affy GeneChip Arrays</a>,
The <a rel="external"
href="http://www.ebi.ac.uk/ega/page.php">European Genome-phenome
Archive (EGA)</a>, and whole genome alignments of individual sequences
from Venter (Levy, 2007), Watson (Wheeler, 2008) and Celera
individuals (Venter, 2001) to the reference sequence. Sources for
other species include Sanger resequencing projects
for <a rel="external"
href="http://www.sanger.ac.uk/Projects/M_musculus/">mouse</a>, and
alignments of sequences from
the <a href="/Rattus_norvegicus/star/index.html"
class="cp-external">STAR consortium</a> for rat. Ancestral alleles
from <a rel="external"
href="http://www.ncbi.nlm.nih.gov/projects/SNP/">dbSNP</a> were
determined through a comparison study of human and chimp DNA (Spencer,
2006).
</p>

<p>
Future directions: Ensembl will be incorporating data from
the <a rel="external" href="http://www.1000genomes.org/page.php">1000
Genomes Project</a> including structural variation data once this has
been submitted to <a rel="external"
href="http://www.ncbi.nlm.nih.gov/projects/SNP/">dbSNP</a>. Ensembl is
also working towards increasing the phenotype data resources and their
relationship to genotypes.
</p>

<p>
Variations and sources can be viewed in the browser through pages such
as:
</p>

<ul>
<li><strong>Gene:</strong> Variation table and Variation image (for
all variations in a gene)</li>
<li><strong>Transcript:</strong> Population comparison and Comparison
image (for all variations in a transcript across different
sequences)</li>
<li><strong>Location:</strong> Region in Detail (Variations can be
drawn using &quot;Configure this page&quot; at the left. The menu allows
display of information in Ensembl databases along with external
sources in DAS format such as <a rel="external"
href="http://projects.tcag.ca/variation/">DGV loci</a>.)</li>
</ul>

<p><strong>Variation Types</strong></p>

<p>
The variation types describe the location or effect on the transcript.
</p>

<ul>

<li><strong>Non-synonymous SNPs</strong> - SNPs that are located in
the coding sequence and result in an amino acid change in the encoded
peptide sequence.</li>

<li><strong>Synonymous SNPs</strong> - In coding sequence, not
resulting in an amino acid change (i.e. silent mutation).</li>

<li><strong>Frameshift variations</strong> - In coding sequence,
resulting in a frameshift.</li>

<li><strong>Stop lost</strong> - In coding sequence, resulting in the
loss of a stop codon.</li>

<li><strong>Stop gained</strong> - In coding sequence, resulting in
the gain of a stop codon (i.e. leading to a shortened peptide
sequence).</li>

<li><strong>Essential splice site</strong> - In the first 2 or the
last 2 basepairs of an intron.</li>

<li><strong>Splice site</strong> - 1-3 bps into an exon or 3-8 bps
into an intron.</li>

<li><strong>Upstream variations</strong> - Within 5 kb upstream of the
5prime end of a transcript.</li>

<li><strong>Regulatory region variations</strong> - In regulatory
region annotated by Ensembl.</li>

<li><strong>5prime UTR variations</strong> - In 5prime UTR
(untranslated region).</li>

<li><strong>Intronic variations</strong> - In intron.</li>

<li><strong>3prime UTR variations</strong> - In 3prime UTR.</li>

<li><strong>Downstream variations</strong> - Within 5 kb downstream of
the 3prime end of a transcript.</li>

<li><strong>Intergenic variations</strong> - More than 5 kb either
upstream or downstream of a transcript.</li>

<li><strong>Complex InDel</strong> - Insertion or deletion that spans
an exon/intron border or a coding sequence/UTR border.</li>

<li><strong>Within mature miRNA</strong> - Located within a
microRNA.</li>

<li><strong>Within non-coding gene</strong> - Located within a gene
that does not code for a protein.</li>

</ul>

<p>
Clicking on any variation on an Ensembl page will open a Variation tab
with information about the flanking sequence and source for the
selected variation. Links to linkage disequilibrium (LD) plots,
phenotype information (for human) from EGA and <a rel="external"
href="http://www.genome.gov/">NHGRI</a> and Ensembl genes and
transcripts that include the variation can be found at the left of
this tab. You may also view multiple genome alignments of various
species, highlighting the
variation. <a href="/info/docs/compara/analyses.html#ancestral"
class="cp-external">Ancestral sequences</a> are included in this
display. 
</p>

<p>
Variation information can also be accessed
using <a href="/info/data/biomart.html"
class="cp-external">BioMart</a> (gene or variation database), and
the <a href="/info/docs/api/index.html" class="cp-external">Perl
API</a> (<a href="/info/docs/api/variation/index.html"
class="cp-external">variation databases</a>).
</p>

<h2>References</h2>

<p>
Levy, S. <i>et al.</i><br />
<strong>The Diploid Genome Sequence of an Individual Human</strong><br />
<em>PLoS Biol.</em> 5(10):e254 (2007)<br />
<a href="http://dx.doi.org/10.1371/journal.pbio.0050254"
rel="external">doi:10.1371/journal.pbio.0050254</a>
</p>

<p>
Redon, R. <i>et al.</i><br />
<strong>Global variation in copy number in the human
genome</strong><br />
<em>Nature</em> 444:444-454 (2006)<br />
<a href="http://dx.doi.org/10.1038/nature05329"
rel="external">doi:10.1038/nature05329</a>
</p>

<p>
Spencer, C. C. A. <i>et al.</i><br />
<strong>The Influence of Recombination on Human Genetic
Diversity</strong><br />
<em>PLoS Genet.</em> 2(9):e148 (2006)<br />
<a href="http://dx.doi.org/10.1371/journal.pgen.0020148"
rel="external">doi:10.1371/journal.pgen.0020148</a>
</p>

<p>
Venter, J. C. <i>et al.</i><br />
<strong>The Sequence of the Human Genome</strong><br />
<em>Science</em> 291(5507):1304-51 (2001)<br />
<a href="http://dx.doi.org/10.1126/science.1058040"
rel="external">doi:10.1126/science.1058040</a>
</p>

<p>
Wheeler, D. A. <i>et al.</i><br />
<strong>The complete genome of an individual by massively parallel DNA
sequencing</strong><br />
<em>Nature</em> 452:872-876 (2008)<br />
<a href="http://dx.doi.org/10.1371/10.1038/nature06884"
rel="external">doi:10.1038/nature06884</a>
</p>

</body>
</html>
