
<html>
<head>
	<meta http-equiv="CONTENT-TYPE" content="text/html; charset=utf-8" />
	<title>Variation Sources</title>
</head>

<body>
<b>
<a href="index.html">About Ensembl Variation</a> |
<a href="data_description.html">Data Description</a> | 
<a href="predicted_data.html">Predicted Data</a> |
<a href="database.html">Database Description</a> | 
<a href="/info/docs/api/variation/index.html">Perl API</a> |
<a href="vep/index.html">Variant Effect Predictor</a>
</b>
<br /><br />

<h1>Ensembl Variation - Sources Documentation</h1>

<h2>List of Variation sources for each species - Ensembl 68</h2>



	<table id="bos_taurus"><tr style="vertical-align:middle">
		<td style="padding-left:0px"><img src="http://static.ensembl.org/img/species/thumb_Bos_taurus.png" alt="Bos taurus" /></td>
		<td style="padding-left:10px"><h3>Bos taurus</h3></td>
	</tr></table>
	
			<table class="ss" style="width:60%">
				<tr><th colspan="2">Source</th><th>Version</th><th>Description</th><th></th></tr>
		 
				<tr class="bg1">
					<td style="width:4px;padding:0px;margin:0px"></td>
					<td><a href="http://www.ncbi.nlm.nih.gov/projects/SNP/">dbSNP</a></td>
					<td style="text-align:center">133</td>
					<td>Variants (including SNPs and indels) imported from dbSNP</td>
					<td style="text-align:center;width:80px"></td>
				</tr>
			
				<tr class="bg2">
					<td style="width:4px;padding:0px;margin:0px;background-color:#00F"></td>
					<td><a href="http://www.ebi.ac.uk/dgva/">DGVa</a></td>
					<td style="text-align:center">06/2012</td>
					<td>Database of Genomic Variants Archive</td>
					<td style="text-align:center;width:80px"><span style="color:#00F">New source</span></td>
				</tr>
			</table><br />
	<table id="canis_familiaris"><tr style="vertical-align:middle">
		<td style="padding-left:0px"><img src="http://static.ensembl.org/img/species/thumb_Canis_familiaris.png" alt="Canis familiaris" /></td>
		<td style="padding-left:10px"><h3>Canis familiaris</h3></td>
	</tr></table>
	
			<table class="ss" style="width:60%">
				<tr><th colspan="2">Source</th><th>Version</th><th>Description</th><th></th></tr>
		 
				<tr class="bg1">
					<td style="width:4px;padding:0px;margin:0px;background-color:#090"></td>
					<td><a href="http://www.ncbi.nlm.nih.gov/projects/SNP/">dbSNP</a></td>
					<td style="text-align:center">131</td>
					<td>Variants (including SNPs and indels) imported from dbSNP [remapped from build CanFam2.0]</td>
					<td style="text-align:center;width:80px"><span style="color:#090">New version</span></td>
				</tr>
			</table><br />
	<table id="danio_rerio"><tr style="vertical-align:middle">
		<td style="padding-left:0px"><img src="http://static.ensembl.org/img/species/thumb_Danio_rerio.png" alt="Danio rerio" /></td>
		<td style="padding-left:10px"><h3>Danio rerio</h3></td>
	</tr></table>
	
			<table class="ss" style="width:60%">
				<tr><th colspan="2">Source</th><th>Version</th><th>Description</th><th></th></tr>
		 
				<tr class="bg1">
					<td style="width:4px;padding:0px;margin:0px"></td>
					<td><a href="http://www.ncbi.nlm.nih.gov/projects/SNP/">dbSNP</a></td>
					<td style="text-align:center">136</td>
					<td>Variants (including SNPs and indels) imported from dbSNP</td>
					<td style="text-align:center;width:80px"></td>
				</tr>
			
				<tr class="bg2">
					<td style="width:4px;padding:0px;margin:0px;background-color:#00F"></td>
					<td><a href="http://www.ebi.ac.uk/dgva/">DGVa</a></td>
					<td style="text-align:center">06/2012</td>
					<td>Database of Genomic Variants Archive</td>
					<td style="text-align:center;width:80px"><span style="color:#00F">New source</span></td>
				</tr>
			</table><br />
	<table id="drosophila_melanogaster"><tr style="vertical-align:middle">
		<td style="padding-left:0px"><img src="http://static.ensembl.org/img/species/thumb_Drosophila_melanogaster.png" alt="Drosophila melanogaster" /></td>
		<td style="padding-left:10px"><h3>Drosophila melanogaster</h3></td>
	</tr></table>
	
			<table class="ss" style="width:60%">
				<tr><th colspan="2">Source</th><th>Version</th><th>Description</th><th></th></tr>
		 
				<tr class="bg1">
					<td style="width:4px;padding:0px;margin:0px"></td>
					<td>DPGP</td>
					<td style="text-align:center">1</td>
					<td>Variation features from DPGP, with Ensembl identifiers</td>
					<td style="text-align:center;width:80px"></td>
				</tr>
			</table><br />
	<table id="equus_caballus"><tr style="vertical-align:middle">
		<td style="padding-left:0px"><img src="http://static.ensembl.org/img/species/thumb_Equus_caballus.png" alt="Equus caballus" /></td>
		<td style="padding-left:10px"><h3>Equus caballus</h3></td>
	</tr></table>
	
			<table class="ss" style="width:60%">
				<tr><th colspan="2">Source</th><th>Version</th><th>Description</th><th></th></tr>
		 
				<tr class="bg1">
					<td style="width:4px;padding:0px;margin:0px"></td>
					<td><a href="http://www.ncbi.nlm.nih.gov/projects/SNP/">dbSNP</a></td>
					<td style="text-align:center">130</td>
					<td>Variants (including SNPs and indels) imported from dbSNP</td>
					<td style="text-align:center;width:80px"></td>
				</tr>
			
				<tr class="bg2">
					<td style="width:4px;padding:0px;margin:0px;background-color:#00F"></td>
					<td><a href="http://www.ebi.ac.uk/dgva/">DGVa</a></td>
					<td style="text-align:center">06/2012</td>
					<td>Database of Genomic Variants Archive</td>
					<td style="text-align:center;width:80px"><span style="color:#00F">New source</span></td>
				</tr>
			</table><br />
	<table id="felis_catus"><tr style="vertical-align:middle">
		<td style="padding-left:0px"><img src="http://static.ensembl.org/img/species/thumb_Felis_catus.png" alt="Felis catus" /></td>
		<td style="padding-left:10px"><h3>Felis catus</h3></td>
	</tr></table>
	
			<table class="ss" style="width:60%">
				<tr><th colspan="2">Source</th><th>Version</th><th>Description</th><th></th></tr>
		 
				<tr class="bg1">
					<td style="width:4px;padding:0px;margin:0px"></td>
					<td><a href="http://www.ncbi.nlm.nih.gov/projects/SNP/">dbSNP</a></td>
					<td style="text-align:center">127</td>
					<td>Variants (including SNPs and indels) imported from dbSNP and mapped to the CAT assembly</td>
					<td style="text-align:center;width:80px"></td>
				</tr>
			</table><br />
	<table id="gallus_gallus"><tr style="vertical-align:middle">
		<td style="padding-left:0px"><img src="http://static.ensembl.org/img/species/thumb_Gallus_gallus.png" alt="Gallus gallus" /></td>
		<td style="padding-left:10px"><h3>Gallus gallus</h3></td>
	</tr></table>
	
			<table class="ss" style="width:60%">
				<tr><th colspan="2">Source</th><th>Version</th><th>Description</th><th></th></tr>
		 
				<tr class="bg1">
					<td style="width:4px;padding:0px;margin:0px"></td>
					<td><a href="http://www.ncbi.nlm.nih.gov/projects/SNP/">dbSNP</a></td>
					<td style="text-align:center">128</td>
					<td>Variants (including SNPs and indels) imported from dbSNP</td>
					<td style="text-align:center;width:80px"></td>
				</tr>
			</table><br />
	<table id="homo_sapiens"><tr style="vertical-align:middle">
		<td style="padding-left:0px"><img src="http://static.ensembl.org/img/species/thumb_Homo_sapiens.png" alt="Homo sapiens" /></td>
		<td style="padding-left:10px"><h3>Homo sapiens</h3></td>
	</tr></table>
	
			<table class="ss" style="width:60%">
				<tr><th colspan="2">Source</th><th>Version</th><th>Description</th><th></th></tr>
		 
				<tr class="bg1">
					<td style="width:4px;padding:0px;margin:0px"></td>
					<td><a href="http://www.ncbi.nlm.nih.gov/projects/SNP/">dbSNP</a></td>
					<td style="text-align:center">135</td>
					<td>Variants (including SNPs and indels) imported from dbSNP</td>
					<td style="text-align:center;width:80px"></td>
				</tr>
			
				<tr class="bg2">
					<td style="width:4px;padding:0px;margin:0px;background-color:#090"></td>
					<td><a href="http://www.ebi.ac.uk/dgva/">DGVa</a></td>
					<td style="text-align:center">06/2012</td>
					<td>Database of Genomic Variants Archive</td>
					<td style="text-align:center;width:80px"><span style="color:#090">New version</span></td>
				</tr>
			
				<tr class="bg1">
					<td style="width:4px;padding:0px;margin:0px;background-color:#090"></td>
					<td><a href="http://www.genome.gov/gwastudies/">NHGRI_GWAS_catalog</a></td>
					<td style="text-align:center">02/06/2012</td>
					<td>Variants associated with phenotype data from the NHGRI GWAS catalog</td>
					<td style="text-align:center;width:80px"><span style="color:#090">New version</span></td>
				</tr>
			
				<tr class="bg2">
					<td style="width:4px;padding:0px;margin:0px;background-color:#090"></td>
					<td><a href="http://www.sanger.ac.uk/genetics/CGP/cosmic/">COSMIC</a></td>
					<td style="text-align:center">59</td>
					<td>Somatic mutations found in human cancers from the COSMIC project</td>
					<td style="text-align:center;width:80px"><span style="color:#090">New version</span></td>
				</tr>
			
				<tr class="bg1">
					<td style="width:4px;padding:0px;margin:0px;background-color:#090"></td>
					<td><a href="http://www.ebi.ac.uk/ega/">EGA</a></td>
					<td style="text-align:center">06/2012</td>
					<td>Variants imported from the European Genome-phenome Archive with phenotype association</td>
					<td style="text-align:center;width:80px"><span style="color:#090">New version</span></td>
				</tr>
			
				<tr class="bg2">
					<td style="width:4px;padding:0px;margin:0px;background-color:#090"></td>
					<td><a href="http://www.uniprot.org/">Uniprot</a></td>
					<td style="text-align:center">13/06/2012</td>
					<td>Variants with protein annotation imported from Uniprot</td>
					<td style="text-align:center;width:80px"><span style="color:#090">New version</span></td>
				</tr>
			
				<tr class="bg1">
					<td style="width:4px;padding:0px;margin:0px"></td>
					<td><a href="http://www.hgmd.cf.ac.uk/ac/index.php">HGMD-PUBLIC</a></td>
					<td style="text-align:center">2012.1</td>
					<td>Variants from HGMD-PUBLIC dataset March 2012</td>
					<td style="text-align:center;width:80px"></td>
				</tr>
			
				<tr class="bg2">
					<td style="width:4px;padding:0px;margin:0px;background-color:#090"></td>
					<td><a href="http://www.omim.org/">OMIM</a></td>
					<td style="text-align:center">18/06/2012</td>
					<td>Variations linked to entries in the Online Mendelian Inheritance in Man (OMIM) database</td>
					<td style="text-align:center;width:80px"><span style="color:#090">New version</span></td>
				</tr>
			
				<tr class="bg1">
					<td style="width:4px;padding:0px;margin:0px"></td>
					<td><a href="http://www.biomedcentral.com/1471-2350/10/6">Open Access GWAS Database</a></td>
					<td style="text-align:center">-</td>
					<td>Johnson & O'Donnell 'An Open Access Database of Genome-wide Association Results' PMID:19161620</td>
					<td style="text-align:center;width:80px"></td>
				</tr>
			
				<tr class="bg2">
					<td style="width:4px;padding:0px;margin:0px"></td>
					<td><a href="https://oi.gene.le.ac.uk/home.php?select_db=LEPRE1">LSDB_LEPRE1</a></td>
					<td style="text-align:center">08/04/2011</td>
					<td>LEPRE1 homepage - Osteogenesis Imperfecta Variant Database - Leiden Open Variation Database</td>
					<td style="text-align:center;width:80px"></td>
				</tr>
			
				<tr class="bg1">
					<td style="width:4px;padding:0px;margin:0px"></td>
					<td><a href="https://oi.gene.le.ac.uk/home.php?select_db=PPIB">LSDB_PPIB</a></td>
					<td style="text-align:center">17/02/2011</td>
					<td>PPIB homepage - Osteogenesis Imperfecta Variant Database - Leiden Open Variation Database</td>
					<td style="text-align:center;width:80px"></td>
				</tr>
			
				<tr class="bg2">
					<td style="width:4px;padding:0px;margin:0px"></td>
					<td><a href="https://oi.gene.le.ac.uk/home.php?select_db=CRTAP">LSDB_CRTAP</a></td>
					<td style="text-align:center">08/04/2011</td>
					<td>CRTAP homepage - Osteogenesis Imperfecta Variant Database - Leiden Open Variation Database</td>
					<td style="text-align:center;width:80px"></td>
				</tr>
			
				<tr class="bg1">
					<td style="width:4px;padding:0px;margin:0px"></td>
					<td><a href="https://oi.gene.le.ac.uk/home.php?select_db=FKBP10">LSDB_FKBP10</a></td>
					<td style="text-align:center">16/05/2011</td>
					<td>FKBP10 homepage - Osteogenesis Imperfecta Variant Database - Leiden Open Variation Database</td>
					<td style="text-align:center;width:80px"></td>
				</tr>
			</table><br />
			<table class="ss" style="width:60%">
				<tr><th colspan="2">Chip Source</th><th>Version</th><th>Description</th><th></th></tr>
		 
				<tr class="bg1">
					<td style="width:4px;padding:0px;margin:0px"></td>
					<td><a href="http://www.affymetrix.com/">Affy GeneChip 100K Array</a></td>
					<td style="text-align:center">-</td>
					<td>Variants from the Affymetrix GeneChip Human Mapping 100K Array Set</td>
					<td style="text-align:center;width:80px"></td>
				</tr>
			
				<tr class="bg2">
					<td style="width:4px;padding:0px;margin:0px"></td>
					<td><a href="http://www.affymetrix.com/">Affy GeneChip 500K Array</a></td>
					<td style="text-align:center">-</td>
					<td>Variants from the Affymetrix GeneChip Human Mapping 500K Array Set</td>
					<td style="text-align:center;width:80px"></td>
				</tr>
			
				<tr class="bg1">
					<td style="width:4px;padding:0px;margin:0px"></td>
					<td><a href="http://www.affymetrix.com/">Affy GenomeWideSNP_6.0</a></td>
					<td style="text-align:center">-</td>
					<td>Variants from the Affymetrix Genome-Wide Human SNP Array 6.0</td>
					<td style="text-align:center;width:80px"></td>
				</tr>
			
				<tr class="bg2">
					<td style="width:4px;padding:0px;margin:0px"></td>
					<td><a href="http://www.illumina.com/">Illumina_CytoSNP12v1</a></td>
					<td style="text-align:center">-</td>
					<td>Variants from the Illumina Cyto SNP-12 v1 whole genome SNP genotyping chip designed for cytogenetic analysis</td>
					<td style="text-align:center;width:80px"></td>
				</tr>
			
				<tr class="bg1">
					<td style="width:4px;padding:0px;margin:0px"></td>
					<td><a href="http://www.illumina.com/">Illumina_Human660W-quad</a></td>
					<td style="text-align:center">-</td>
					<td>Variants from the Illumina Human 660W-Quad whole genome SNP genotyping chip designed for association studies</td>
					<td style="text-align:center;width:80px"></td>
				</tr>
			
				<tr class="bg2">
					<td style="width:4px;padding:0px;margin:0px"></td>
					<td><a href="http://www.illumina.com/">Illumina_Human1M-duoV3</a></td>
					<td style="text-align:center">-</td>
					<td>Variants from the Illumina Human 1M-DuoV3 whole genome SNP genotyping chip designed for association studies</td>
					<td style="text-align:center;width:80px"></td>
				</tr>
			
				<tr class="bg1">
					<td style="width:4px;padding:0px;margin:0px"></td>
					<td><a href="http://www.affymetrix.com/">Affy GenomeWideSNP_6.0 CNV</a></td>
					<td style="text-align:center">-</td>
					<td>Copy Number Variation (CNV) probes from the Affymetrix Genome-Wide Human SNP Array 6.0</td>
					<td style="text-align:center;width:80px"></td>
				</tr>
			
				<tr class="bg2">
					<td style="width:4px;padding:0px;margin:0px;background-color:#00F"></td>
					<td><a href="http://support.illumina.com/array/array_kits/humancardio-metabo_beadchip_kit.ilmn">Illumina_Cardio-Metabo_Chip</a></td>
					<td style="text-align:center">1</td>
					<td>Illumina Human Cardio-Metabo_Chip</td>
					<td style="text-align:center;width:80px"><span style="color:#00F">New source</span></td>
				</tr>
			
				<tr class="bg1">
					<td style="width:4px;padding:0px;margin:0px;background-color:#00F"></td>
					<td><a href="http://support.illumina.com/array/array_kits/humanomni1-quad_mu_beadchip_kit/documentation.ilmn">Illumina_HumanOmni1-Quad</a></td>
					<td style="text-align:center">1</td>
					<td>Illumina Human_HumanOmni1-Quad</td>
					<td style="text-align:center;width:80px"><span style="color:#00F">New source</span></td>
				</tr>
			</table><br />
	<table id="macaca_mulatta"><tr style="vertical-align:middle">
		<td style="padding-left:0px"><img src="http://static.ensembl.org/img/species/thumb_Macaca_mulatta.png" alt="Macaca mulatta" /></td>
		<td style="padding-left:10px"><h3>Macaca mulatta</h3></td>
	</tr></table>
	
			<table class="ss" style="width:60%">
				<tr><th colspan="2">Source</th><th>Version</th><th>Description</th><th></th></tr>
		 
				<tr class="bg1">
					<td style="width:4px;padding:0px;margin:0px"></td>
					<td><a href="http://www.ncbi.nlm.nih.gov/projects/SNP/">dbSNP</a></td>
					<td style="text-align:center">136</td>
					<td>Variants (including SNPs and indels) imported from dbSNP</td>
					<td style="text-align:center;width:80px"></td>
				</tr>
			
				<tr class="bg2">
					<td style="width:4px;padding:0px;margin:0px"></td>
					<td><a href="http://www.ebi.ac.uk/dgva/">DGVa</a></td>
					<td style="text-align:center">10/2011</td>
					<td>Database of Genomic Variants Archive</td>
					<td style="text-align:center;width:80px"></td>
				</tr>
			</table><br />
	<table id="monodelphis_domestica"><tr style="vertical-align:middle">
		<td style="padding-left:0px"><img src="http://static.ensembl.org/img/species/thumb_Monodelphis_domestica.png" alt="Monodelphis domestica" /></td>
		<td style="padding-left:10px"><h3>Monodelphis domestica</h3></td>
	</tr></table>
	
			<table class="ss" style="width:60%">
				<tr><th colspan="2">Source</th><th>Version</th><th>Description</th><th></th></tr>
		 
				<tr class="bg1">
					<td style="width:4px;padding:0px;margin:0px"></td>
					<td><a href="http://www.ncbi.nlm.nih.gov/projects/SNP/">dbSNP</a></td>
					<td style="text-align:center">128</td>
					<td>Variants (including SNPs and indels) imported from dbSNP</td>
					<td style="text-align:center;width:80px"></td>
				</tr>
			</table><br />
	<table id="mus_musculus"><tr style="vertical-align:middle">
		<td style="padding-left:0px"><img src="http://static.ensembl.org/img/species/thumb_Mus_musculus.png" alt="Mus musculus" /></td>
		<td style="padding-left:10px"><h3>Mus musculus</h3></td>
	</tr></table>
	
			<table class="ss" style="width:60%">
				<tr><th colspan="2">Source</th><th>Version</th><th>Description</th><th></th></tr>
		 
				<tr class="bg1">
					<td style="width:4px;padding:0px;margin:0px"></td>
					<td><a href="http://www.ncbi.nlm.nih.gov/projects/SNP/">dbSNP</a></td>
					<td style="text-align:center">132</td>
					<td>Variants (including SNPs and indels) imported from dbSNP [remapped from build NCBIM37]</td>
					<td style="text-align:center;width:80px"></td>
				</tr>
			
				<tr class="bg2">
					<td style="width:4px;padding:0px;margin:0px;background-color:#090"></td>
					<td><a href="http://www.ebi.ac.uk/dgva/">DGVa</a></td>
					<td style="text-align:center">06/2012</td>
					<td>Database of Genomic Variants Archive</td>
					<td style="text-align:center;width:80px"><span style="color:#090">New version</span></td>
				</tr>
			</table><br />
	<table id="ornithorhynchus_anatinus"><tr style="vertical-align:middle">
		<td style="padding-left:0px"><img src="http://static.ensembl.org/img/species/thumb_Ornithorhynchus_anatinus.png" alt="Ornithorhynchus anatinus" /></td>
		<td style="padding-left:10px"><h3>Ornithorhynchus anatinus</h3></td>
	</tr></table>
	
			<table class="ss" style="width:60%">
				<tr><th colspan="2">Source</th><th>Version</th><th>Description</th><th></th></tr>
		 
				<tr class="bg1">
					<td style="width:4px;padding:0px;margin:0px"></td>
					<td>ENSEMBL</td>
					<td style="text-align:center">-</td>
					<td>Variants called by the ENSEMBL pipeline</td>
					<td style="text-align:center;width:80px"></td>
				</tr>
			
				<tr class="bg2">
					<td style="width:4px;padding:0px;margin:0px"></td>
					<td><a href="http://www.ncbi.nlm.nih.gov/projects/SNP/">dbSNP</a></td>
					<td style="text-align:center">-</td>
					<td>Variants (including SNPs and indels) imported from dbSNP</td>
					<td style="text-align:center;width:80px"></td>
				</tr>
			</table><br />
	<table id="pan_troglodytes"><tr style="vertical-align:middle">
		<td style="padding-left:0px"><img src="http://static.ensembl.org/img/species/thumb_Pan_troglodytes.png" alt="Pan troglodytes" /></td>
		<td style="padding-left:10px"><h3>Pan troglodytes</h3></td>
	</tr></table>
	
			<table class="ss" style="width:60%">
				<tr><th colspan="2">Source</th><th>Version</th><th>Description</th><th></th></tr>
		 
				<tr class="bg1">
					<td style="width:4px;padding:0px;margin:0px"></td>
					<td><a href="http://www.ncbi.nlm.nih.gov/projects/SNP/">dbSNP</a></td>
					<td style="text-align:center">136</td>
					<td>Variants (including SNPs and indels) imported from dbSNP</td>
					<td style="text-align:center;width:80px"></td>
				</tr>
			</table><br />
	<table id="pongo_abelii"><tr style="vertical-align:middle">
		<td style="padding-left:0px"><img src="http://static.ensembl.org/img/species/thumb_Pongo_abelii.png" alt="Pongo abelii" /></td>
		<td style="padding-left:10px"><h3>Pongo abelii</h3></td>
	</tr></table>
	
			<table class="ss" style="width:60%">
				<tr><th colspan="2">Source</th><th>Version</th><th>Description</th><th></th></tr>
		 
				<tr class="bg1">
					<td style="width:4px;padding:0px;margin:0px"></td>
					<td><a href="http://www.ncbi.nlm.nih.gov/projects/SNP/">dbSNP</a></td>
					<td style="text-align:center">136</td>
					<td>Variants (including SNPs and indels) imported from dbSNP</td>
					<td style="text-align:center;width:80px"></td>
				</tr>
			</table><br />
	<table id="rattus_norvegicus"><tr style="vertical-align:middle">
		<td style="padding-left:0px"><img src="http://static.ensembl.org/img/species/thumb_Rattus_norvegicus.png" alt="Rattus norvegicus" /></td>
		<td style="padding-left:10px"><h3>Rattus norvegicus</h3></td>
	</tr></table>
	
			<table class="ss" style="width:60%">
				<tr><th colspan="2">Source</th><th>Version</th><th>Description</th><th></th></tr>
		 
				<tr class="bg1">
					<td style="width:4px;padding:0px;margin:0px"></td>
					<td><a href="http://www.ncbi.nlm.nih.gov/projects/SNP/">dbSNP</a></td>
					<td style="text-align:center">136</td>
					<td>Variants (including SNPs and indels) imported from dbSNP</td>
					<td style="text-align:center;width:80px"></td>
				</tr>
			</table><br />
	<table id="saccharomyces_cerevisiae"><tr style="vertical-align:middle">
		<td style="padding-left:0px"><img src="http://static.ensembl.org/img/species/thumb_Saccharomyces_cerevisiae.png" alt="Saccharomyces cerevisiae" /></td>
		<td style="padding-left:10px"><h3>Saccharomyces cerevisiae</h3></td>
	</tr></table>
	
			<table class="ss" style="width:60%">
				<tr><th colspan="2">Source</th><th>Version</th><th>Description</th><th></th></tr>
		 
				<tr class="bg1">
					<td style="width:4px;padding:0px;margin:0px"></td>
					<td>SGRP</td>
					<td style="text-align:center">-</td>
					<td>Variation features from SGRP, with Ensembl identifiers</td>
					<td style="text-align:center;width:80px"></td>
				</tr>
			</table><br />
	<table id="sus_scrofa"><tr style="vertical-align:middle">
		<td style="padding-left:0px"><img src="http://static.ensembl.org/img/species/thumb_Sus_scrofa.png" alt="Sus scrofa" /></td>
		<td style="padding-left:10px"><h3>Sus scrofa</h3></td>
	</tr></table>
	
			<table class="ss" style="width:60%">
				<tr><th colspan="2">Source</th><th>Version</th><th>Description</th><th></th></tr>
		 
				<tr class="bg1">
					<td style="width:4px;padding:0px;margin:0px"></td>
					<td><a href="http://www.ncbi.nlm.nih.gov/projects/SNP/">dbSNP</a></td>
					<td style="text-align:center">136</td>
					<td>Variants (including SNPs and indels) imported from dbSNP</td>
					<td style="text-align:center;width:80px"></td>
				</tr>
			</table><br />
			<table class="ss" style="width:60%">
				<tr><th colspan="2">Chip Source</th><th>Version</th><th>Description</th><th></th></tr>
		 
				<tr class="bg1">
					<td style="width:4px;padding:0px;margin:0px"></td>
					<td><a href="http://www.illumina.com/products/porcine_snp60_whole_genome_genotyping_kits.ilmn">Illumina_PorcineSNP60</a></td>
					<td style="text-align:center">60</td>
					<td>PorcineSNP60 BeadChip</td>
					<td style="text-align:center;width:80px"></td>
				</tr>
			</table><br />
	<table id="taeniopygia_guttata"><tr style="vertical-align:middle">
		<td style="padding-left:0px"><img src="http://static.ensembl.org/img/species/thumb_Taeniopygia_guttata.png" alt="Taeniopygia guttata" /></td>
		<td style="padding-left:10px"><h3>Taeniopygia guttata</h3></td>
	</tr></table>
	
			<table class="ss" style="width:60%">
				<tr><th colspan="2">Source</th><th>Version</th><th>Description</th><th></th></tr>
		 
				<tr class="bg1">
					<td style="width:4px;padding:0px;margin:0px"></td>
					<td><a href="http://www.ncbi.nlm.nih.gov/projects/SNP/">dbSNP</a></td>
					<td style="text-align:center">131</td>
					<td>Variants (including SNPs and indels) imported from dbSNP and mapped to the Taeniopygia_guttata-3.2.4 assembly</td>
					<td style="text-align:center;width:80px"></td>
				</tr>
			</table><br />
	<table id="tetraodon_nigroviridis"><tr style="vertical-align:middle">
		<td style="padding-left:0px"><img src="http://static.ensembl.org/img/species/thumb_Tetraodon_nigroviridis.png" alt="Tetraodon nigroviridis" /></td>
		<td style="padding-left:10px"><h3>Tetraodon nigroviridis</h3></td>
	</tr></table>
	
			<table class="ss" style="width:60%">
				<tr><th colspan="2">Source</th><th>Version</th><th>Description</th><th></th></tr>
		 
				<tr class="bg1">
					<td style="width:4px;padding:0px;margin:0px"></td>
					<td><a href="http://www.ncbi.nlm.nih.gov/projects/SNP/">dbSNP</a></td>
					<td style="text-align:center">132</td>
					<td>Variants (including SNPs and indels) imported from dbSNP and mapped to the TETRAODON 8.0 assembly</td>
					<td style="text-align:center;width:80px"></td>
				</tr>
			</table>

<div style="border:1px solid #000;padding:5px;width:400px;margin-top:40px">
	<b>Legend</b><br />
	<table>
		<tr>
			<td style="width:5px;background-color:#090"></td>
			<td style="color:#090">New version of the data source in this release for the species</td>
		</tr>
		<tr>
			<td style="width:5px;background-color:#00F"></td>
			<td style="color:#00F">New data source in this release for the species</td>
		</tr>
	</table>
</div>
</body>
</html>
