
<html>
<head>
	<meta http-equiv="CONTENT-TYPE" content="text/html; charset=utf-8" />
	<title>Variation Sources Documentation</title>
</head>

<body>
<h1>Ensembl Variation Sources Documentation</h1>

<h2>List of Variation sources for each species - Ensembl 63</h2>



<h3 id="bos_taurus"># Bos taurus</h3>
			<table class="ss" style="width:60%">
				<tr><th>Source</th><th>Version</th><th>Description</th></tr>
		 
			<tr class="bg1">
				<td><a href="http://www.ncbi.nlm.nih.gov/projects/SNP/">dbSNP</a></td>
				<td>130</td>
				<td>Variants (including SNPs and indels) imported from dbSNP</td>
			</tr>
		</table><br />
<h3 id="canis_familiaris"># Canis familiaris</h3>
			<table class="ss" style="width:60%">
				<tr><th>Source</th><th>Version</th><th>Description</th></tr>
		 
			<tr class="bg1">
				<td><a href="http://www.ncbi.nlm.nih.gov/projects/SNP/">dbSNP</a></td>
				<td>126</td>
				<td>Variants (including SNPs and indels) imported from dbSNP</td>
			</tr>
		
			<tr class="bg2">
				<td><a href="http://www.ebi.ac.uk/dgva/">DGVa</a></td>
				<td>05/2011</td>
				<td>Database of Genomic Variants Archive</td>
			</tr>
		</table><br />
<h3 id="danio_rerio"># Danio rerio</h3>
			<table class="ss" style="width:60%">
				<tr><th>Source</th><th>Version</th><th>Description</th></tr>
		 
			<tr class="bg1">
				<td><a href="http://www.ncbi.nlm.nih.gov/projects/SNP/">dbSNP</a></td>
				<td>130</td>
				<td>Variants (including SNPs and indels) imported from dbSNP [remapped from build 7]</td>
			</tr>
		
			<tr class="bg2">
				<td>Sanger</td>
				<td>-</td>
				<td>Variants called by the Stemple Lab at Sanger for the SATMap project</td>
			</tr>
		</table><br />
<h3 id="drosophila_melanogaster"># Drosophila melanogaster</h3>
			<table class="ss" style="width:60%">
				<tr><th>Source</th><th>Version</th><th>Description</th></tr>
		 
			<tr class="bg1">
				<td>DPGP</td>
				<td>1</td>
				<td>Variation features from DPGP, with Ensembl identifiers</td>
			</tr>
		</table><br />
<h3 id="equus_caballus"># Equus caballus</h3>
			<table class="ss" style="width:60%">
				<tr><th>Source</th><th>Version</th><th>Description</th></tr>
		 
			<tr class="bg1">
				<td><a href="http://www.ncbi.nlm.nih.gov/projects/SNP/">dbSNP</a></td>
				<td>130</td>
				<td>Variants (including SNPs and indels) imported from dbSNP</td>
			</tr>
		</table><br />
<h3 id="felis_catus"># Felis catus</h3>
			<table class="ss" style="width:60%">
				<tr><th>Source</th><th>Version</th><th>Description</th></tr>
		 
			<tr class="bg1">
				<td>dbSNP</td>
				<td>127</td>
				<td>Variants (including SNPs and indels) imported from dbSNP [http://www.ncbi.nlm.nih.gov/projects/SNP/] and mapped to the CAT assembly</td>
			</tr>
		</table><br />
<h3 id="gallus_gallus"># Gallus gallus</h3>
			<table class="ss" style="width:60%">
				<tr><th>Source</th><th>Version</th><th>Description</th></tr>
		 
			<tr class="bg1">
				<td><a href="http://www.ncbi.nlm.nih.gov/projects/SNP/">dbSNP</a></td>
				<td>128</td>
				<td>Variants (including SNPs and indels) imported from dbSNP</td>
			</tr>
		</table><br />
<h3 id="homo_sapiens"># Homo sapiens</h3>
			<table class="ss" style="width:60%">
				<tr><th>Source</th><th>Version</th><th>Description</th></tr>
		 
			<tr class="bg1">
				<td>dbSNP</td>
				<td>132</td>
				<td>Variants (including SNPs and indels) imported from dbSNP [http://www.ncbi.nlm.nih.gov/projects/SNP/]</td>
			</tr>
		
			<tr class="bg2">
				<td><a href="http://www.affymetrix.com/">Affy GeneChip 100K Array</a></td>
				<td>-</td>
				<td>Variants from the Affymetrix GeneChip Human Mapping 100K Array Set</td>
			</tr>
		
			<tr class="bg1">
				<td><a href="http://www.affymetrix.com/">Affy GeneChip 500K Array</a></td>
				<td>-</td>
				<td>Variants from the Affymetrix GeneChip Human Mapping 500K Array Set</td>
			</tr>
		
			<tr class="bg2">
				<td><a href="http://www.affymetrix.com/">Affy GenomeWideSNP_6.0</a></td>
				<td>-</td>
				<td>Variants from the Affymetrix Genome-Wide Human SNP Array 6.0</td>
			</tr>
		
			<tr class="bg1">
				<td><a href="http://www.illumina.com/">Illumina_CytoSNP12v1</a></td>
				<td>-</td>
				<td>Variants from the Illumina Cyto SNP-12 v1 whole genome SNP genotyping chip designed for cytogenetic analysis</td>
			</tr>
		
			<tr class="bg2">
				<td><a href="http://www.illumina.com/">Illumina_Human660W-quad</a></td>
				<td>-</td>
				<td>Variants from the Illumina Human 660W-Quad whole genome SNP genotyping chip designed for association studies</td>
			</tr>
		
			<tr class="bg1">
				<td><a href="http://www.illumina.com/">Illumina_Human1M-duoV3</a></td>
				<td>-</td>
				<td>Variants from the Illumina Human 1M-DuoV3 whole genome SNP genotyping chip designed for association studies</td>
			</tr>
		
			<tr class="bg2">
				<td><a href="http://www.affymetrix.com/">Affy GenomeWideSNP_6.0 CNV</a></td>
				<td>-</td>
				<td>Copy Number Variation (CNV) probes from the Affymetrix Genome-Wide Human SNP Array 6.0</td>
			</tr>
		
			<tr class="bg1">
				<td><a href="http://www.ebi.ac.uk/dgva/">DGVa</a></td>
				<td>05/2011</td>
				<td>Database of Genomic Variants Archive</td>
			</tr>
		
			<tr class="bg2">
				<td>ENSEMBL</td>
				<td>-</td>
				<td>Variants called by Ensembl</td>
			</tr>
		
			<tr class="bg1">
				<td><a href="http://www.genome.gov/gwastudies/">NHGRI_GWAS_catalog</a></td>
				<td>13/05/2011</td>
				<td>Variants associated with phenotype data from the NHGRI GWAS catalog</td>
			</tr>
		
			<tr class="bg2">
				<td><a href="http://www.sanger.ac.uk/genetics/CGP/cosmic/">COSMIC</a></td>
				<td>53</td>
				<td>Somatic mutations found in human cancers from the COSMIC project</td>
			</tr>
		
			<tr class="bg1">
				<td><a href="http://www.ebi.ac.uk/ega/">EGA</a></td>
				<td>05/2011</td>
				<td>Variants imported from the European Genome-phenome Archive with phenotype association</td>
			</tr>
		
			<tr class="bg2">
				<td><a href="http://www.uniprot.org/">Uniprot</a></td>
				<td>05/2011</td>
				<td>Variants annotated by Uniprot</td>
			</tr>
		
			<tr class="bg1">
				<td><a href="http://www.hgmd.cf.ac.uk/ac/index.php">HGMD-PUBLIC</a></td>
				<td>01/2011</td>
				<td>Variants from HGMD-PUBLIC data set January 2011 [remapped from build NCBI36]</td>
			</tr>
		
			<tr class="bg2">
				<td><a href="http://www.ncbi.nlm.nih.gov/omim/">OMIM</a></td>
				<td>05/2011</td>
				<td>Variations linked to entries in the Online Mendelian Inheritance in Man (OMIM) database</td>
			</tr>
		
			<tr class="bg1">
				<td><a href="http://www.biomedcentral.com/1471-2350/10/6">Open Access GWAS Database</a></td>
				<td>-</td>
				<td>Johnson & O'Donnell 'An Open Access Database of Genome-wide Association Results' PMID:19161620</td>
			</tr>
		
			<tr class="bg2">
				<td><a href="https://oi.gene.le.ac.uk/home.php?select_db=LEPRE1">LSDB_LEPRE1</a></td>
				<td>08/04/2011</td>
				<td>LEPRE1 homepage - Osteogenesis Imperfecta Variant Database - Leiden Open Variation Database</td>
			</tr>
		
			<tr class="bg1">
				<td><a href="https://oi.gene.le.ac.uk/home.php?select_db=PPIB">LSDB_PPIB</a></td>
				<td>17/02/2011</td>
				<td>PPIB homepage - Osteogenesis Imperfecta Variant Database - Leiden Open Variation Database</td>
			</tr>
		
			<tr class="bg2">
				<td><a href="https://oi.gene.le.ac.uk/home.php?select_db=CRTAP">LSDB_CRTAP</a></td>
				<td>08/04/2011</td>
				<td>CRTAP homepage - Osteogenesis Imperfecta Variant Database - Leiden Open Variation Database</td>
			</tr>
		
			<tr class="bg1">
				<td><a href="https://oi.gene.le.ac.uk/home.php?select_db=FKBP10">LSDB_FKBP10</a></td>
				<td>16/05/2011</td>
				<td>FKBP10 homepage - Osteogenesis Imperfecta Variant Database - Leiden Open Variation Database</td>
			</tr>
		</table><br />
<h3 id="monodelphis_domestica"># Monodelphis domestica</h3>
			<table class="ss" style="width:60%">
				<tr><th>Source</th><th>Version</th><th>Description</th></tr>
		 
			<tr class="bg1">
				<td>dbSNP</td>
				<td>128</td>
				<td>Variants (including SNPs and indels) imported from dbSNP [http://www.ncbi.nlm.nih.gov/projects/SNP/]</td>
			</tr>
		</table><br />
<h3 id="mus_musculus"># Mus musculus</h3>
			<table class="ss" style="width:60%">
				<tr><th>Source</th><th>Version</th><th>Description</th></tr>
		 
			<tr class="bg1">
				<td>dbSNP</td>
				<td>132</td>
				<td>Variants (including SNPs and indels) imported from dbSNP [http://www.ncbi.nlm.nih.gov/projects/SNP/]</td>
			</tr>
		
			<tr class="bg2">
				<td>ENSEMBL:Sanger</td>
				<td>-</td>
				<td>Variants called by the Ensembl pipeline</td>
			</tr>
		
			<tr class="bg1">
				<td><a href="http://www.ebi.ac.uk/dgva/">DGVa</a></td>
				<td>05/2011</td>
				<td>Database of Genomic Variants Archive</td>
			</tr>
		</table><br />
<h3 id="ornithorhynchus_anatinus"># Ornithorhynchus anatinus</h3>
			<table class="ss" style="width:60%">
				<tr><th>Source</th><th>Version</th><th>Description</th></tr>
		 
			<tr class="bg1">
				<td>ENSEMBL</td>
				<td>-</td>
				<td>Variants called by the ENSEMBL pipeline</td>
			</tr>
		
			<tr class="bg2">
				<td><a href="http://www.ncbi.nlm.nih.gov/projects/SNP/">dbSNP</a></td>
				<td>-</td>
				<td>Variants (including SNPs and indels) imported from dbSNP</td>
			</tr>
		</table><br />
<h3 id="pan_troglodytes"># Pan troglodytes</h3>
			<table class="ss" style="width:60%">
				<tr><th>Source</th><th>Version</th><th>Description</th></tr>
		 
			<tr class="bg1">
				<td><a href="http://www.ncbi.nlm.nih.gov/projects/SNP/">dbSNP</a></td>
				<td>125</td>
				<td>Variants (including SNPs and indels) imported from dbSNP</td>
			</tr>
		</table><br />
<h3 id="pongo_abelii"># Pongo abelii</h3>
			<table class="ss" style="width:60%">
				<tr><th>Source</th><th>Version</th><th>Description</th></tr>
		 
			<tr class="bg1">
				<td>ENSEMBL</td>
				<td>-</td>
				<td>Variants called by the ENSEMBL pipeline</td>
			</tr>
		</table><br />
<h3 id="rattus_norvegicus"># Rattus norvegicus</h3>
			<table class="ss" style="width:60%">
				<tr><th>Source</th><th>Version</th><th>Description</th></tr>
		 
			<tr class="bg1">
				<td><a href="http://www.ncbi.nlm.nih.gov/projects/SNP/">dbSNP</a></td>
				<td>130</td>
				<td>Variants (including SNPs and indels) imported from dbSNP</td>
			</tr>
		
			<tr class="bg2">
				<td>ENSEMBL:celera</td>
				<td>-</td>
				<td>Variants from Celera called by the ENSEMBL pipeline</td>
			</tr>
		
			<tr class="bg1">
				<td>ENSEMBL:STAR-4strain</td>
				<td>-</td>
				<td>Variants in STAR-4strain called by the ENSEMBL pipeline</td>
			</tr>
		
			<tr class="bg2">
				<td>ENSEMBL:STAR-gtype</td>
				<td>-</td>
				<td>Variants in STAR-gtype called by the ENSEMBL pipeline</td>
			</tr>
		
			<tr class="bg1">
				<td>ENSEMBL:RNB1</td>
				<td>-</td>
				<td>Variants in RNB1 called by the ENSEMBL pipeline</td>
			</tr>
		</table><br />
<h3 id="saccharomyces_cerevisiae"># Saccharomyces cerevisiae</h3>
			<table class="ss" style="width:60%">
				<tr><th>Source</th><th>Version</th><th>Description</th></tr>
		 
			<tr class="bg1">
				<td>SGRP</td>
				<td>-</td>
				<td>Variation features from SGRP, with Ensembl identifiers</td>
			</tr>
		</table><br />
<h3 id="sus_scrofa"># Sus scrofa</h3>
			<table class="ss" style="width:60%">
				<tr><th>Source</th><th>Version</th><th>Description</th></tr>
		 
			<tr class="bg1">
				<td><a href="http://www.ncbi.nlm.nih.gov/projects/SNP/">dbSNP</a></td>
				<td>128</td>
				<td>Variants (including SNPs and indels) imported from dbSNP [remapped from NCBI Sus scrofa build 1.1]</td>
			</tr>
		
			<tr class="bg2">
				<td><a href="http://www.illumina.com/products/porcine_snp60_whole_genome_genotyping_kits.ilmn">Illumina_PorcineSNP60</a></td>
				<td>60</td>
				<td>PorcineSNP60 BeadChip</td>
			</tr>
		
			<tr class="bg1">
				<td><a href="http://www.ebi.ac.uk/dgva/">DGVa</a></td>
				<td>05/2011</td>
				<td>Database of Genomic Variants Archive</td>
			</tr>
		</table><br />
<h3 id="taeniopygia_guttata"># Taeniopygia guttata</h3>
			<table class="ss" style="width:60%">
				<tr><th>Source</th><th>Version</th><th>Description</th></tr>
		 
			<tr class="bg1">
				<td>dbSNP</td>
				<td>131</td>
				<td>Variants (including SNPs and indels) imported from dbSNP [http://www.ncbi.nlm.nih.gov/projects/SNP/] and mapped to the Taeniopygia_guttata-3.2.4 assembly</td>
			</tr>
		</table><br />
<h3 id="tetraodon_nigroviridis"># Tetraodon nigroviridis</h3>
			<table class="ss" style="width:60%">
				<tr><th>Source</th><th>Version</th><th>Description</th></tr>
		 
			<tr class="bg1">
				<td>dbSNP</td>
				<td>132</td>
				<td>Variants (including SNPs and indels) imported from dbSNP [http://www.ncbi.nlm.nih.gov/projects/SNP/] and mapped to the TETRAODON 8.0 assembly</td>
			</tr>
		</table>

</body>
</html>
