<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="CONTENT-TYPE" CONTENT="text/html; charset=utf-8">
	<TITLE>e! Variation schema </TITLE>
</HEAD>
<BODY>
<H1>Ensembl Variation Schema Documentation</H1>
<H2>Introduction</H2>
<P>This document gives a high-level description of the tables that
make up the Ensembl variation schema. Tables are grouped into logical
groups, and the purpose of each table is explained. It is intended to
allow people to familiarise themselves with the schema when
encountering it for the first time, or when they need to use some
tables that they've not used before. Note that this document
makes no attempt to enumerate all of the names, types and contents of
every single table.</P>
<P>This document refers to version <STRONG>55</STRONG> of the Ensembl
variation schema. 
</P>
<P>A PDF document of the schema is available <A HREF="variation-database-schema.pdf">here</A>.</P>
<HR>
<P><B>Samples and populations</B></P>
<UL>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#individual">individual</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#individual_population">individual_population</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#individual_type">individual_type</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#sample">sample</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#sample_synonym">sample_synonym</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#population">population</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#population_structure">population_structure</A></P>
</UL>
<P STYLE="margin-bottom: 0cm"><BR></P>
<P><B>Simple variations</B></P>
<UL>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#source">source </A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#variation">variation</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#variation_synonym">variation_synonym</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#flanking_sequence">flanking_sequence</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#failed_variation">failed_variation</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#failed_description">failed_description</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#allele">allele</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#variation_feature">variation_feature</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#transcript_variation">transcript_variation</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#tagged_variation_feature">tagged_variation_feature</A></P>
</UL>
<P STYLE="margin-bottom: 0cm"><BR></P>
<P><B>Multi-marker variations</B></P>
<UL>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#allele_group">allele_group</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#allele_group_allele">allele_group_allele</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#variation_group">variation_group</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#variation_group_feature">variation_group_feature</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#variation_group_variation">variation_group_variation</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#httag">httag</A></P>
</UL>
<P STYLE="margin-bottom: 0cm"><BR></P>
<P><B>Genotypes, read coverage and annotations</B></P>
<UL>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#compressed_genotype_single_bp">compressed_genotype_single_bp</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#individual_genotype_multiple_bp">individual_genotype_multiple_bp</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#population_genotype">population_genotype</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#read_coverage">read_coverage</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#variation_annotation">variation_annotation</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#phenotype">phenotype</A></P>
</UL>
<P STYLE="margin-bottom: 0cm"><BR></P>
<P><B>Miscellaneous</B></P>
<UL>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#meta">meta</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#meta_coord">meta_coord</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#seq_region">seq_region</A></P>
</UL>
<HR>
<H2>Samples and populations</H2>
<H3><A NAME="individual"/>individual</H3>
<P>Stores information about an identifiable individual, including
gender and the identifiers of the individual's parents (if known).</P>
<P><B>See also:</B></P>
<P>Tables:</P>
<UL>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#sample">sample</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#individual_type">individual_type</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#individual_population">individual_population</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#individual_genotype_multiple_bp">individual_genotype_multiple_bp</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#compressed_genotype_single_bp">compressed_genotype_single_bp</A></P>
</UL>
<H3><A NAME="individual_population"/>individual_population</H3>
<P>This table resolves the many-to-many relationship between the
individual and population tables; i.e. samples may belong to more
than one population. Hence it is composed of rows of individual and
population identifiers.</P>
<P><B>See also:</B></P>
<P>Tables:</P>
<UL>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#individual">individual</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#population">population</A></P>
</UL>
<H3><A NAME="individual_type"/>individual_type</H3>
<P>This table contains a set of common individual types and their
descriptions. These types are currently: “fully_inbred”,
“partly_inbred”, “outbred” and “mutant”.</P>
<P><B>See also:</B></P>
<P>Tables:</P>
<UL>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#individual">individual</A></P>
</UL>
<H3><A NAME="sample"/>sample</H3>
<P>Sample is used as a generic catch-all term to cover
individuals, populations and strains; it contains a name and description, as
well as a size if applicable to the population.</P>
<P><B>See also:</B></P>
<P>Tables:</P>
<UL>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#individual">individual</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#population">population</A></P>
</UL>
<H3><A NAME="sample_synonym"/>sample_synonym</H3>
<P>Used to store alternative names for populations when data comes
from multiple sources.</P>
<P><B>See also:</B></P>
<P>Tables:</P>
<UL>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#sample">sample</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#population">population</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#source">source</A></P>
</UL>
<H3><A NAME="population"/>population</H3>
<P>A table consisting simply of sample_ids representing populations;
all data relating to the populations are stored in separate tables
(see below).</P>
<P><B>See also:</B></P>
<P>Tables:</P>
<UL>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#sample">sample</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#sample_synonym">sample_synonym</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#individual_population">individual_population</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#population_structure">population_structure</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#population_genotype">population_genotype</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#allele">allele</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#allele_group">allele_group</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#tagged_variation_feature">tagged_variation_feature</A></P>
</UL>
<H3><A NAME="population_structure"/>population_structure</H3>
<P>This table stores hierarchical relationships between populations
by relating them as populations and sub-populations.</P>
<P><B>See also:</B></P>
<P>Tables:</P>
<UL>
	<LI><P><A HREF="#population">population</A></P>
</UL>
<HR>
<H2>Simple variations</H2>
<H3><A NAME="source"/>source</H3>
<P>This table contains details of the source from which a variation
is derived. Most commonly this is NCBI's dbSNP; other sources include
SNPs called by Ensembl.</P>
<P><B>See also:</B></P>
<P>Tables:</P>
<UL>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#variation">variation</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#variation_synonym">variation_synonym</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#variation_feature">variation_feature</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#variation_annotation">variation_annotation</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#variation_group">variation_group</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#allele_group">allele_group</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#sample_synonym">sample_synonym</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#httag">httag</A></P>
</UL>
<H3><A NAME="variation"/>variation</H3>
<P>This is the schema's generic representation of a variation,
defined as a genetic feature that varies between individuals of the
same species. The most common type is the single nucleotide variation
(SNP) though the schema also accommodates copy number variations
(CNVs) and structural variations (SVs). A variation is defined by its
flanking sequence rather than its mapped location on a chromosome; a
variation may in fact have multiple mappings across a genome. This
table stores a variation's name (commonly an ID of the form e.g.
rs123456, assigned by dbSNP), along with a validation status and
ancestral (or reference) allele.</P>
<P><B>See also:</B></P>
<P>Tables:</P>
<UL>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#variation_synonym">variation_synonym</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#flanking_sequence">flanking_sequence</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#failed_variation">failed_variation</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#variation_feature">variation_feature</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#variation_group_variation">variation_group_variation</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#allele">allele</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#allele_group_allele">allele_group_allele</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#individual_genotype_multiple_bp">individual_genotype_multiple_bp</A></P>
</UL>
<H3><A NAME="variation_synonym"/>variation_synonym</H3>
<P>This table allows for a variation to have multiple IDs, generally
given by multiple sources.</P>
<P><B>See also:</B></P>
<P>Tables:</P>
<UL>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#source">source</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#variation">variation</A></P>
</UL>
<H3><A NAME="flanking_sequence"/>flanking_sequence</H3>
<P>This table contains the upstream and downstream sequence
surrounding a variation. Since each variation is defined by its
flanking sequence, this table has a one-to-one relationship with the
variation table.</P>
<P><B>See also:</B></P>
<P>Tables:</P>
<UL>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#seq_region">seq_region</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#variation">variation</A></P>
</UL>
<H3><A NAME="failed_variation"/>failed_variation</H3>
<P>For various reasons it may be necessary to store information about
a variation that has failed quality checks in the Variation pipeline.
This table acts as a flag for such failures.</P>
<P><B>See also:</B></P>
<P>Tables:</P>
<UL>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#failed_description">failed_description</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#variation">variation</A></P>
</UL>
<H3><A NAME="failed_description"/>failed_description</H3>
<P>This table contains descriptions of reasons for a variation being
flagged as failed.</P>
<P><B>See also:</B></P>
<P>Tables:</P>
<UL>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#failed_description">failed_description</A></P>
</UL>
<H3><A NAME="allele"/>allele</H3>
<P>This table stores information about each of a variation's alleles,
along with population frequencies.</P>
<P><B>See also:</B></P>
<P>Tables:</P>
<UL>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#variation">variation</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#population">population</A></P>
</UL>
<H3><A NAME="variation_feature"/>variation_feature</H3>
<P>This table represents mappings of variations to genomic locations.
It stores an allele string representing the different possible
alleles that are found at that locus e.g. “A/T” for a SNP, as
well as a “worst case” consequence of the mutation. It also acts
as part of the relationship between variations and transcripts.</P>
<P><B>See also:</B></P>
<P>Tables:</P>
<UL>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#variation">variation</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#tagged_variation_feature">tagged_variation_feature</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#transcript_variation">transcript_variation</A></P>
</UL>
<H3><A NAME="transcript_variation"/>transcript_variation</H3>
<P>This table relates a variation_feature to a transcript (see Core
documentation). It contains the consequence of the variation e.g.
intronic, non-synonymous coding, stop etc, along with the change in
amino acid in the resulting protein if applicable.</P>
<P><B>See also:</B></P>
<P>Tables:</P>
<UL>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#variation_feature">variation_feature</A></P>
</UL>
<H3><A NAME="tagged_variation_feature"/>tagged_variation_feature</H3>
<P>This table lists variation features that are tagged by another
variation feature. Tag pairs are defined as having an r&sup2;
&gt; 0.99.</P>
<P><B>See also:</B></P>
<P>Tables:</P>
<UL>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#variation_feature">variation_feature</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#population">population</A></P>
</UL>
<HR>
<H2>Multi-marker variations</H2>
<P>These tables are designed to represent alleles and haplotypes that
consist of more than one variation.</P>
<H3><A NAME="allele_group"/>allele_group</H3>
<P>This table, along with allele_group_allele, represents a
particular multi-marker allele of a given multi-marker variation, or
haplotype. It stores an associated population frequency.</P>
<P><B>See also:</B></P>
<P>Tables:</P>
<UL>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#allele_group_allele">allele_group_allele</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#variation_group">variation_group</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#population">population</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#source">source</A></P>
</UL>
<H3><A NAME="allele_group_allele"/>allele_group_allele</H3>
<P>This table represents an allele of one variation in a multi-marker
variation, or haplotype. It stores a string of the allele.</P>
<P><B>See also:</B></P>
<P>Tables:</P>
<UL>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#allele_group">allele_group</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#variation">variation</A></P>
</UL>
<H3><A NAME="variation_group"/>variation_group</H3>
<P>This table represents the equivalent of a variation for a
multi-marker variation.</P>
<P><B>See also:</B></P>
<P>Tables:</P>
<UL>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#variation_group_variation">variation_group_variation</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#allele_group">allele_group</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#variation">variation</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#source">source</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#httag">httag</A></P>
</UL>
<H3><A NAME="variation_group_variation"/>variation_group_variation</H3>
<P>This table represents an individual variation that makes up a
multi-marker variation, and resolves the many-to-many relationship
between variation and variation_group.</P>
<P><B>See also:</B></P>
<P>Tables:</P>
<UL>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#variation_group">variation_group</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#variation">variation</A></P>
</UL>
<H3><A NAME="variation_group_feature"/>variation_group_feature</H3>
<P>This table represents the equivalent of a variation_feature for
multi-marker variations, mapping a haplotype to a chromosomal
coordinate system.</P>
<P><B>See also:</B></P>
<P>Tables:</P>
<UL>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#variation_group">variation_group</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#seq_region">seq_region</A></P>
</UL>
<H3><A NAME="httag"/>httag</H3>
<P>This table represents the equivalent of a tagged_variation_feature
for multi-marker variations, representing an instance where a
haplotype is tagged by or tags another marker.</P>
<P><B>See also:</B></P>
<P>Tables:</P>
<UL>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#variation_group">variation_group</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#source">source</A></P>
</UL>
<HR>
<H2>Genotypes, read coverage and annotations</H2>
<H3><A NAME="compressed_genotype_single_bp"/>compressed_genotype_single_bp</H3>
<P>This table holds genotypes compressed using the pack() method in
Perl. These genotypes are mapped to particular genomic locations
rather than variation objects. The data have been compressed to reduce table
size and increase the speed of the web code.</P>
<P><B>See also:</B></P>
<P>Tables:</P>
<UL>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#individual">individual</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#seq_region">seq_region</A></P>
</UL>
<H3><A NAME="individual_genotype_multiple_bp"/>individual_genotype_multiple_bp</H3>
<P>This table holds uncompressed genotypes for given variations.</P>
<P><B>See also:</B></P>
<P>Tables:</P>
<UL>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#individual">individual</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#variation">variation</A></P>
</UL>
<H3><A NAME="population_genotype"/>population_genotype</H3>
<P>This table stores alleles and frequencies for variations in
given populations.</P>
<P><B>See also:</B></P>
<P>Tables:</P>
<UL>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#population">population</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#variation">variation</A></P>
</UL>
<H3><A NAME="read_coverage"/>read_coverage</H3>
<P>This table stores the read coverage in the resequencing of
individuals. Each row contains an individual ID, chromosomal
coordinates and a read coverage level.</P>
<P><B>See also:</B></P>
<P>Tables:</P>
<UL>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#individual">individual</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#seq_region">seq_region</A></P>
</UL>
<H3><A NAME="variation_annotation"/>variation_annotation</H3>
<P>This table stores information linking genotypes and phenotypes. It
stores various fields pertaining to the study conducted, along with
the associated gene, risk allele frequency and a p-value.</P>
<P><B>See also:</B></P>
<P>Tables:</P>
<UL>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#variation">variation</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#phenotype">phenotype</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#source">source</A></P>
</UL>
<H3><A NAME="phenotype"/>phenotype</H3>
<P>This table stores details of the phenotypes associated with
variation annotations.</P>
<P><B>See also:</B></P>
<P>Tables:</P>
<UL>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#variation_annotation">variation_annotation</A></P>
</UL>
<HR>
<H2>Miscellaneous</H2>
<P>Tables that don't fit anywhere else.</P>
<H3><A NAME="seq_region"/>seq_region</H3>
<P>This table stores the relationship between Ensembl's internal
coordinate system identifiers and traditional chromosome names.</P>
<P><B>See also:</B></P>
<P>Tables:</P>
<UL>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#variation_feature">variation_feature</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#variation_group_feature">variation_group_feature</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#flanking_sequence">flanking_sequence</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#compressed_genotype_single_bp">compressed_genotype_single_bp</A></P>
	<LI><P STYLE="margin-bottom: 0cm"><A HREF="#read_coverage">read_coverage</A></P>
</UL>
<H3><A NAME="seq_region_attrib"/>meta</H3>
<P>This table stores various metadata relating to the database,
generally used by the Ensembl web code.</P>
<H3><A NAME="meta_coord"/>meta_coord</H3>
<P>This table gives the coordinate system used by various tables in
the database.</P>
<HR>
</BODY>
</HTML>
