<h2>About the Chicken genome</h2>

<h3>Assembly</h3>

<p><img src="/img/species/pic_Gallus_gallus.png" height="95" width="100" class="float-left" alt="Chicken" title="B...ook-buk-buk!" />

This site displays the 2.1 (May 2006) release of the <a href="http://genome.wustl.edu/genome.cgi?GENOME=Gallus%20gallus">chicken genome</a>. The analysis of the chicken sequence involves an international group of scientists including individuals from the US, UK, Europe and China, headed by the Genome Sequencing Center at Washington University School of Medicine, St. Louis.</p>
<ul><li><a href="credits.html">Full list of credits</a></li></ul>

<p>Of the 1.05 Gb genome, approximately 95% of the sequence has been anchored to chromosomes, which include autosomes 1-28 and 32, two additional linkage groups, and sex chromosomes W and Z. (In contrast to mammals, the female chicken is heterogametic (ZW) and the male is homogametic (ZZ).) The remaining unanchored contigs have been concatenated into the virtual chromosome "chrUn", separated by gaps of 100 bp. All unknown gap sizes have been set to 100 bp. The N50 ultracontig size is 15.5Mb (n=19); the longest ultracontig is 80.3Mb on chicken chromosome 3. The N50 supercontig size is 11.1Mb (n=26); the longest supercontig is 52Mb on chr2. The N50 contig size is 45kb (n=5863); the longest contig is 625kb. Q20 base redundancy: 7.1X</p>


<h3>Annotation</h3>

<p>The gene set for Chicken was built using a modified version of the 
standard Ensembl genebuild pipeline. Gene models are based on genewise 
alignments of chicken proteins as well as genetically distant proteins 
from other species. To improve the accuracy of models generated from 
distant species, the Genewise alignments were made to stretches of 
genomic sequence rather than to 'miniseqs'. The protein based gene 
models were then extended using Chicken cDNA and EST alignments, the 
cDNAs were added to models made from chicken proteins and the ESTs to 
models based on proteins from other species. The resulting gene models 
were then assessed by generating sets of potential orthologs to genes 
from other vertebrate species. Potentially missing predictions and 
partial gene predictions were identified by examining the orthologs, 
which were then used to build new gene models.</p>
