<html>
<head>
<meta name="order" content="99" />
<title>Projects using Ensembl</title>
</head>

<body>

<h1>Projects using Ensembl</h1>


<p>
The Ensembl project is both a source of genome sequence related data
(<a href="http://www.ensembl.org/">http://www.ensembl.org</a>) and an open source 
software system that can be used to organise any such data.
</p>

<h2>User-contributed code</h2>

<p>Whilst we have developed a comprehensive Perl API in-house, we welcome contributions
in other programming languages from the community.</p>

<h3>Ruby</h3>

<p>A Ruby API has been developed by Jan Aerts. A small example, showing what it can do 
(including coordinate transformation and reflecting on the types of associations for a 
given class) can be found on the wiki page.</p>

<ul>
<li><a href="http://github.com/jandot/ruby-ensembl-api">repository</a></li>
<li><a href="http://wiki.github.com/jandot/ruby-ensembl-api">wiki</a></li>
</ul>

<p>Note that we do not support these user-contributed packages - please contact
the original developer if you have any questions or comments.</p>

<h2>Ensembl-powered Websites</h2>

<p>
Some projects make use of the Ensembl software system, or are
derived from it.
</p>

<p class="center">
<img src="/img/empowered.png" alt="'Powered by Ensembl' logo" width="98" height="20" />
</p>

<table class="ss tint">
<tr>
    <th>Project</th>
    <th>Description</th>
    <th>Code used</th>
</tr>

<tr>
    <td><strong><a href="http://vega.sanger.ac.uk/">Vega</a></strong></td>
    <td>Website to present curated vertebrate annotation</td>
    <td>Web front-end derived from Ensembl webcode;<br />Ensembl database and Perl API</td>
</tr>

<tr class="bg2">
    <td><strong><a href="http://www.vectorbase.org/">VectorBase</a></strong></td>
    <td>Bioinformatic Resource Centre for Invertebrate Vectors of Human Pathogens</td>
    <td>Customised Ensembl webcode, Ensembl Compara pipeline, Ensembl annotation pipeline</td>
</tr>

<tr>
    <td><strong><!-- DNS error<a href="http://decifr.ccgb.umn.edu/">-->DeCIFR legume project</strong></td>
    <td>Automated annotation and display of legume genomes</td>
    <td>Customised Ensembl pipeline, API, and webcode</td>
</tr>

<tr class="bg2">
    <td><strong><a href="http://atensembl.arabidopsis.info/">AtEnsembl</a></strong></td>
    <td><i>Arabidopsis thaliana</i> genome browser</td>
    <td>Customised Ensembl pipeline, API, and webcode</td>
</tr>

<tr>
    <td><strong>Otter</strong></td>
    <td>Database backend for interactive curation of annotation</td>
    <td>Otter is an extension of the Ensembl database schema</td>
</tr>

<tr class="bg2">
    <td><strong><a href="http://www.gramene.org/">Gramene</a></strong></td>
    <td>A Comparative Mapping Resource for Grains</td>
    <td>Customised webcode, Ensembl Compara pipeline,  Ensembl database used for annotation, import of the Arabidopsis Ensembl database from NASC.</td>
</tr>

<tr>
    <td><strong><a href="http://www.cadre-genomes.org.uk">CADRE</a></strong></td>
    <td>Central Aspergillus Data REpository</td>
    <td></td>
</tr>

<tr class="bg2">
    <td><strong><a href="http://public-contigbrowser.sigenae.org:9090/">Sigenae</a></strong></td>
    <td>Sigenae EST ContigBrowser</td>
    <td></td>
</tr>

<tr>
    <td><strong><a href="http://www.sanger.ac.uk/genetics/CGP/cosmic/">COSMIC (Catalogue of Somatic Mutations in Cancer)</a></strong></td>
    <td>Web display of somatic sequence variant/mutation data</td>
    <td>Customised Ensembl drawing code</td>
</tr>

<tr class="bg2">
    <td><strong><a href="http://www.sanger.ac.uk/perl/MVP/mvp/">MVP (Methylation Variable Positions)</a></strong></td>
    <td>Web display of MVPs analysed in the Human Epigenome Project.</td>
    <td>Customised Ensembl drawing code</td>
</tr>

<tr>
    <td><strong><a href="http://www.sanger.ac.uk/perl/SPGE/geexview/"><i>S. pombe</i> Gene Expression Viewer</a></strong></td>
    <td>Web display of <i>S. pombe</i> gene expression under various environmental conditions.</td>
    <td>Customised Ensembl drawing code</td>
</tr>

<tr class="bg2">
    <td><strong><a href="http://pupasuite.bioinfo.cipf.es/">PupaSuite</a></strong></td>
    <td>Interactive web tool for the selection of optimal sets of SNP for 
        genotyping purposes, with prediction of functional effect and visualisation facilities.</td>
    <td>Web front end to a CGI that uses APIs.</td>
</tr>

<tr>
    <td><strong><a href="http://isacgh.bioinfo.cipf.es/">ISACGH</a></strong></td>
    <td>Web tool to detect chromosomal copy number alterations and relate them 
        to gene expression and gene function.</td>
    <td>Web front end to a CGI that uses APIs. DAS server.</td>
</tr>

<tr class="bg2">
    <td><strong><a href="http://www.cmbi.ru.nl/phylopat/">Phylopat</a></strong></td>
    <td>Web tool to show the presence or absence of certain genes or proteins in a set of 
      eukaryote species. 
        </td>
    <td>Uses orthologue data extracted from EnsMart.</td>
</tr>

<tr>
    <td><strong><!-- 500 error<a href="http://www.chromhome.org/">-->Chromhome<!--</a>--></strong></td>
    <td>A database of chromosome map data, particularly chromosome painting data 
      between species.</td>
    <td>Ensembl karyotype drawing code.</td>
</tr>

</table>

<p class="center">
<img src="/img/empowered.png" alt="'Powered by Ensembl' logo" width="98" height="20" />
</p>

<p>If you are using Ensembl code in your project, you might like to <a href="/img/empowered_lg.png">download the 'empowered' logo</a> (high-resolution PNG, suitable for use in print).</p>

</body>
</html>

