<html>
  <head>
    <title>Configuring your species</title>
<meta name="order" content="2" />
  </head>
<body>
  <h1>Configuring your Ensembl species</h1>

<p>Most of the time you will want to run your Ensembl mirror site with the same set
of species as the equivalent public site. In this situation, you just need to set up
all the required databases and configure the relevant plugin, e.g. 
<kbd>public-plugins/ensembl</kbd> for the main Ensembl website (www.ensembl.org).</p> 

<p>However if you wish to show only a subset of species, or perhaps add one of your
own, you will need to do some extra manual configuration.</p>

<h2>Removing species</h2>

<p>It is obviously simpler to omit species from your Ensembl mirror 
than to add new ones, so we'll look at this first.</p> 

If you only want to omit a few species, simply add the
required <kbd>delete</kbd> lines to the <kbd>update_conf</kbd> method in
<kbd>/public-plugins/mirror/conf/SiteDefs.pm</kbd>:</p>

<pre class="code">
delete($SiteDefs::__species_aliases{'Mus_musculus'});
delete($SiteDefs::__species_aliases{'Rattus_norvegicus'});
</pre>

<p>If on the other hand you want to omit most species and just use one or two of 
the Ensembl species, it is more efficient to omit the entire species-related plugin
(e.g. <kbd>public-plugins/ensembl</kbd>) and copy the relevant bits of code into
your own plugin as if you were setting up new species (see below).</p> 

<p><strong>Important note</strong>: the comparative genomics code in particular
relies on the default primary and secondary comparison species being configured 
in SiteDefs.pm (e.g. Human and Mouse for Ensembl Vertebrates). If you
omit either of these species, you need to set others in their place or some
pages will not work. E.g.</p>

<pre class="code">
$SiteDefs::ENSEMBL_SECONDARY_SPECIES  = 'Canis_familiaris'; 
</pre>

<h2>Adding new species</h2>

<p>To add a species, you obviously need a complete Ensembl database
in the same schema as the rest of your site, which is outside the 
scope of this tutorial. However assuming you have the data already,
the following changes will add it to the website (substituting the
details of your own species for the ones in the examples, obviously!).</p>

<h3>SiteDefs.pm</h3>

<p>Firstly, add it to the species aliases in the <kbd>update_conf</kbd> 
method in <kbd>/public-plugins/mirror/conf/SiteDefs.pm</kbd>:</p>

<pre class="code">
$SiteDefs::__species_aliases{'Canis_latrans'}           = [qw(cl coyote)];
$SiteDefs::__species_aliases{'Geococcyx_californianus'} = [qw(gc roadrunner)];
</pre>

<p>Note that all aliases should be unique, so check in 
<kbd>/public-plugins/ensembl/conf/SiteDefs.pm</kbd> to see which 
are already in use. These aliases allow quick access to pages, so that users
can type, e.g. www.ensembl.org/worm/Transcript?t=ZK792.7 instead of
www.ensembl.org/Caenorhabditis_elegans/Transcript?t=ZK792.7</p>

<h3 id="ini-file">Species ini files</h3>
<p>
    The species configuration (database names, etc.) is
     kept in the conf/ini_files directory as an "ini" file. 
     There is a generic file called DEFAULTS.ini which holds default
     values for all the species.   (This is simply to save having
     to make the same setting in multiple species ini files.)
     There needs to be a species-specific .ini file for each
     species you want to display on your Ensembl website 
     (e.g. Homo_sapiens.ini).
     You can override settings in DEFAULTS.ini by redefining the setting
      in a particular species' ini file.
     The MULTI.ini contains multi-species settings. It has
     database connection information for the Compara and
     Mart databases.
</p>
<p>To create a new species-specific ini file, copy, rename and modify the template file in <code>public-plugins/mirror/conf/ini-files/Genus_species.ini</code>.</p>
<ul>
<li>Under [databases] list all the ones configured for this species.  <br />
You can use "%" instead of the release number if it is idential to the &lt;ENSEMBL_VERSION&gt; (configured in <kbd>conf/SiteDefs.pm</kbd>). This is equivalent to the schema version for the main ensembl site. <br />
</li>
<li> If you want to configure additional data you can use an existing ini file from the <code>public-plugins/ensembl/conf/ini-files/</code> directory as a guide.
</li>
</ul>

<p>If your species has chromosomes, make sure you have the correct names
in the array ENSEMBL_CHROMOSOMES, as this is used to draw karyotypes.</p>

<p>If your new species lies in a taxonomic group outside the standard Ensembl
eukaryote tree, or if you simply wish to alter the groupings in the dropdown
species list on the home page, see the instructions on 
<a href="specieslist.html">customising the species list on the home page</a>.</p>

<h3>Static content</h3>

<p><img src="/img/info/Canis_latrans.png" style="float:left;width:64px;height:64px;margin-right:10px" alt="Canis latrans" />
Species home pages are now dynamically generated, so only a minimum of maintenance
is required. Much of the content is now generated from the databases and ini files.</p>

<p>By default, the species homepage links to a page called 'more information and statistics',
which includes some optional text about the species. Examples of the kind of text used
can be found in public-plugins/ensembl/htdocs/ssi/species.</p> 

<p>Finally, if your new species is a eukaryote you may want to display a thumbnail image
on the home page and in search menus, etc. If so, you will need a square image in the
following sizes and file locations (inside your plugin):</p>
<ul>
<li>htdocs/i/species/16/ - 16 x 16 pixels</li> 
<li>htdocs/i/species/48/ - 48 x 48 pixels</li> 
<li>htdocs/i/species/64/ - 64 x 64 pixels</li> 
</ul>

<p>Each file should be named after the species, e.g. Canis_latrans.png</p>


</body>
</html>
