<h3>Assembly</h3>

<p class="space-below"><img src="/img/species/pic_Anolis_carolinensis.png" style="height:95px;width:125px" alt="green anole lizard" class="float-left" title="Bob, bob!" />

The AnoCar1.0 assembly is a Whole Genome Shotgun (WGS) assembly at
6.3x. The Green anole lizard, <i>Anolis carolinensis</i>, is the first
reptilian genome to be sequenced. The project coordination, genome
sequencing and assembly is provided by the 
<a href="http://www.broad.mit.edu/">Broad Institute</a>.</p>

<p>The N50 size is the length such that 50% of the assembled genome lies
in blocks of the N50 size or longer. The N50 size of the contigs is
83.2 kb and the N50 of the scaffolds is 2399.2 kb. The total length of
all contigs is 1.74 Gb. When the gaps between contigs in scaffolds are
included, the total span of the assembly is 1.77 Gb.</p>

<h3>Annotation</h3>
<p class="space-below">The current gene set for anole was built using a 
modified version of the standard Ensembl genebuild pipeline. The 
species-specific sequence resources (cDNAs and proteins) for anole are 
very limited, hence the majority of the gene models are based on genewise 
alignments of proteins from other species, mainly mammals and birds.</p>

<p class="space-below">We used the top three levels of protein existence 
criteria in <a href="http://www.uniprot.org/">Uniprot</a>
to filter out all predicted and uncertain proteins. Chicken proteins 
annotated by Ensembl were also aligned to the genome. To improve the 
accuracy of models generated from these proteins, the genewise alignments 
were made to stretches of genomic sequence rather than to 'miniseqs'. 
Additional genes were predicted using anole EST evidence of which a subset 
was included in the final gene set. ESTs were also used to add UTRs to the 
gene models where possible.</p>

<p>Using annotations from the Oxford Functional Genomics group, we have
now refined the existing gene models and included additional genes to
the set.</p>
