<h3>Assembly</h3>

<p><img src="/img/species/pic_Pan_troglodytes.png" alt="chimpanzee" title="Oo ooo!" class="float-right" style="width:100px;height:126px" />
This site provides a data set based on the March 
2006 Pan_troglodytes-2.1 6x shotgun assembly from the Chimpanzee Sequencing 
Consortium headed by the <a href="http://www.genome.wustl.edu/">GSC</a> (St. Louis) 
and <a href="http://www.broad.mit.edu/">The Broad Institute</a> (MIT).</p>  
   
<p>The chimpanzee 2.1 assembly is a merge of the initial  4X made in 
collaboration with the Broad Institute at MIT and Harvard and an 
additional (2X) whole genome coverage from the WUGSC (St. Louis) 
utilizing a combination of whole genome plasmid reads as well as fosmid 
and BAC end sequences.</p>
 
<p>This release of the assembly has the following properties:</p>
<ul>
  <li>246876 contigs, having N50 length 30.8 Kb</li>
	<li>contig length total 2.92 Gb</li>
	<li>chromosome length total 3.35 Gb</li>
</ul>

<h3>Annotation</h3>

<p class="space-below">The genome was aligned to human NCBI36 by UCSC using BLASTz. These 
alignments were used to transfer human ensembl gene structures (Human 
Build 36f) to chimpanzee. 92% of the chimp-specific proteins were aligned 
to the chimp genome in a first layer of annotation. The 8% missing correspond 
to fragments or proteins that contain stop codons in the assembled genome</p>

<p class="space-below">More than 2000 chimp-specific protein sequences were used during the
gene build process, and were aligned using a combination of
Genewise and Exonerate. Owing to the small number of proteins (many of
which aligned in the same location) an additional layer of gene structures
was added by projection of human genes. The high-quality
annotation of the human genome and the high degree of similarity between
the human and chimpanzee genomes enables us to identify genes in chimpanzee
by transfer of human genes to the corresponding location in chimp.</p>

<p class="space-below">The protein-coding transcripts of the human gene structures are
projected through the WGA onto the chromosomes in the chimp genome.
Small insertions/deletions that disrupt the reading-frame of the
resultant transcripts are corrected for by inserting "frame-shift"
introns into the structure.</p>

<p class="space-below">For some human exons and parts of exons, the corresponding chimp
sequence is missing from the assembly. In most of these cases, the
missing exon is omitted from the chimpanzee gene model. In a small
number of cases however, where BLASTZ has aligned the human sequence to
a gap in the chimp sequence, the exon is placed in the gap, resulting on
a run of X's of the correct length in the translation.</p>

<p>Some human transcripts fail to transfer cleanly (due to, for example,
missing alignment in the othologous regions). We have attempted to
recover these using Exonerate. The single best exonerate alignment to
chimp is chosen for each "missing" human transcript, and transcripts
with less that 50% identity to the source or 50% coverage of the source
are discarded.</p>

