
  <h3>About the <i>Tupaia belangeri</i> genome</h3>

<h4>Assembly</h4>

<p><img src="/img/species/pic_Tupaia_belangeri.png" style="height:83px;width:120px" class="float-left" alt="Treeshrew" title="rustle rustle" />
This is the first release of the low-coverage 2X assembly of the northern
treeshrew(<i>Tupaia belangeri</i>). The genome sequencing
and assembly is provided by the 
<a href="http://www.broad.mit.edu/">Broad Institute</a>.
</p>

<p>The N50 size is the length such that 50% of the assembled genome lies
in blocks of the N50 size or longer. The N50 length for supercontigs
is 88.86 kb and is 2.97 kb for contigs. The total number of bases in
supercontigs is 3.66 Gb and in contigs is 2.14 Gb.
</p>


<h4>Annotation</h4>
<p>
Owing to the fragmentary nature of this preliminary assembly, it was necessary to arrange some scaffolds into "gene-scaffold" super-structures, in order to present complete genes. There are 6153 such gene-scaffolds which consist of 1.58 Gb , with identifiers of the form "GeneScaffold_1".
</p>

<ul><li><a href="/info/docs/genebuild/2x_genomes.html">Details of the gene-scaffold construction and subsequent gene-build</a></li></ul>


<h4>Mammalian Genome Project</h4>
<p><i>Tupaia belangeri</i> is one of 24 mammals that will be sequenced as part
of the <a href="http://www.broad.mit.edu/mammals/">Mammalian Genome Project</a>, funded by the National Institutes of
Health (<a href="http://www.nih.gov/">NIH</a>). A group of species were chosen to maximise the branch
length of the evolutionary tree while representing the diversity of
mammalian species. Low-coverage 2X assemblies will be produced for
these mammals and used in alignments for cross-species comparison. The
aim is to increase our understanding of functional elements, especially
in the human genome.
</p>


