<h3>Assembly</h3>

<p class="space-below"><img src="/img/species/pic_Meleagris_gallopavo.png" style="width:300px;height:210px"
class="float-left" alt="Turkey" title="gobble gobble!" />

<p class="space-below">UMD 2.0 is an assembly of the domesticated turkey genome. The first sequences have been generated with the GS-Titanium platform in VBI's (<a href="https://www.vbi.vt.edu/">Virginia Bioinformatics Institute</a>) Core Lab.
Additionnally, the USDA (<a href="http://www.ars.usda.gov/main/site_main.htm?MODECODE=12-00-00-00">United States Department of
Agriculture</a>) Beltsville group provided 30x coverage with
Illumina 70 bp reads. And the Salzberg group at UMD (<a href="http://www.cbcb.umd.edu/">University
of MarylandM</a>) assisted with assembly and annotation of the
integrated 454/Illumina/BAC end sequence data set into a "draft
genome sequence".</p>

<p>The whole composite covers 90% of the genome.
The turkey genome is composed of 33 chromosomes, assembled from
176.691 contigs.
The total number of bases in the genome is 1.08Gb.</p>

<h3>Annotation</h3>
<p>The gene set for turkey was built using the Ensembl genebuild
pipeline. Gene models are based on genewise alignments of
chicken proteins as well as genetically distant proteins from
other species, including most vertebrates from Uniprot. To
improve the accuracy of models generated from distant species,
the Genewise alignments were made to stretches of genomic
sequence rather than to 'miniseqs'. The protein based gene
models were then extended using turkey cDNA and EST alignments
as well as chicken cDNA. The resulting gene models were then
assessed by generating sets of potential orthologs to genes from
other vertebrate species. Potentially missing predictions and
partial gene predictions were identified by examining the
orthologs, which were then used to build new gene models.</p>

