<h3>Assembly</h3>

<p><img src="/img/species/pic_Cavia_porcellus.png" style="height:91px;width:100px" class="float-left" alt="Guinea Pig" title="ooeeeeek!" />
This is the first release of the high-coverage 6.79X assembly of the
Guinea Pig (<i>Cavia porcellus</i>). The genome sequencing
and assembly is provided by the 
<a href="http://www.broad.mit.edu/">Broad Institute</a>.
</p>

<p>
The N50 size is the length such that 50% of the assembled genome lies in blocks of the N50 size or longer. The N50 length for supercontigs is 80.58 kb and the N50 of the supercontigs is 27.41 Mb. The total length of all contigs is 2.66 Gb. When the gaps between contigs in chromosomes are included, the total span of the assembly is 2.72 Gb.
</p>

<h3 style="clear:left">Annotation</h3>
<p>
Due to the lack of Guineapig protein and cDNA resources, the majority of 
the gene models are based on Genewise alignments of proteins from 
mammals. Alignments made from Guineapig proteins had UTR added using 
Guineapig cDNAs. The gene models were assessed by generating sets of 
potential orthologs to genes from other mammalian species. Potentially 
missing predictions and partial gene predictions were identified by 
examining the orthologs, and exonerate used to build new gene models for 
these based on the human ortholog peptide sequence.
</p>

<h3>Mammalian Genome Project</h3>
<p><i>Cavia porcellus</i> is one of 24 mammals that will be sequenced as part
of the <a href="http://www.broad.mit.edu/mammals/">Mammalian Genome Project</a>, funded by the National Institutes of
Health (<a href="http://www.nih.gov/">NIH</a>). A group of species were chosen to maximise the branch
length of the evolutionary tree while representing the diversity of
mammalian species. Low-coverage 2X assemblies will be produced for
these mammals and used in alignments for cross-species comparison. A 
subset of these species are being sequenced in further depth. The
aim is to increase our understanding of functional elements, especially
in the human genome.
</p>

