<h3>Assembly</h3>

<p class="space-below"><img src="/img/species/pic_Sus_scrofa.png" style="height:75px;width:125px" class="float-left" alt="Pig" title="Oink" />
This is the first release of the 
<a href="http://www.sanger.ac.uk/Projects/S_scrofa/">high-coverage Sscrofa9 (April 2009) assembly</a> 
for chromosomes 1 to 18 and X of the pig genome,  <i>Sus scrofa</i>.  Using the 
integrated highly contiguous physical map of the pig genome 
(Humphray et al, Genome Biology 2007,8:R139) as a template for sequencing, the  
<a href="http://www.piggenome.org/">International Swine Genome Sequencing Consortium</a>
has secured <a href="http://www.sanger.ac.uk/Info/Press/2006/060116.shtml">first phase funding</a> 
from the USDA to achieve a draft 4x sequence depth across the genome through a minimal tilepath
BAC by BAC approach.  The map is localised to the genome through the
integration of the UIUC RH markers, the tilepath is being selected from the 
<a href="http://www.sanger.ac.uk/Projects/S_scrofa/library_details.shtml#CHORI242">CH-242 library</a> 
generated from a single Duroc sow.</a>

<h3>Annotation</h3>
<p>
The Pig Sscrofa9 was annotated using a standard Ensembl mammalian pipeline. Predictions 
from vertebrates as well as pig proteins have been used to create the gene set, along with 
human 1:1 orthologs aligned to the pig genome using Exonerate.  Pig cDNAs have been used 
to add UTRs to pig protein based predictions, and pig ESTs were used to add UTR to predictions 
made from non-pig proteins.
</p>

