<h3>Assembly</h3>

<p class="space-below"><img src="/img/species/pic_Callithrix_jacchus.png" style="height:100px;width:120px" alt="common marmoset" class="float-left" />
Callithrix jacchus-3.2 is an assembly of the 
<a href="http://genome.wustl.edu/genome.cgi?GENOME=Callithrix%20jacchus">common marmoset genome</a>.
 It was sequenced to 6X coverage using a female marmoset (animal id #186/17066) DNA source kindly 
provided by Dr. Suzette Tardif, of the  <a href ="http://www.sfbr.org/SNPRC/index.aspx">Southwestern National Primate Research Center</a> 
(San Antonio, Texas). The DNA source for the CHORI-259 BAC library was male and the full brother of the 
female used for whole genome shotgun sequencing (#186/17066). The combined sequence reads were assembled 
using the PCAP software <a href = "http://www.ncbi.nlm.nih.gov/pubmed/12952883">(Genome Res. 13(9):2164-70 2003)</a> 
and filtered for all known non-marmoset sequence contaminants as well as singleton contigs and supercontigs 
less than 2kb in length.
</p>

<p class= "space-below">The sequence assembly was provided by the Washington University School
of Medicine, Genome Sequencing Center, St Louis, Missouri.
</p>

<p class="space-below">The N50 size is the length such that 50% of the assembled genome lies
in blocks of the N50 size or longer. The N50 size of the contigs is 29.2 kb. The total length 
of all contigs is 2.76 Gb.
</p>

<p>The 3.2 assembly is a preliminary assembly for marmoset.
       A higher coverage assembly is in process, which will lead to a
       new genebuild.</p>

<h3>Annotation</h3>
<p>
The gene set for marmoset was built using the Ensembl pipeline.
       The species-specific resources for marmoset being relatively
       limited, most of the evidences are taken from primates, and
       particularly human. The gene structures are mainly based on
       Genewise alignments of primate proteins and gene predictions
       based on human cDNAs. Additionally, human genes were aligned
       thanks to the projection of the gene structures, but also direct
       alignment using exonerate.
       Both human and marmoset cDNAs were used to add UTR structures.
</p>
