<h3>Assembly</h3>

<p class="space-below"><img src="/img/species/pic_Gorilla_gorilla.png" style="width:150px;height:200px;float:left;margin-left:10px;margin-bottom:10px;float:right" alt="Gorilla photo" title="Wild? I was absolutely livid!"/>
This is the third release of the draft assembly of the Western lowland gorilla 
<em>Gorilla gorilla gorilla</em>. The DNA sample came from a 30-year-old female, 
Kamilah, owned by the San Diego Wild Animal Park, 
and sequencing and assembly is provided by the 
<a href="http://www.sanger.ac.uk/">Welcome Trust Sanger Institute</a>.
</p>

<p class="space-below">Sequencing was undertaken using two separate methods: traditional
capillary whole-genome shotgun (WGS) sequencing and Solexa
new-technology sequencing. Results from the two methods were used in
the first, second and third draft gorilla assemblies.</p>

<p class="space-below">The first draft assembly (gorGor1) was released in September 2008. This
initial draft assembly was a 2.1x coverage assembly. It was created from
WGS capillary reads using the <a href="http://www.sanger.ac.uk/Software/production/phusion/">Phusion assembler</a>,
with these capillary reads' sequencing errors being corrected by taking the 
consensus of Solexa data aligned to it.</p>

<p class="space-below">To create the second draft assemby, Solexa data, sequenced at roughly 35x,
was assembled into contigs using 
<a href="http://genome.cshlp.org/content/19/6/1117.full">Abyss</a>.
The resulting contigs of length 50bp or longer were then assembled along with the WGS
capillary data using the <a href="http://www.sanger.ac.uk/Software/production/phusion/">Phusion assembler</a>. Next, the Solexa read pairs
were aligned to the human reference genome using
<a href="http://genome.cshlp.org/content/18/11/1851.full">Maq</a> to identify 
syntenic regions and breakpoints between human and gorilla. Using human-gorilla 
synteny as a guide, longer gorilla supercontigs were constructed using
<a href="http://genome.cshlp.org/content/18/5/821.full">Velvet</a> and other
assembly tools.</p>

<p class="space-below">In the third draft assembly (gorGor3) for the current release, gorilla
supercontigs which could be ordered with respect to the human reference
genome were assembled into simulated chromosomes, while incorporating
the chromosome 2 split (as in chimpanzee) and the reciprocal
translocation between chromosomes 5 and 17.</p>
 
<p>The total length of the gorGor3 assembly is 3.04Gb. The N50 size for
contigs is 11657 bp and the N50 size for supercontigs is 913458 bp.</p>

<h3>Annotation</h3>

<p>
Gene annotation in gorilla has been generated by projection of genes from the human 
reference genome as well as alignment of proteins from three major sources (in 
descending order of their contribution to the final gene set):
</p>

<ol>
<li>Ensembl human translations from Ensembl release 56</li>
<li>Uniprot mammalian and vertebrate proteins with evidence at either the protein 
or transcript level for their existence; and</li>
<li><em>Gorilla gorilla</em> proteins obtained from UniprotKB.</li>
</ol>

<p class="space-below">
Projection of human genes to gorilla began with the alignment of gorilla
genome to the latest human reference genome (GRCh37 assembly) using
BLASTz. These alignments were used to project human Ensembl gene
structures (Ensembl version 56) to the corresponding location in
gorilla. About 60% of human protein-coding genes were projected onto the
gorilla genome. Small insertions/deletions that disrupt the
reading-frame of the resultant projected transcripts are corrected for
by inserting "frame-shift" introns into the structure. For some human
exons and parts of exons, the corresponding gorilla sequence is missing
from the assembly. In most of these cases, the missing exon is omitted
from the gorilla gene model. In a small number of cases however, where
BLASTz has aligned the human sequence to a gap in the gorilla sequence,
the exon is placed in the gap, resulting in a run of X's of the correct
length in the translation.
</p>

<p class="space-below">
Ensembl human translations were also aligned to the gorilla genome using
Exonerate. The alignment of mammalian/vertebrate proteins and
gorilla-specific proteins followed procedures in the standard Ensembl
genebuild pipeline using Genewise.
</p>

<p class="space-below">
The gene-building procedure on the gorGor3 assembly identified 20803
protein coding genes and 1553 pseudogenes.
</p>

<p><a href="http://vega.sanger.ac.uk/Gorilla_gorilla/"> <img src="/img/vega_small.gif" alt="Vega logo" style="width:83px;height:30px" title="Vega - Vertebrate Genome Annotation database" /></a>
Additional manual annotation of this genome can be found in
<a href="http://vega.sanger.ac.uk/Gorilla_gorilla/">Vega</a></p>
