<h3>Assembly</h3>

<p class="space-below"><img src="/img/species/pic_Choloepus_hoffmanni.png" style="height:233px;width:200px" class="float-right" alt="Sloth" title="What! You got to go there?">
This is the first release of the low-coverage 2.05X assembly of the 
two-toed sloth  (<i>Choloepus hoffmanni</i>). The genome sequencing and 
assembly is provided by the <a href="http://www.broad.mit.edu/node/347">Broad Institute</a>.</p>

<p class="space-below">The N50 size is the length such that 50% of the assembled genome lies 
in blocks of the N50 size or longer. The contigs have an N50 length of 
2461 bp and 8.6 kb for supercontigs. The total number of bases in 
supercontigs is 2.34 Gb and in contigs is 1.95 Gb.</p>


<p class="space-below">The project coordination and genome sequencing and assembly is provided by
the <a href="http://www.hgsc.bcm.tmc.edu/projects/bovine/">Human Genome
Sequencing Center at Baylor College of Medicine</a></p>

 
<h3>Annotation</h3>
<p class="space-below">Owing to the fragmentary nature of this preliminary assembly, it was 
necessary to arrange some scaffolds into "gene-scaffold" 
super-structures, in order to present complete genes. There are 8139 
such gene-scaffolds, with identifiers of the form "GeneScaffold_1".</p>

<ul><li><a href="/info/docs/genebuild/2x_genomes.html">Details of the gene-scaffold construction and subsequent gene-build</a></li></ul>

<h3>Mammalian Genome Project</h3>

<p><i>Choloepus hoffmanni</i> is one of 24 mammals that will be sequenced as part of the
<a href="http://www.broad.mit.edu/mammals/">Mammalian Genome Project</a>, funded
by the National Institutes of Health (<a href="http://www.nih.gov/">NIH</a>). A
group of species were chosen to maximise the branch length of
the evolutionary tree while representing the diversity of mammalian
species. Low-coverage 2X assemblies will be produced for these mammals
and used in alignments for cross-species comparison. The aim is to
increase our understanding of functional elements, especially in the
human genome.</p>

