<p>The Ensembl GRCz10 assembly was annotated using Ensembl's automatic annotation 
pipeline. Predictions from zebrafish proteins have been given priority over 
predictions from other non-mammalian vertebrate species. All Uniprot proteins 
were filtered to remove predictions (PE level 3 and above). Aligned zebrafish cDNAs 
and zebrafish RNASeq data have been used to add UTRs. RNASeq data from embryonic 
and olfactory epithelium tissues were also used to produce gene models. Genes are 
named based on the alignment of their coding regions to known entries in public 
databases; ZFIN genes have priority in this process.</p>

<p>The Ensembl annotations were then merged with Vega annotations at the transcript 
level. Transcripts were merged if they shared the same internal exon-intron boundaries 
(i.e. had identical splicing pattern) with slight differences in the terminal exons 
allowed. Importantly, all Vega source transcripts (regardless of merge status) were 
included in the final merged gene set.</p>

<ul>
<li><a href="/info/genome/genebuild/2015_04_zebrafish_genebuild.pdf">Detailed information on genebuild</a> (PDF)</li>
</ul>
