<p>FicAlb_1.4 was annotated using a standard Ensembl gene annotation system, incorporating RNAseq data provided by Uppsala University. The annotation process is described in the document below.</p>

<h3>RNAseq data set</h3>

<p>In addition to the Ensembl gene set, we produced RNAseq-based gene models and an indexed BAM file for each of the nine tissue sample used by the RNAseq pipeline and also for the merged data from all tissues. Each RNAseq-based gene model represents only the best supported transcript model. We did a BLASTp of these transcript models against UniProt proteins of protein existence level 1 and 2 in order to annotate the open reading frame. The best BLAST hit is displayed as a transcript supporting evidence.</p>

<p>A total of 12238 gene models in the Ensembl gene set include RNAseq data as supporting evidence. Of these 12238 gene models, 6793 are annotated exclusively from RNAseq data with a further 5445 gene models annotated using a combination of RNAseq data and other protein evidence. The RNAseq data were used to add UTR to protein coding models.</p>

<p>The tissue-specific sets of transcript models built using our RNAseq pipeline are as follows:</p>

<table class="ss tint">
  <tr><th>Tissue</th><th>Number of gene models</th></tr>
  <tr class="bg1"><td>brain</td><td>16758</td></tr>
  <tr class="bg2"><td>embryo</td><td>17711</td></tr>
  <tr class="bg1"><td>kidney</td><td>16432</td></tr>
  <tr class="bg2"><td>liver</td><td>14916</td></tr>
  <tr class="bg1"><td>lung</td><td>15946</td></tr>
  <tr class="bg2"><td>merged</td><td>22302</td></tr>
  <tr class="bg1"><td>muscle</td><td>14963</td></tr>
  <tr class="bg2"><td>ovary</td><td>19306</td></tr>
  <tr class="bg1"><td>skin</td><td>17358</td></tr>
  <tr class="bg2"><td>testis</td><td>18812</td></tr>
</table>

<p>Additional splice junctions (introns) may have been identified by our pipeline and not included in the best supported transcript model. We therefore provide users with the full set of introns identified by our RNAseq pipeline to enable further analysis. These introns were identified by searching for reads that splice when mapped to the genome.</p>

<ul>
  <li><a href="/info/genome/genebuild/2013_08_flycatcher_genebuild.pdf">Detailed information on genebuild</a> (PDF)</li>
</ul>