<h3>Annotation</h3>

<p class="space-below">
The mouse assembly GRCm38 was annotated using Ensembl's automatic 
annotation pipeline. This includes an updated mouse-specific repeat 
library, <a href="http://www.ncbi.nlm.nih.gov/RefSeq">RefSeq</a> and 
<a href="http://www.uniprot.org/protein">Uniprot</a> 
protein sequence data for annotating the 
coding regions of protein-coding genes, as well as mouse cDNAs and ESTs 
for annotation untranslated regions (UTRs) of protein-coding genes. 
Extensive data quality checks have been performed to remove 
gene/transcript models with erroneous structures (e.g. interlocking 
transcripts with long introns on the same strand) or supported by 
dubious evidence (e.g. cDNA fragments with short, wrongly annotated 
open-reading frames).
</p>

<p class="space-below">
In release 68 (July 2012), we continue to display a joint gene set based 
on the merge between the automatic annotation from Ensembl and the 
manually curated annotation from Havana. This refined gene set 
corresponds to GENCODE release 12. The Consensus Coding Sequence (CCDS) 
identifiers have also been mapped to the annotations. More information 
about the CCDS project.
</p>

<p class="space-below">
Updated manual annotation from Havana is merged into the Ensembl 
annotation every release. Transcripts from the two annotation sources 
are merged if they share the same internal exon-intron boundaries (i.e. 
have identical splicing pattern) with slight differences in the terminal 
exons allowed. Importantly, all Havana transcripts are included in the 
final Ensembl/Havana merged (GENCODE) gene set. In this release, 23532 
Ensembl gene models and 46246 Havana genes were merged together to 
create the final set of 57891 genes.
</p>

<p class="space-below">
As in all previous releases since October 2007, in Ensembl release 68, 
we provide a combined Ensembl-Vega merged gene set.  Specifically, the 
frozen Ensembl annotations from May 2012 were merged with the latest 
<a href="http://Vega.sanger.ac.uk/Mus_musculus/Info/Index">Vega</a> 
manual annotations (April 2012) at the transcript level. The Vega 
annotations were generated mainly by the HAVANA team at the 
<a href="http://www.sanger.ac.uk">Wellcome Trust Sanger Institute</a>. 
Transcripts from the two annotation sources were 
merged if they shared the same internal exon-intron boundaries (i.e. had 
identical splicing pattern) with slight differences in the terminal 
exons allowed. Importantly, all Vega source transcripts (regardless of 
merge status) were included in the final merged gene set.  In addition, 
the gene set was also updated in release 66 to maintain its consistency 
with the 22119 <a href="<http://www.ncbi.nlm.nih.gov/projects/CCDS/CcdsBrowse.cgi">CCDS</a> 
gene models at the time of production (April 2012).
</p>

<p class="space-below">
As a result, the release 68 gene set consists of 37315 genes and 90956 
transcripts. Of the 90956 transcripts, 17.6% (16054) were the result of 
merging Ensembl and Vega annotations, 15.6% (14197) originated from 
Ensembl, 60.6% (55162) originated from Vega, and a remaining ~6.2% were 
incorporated from other sources (e.g. non-coding RNAs/transcripts 
imported from <a href="http://imgt.cines.fr/">IMGT</a> data). 
Most of the non-merged Vega transcripts were alternative splice variants 
and/or non-coding transcripts complementing Ensembl annotations which focus 
on providing a conservative set of protein-coding genes/transcripts.
</p>

<ul>
<li><a href="/Mus_musculus/2012_05_mouse_genebuild.pdf">Detailed information on genebuild</a> (PDF)</li>
</ul>

<p class="space-below">
<a href="http://vega.sanger.ac.uk/Mus_musculus/"> <img src="/img/vega_small.gif" alt="Vega logo" style="width:83px;height:30px" title="Vega - Vertebrate Genome Annotation database" /></a>
Additional manual annotation of this genome can be found in
<a href="http://vega.sanger.ac.uk/Mus_musculus/">Vega</a></p>

<a href="http://www.heroic-ip.eu/"><img src="/img/credits/heroic_logo.jpg" alt="HEROIC" style="width:100px;height:26px;float:left;margin-right:5px" /></a>
<p>Additional functional genomics data produced by the HEROIC project (High-throughput Epigenetic 
Regulatory Organisation In Chromatin) is available to download from the 
<a href="http://projects.ensembl.org/heroic/">Ensembl Projects HEROIC portal</a>.</p>
