<h3>Annotation</h3>

<p class="space-below">
The mouse assembly GRCm38 was annotated using Ensembl's automatic annotation system. This includes an updated mouse-specific repeat library, RefSeq and Uniprot protein sequence data for annotating the coding regions of protein-coding genes, as well as mouse cDNAs and ESTs for annotation untranslated regions (UTRs) of protein-coding genes. 
</p>

<p class="space-below">
In Ensembl release 68 (July 2012), we continue to display a joint gene set based on the merge between the automatic annotation from Ensembl (May 2012) and the manually curated annotation from <a href="http://vega.sanger.ac.uk/index.html">Havana</a>. Transcripts from the two annotation sources were merged if they shared the same internal exon-intron boundaries (i.e. had identical splicing pattern) with slight differences in the terminal exons allowed. Importantly, all Vega source transcripts (regardless of merge status) were included in the final merged gene set. This gene set was provided to <a href="http://www.gencodegenes.org/">GENCODE</a> to produce the GENCODE 13 release.
</p>

<p class="space-below">
The Ensembl release 68 gene set consists of 23140 protein-coding genes (including polymorphic pseudogenes, immunoglobulin and T-cell receptor genes), 3132 long noncoding RNA genes, 5500 pseudogenes, and 5506 short noncoding RNA genes. An additional 37 genes were imported for the mitochondrion. This makes a total of 37315 genes and 90956 transcripts, across both the primary assembly and alternate sequences. Of the 90956 transcripts, 17.6% (16054) were the result of merging Ensembl and Vega annotations, 15.6% (14197) originated from Ensembl, 60.6% (55162) originated from Vega, and a remaining ~6.2% were incorporated from other sources (e.g. non-coding RNAs/transcripts imported from IMGT data). Most of the non-merged Vega transcripts were alternative splice variants and/or non-coding transcripts complementing Ensembl annotations which focus on providing a conservative set of protein-coding genes/transcripts.
</p>

<p class="space-below">
In addition to the gene set, we display alignments of mouse cDNA and EST sequences. The mouse cDNA alignments are updated for every Ensembl release. We also display alignments of sequences from UniProt, UniGene and the ENA vertebrate RNA collection, and ab initio gene predictions from Genscan. The Consensus Coding Sequence (CCDS) identifiers have also been mapped to the new assembly. More information about the <a href="http://www.ncbi.nlm.nih.gov/CCDS/CcdsBrowse.cgi">CCDS</a> project.
</p>

<ul>
<li><a href="/Mus_musculus/2012_05_mouse_genebuild.pdf">Detailed information on genebuild</a> (PDF)</li>
</ul>

<p class="space-below">
<a href="http://vega.sanger.ac.uk/Mus_musculus/Info/Index"> <img src="/img/vega_small.gif" alt="Vega logo" style="width:83px;height:30px;margin-right:23px" title="Vega - Vertebrate Genome Annotation database" /></a>
Additional manual annotation of this genome can be found in
<a href="http://vega.sanger.ac.uk/Mus_musculus/Info/Index">Vega</a></p>

<a href="http://www.heroic-ip.eu/"><img src="/img/credits/heroic_logo.jpg" alt="HEROIC" style="width:100px;height:26px;float:left;margin-right:10px" /></a>
<p>Additional functional genomics data produced by the HEROIC project (High-throughput Epigenetic 
Regulatory Organisation In Chromatin) is available to download from the 
<a href="http://projects.ensembl.org/heroic/">Ensembl Projects HEROIC portal</a>.</p>
