<h3>Annotation</h3>

<p class="space-below">
The Ensembl mouse gene annotations have been vastly improved by using 
updated Ensembl genebuild pipeline code and incorporating new data 
resources which have become available since the last NCBIM37 genebuild 
(April 2007). The new resources include an updated mouse-specific 
repeat library, additional RefSeq and Uniprot protein sequence data for 
predicting the coding regions of protein-coding genes, as well as new 
cDNAs and ESTs for annotating untranslated regions (UTRs) of 
protein-coding genes. Extensive data quality checks have been performed 
to remove gene/transcript models with erroneous structures (e.g. 
interlocking transcripts with long introns on the same strand) or 
supported by dubious evidence (e.g. cDNA fragments with short, wrongly 
annotated open-reading frames).
</p>

<p class="space-below">
The Ensembl annotations were then merged with  
<a href="http://Vega.sanger.ac.uk/Mus_musculus/Info/Index">Vega</a> 
annotations (mainly generated by the HAVANA team at the Wellcome Trust 
Sanger Institute) at the transcript level. As in previous releases 
since October 2007, in release 61 we provide the combined Ensembl-Vega 
gene set. Transcripts from the two annotation sources were merged if 
they shared the same internal exon-intron boundaries (i.e. had identical 
splicing pattern) with slight differences in the terminal exons allowed. 
Importantly, all Vega source transcripts (regardless of merge status) 
were included in the final merged gene set.
</p>

<p class="space-below">
The combined Ensembl-Vega gene set consists of 36817 genes and 93809 
transcripts. Of the 93809 transcripts, 18.96% (17788) were the result of 
merging Ensembl and Vega annotations, 22.18% (20809) originated from 
Ensembl, 52.04% (48821) originated from Vega, and a remaining ~6.8% were 
incorporated from other sources (e.g. immunoglobulin gene 
segments/transcripts imported from IMGT data). Most of the non-merged 
Vega transcripts were alternative splice variants and/or non-coding 
transcripts complementing Ensembl annotations which focus on providing a 
conservative set of protein-coding genes/transcripts.
</p>

<p class="space-below">
Since September 2006, the mouse genome sequence has been considered 
sufficiently stable to start identifying those gene placements that are 
identical in the three major genome browsers. The
<a href="http://www.ncbi.nlm.nih.gov/projects/CCDS/CcdsBrowse.cgi">CCDS project</a> 
was established and Ensembl is collaborating with
<a href="http://Vega.sanger.ac.uk/Mus_musculus/Info/Index">HAVANA</a>, 
<a href="http://www.ncbi.nlm.nih.gov/RefSeq/">NCBI</a> and 
<a href="http://genome.ucsc.edu/">UCSC</a> to identify a core set of 
protein coding regions that are consistently annotated and of high quality. 
All CCDS annotations (as of 22 September 2010) are included in the 
Ensembl-HAVANA gene set, and the CCDS identifiers are shown where applicable.
</p>
<ul>
<li><a href="/Mus_musculus/2010_11_mouse_genebuild.pdf">Detailed information on genebuild</a> (PDF)</li>
</ul>

<p class="space-below">
<a href="http://vega.sanger.ac.uk/Mus_musculus/"> <img src="/img/vega_small.gif" alt="Vega logo" style="width:83px;height:30px" title="Vega - Vertebrate Genome Annotation database" /></a>
Additional manual annotation of this genome can be found in
<a href="http://vega.sanger.ac.uk/Mus_musculus/">Vega</a></p>

<a href="http://www.heroic-ip.eu/"><img src="/img/credits/heroic_logo.jpg" alt="HEROIC" style="width:100px;height:26px;float:left;margin-right:5px" /></a>
<p>Additional functional genomics data produced by the HEROIC project (High-throughput Epigenetic 
Regulatory Organisation In Chromatin) is available to download from the 
<a href="http://projects.ensembl.org/heroic/">Ensembl Projects HEROIC portal</a>.</p>
