<p>The gene set for rabbit was built using the Ensembl genebuild
pipeline. Gene models are based on genewise alignments of glire
proteins as well as genetically distant proteins from other
species, including most mammal proteins from Uniprot. To improve
the accuracy of models generated from distant species, the
Genewise alignments were made to stretches of genomic sequence
rather than to 'miniseqs'. Additionnaly, human and mouse
one-to-one orthologues were aligned using exonerate, combined
with rabbit-specific proteins. The protein based gene models
were then extended using rabbit cDNA and EST alignments. The
resulting gene models were then assessed by generating sets of
potential orthologs to genes from other vertebrate species.
Potentially missing predictions and partial gene predictions
were identified by examining the orthologs, which were then used
to build new gene models. </p>
<ul>
<li><a href="/info/docs/genebuild/2011_01_rabbit_genebuild.pdf">Detailed information on genebuild</a> (PDF)</li>
</ul>


<h3>Mammalian Genome Project</h3>
<p><i>Oryctolagus cuniculus</i> is one of 24 mammals that will be sequenced as 
part of the 
<a href="http://www.broad.mit.edu/mammals/">Mammalian Genome Project</a>, funded by the National Institutes of Health (<a href="http://www.nih.gov/">NIH</a>).  
The species were chosen to 
maximise the branch length of the evolutionary
tree while representing the diversity of mammalian species.
Low-coverage 2X assemblies will be produced for these mammals and used 
in alignments for cross-species comparison.
The aim is to increase our understanding of functional elements, 
especially in the human genome.
</p>

<h3>RNASeq data set</h3>

<p>
In addition to the main set, we have predicted gene models for each tissue
type using the RNA-Seq pipeline. We did a BLASTp of these models against
UniProt vertebrate proteins of protein existence level 1 and 2 in order to
confirm the open reading frame (ORF). The best BLAST hit is displayed as a
transcript supporting evidence.
</p>

<p>
The tissue-specific sets of transcript models built using our RNAseq
pipeline are as follows:
</p>

<table class="ss tint">
  <tr><th>Tissue</th><th>Number of gene models</th></tr>
  <tr class="bg1"><td>blood</td><td>11980</td></tr>
  <tr class="bg2"><td>brain</td><td>16017</td></tr>
  <tr class="bg1"><td>heart</td><td>14476</td></tr>
  <tr class="bg2"><td>kidney</td><td>14813</td></tr>
  <tr class="bg1"><td>liver</td><td>13907</td></tr>
  <tr class="bg2"><td>lung</td><td>15473</td></tr>
  <tr class="bg1"><td>merged</td><td>22042</td></tr>
  <tr class="bg2"><td>ovary</td><td>15958</td></tr>
  <tr class="bg1"><td>skeletal muscle</td><td>13593</td></tr>
  <tr class="bg2"><td>skin</td><td>15398</td></tr>
  <tr class="bg1"><td>testis</td><td>19135</td></tr>
</table>

