<html>
<head>
<meta name="author" content="helpdesk" />
<meta name="navigation" content="eFG introduction" />
<title>Introduction to eFG</title>
</head>

<body>

<h2>Introduction to Ensembl Functional Genomics</h2> 




<h3 class="boxed">Installation Requirements</h3>

<p>Given that you have already followed the instructions for installation of the Ensembl core and functional genomics APIs (http://www.ensembl.org/info/using/api/api_installation.html), the next step is to set up the eFG specific requirements.  This is an exhaustive list and may not be necessary if you do not intend to use the full functionality of eFG.  Install the following as required:</p>

<ul>
1.  R<br>
Required for normalisation, QC reports and analysis.<br>
Available from http://cran.r-project.org/<br>
<br>
2.  RMySQL -  R package<br>
Required for normalisation, QC reports and analysis.<br>
Available from http://cran.r-project.org/<br>
<br>
3.  BioConductor - R platform<br>
Required for normalisation, QC reports and analysis.<br>
Available from http://www.bioconductor.org/download<br>
<br>
4.  Ringo - BioConductor package<br>
Required for normalisation, QC reports and analysis.<br>
Available from http://bioconductor.org/packages/2.1/bioc/html/Ringo.html<br>
<br>
5.  Tab2MAGE & MAGE.dtd<br>
Required for meta data import and subsequent submission to ArrayExpress.	<br>
Tab2MAGE is available from http://sourceforge.net/project/showfiles.php?group_id=120325<br>
MAGE.dtd v1.1 is available from http://mged.sourceforge.net/software/docs.php<br>
The MAGE.dtd should be placed in the root data directory ($EFG_DATA).<br>
<br>
Something about hardware here..................................
</ul>


<h3 class="boxed">Set Up</h3>

<p>The eFG system uses a shell environment to set global variables and help perform common tasks.  You will need to edit the .efg file accordingly:</p>


<font face='courier'>efg@bc-9-1-02>more ensembl-functgenomics/scripts/.efg<br>
#!/usr/local/bin/bash<br>
<br>
echo "Setting up the Ensembl Function Genomics environment..."<br>
<br>
### ENV VARS ###<br>
<br>
#Prompt<br>
export PS1='efg@$PS1HOST>'<br>
<br>
#Code/Data Directories<br>
export SRC=~/src                                #Root source code directory. EDIT<br>
export EFG_SRC=$SRC/ensembl-functgenomics       #eFG source directory<br>
export EFG_SQL=$EFG_SRC/sql                     #eFG SQL<br>
export EFG_DATA=/your/data/dir/efg              #Data directory. EDIT<br>
export PATH=$PATH:$EFG_SRC/scripts              #eFG scripts directory<br>
export PERL5LIB=$EFG_SRC/modules:$PERL5LIB      #Update PERL5LIB. EDIT add ensembl(core) etc. if required<br>
<br>
#Your efg DB connection params<br>
export WRITE_USER='write_user'                  #EDIT<br>
export READ_USER='read_user'                    #EDIT<br>
export HOST='efg-host'                          #EDIT<br>
export PORT=3306                                #EDIT<br>
export MYSQL_ARGS="-h${HOST} -P${PORT}"   <br>      
<br>
#Your ensembl core DB connection params, read only<br>
export CORE_USER='anonymous'                    #EDIT if required<br>
export CORE_HOST='ensembldb.ensembl.org'        #EDIT if required<br>
export CORE_PORT=3306                           #EDIT if required<br>
<br>
#Default norm and analysis methods<br>
export NORM_METHOD='VSN_GLOG'                   #EDIT if required e.g. T.Biweight, Loess<br>
export PEAK_METHOD='Nessie'                     #EDIT if required e.g. TileMap, MPeak, Chipotle<br>
<br>
#R config<br>
export R_LIBS=${R_LIBS:=$SRC/R-modules}         #EDIT if required<br>
export R_PATH=/software/bin/R                   #Location of local version of R. EDIT<br>
export R_FARM_PATH=/software/R-2.4.0/bin/R      #Location of farm installed R. EDIT<br>
export R_BSUB_OPTIONS="-R'select[type==LINUX64 && mem>6000] rusage[mem=6000]' -q bigmem" #EDIT<br>
<br>
</font>
<p>
As is indicated at the head of the .efg file, to enable easy access to the eFG environment it is useful to add the following to your .*rc login file:</p>

<font face='courier'>alias efg='. ~/src/ensembl-efg/.efg'<br></font>
<p>
Once this is done simple type 'efg' to enter the envornment, which will give you access to some helper functions such as, CreateDB:</p>
<font face='courier'>
efg@bc-9-1-02>CreateDB my_homo_sapiens_funcgen_47_36i password<br>
Creating DB my_homo_sapiens_funcgen_47_36i<br>
</font>
<p>
It is desirable to maintain the standard Ensembl nomenclature for a database and simply prefix it with some descriptive tag.  Failure to do so may cause problems in dynamically detecting the correct core DB to use.  The CreateDB function also supports overwriting of a particular instance of an eFG DB by specifying a third 'drop' argument:</p>
<font face='courier'>
efg@bc-9-1-02>CreateDB my_homo_sapiens_funcgen_47_36i password drop<br>
Dropping DB my_homo_sapiens_funcgen_47_36i<br>
Creating DB my_homo_sapiens_funcgen_47_36i<br>
</font>
<p>
Once you have set up the environment, you are now ready to import data or query the central ensembl or a local copy of an eFG DB.  </p>
<p>
<b>Note:</b>  It is not necessary to set up the environment if you simply want to query a remote eFG DB i.e. The eFG Dbs available at ensembldb.ensembl.org.  However, you may find that some of the tools scripts will require explicit definition of some of the above environment variables via the command line.  
</p>
<h3 class="boxed">Tools scripts</h3>

<p>There are various types of data import, export and transformation which can be performed using the scripts available in the scripts directory.  These encompass simple cell and feature type imports, through to array design and full experiment imports.  Most of the more common tasks have template shell scripts with required parameters set and others left for editing.   Here follows a list of the main types of tool script:</p>

<ul>	
1. Feature and CellType import.  This is done explicitly to avoid propogation of duplicated types through the database.<br>  
See:	run_import_type.sh<br>
<br>
2. Array design import.  Importing arrays prior to receipt of experimental can be beneficial.  During the initial import of an array an exhaustive process is undertaken to resolve replicate probe and build a cache external to the DB. Depending on the size of the array, this process can take quite a few hours.  The benefit of this process apart from the correct handling of replicate probes is that subsequent imports based on the given array are much faster.  eFG also supports import for custom pre-production arrays to facilitate analysis and design turnover during development. <br> 
See:	run_import_design.sh and run_import_array_from_fasta.sh<br>
<br>
3. Experiment import.  eFG supports import of experiments using the Nimblegen tiling array and Sanger PCR array platforms.  See:	 run_NIMBLEGEN.sh and run_Sanger.sh <br>
<br>
4. Feature level import.  It is possible to perform a direct feature import.  This accommodates pre-analysed data or technologies which are not fully supported by eFG. e.g. Hitlists, Wiggletracks, Short reads.<br>
See:	run_Solexa.sh, run_import_hitlist.sh and run_import_wiggle.sh<br>
<br>
5. Feature mapping.  eFG utilises the assembly mapping information in the core DB to remap feature up to new versions of a given genome. <br>
 See:	 run_remap_features.sh and run_project_feature_set.sh <br>
<br>
<b>Note:</b>  This has been shown to give spurious results when remapping probe level features, dependent on the nature of the assembly change.  A more exhaustive full genomic mapping is also possible via the eFG probe mapping pipeline.<br>
<br>
6. Data export.  Various flavours of data export including GFF and custom formats.<br>
See:	run_dump_gff_features.sh and get_data.pl<br>
<br>
7. Data rollback.  Removes part or all of an experimental import and it's associated data.<br>
See:	roll_back_experiment.pl<br>
<br>	
</ul>


<h3 class="boxed">Importing an experiment</h3>

<p>Prior to running your first experiment import, you will likely need to import the necessary features types first.</p>
<font face='courier'>efg@bc-9-1-02>more run_import_type.sh <br>
#!/bin/sh<br>
<br>
PASS=$1<br>
shift<br>
<br>
$EFG_SRC/scripts/import_type.pl\<br>
        -type FeatureType\<br>
        -name H3K4me3\<br>
        -dbname your_homo_sapiens_funcgen_48_36j\<br>
        -description 'Histone 3 Lysine 4 Tri-methyl'\<br>
        -class HISTONE\<br>
        -pass $PASS<br>
<br></font>
<p>               
Feature type names should correspond to a recognised ontology or nomenclature where appropriate e.g. Brno nomenclature for histones.  The class parameter is not required for CellType imports.</p>
To import an experiment you must first create an input directory for the array vendor and your experiment e.g.<br>
<br>
<font face='courier'>mkdir $EFG_DATA/input/NIMBLEGEN/EXPERIMENT_NAME<br></font>    
</p>
<p>The eFG system currently expects only one experiment per input directory.  If your DVD contains more than one experiment, you will need to split the files up, recreating any meta files accordingly e.g. DesignNotes.txt, SampleKey.txt.  A Nimblegen experiment import can be done using the appropriate run script:</p>
<br>
<font face='courier'>efg@bc-9-1-02>more run_NIMBLEGEN.sh <br>
#!/bin/sh<br>
<br>
PASS=$1<br>
shift<br>
<br>
$EFG_SRC/scripts/parse_and_import.pl\<br>
       -name 'DVD_OR_EXPERIMENT_NAME'\		Name of the data directory<br>
       -format tiled\					Array format<br>
	-vendor NIMBLEGEN<br>
       -location Hinxton\				Your group location<br>
       -contact 'your@email.com'\<br>
       -species homo_sapiens\<br>
       -fasta\						Flag to dump the array as a fasta file, useful for remapping<br>
       -port 3306\<br>
       -host dbhost\
       -dbname 'your_homo_sapiens_funcgen_47_36i'\<br>
       -array_set\					Flag to treat every chip/slide as part of one array<br>
       -array_name "DESIGN_NAME"\<br>
	-cell_type e.g. U2OS\<br>
	-feature_type e.g. H3K4me3\<br>
       -group efg\					Your groupname<br>
       -data_version 41_36c\				The Ensembl data version corresponding to your data<br>
       -verbose\<br>
       -tee\<br>
       -pass $PASS\<br>
       -recover					Enables recovery mode for failed/partial imports<br>
<br></font>    
<p>
Running the above script will perform a preliminary import.  This involves validation checks and import of some basic meta data.  The meta data gleaned from the import parameters and available files is automatically populated within a tab2mage file located in the output directory, at which point the import stops to allow manual curation of the tab2mage file.  Due to the lack of comprehensive meta data associated with an experiment DVD, it is necessary to inspect, correct and annotate the tab2mage file where possible.  Failure to do so may result in permanent loss of meta data, an inability to submit to ArrayExpress and a corrupted import which ultimately may prevent any further analysis.</p>
<p>
There are three main areas to be addressed, most of which may have been automatically populated.  Fields which need attention are marked with three question marks e.g. <b>???</b></p>
<ul>
1. Experiment section:  This includes information about the laboratory where the experiment was performed along with some field which are required for ArrayExpress submission.<br>
<br>
2. Protocol section:  This is automatically populated from the defaults present in the vendor definitions package e.g. NimblegenDefs.pm. These should be checked and edited accordingly, propagating the changes to the Hybridisation section.<br>
<br>
3. Hybridisation section:  This is possibly the most important section to scrutinise, any inconsistencies here may result in import failure or anomalous "ResultSet" generation.   This section provides for capturing information about multi cell or feature type experiments, aswell as the relationship between individual slides or in eFG parlance, "ExperimentalChips".  The submitter should cross reference this section with the meta files supplied by the vendor e.g. SampleKey.txt, DesignNotes.txt.<br>
<br>
The following fields should be checked explicitly:<br>
<br>
BioSource					CellType or specific source sample name if known.<br>
Sample					Biological replicate name.<br>
Extract					Technical replicate name.<br>
LabeledExtract				Control/Experimental channel sample.<br>
Immunoprecipitate				Description of IP, blank for control channel.<br>
Hybridization				Description of Hybridisation.<br>
BioSourceMaterial				e.g. cell, tissue MGED?<br>
Dye						e.g. Cy3/5<br>
BioMaterialCharacteristics[StrainOrLine]<br>
BioMaterialCharacteristics[CellType]	<br>
FactorValue[StrainOrLine]	<br>
FactorValue[Immunoprecipitate]		e.g. anti-H3ac antibody<br>
<p>
Some standard naming formats have been put in place to aid validation. Primarily these are the replicate names which have been given the BRN and TRN denominations for biological replicates and techniocal replicates respectively e.g. </p>
EXPERIMENT_BR1_TR1<br>
EXPERIMENT_BR1_TR2<br>
EXPERIMENT_BR2_TR1<br>
<br>
<p>
Here we see two biological replicate for "EXPERIMENT", the first having two technical replicates and the second having just one.  The other naming convention adopted is that of the "FactorValue[Immunoprecipitate]" field.  This must follow the format of the example above i.e. anti-"FeatureType Name" antibody  </p>
<p>
This is parsed during validation and used to store chip/slide level information.  Replicates with mismatching feature type names in this field will fail validation.</p>
<br>
4. <br>
<br>
<b>Note:</b>  The autogenerated tab2mage file is simply a template, you may add further valid fields to any section of the file, or supply your own tab2mage file.  However, the validation checks will still be necessary for correct import, therefore it is advised to follow the standard naming rules given above.<br>
</body>
</html>

