<html>
<head>
<title>Homo sapiens site map</title>
</head>

<body>

<h2><i>Homo sapiens</i> site map</h2>

    <div class="col-wrapper">
    <div class="col3">
    
            <h3 class="boxed">Find a sequence or feature</h3>
        
            <dl class="species-map">
            <dt><a href="/Homo_sapiens/mapview?chr=1">MapView</a></dt>
                <dd>Explore a chromosome</dd>
                <dt><a href="/Homo_sapiens/idhistoryview?gene=ENSG00000139618">IDHistoryView</a></dt>
                <dd>Track the history of an Ensembl stable ID</dd>
                <dt><a href="/Homo_sapiens/historyview">HistoryView</a></dt>
                <dd>Convert a list of old Ensembl stable IDs</dd>
                </dl>

            <h3 class="boxed">Genes and Sequences</h3>
        
            <dl class="species-map">
            <dt><a href="/Homo_sapiens/geneview?gene=ENSG00000139618">GeneView</a></dt>
                <dd>Gene report</dd>
                <dt><a href="/Homo_sapiens/geneseqview?gene=ENSG00000139618">GeneSeqView</a></dt>
                <dd>Displays the sequence section from GeneView</dd>
                <dt><a href="/Homo_sapiens/genespliceview?gene=ENSG00000128573">GeneSpliceView</a></dt>
                <dd>Alternative splicing for a given gene</dd>
                <dt><a href="/Homo_sapiens/generegulationview?gene=ENSG00000124334">GeneRegulationView</a></dt>
                <dd>Regulatory factors for a given gene</dd>
                <dt><a href="/Homo_sapiens/exonview?exon=ENSE00001428812">ExonView</a></dt>
                <dd>Exon report</dd>
                <dt><a href="/Homo_sapiens/transview?transcript=ENST00000342929">TransView</a></dt>
                <dd>Transcript report</dd>
                <dt><a href="/Homo_sapiens/contigview?region=AL138722.15.1.44776;start=101;end=44776">ContigView</a></dt>
                <dd>Small-scale Sequence display</dd>
                <dt><a href="/Homo_sapiens/cytoview?region=AC016942.13.1.152014;start=1;end=150014">CytoView</a></dt>
                <dd>Large-scale Sequence display</dd>
                <dt><a href="/Homo_sapiens/markerview?marker=ECD13863">MarkerView</a></dt>
                <dd>Information about a chromosome marker</dd>
                </dl>
</div>

    <div class="col3">
    

    <h3 class="boxed">Import/Export data</h3>
    
    <dl class="species-map">
<dt><a href="/info/data/external_data/index.html">DAS</a> (Distributed Annotation Server)</dt>

    <dd>Import your data into Ensembl</dd>
    <dt><a href="/Homo_sapiens/exportview">ExportView</a></dt>
    <dd>Export sequence data to file</dd>
    </dl>
    
            <h3 class="boxed">Proteins</h3>
        
            <dl class="species-map">
            <dt><a href="/Homo_sapiens/protview?peptide=ENSP00000342827">ProtView</a></dt>
                <dd>Protein Report</dd>
                <dt><a href="/Homo_sapiens/domainview?domainentry=IPR006207">DomainView</a></dt>
                <dd>Protein Domain report</dd>
                <dt><a href="/Homo_sapiens/familyview?family=ENSF00000000593">FamilyView</a></dt>
                <dd>Protein Family report</dd>
                </dl>

            <h3 class="boxed">Feature displays</h3>
        
            <dl class="species-map">
            <dt><a href="/Homo_sapiens/featureview?type=OligoProbe;id=230777_s_at">FeatureView</a></dt>
                <dd>Display Ensembl features (on a karyotype if available)</dd>
                <dt><a href="/Homo_sapiens/karyoview">KaryoView</a></dt>
                <dd>Map your own data onto a karyotype</dd>
                </dl>
</div>

    <div class="col3">
    
            <h3 class="boxed">Variation</h3>
        
            <dl class="species-map">
            <dt><a href="/Homo_sapiens/genesnpview?gene=ENSG00000128573">GeneSNPView</a></dt>
                <dd>Lists all SNPs for a gene</dd>
                <dt><a href="/Homo_sapiens/transcriptsnpview?transcript=ENST00000342929">TranscriptSNPView</a></dt>
                <dd>Compare a transcript's SNPs in individuals / strains / populations</dd>
                <dt><a href="/Homo_sapiens/snpview?snp=rs10264683">SNPView</a></dt>
                <dd>SNP report</dd>
                <dt><a href="/Homo_sapiens/ldview?snp=rs11571658;bottom=%7Copt_pop_CSHL-HAPMAP%3AHapMap-JPT%3Aon">LDView</a></dt>
                <dd>Diagram of LD values for a SNP</dd>
                <dt><a href="/Homo_sapiens/ldtableview?_format=HTML;dump=ashtml;snp=rs11571658;bottom=%7Copt_pop_CSHL-HAPMAP%3AHapMap-JPT%3Aon%7C">LDTableView</a></dt>
                <dd>Table of LD values for a SNP</dd>
                </dl>

            <h3 class="boxed">Comparative genomics</h3>
        
            <dl class="species-map">
            <dt><a href="/Homo_sapiens/syntenyview?chr=19;other_species=Pan_troglodytes">SyntenyView</a></dt>
                <dd>Compare syntenous regions</dd>
                <dt><a href="/Homo_sapiens/multicontigview?l=1:112714149-112716251;s1=Mus_musculus">MultiContigView</a></dt>
                <dd>Compare syntenous sequences (assembled genomes only) - accessible through SyntenyView</dd>
                <dt><a href="/Homo_sapiens/sequencealignview?l=20:33052876-33145279">SequenceAlignView</a></dt>
                <dd>Compare the DNA sequence between strains or individuals</dd>
                <dt><a href="/Homo_sapiens/genetreeview?gene=ENSG00000139618">GeneTreeView</a></dt>
                <dd>Tree of orthologues for a gene</dd>
                <dt><a href="/Homo_sapiens/geneseqalignview?gene=ENSG00000139618">GeneSeqAlignView</a></dt>
                <dd>Compare the DNA sequence of orthologues</dd>
                </dl>
</div>


    </div>
    

</body>
</html>
