<html>
<head>
<meta name="author" content="webteam" />
<meta name="navigation" content="Data" />
<meta name="update-period" content="2005-05-27" />
<title>Data Downloads</title>
</head>

<body>

<h2>Data Downloads</h2>

<p>Ensembl data is available through a number of routes:</p>

<dl class="spaced">
<dt>Export from Ensembl web pages</dt>
<dd>Many of the pages displaying Ensembl genomic data offer an export 
option, suitable for small amounts of data. <a href="#export">More...</a></dd>
<dt>Public MySQL server</dt>
<dd>For larger amounts of data and more detailed analysis, we recommend you use our 
publicly-accessible MySQL server. <a href="#mysql">More...</a></dd>
<dt>FTP</dt>
<dd>If required, entire databases can be downloaded from our FTP site 
in a variety of formats, from flat files to MySQL dumps. <a href="#ftp">More...</a></dd>
<dt>DAS</dt>
<dd>Ensembl now acts as a DAS (Distributed Annotation System) reference server.
<a href="#das">More...</a></dd>
<dt>BioMart</dt>
<dd>More complex datasets can be retrieved using the BioMart data-mining tool.
<a href="#mart">More...</a></dd>
</dl>

<p>All data produced by the Ensembl project is 
<a href="/info/about/disclaimer.html">freely available</a> for your own use.</p>

<h3 class="boxed" id="export">Exporting from Ensembl pages</h3>

<p>This is the ideal option if you want a short sequence as a FASTA file, or a GFF file
of a few features in a region. Simply find your desired feature or region, click on one of
the "Export..." links in the left-hand menu, and select your output options.</p> 

<p>We respectfully request that users do not script against the export pages on the Ensembl 
website, as this degrades the service for other web servers. The public MySQL server 
(see below) is provided specifically for this purpose. Thank you.</p>

<h3 class="boxed" id="mysql">Public MySQL server</h3>

<p>Ensembl maintains a public MySQL database server, <strong>ensembldb.ensembl.org</strong>,
which you can access as user 'anonymous'.</p>

<p>There are two options for retrieving data from this database:</p>

<dl class="spaced">
<dt>Perl API</dt>
<dd>The easiest option is to use the Ensembl <a href="/info/using/api/">Perl API</a>. 
The API uses an object-oriented approach to model real biological objects such as genes, 
transcripts and 'slices' of DNA sequence, making it straightforward for you to write 
scripts that retrieve and analyze data.</dd>
<dt>MySQL client</dt>
<dd>Alternatively, you can use a MySQL <a href="http://dev.mysql.com/downloads/mysql/5.0.html#downloads">client program</a> to query the database directly. However this does require knowledge of the
complex <a href="/info/using/api/">database schemas</a> used by Ensembl, so we only recommend 
it if you are unable to install our API (e.g. if you have no access to Perl).<br />
<strong>Important note:</strong> From release 48 you need a MySQL5-compatible client and 
to connect to port 5306; releases 47 and earlier are on MySQL4 and use the standard port (3306).</dd>
</dl>


<h3 class="boxed" id="ftp">FTP</h3>

<p>All of the data used on the main Ensembl website can be downloaded from our public 
<a href="ftp_site.html">FTP server</a>. Please be aware that these files can run to many 
gigabytes of data.</p>

<p>For a list of which databases are needed for an Ensembl mirror site, please see the
<a href="/info/webcode/">website installation instructions</a>.</p>

<h3 class="boxed" id="das">Ensembl DAS</h3>

<p>As well as acting as a DAS client to incorporate external data, Ensembl now acts as
a DAS server, giving access to genomic sequences, the latest Ensembl gene predictions, and for 
some species, karyotypes and ditags. 
<a href="/info/using/external_data/das/ensembl_das.html">How to use Ensembl-DAS</a>.</p>

<h3 class="boxed" id="mart">BioMart</h3>

<p>Ensembl supports downloading of many more correlation tables via the highly customisable 
<a href="/biomart/martview/">BioMart data mining tool</a>. You may find exploring this 
web-based data mining tool easier than extracting information from our normalised database dumps.</p>

</body>
</html>
